

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Bio.Blast.Applications module &mdash; Biopython 1.74 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Bio.Blast.NCBIWWW module" href="Bio.Blast.NCBIWWW.html" />
    <link rel="prev" title="Bio.Blast package" href="Bio.Blast.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search"  style="background: #10100F" >
          

          
            <a href="index.html" class="icon icon-home"> Biopython
          

          
            
            <img src="_static/biopython_logo.svg" class="logo" alt="Logo"/>
          
          </a>

          
            
            
              <div class="version">
                1.74
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">API Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="Bio.html">Bio package</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="Bio.html#subpackages">Subpackages</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="Bio.Affy.html">Bio.Affy package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Align.html">Bio.Align package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.AlignIO.html">Bio.AlignIO package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Alphabet.html">Bio.Alphabet package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Application.html">Bio.Application package</a></li>
<li class="toctree-l3 current"><a class="reference internal" href="Bio.Blast.html">Bio.Blast package</a><ul class="current">
<li class="toctree-l4 current"><a class="reference internal" href="Bio.Blast.html#submodules">Submodules</a></li>
<li class="toctree-l4"><a class="reference internal" href="Bio.Blast.html#module-Bio.Blast">Module contents</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="Bio.CAPS.html">Bio.CAPS package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Cluster.html">Bio.Cluster package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Compass.html">Bio.Compass package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Crystal.html">Bio.Crystal package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Data.html">Bio.Data package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Emboss.html">Bio.Emboss package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Entrez.html">Bio.Entrez package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.ExPASy.html">Bio.ExPASy package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.FSSP.html">Bio.FSSP package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.GenBank.html">Bio.GenBank package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Geo.html">Bio.Geo package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Graphics.html">Bio.Graphics package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.HMM.html">Bio.HMM package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.KDTree.html">Bio.KDTree package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.KEGG.html">Bio.KEGG package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Medline.html">Bio.Medline package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.NMR.html">Bio.NMR package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Nexus.html">Bio.Nexus package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.PDB.html">Bio.PDB package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Pathway.html">Bio.Pathway package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Phylo.html">Bio.Phylo package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.PopGen.html">Bio.PopGen package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Restriction.html">Bio.Restriction package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SCOP.html">Bio.SCOP package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SVDSuperimposer.html">Bio.SVDSuperimposer package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SearchIO.html">Bio.SearchIO package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SeqIO.html">Bio.SeqIO package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SeqUtils.html">Bio.SeqUtils package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Sequencing.html">Bio.Sequencing package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Statistics.html">Bio.Statistics package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SubsMat.html">Bio.SubsMat package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SwissProt.html">Bio.SwissProt package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.TogoWS.html">Bio.TogoWS package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.UniGene.html">Bio.UniGene package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.UniProt.html">Bio.UniProt package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Wise.html">Bio.Wise package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.codonalign.html">Bio.codonalign package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.motifs.html">Bio.motifs package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.phenotype.html">Bio.phenotype package</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="Bio.html#submodules">Submodules</a></li>
<li class="toctree-l2"><a class="reference internal" href="Bio.html#module-Bio">Module contents</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="BioSQL.html">BioSQL package</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Biopython</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
          <li><a href="Bio.html">Bio package</a> &raquo;</li>
        
          <li><a href="Bio.Blast.html">Bio.Blast package</a> &raquo;</li>
        
      <li>Bio.Blast.Applications module</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
        
      </li>
    
  </ul>

  
  <div class="rst-breadcrumbs-buttons" role="navigation" aria-label="breadcrumb navigation">
      
        <a href="Bio.Blast.NCBIWWW.html" class="btn btn-neutral float-right" title="Bio.Blast.NCBIWWW module" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="Bio.Blast.html" class="btn btn-neutral float-left" title="Bio.Blast package" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
  </div>
  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="module-Bio.Blast.Applications">
<span id="bio-blast-applications-module"></span><h1>Bio.Blast.Applications module<a class="headerlink" href="#module-Bio.Blast.Applications" title="Permalink to this headline">¶</a></h1>
<p>Definitions for interacting with BLAST related applications.</p>
<p>Wrappers for the new NCBI BLAST+ tools (written in C++):</p>
<blockquote>
<div><ul class="simple">
<li><p>NcbiblastpCommandline - Protein-Protein BLAST</p></li>
<li><p>NcbiblastnCommandline - Nucleotide-Nucleotide BLAST</p></li>
<li><p>NcbiblastxCommandline - Translated Query-Protein Subject BLAST</p></li>
<li><p>NcbitblastnCommandline - Protein Query-Translated Subject BLAST</p></li>
<li><p>NcbitblastxCommandline - Translated Query-Protein Subject BLAST</p></li>
<li><p>NcbipsiblastCommandline - Position-Specific Initiated BLAST</p></li>
<li><p>NcbirpsblastCommandline - Reverse Position Specific BLAST</p></li>
<li><p>NcbirpstblastnCommandline - Translated Reverse Position Specific BLAST</p></li>
<li><p>NcbideltablastCommandline - Protein-Protein domain enhanced lookup time accelerated blast</p></li>
<li><p>NcbiblastformatterCommandline - Convert ASN.1 to other BLAST output formats</p></li>
<li><p>NcbimakeblastdbCommandline - Application to create BLAST databases</p></li>
</ul>
</div></blockquote>
<p>For further details, see:</p>
<p>Camacho et al. BLAST+: architecture and applications
BMC Bioinformatics 2009, 10:421
<a class="reference external" href="https://doi.org/10.1186/1471-2105-10-421">https://doi.org/10.1186/1471-2105-10-421</a></p>
<dl class="class">
<dt id="Bio.Blast.Applications.NcbiblastformatterCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Blast.Applications.</code><code class="sig-name descname">NcbiblastformatterCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='blast_formatter'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Blast.Applications.NcbiblastformatterCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Blast.Applications._NcbibaseblastCommandline</span></code></p>
<p>Wrapper for the NCBI BLAST+ program blast_formatter.</p>
<p>With the release of BLAST 2.2.24+ (i.e. the BLAST suite rewritten in C++
instead of C), the NCBI added the ASN.1 output format option to all the
search tools, and extended the blast_formatter to support this as input.</p>
<p>The blast_formatter command allows you to convert the ASN.1 output into
the other output formats (XML, tabular, plain text, HTML).</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Blast.Applications</span> <span class="k">import</span> <span class="n">NcbiblastformatterCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span> <span class="o">=</span> <span class="n">NcbiblastformatterCommandline</span><span class="p">(</span><span class="n">archive</span><span class="o">=</span><span class="s2">&quot;example.asn&quot;</span><span class="p">,</span> <span class="n">outfmt</span><span class="o">=</span><span class="mi">5</span><span class="p">,</span> <span class="n">out</span><span class="o">=</span><span class="s2">&quot;example.xml&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span>
<span class="go">NcbiblastformatterCommandline(cmd=&#39;blast_formatter&#39;, out=&#39;example.xml&#39;, outfmt=5, archive=&#39;example.asn&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">cline</span><span class="p">)</span>
<span class="go">blast_formatter -out example.xml -outfmt 5 -archive example.asn</span>
</pre></div>
</div>
<p>You would typically run the command line with cline() or via the Python
subprocess module, as described in the Biopython tutorial.</p>
<p>Note that this wrapper is for the version of blast_formatter from BLAST
2.2.24+ (or later) which is when the NCBI first announced the inclusion
this tool. There was actually an early version in BLAST 2.2.23+ (and
possibly in older releases) but this did not have the -archive option
(instead -rid is a mandatory argument), and is not supported by this
wrapper.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Blast.Applications.NcbiblastnCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Blast.Applications.</code><code class="sig-name descname">NcbiblastnCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='blastn'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Blast.Applications.NcbiblastnCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Blast.Applications._NcbiblastMain2SeqCommandline</span></code></p>
<p>Wrapper for the NCBI BLAST+ program blastn (for nucleotides).</p>
<p>With the release of BLAST+ (BLAST rewritten in C++ instead of C), the NCBI
replaced the old blastall tool with separate tools for each of the searches.
This wrapper therefore replaces BlastallCommandline with option -p blastn.</p>
<p>For example, to run a search against the “nt” nucleotide database using the
FASTA nucleotide file “m_code.fasta” as the query, with an expectation value
cut off of 0.001, saving the output to a file in XML format:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Blast.Applications</span> <span class="k">import</span> <span class="n">NcbiblastnCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span> <span class="o">=</span> <span class="n">NcbiblastnCommandline</span><span class="p">(</span><span class="n">query</span><span class="o">=</span><span class="s2">&quot;m_cold.fasta&quot;</span><span class="p">,</span> <span class="n">db</span><span class="o">=</span><span class="s2">&quot;nt&quot;</span><span class="p">,</span> <span class="n">strand</span><span class="o">=</span><span class="s2">&quot;plus&quot;</span><span class="p">,</span>
<span class="gp">... </span>                              <span class="n">evalue</span><span class="o">=</span><span class="mf">0.001</span><span class="p">,</span> <span class="n">out</span><span class="o">=</span><span class="s2">&quot;m_cold.xml&quot;</span><span class="p">,</span> <span class="n">outfmt</span><span class="o">=</span><span class="mi">5</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span>
<span class="go">NcbiblastnCommandline(cmd=&#39;blastn&#39;, out=&#39;m_cold.xml&#39;, outfmt=5, query=&#39;m_cold.fasta&#39;, db=&#39;nt&#39;, evalue=0.001, strand=&#39;plus&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">cline</span><span class="p">)</span>
<span class="go">blastn -out m_cold.xml -outfmt 5 -query m_cold.fasta -db nt -evalue 0.001 -strand plus</span>
</pre></div>
</div>
<p>You would typically run the command line with cline() or via the Python
subprocess module, as described in the Biopython tutorial.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Blast.Applications.NcbiblastpCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Blast.Applications.</code><code class="sig-name descname">NcbiblastpCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='blastp'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Blast.Applications.NcbiblastpCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Blast.Applications._NcbiblastMain2SeqCommandline</span></code></p>
<p>Create a commandline for the NCBI BLAST+ program blastp (for proteins).</p>
<p>With the release of BLAST+ (BLAST rewritten in C++ instead of C), the NCBI
replaced the old blastall tool with separate tools for each of the searches.
This wrapper therefore replaces BlastallCommandline with option -p blastp.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Blast.Applications</span> <span class="k">import</span> <span class="n">NcbiblastpCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span> <span class="o">=</span> <span class="n">NcbiblastpCommandline</span><span class="p">(</span><span class="n">query</span><span class="o">=</span><span class="s2">&quot;rosemary.pro&quot;</span><span class="p">,</span> <span class="n">db</span><span class="o">=</span><span class="s2">&quot;nr&quot;</span><span class="p">,</span>
<span class="gp">... </span>                              <span class="n">evalue</span><span class="o">=</span><span class="mf">0.001</span><span class="p">,</span> <span class="n">remote</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">ungapped</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span>
<span class="go">NcbiblastpCommandline(cmd=&#39;blastp&#39;, query=&#39;rosemary.pro&#39;, db=&#39;nr&#39;, evalue=0.001, remote=True, ungapped=True)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">cline</span><span class="p">)</span>
<span class="go">blastp -query rosemary.pro -db nr -evalue 0.001 -remote -ungapped</span>
</pre></div>
</div>
<p>You would typically run the command line with cline() or via the Python
subprocess module, as described in the Biopython tutorial.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Blast.Applications.NcbiblastxCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Blast.Applications.</code><code class="sig-name descname">NcbiblastxCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='blastx'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Blast.Applications.NcbiblastxCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Blast.Applications._NcbiblastMain2SeqCommandline</span></code></p>
<p>Wrapper for the NCBI BLAST+ program blastx (nucleotide query, protein database).</p>
<p>With the release of BLAST+ (BLAST rewritten in C++ instead of C), the NCBI
replaced the old blastall tool with separate tools for each of the searches.
This wrapper therefore replaces BlastallCommandline with option -p blastx.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Blast.Applications</span> <span class="k">import</span> <span class="n">NcbiblastxCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span> <span class="o">=</span> <span class="n">NcbiblastxCommandline</span><span class="p">(</span><span class="n">query</span><span class="o">=</span><span class="s2">&quot;m_cold.fasta&quot;</span><span class="p">,</span> <span class="n">db</span><span class="o">=</span><span class="s2">&quot;nr&quot;</span><span class="p">,</span> <span class="n">evalue</span><span class="o">=</span><span class="mf">0.001</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span>
<span class="go">NcbiblastxCommandline(cmd=&#39;blastx&#39;, query=&#39;m_cold.fasta&#39;, db=&#39;nr&#39;, evalue=0.001)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">cline</span><span class="p">)</span>
<span class="go">blastx -query m_cold.fasta -db nr -evalue 0.001</span>
</pre></div>
</div>
<p>You would typically run the command line with cline() or via the Python
subprocess module, as described in the Biopython tutorial.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Blast.Applications.NcbideltablastCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Blast.Applications.</code><code class="sig-name descname">NcbideltablastCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='deltablast'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Blast.Applications.NcbideltablastCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Blast.Applications._Ncbiblast2SeqCommandline</span></code></p>
<p>Create a commandline for the NCBI BLAST+ program deltablast (for proteins).</p>
<p>This is a wrapper for the deltablast command line command included in
the NCBI BLAST+ software (not present in the original BLAST).</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Blast.Applications</span> <span class="k">import</span> <span class="n">NcbideltablastCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span> <span class="o">=</span> <span class="n">NcbideltablastCommandline</span><span class="p">(</span><span class="n">query</span><span class="o">=</span><span class="s2">&quot;rosemary.pro&quot;</span><span class="p">,</span> <span class="n">db</span><span class="o">=</span><span class="s2">&quot;nr&quot;</span><span class="p">,</span>
<span class="gp">... </span>                              <span class="n">evalue</span><span class="o">=</span><span class="mf">0.001</span><span class="p">,</span> <span class="n">remote</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span>
<span class="go">NcbideltablastCommandline(cmd=&#39;deltablast&#39;, query=&#39;rosemary.pro&#39;, db=&#39;nr&#39;, evalue=0.001, remote=True)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">cline</span><span class="p">)</span>
<span class="go">deltablast -query rosemary.pro -db nr -evalue 0.001 -remote</span>
</pre></div>
</div>
<p>You would typically run the command line with cline() or via the Python
subprocess module, as described in the Biopython tutorial.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Blast.Applications.NcbimakeblastdbCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Blast.Applications.</code><code class="sig-name descname">NcbimakeblastdbCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='makeblastdb'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Blast.Applications.NcbimakeblastdbCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Wrapper for the NCBI BLAST+ program makeblastdb.</p>
<p>This is a wrapper for the NCBI BLAST+ makeblastdb application
to create BLAST databases. By default, this creates a blast database
with the same name as the input file. The default output location
is the same directory as the input.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Blast.Applications</span> <span class="k">import</span> <span class="n">NcbimakeblastdbCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span> <span class="o">=</span> <span class="n">NcbimakeblastdbCommandline</span><span class="p">(</span><span class="n">dbtype</span><span class="o">=</span><span class="s2">&quot;prot&quot;</span><span class="p">,</span>
<span class="gp">... </span>                                   <span class="n">input_file</span><span class="o">=</span><span class="s2">&quot;NC_005816.faa&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span>
<span class="go">NcbimakeblastdbCommandline(cmd=&#39;makeblastdb&#39;, dbtype=&#39;prot&#39;, input_file=&#39;NC_005816.faa&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">cline</span><span class="p">)</span>
<span class="go">makeblastdb -dbtype prot -in NC_005816.faa</span>
</pre></div>
</div>
<p>You would typically run the command line with cline() or via the Python
subprocess module, as described in the Biopython tutorial.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Blast.Applications.NcbipsiblastCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Blast.Applications.</code><code class="sig-name descname">NcbipsiblastCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='psiblast'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Blast.Applications.NcbipsiblastCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Blast.Applications._Ncbiblast2SeqCommandline</span></code></p>
<p>Wrapper for the NCBI BLAST+ program psiblast.</p>
<p>With the release of BLAST+ (BLAST rewritten in C++ instead of C), the NCBI
replaced the old blastpgp tool with a similar tool psiblast. This wrapper
therefore replaces BlastpgpCommandline, the wrapper for blastpgp.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Blast.Applications</span> <span class="k">import</span> <span class="n">NcbipsiblastCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span> <span class="o">=</span> <span class="n">NcbipsiblastCommandline</span><span class="p">(</span><span class="n">help</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span>
<span class="go">NcbipsiblastCommandline(cmd=&#39;psiblast&#39;, help=True)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">cline</span><span class="p">)</span>
<span class="go">psiblast -help</span>
</pre></div>
</div>
<p>You would typically run the command line with cline() or via the Python
subprocess module, as described in the Biopython tutorial.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Blast.Applications.NcbirpsblastCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Blast.Applications.</code><code class="sig-name descname">NcbirpsblastCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='rpsblast'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Blast.Applications.NcbirpsblastCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Blast.Applications._NcbiblastCommandline</span></code></p>
<p>Wrapper for the NCBI BLAST+ program rpsblast.</p>
<p>With the release of BLAST+ (BLAST rewritten in C++ instead of C), the NCBI
replaced the old rpsblast tool with a similar tool of the same name. This
wrapper replaces RpsBlastCommandline, the wrapper for the old rpsblast.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Blast.Applications</span> <span class="k">import</span> <span class="n">NcbirpsblastCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span> <span class="o">=</span> <span class="n">NcbirpsblastCommandline</span><span class="p">(</span><span class="n">help</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span>
<span class="go">NcbirpsblastCommandline(cmd=&#39;rpsblast&#39;, help=True)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">cline</span><span class="p">)</span>
<span class="go">rpsblast -help</span>
</pre></div>
</div>
<p>You would typically run the command line with cline() or via the Python
subprocess module, as described in the Biopython tutorial.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Blast.Applications.NcbirpstblastnCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Blast.Applications.</code><code class="sig-name descname">NcbirpstblastnCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='rpstblastn'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Blast.Applications.NcbirpstblastnCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Blast.Applications._NcbiblastCommandline</span></code></p>
<p>Wrapper for the NCBI BLAST+ program rpstblastn.</p>
<p>With the release of BLAST+ (BLAST rewritten in C++ instead of C), the NCBI
replaced the old rpsblast tool with a similar tool of the same name, and a
separate tool rpstblastn for Translated Reverse Position Specific BLAST.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Blast.Applications</span> <span class="k">import</span> <span class="n">NcbirpstblastnCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span> <span class="o">=</span> <span class="n">NcbirpstblastnCommandline</span><span class="p">(</span><span class="n">help</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span>
<span class="go">NcbirpstblastnCommandline(cmd=&#39;rpstblastn&#39;, help=True)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">cline</span><span class="p">)</span>
<span class="go">rpstblastn -help</span>
</pre></div>
</div>
<p>You would typically run the command line with cline() or via the Python
subprocess module, as described in the Biopython tutorial.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Blast.Applications.NcbitblastnCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Blast.Applications.</code><code class="sig-name descname">NcbitblastnCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='tblastn'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Blast.Applications.NcbitblastnCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Blast.Applications._NcbiblastMain2SeqCommandline</span></code></p>
<p>Wrapper for the NCBI BLAST+ program tblastn.</p>
<p>With the release of BLAST+ (BLAST rewritten in C++ instead of C), the NCBI
replaced the old blastall tool with separate tools for each of the searches.
This wrapper therefore replaces BlastallCommandline with option -p tblastn.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Blast.Applications</span> <span class="k">import</span> <span class="n">NcbitblastnCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span> <span class="o">=</span> <span class="n">NcbitblastnCommandline</span><span class="p">(</span><span class="n">help</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span>
<span class="go">NcbitblastnCommandline(cmd=&#39;tblastn&#39;, help=True)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">cline</span><span class="p">)</span>
<span class="go">tblastn -help</span>
</pre></div>
</div>
<p>You would typically run the command line with cline() or via the Python
subprocess module, as described in the Biopython tutorial.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Blast.Applications.NcbitblastxCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Blast.Applications.</code><code class="sig-name descname">NcbitblastxCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='tblastx'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Blast.Applications.NcbitblastxCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Blast.Applications._NcbiblastMain2SeqCommandline</span></code></p>
<p>Wrapper for the NCBI BLAST+ program tblastx.</p>
<p>With the release of BLAST+ (BLAST rewritten in C++ instead of C), the NCBI
replaced the old blastall tool with separate tools for each of the searches.
This wrapper therefore replaces BlastallCommandline with option -p tblastx.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Blast.Applications</span> <span class="k">import</span> <span class="n">NcbitblastxCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span> <span class="o">=</span> <span class="n">NcbitblastxCommandline</span><span class="p">(</span><span class="n">help</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span>
<span class="go">NcbitblastxCommandline(cmd=&#39;tblastx&#39;, help=True)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">cline</span><span class="p">)</span>
<span class="go">tblastx -help</span>
</pre></div>
</div>
<p>You would typically run the command line with cline() or via the Python
subprocess module, as described in the Biopython tutorial.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="Bio.Blast.NCBIWWW.html" class="btn btn-neutral float-right" title="Bio.Blast.NCBIWWW module" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="Bio.Blast.html" class="btn btn-neutral float-left" title="Bio.Blast package" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 1999-2017, The Biopython Contributors

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>