

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Bio.Align.Applications package &mdash; Biopython 1.74 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Bio.Align.AlignInfo module" href="Bio.Align.AlignInfo.html" />
    <link rel="prev" title="Bio.Align package" href="Bio.Align.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search"  style="background: #10100F" >
          

          
            <a href="index.html" class="icon icon-home"> Biopython
          

          
            
            <img src="_static/biopython_logo.svg" class="logo" alt="Logo"/>
          
          </a>

          
            
            
              <div class="version">
                1.74
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">API Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="Bio.html">Bio package</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="Bio.html#subpackages">Subpackages</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="Bio.Affy.html">Bio.Affy package</a></li>
<li class="toctree-l3 current"><a class="reference internal" href="Bio.Align.html">Bio.Align package</a><ul class="current">
<li class="toctree-l4 current"><a class="reference internal" href="Bio.Align.html#subpackages">Subpackages</a></li>
<li class="toctree-l4"><a class="reference internal" href="Bio.Align.html#submodules">Submodules</a></li>
<li class="toctree-l4"><a class="reference internal" href="Bio.Align.html#module-Bio.Align">Module contents</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="Bio.AlignIO.html">Bio.AlignIO package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Alphabet.html">Bio.Alphabet package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Application.html">Bio.Application package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Blast.html">Bio.Blast package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.CAPS.html">Bio.CAPS package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Cluster.html">Bio.Cluster package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Compass.html">Bio.Compass package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Crystal.html">Bio.Crystal package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Data.html">Bio.Data package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Emboss.html">Bio.Emboss package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Entrez.html">Bio.Entrez package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.ExPASy.html">Bio.ExPASy package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.FSSP.html">Bio.FSSP package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.GenBank.html">Bio.GenBank package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Geo.html">Bio.Geo package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Graphics.html">Bio.Graphics package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.HMM.html">Bio.HMM package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.KDTree.html">Bio.KDTree package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.KEGG.html">Bio.KEGG package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Medline.html">Bio.Medline package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.NMR.html">Bio.NMR package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Nexus.html">Bio.Nexus package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.PDB.html">Bio.PDB package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Pathway.html">Bio.Pathway package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Phylo.html">Bio.Phylo package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.PopGen.html">Bio.PopGen package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Restriction.html">Bio.Restriction package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SCOP.html">Bio.SCOP package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SVDSuperimposer.html">Bio.SVDSuperimposer package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SearchIO.html">Bio.SearchIO package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SeqIO.html">Bio.SeqIO package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SeqUtils.html">Bio.SeqUtils package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Sequencing.html">Bio.Sequencing package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Statistics.html">Bio.Statistics package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SubsMat.html">Bio.SubsMat package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SwissProt.html">Bio.SwissProt package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.TogoWS.html">Bio.TogoWS package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.UniGene.html">Bio.UniGene package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.UniProt.html">Bio.UniProt package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Wise.html">Bio.Wise package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.codonalign.html">Bio.codonalign package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.motifs.html">Bio.motifs package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.phenotype.html">Bio.phenotype package</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="Bio.html#submodules">Submodules</a></li>
<li class="toctree-l2"><a class="reference internal" href="Bio.html#module-Bio">Module contents</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="BioSQL.html">BioSQL package</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Biopython</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
          <li><a href="Bio.html">Bio package</a> &raquo;</li>
        
          <li><a href="Bio.Align.html">Bio.Align package</a> &raquo;</li>
        
      <li>Bio.Align.Applications package</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
        
      </li>
    
  </ul>

  
  <div class="rst-breadcrumbs-buttons" role="navigation" aria-label="breadcrumb navigation">
      
        <a href="Bio.Align.AlignInfo.html" class="btn btn-neutral float-right" title="Bio.Align.AlignInfo module" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="Bio.Align.html" class="btn btn-neutral float-left" title="Bio.Align package" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
  </div>
  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="bio-align-applications-package">
<h1>Bio.Align.Applications package<a class="headerlink" href="#bio-align-applications-package" title="Permalink to this headline">¶</a></h1>
<div class="section" id="module-Bio.Align.Applications">
<span id="module-contents"></span><h2>Module contents<a class="headerlink" href="#module-Bio.Align.Applications" title="Permalink to this headline">¶</a></h2>
<p>Alignment command line tool wrappers.</p>
<dl class="class">
<dt id="Bio.Align.Applications.MuscleCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Align.Applications.</code><code class="sig-name descname">MuscleCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='muscle'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Align.Applications.MuscleCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for the multiple alignment program MUSCLE.</p>
<p><a class="reference external" href="http://www.drive5.com/muscle/">http://www.drive5.com/muscle/</a></p>
<p class="rubric">Notes</p>
<p>Last checked against version: 3.7, briefly against 3.8</p>
<p class="rubric">References</p>
<p>Edgar, Robert C. (2004), MUSCLE: multiple sequence alignment with high
accuracy and high throughput, Nucleic Acids Research 32(5), 1792-97.</p>
<p>Edgar, R.C. (2004) MUSCLE: a multiple sequence alignment method with
reduced time and space complexity. BMC Bioinformatics 5(1): 113.</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Align.Applications</span> <span class="k">import</span> <span class="n">MuscleCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">muscle_exe</span> <span class="o">=</span> <span class="sa">r</span><span class="s2">&quot;C:\Program Files\Aligments\muscle3.8.31_i86win32.exe&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">in_file</span> <span class="o">=</span> <span class="sa">r</span><span class="s2">&quot;C:\My Documents\unaligned.fasta&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">out_file</span> <span class="o">=</span> <span class="sa">r</span><span class="s2">&quot;C:\My Documents\aligned.fasta&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">muscle_cline</span> <span class="o">=</span> <span class="n">MuscleCommandline</span><span class="p">(</span><span class="n">muscle_exe</span><span class="p">,</span> <span class="nb">input</span><span class="o">=</span><span class="n">in_file</span><span class="p">,</span> <span class="n">out</span><span class="o">=</span><span class="n">out_file</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">muscle_cline</span><span class="p">)</span>
<span class="go">&quot;C:\Program Files\Aligments\muscle3.8.31_i86win32.exe&quot; -in &quot;C:\My Documents\unaligned.fasta&quot; -out &quot;C:\My Documents\aligned.fasta&quot;</span>
</pre></div>
</div>
<p>You would typically run the command line with muscle_cline() or via
the Python subprocess module, as described in the Biopython tutorial.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Align.Applications.ClustalwCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Align.Applications.</code><code class="sig-name descname">ClustalwCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='clustalw'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Align.Applications.ClustalwCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for clustalw (version one or two).</p>
<p><a class="reference external" href="http://www.clustal.org/">http://www.clustal.org/</a></p>
<p class="rubric">Notes</p>
<p>Last checked against versions: 1.83 and 2.1</p>
<p class="rubric">References</p>
<p>Larkin MA, Blackshields G, Brown NP, Chenna R, McGettigan PA,
McWilliam H, Valentin F, Wallace IM, Wilm A, Lopez R, Thompson JD,
Gibson TJ, Higgins DG. (2007). Clustal W and Clustal X version 2.0.
Bioinformatics, 23, 2947-2948.</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Align.Applications</span> <span class="k">import</span> <span class="n">ClustalwCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">in_file</span> <span class="o">=</span> <span class="s2">&quot;unaligned.fasta&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">clustalw_cline</span> <span class="o">=</span> <span class="n">ClustalwCommandline</span><span class="p">(</span><span class="s2">&quot;clustalw2&quot;</span><span class="p">,</span> <span class="n">infile</span><span class="o">=</span><span class="n">in_file</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">clustalw_cline</span><span class="p">)</span>
<span class="go">clustalw2 -infile=unaligned.fasta</span>
</pre></div>
</div>
<p>You would typically run the command line with clustalw_cline() or via
the Python subprocess module, as described in the Biopython tutorial.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Align.Applications.ClustalOmegaCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Align.Applications.</code><code class="sig-name descname">ClustalOmegaCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='clustalo'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Align.Applications.ClustalOmegaCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for clustal omega.</p>
<p><a class="reference external" href="http://www.clustal.org/omega">http://www.clustal.org/omega</a></p>
<p class="rubric">Notes</p>
<p>Last checked against version: 1.2.0</p>
<p class="rubric">References</p>
<p>Sievers F, Wilm A, Dineen DG, Gibson TJ, Karplus K, Li W, Lopez R,
McWilliam H, Remmert M, Söding J, Thompson JD, Higgins DG (2011).
Fast, scalable generation of high-quality protein multiple
sequence alignments using Clustal Omega.
Molecular Systems Biology 7:539 <a class="reference external" href="https://doi.org/10.1038/msb.2011.75">https://doi.org/10.1038/msb.2011.75</a></p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Align.Applications</span> <span class="k">import</span> <span class="n">ClustalOmegaCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">in_file</span> <span class="o">=</span> <span class="s2">&quot;unaligned.fasta&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">out_file</span> <span class="o">=</span> <span class="s2">&quot;aligned.fasta&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">clustalomega_cline</span> <span class="o">=</span> <span class="n">ClustalOmegaCommandline</span><span class="p">(</span><span class="n">infile</span><span class="o">=</span><span class="n">in_file</span><span class="p">,</span> <span class="n">outfile</span><span class="o">=</span><span class="n">out_file</span><span class="p">,</span> <span class="n">verbose</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">auto</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">clustalomega_cline</span><span class="p">)</span>
<span class="go">clustalo -i unaligned.fasta -o aligned.fasta --auto -v</span>
</pre></div>
</div>
<p>You would typically run the command line with clustalomega_cline() or via
the Python subprocess module, as described in the Biopython tutorial.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Align.Applications.PrankCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Align.Applications.</code><code class="sig-name descname">PrankCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='prank'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Align.Applications.PrankCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for the multiple alignment program PRANK.</p>
<p><a class="reference external" href="http://www.ebi.ac.uk/goldman-srv/prank/prank/">http://www.ebi.ac.uk/goldman-srv/prank/prank/</a></p>
<p class="rubric">Notes</p>
<p>Last checked against version: 081202</p>
<p class="rubric">References</p>
<p>Loytynoja, A. and Goldman, N. 2005. An algorithm for progressive
multiple alignment of sequences with insertions. Proceedings of
the National Academy of Sciences, 102: 10557–10562.</p>
<p>Loytynoja, A. and Goldman, N. 2008. Phylogeny-aware gap placement
prevents errors in sequence alignment and evolutionary analysis.
Science, 320: 1632.</p>
<p class="rubric">Examples</p>
<p>To align a FASTA file (unaligned.fasta) with the output in aligned
FASTA format with the output filename starting with “aligned” (you
can’t pick the filename explicitly), no tree output and no XML output,
use:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Align.Applications</span> <span class="k">import</span> <span class="n">PrankCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">prank_cline</span> <span class="o">=</span> <span class="n">PrankCommandline</span><span class="p">(</span><span class="n">d</span><span class="o">=</span><span class="s2">&quot;unaligned.fasta&quot;</span><span class="p">,</span>
<span class="gp">... </span>                               <span class="n">o</span><span class="o">=</span><span class="s2">&quot;aligned&quot;</span><span class="p">,</span> <span class="c1"># prefix only!</span>
<span class="gp">... </span>                               <span class="n">f</span><span class="o">=</span><span class="mi">8</span><span class="p">,</span> <span class="c1"># FASTA output</span>
<span class="gp">... </span>                               <span class="n">notree</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">noxml</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">prank_cline</span><span class="p">)</span>
<span class="go">prank -d=unaligned.fasta -o=aligned -f=8 -noxml -notree</span>
</pre></div>
</div>
<p>You would typically run the command line with prank_cline() or via
the Python subprocess module, as described in the Biopython tutorial.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Align.Applications.MafftCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Align.Applications.</code><code class="sig-name descname">MafftCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='mafft'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Align.Applications.MafftCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for the multiple alignment program MAFFT.</p>
<p><a class="reference external" href="http://align.bmr.kyushu-u.ac.jp/mafft/software/">http://align.bmr.kyushu-u.ac.jp/mafft/software/</a></p>
<p class="rubric">Notes</p>
<p>Last checked against version: MAFFT v6.717b (2009/12/03)</p>
<p class="rubric">References</p>
<p>Katoh, Toh (BMC Bioinformatics 9:212, 2008) Improved accuracy of
multiple ncRNA alignment by incorporating structural information into
a MAFFT-based framework (describes RNA structural alignment methods)</p>
<p>Katoh, Toh (Briefings in Bioinformatics 9:286-298, 2008) Recent
developments in the MAFFT multiple sequence alignment program
(outlines version 6)</p>
<p>Katoh, Toh (Bioinformatics 23:372-374, 2007)  Errata PartTree: an
algorithm to build an approximate tree from a large number of
unaligned sequences (describes the PartTree algorithm)</p>
<p>Katoh, Kuma, Toh, Miyata (Nucleic Acids Res. 33:511-518, 2005) MAFFT
version 5: improvement in accuracy of multiple sequence alignment
(describes [ancestral versions of] the G-INS-i, L-INS-i and E-INS-i
strategies)</p>
<p>Katoh, Misawa, Kuma, Miyata (Nucleic Acids Res. 30:3059-3066, 2002)</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Align.Applications</span> <span class="k">import</span> <span class="n">MafftCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mafft_exe</span> <span class="o">=</span> <span class="s2">&quot;/opt/local/mafft&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">in_file</span> <span class="o">=</span> <span class="s2">&quot;../Doc/examples/opuntia.fasta&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mafft_cline</span> <span class="o">=</span> <span class="n">MafftCommandline</span><span class="p">(</span><span class="n">mafft_exe</span><span class="p">,</span> <span class="nb">input</span><span class="o">=</span><span class="n">in_file</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">mafft_cline</span><span class="p">)</span>
<span class="go">/opt/local/mafft ../Doc/examples/opuntia.fasta</span>
</pre></div>
</div>
<p>If the mafft binary is on the path (typically the case on a Unix style
operating system) then you don’t need to supply the executable location:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Align.Applications</span> <span class="k">import</span> <span class="n">MafftCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">in_file</span> <span class="o">=</span> <span class="s2">&quot;../Doc/examples/opuntia.fasta&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mafft_cline</span> <span class="o">=</span> <span class="n">MafftCommandline</span><span class="p">(</span><span class="nb">input</span><span class="o">=</span><span class="n">in_file</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">mafft_cline</span><span class="p">)</span>
<span class="go">mafft ../Doc/examples/opuntia.fasta</span>
</pre></div>
</div>
<p>You would typically run the command line with mafft_cline() or via
the Python subprocess module, as described in the Biopython tutorial.</p>
<p>Note that MAFFT will write the alignment to stdout, which you may
want to save to a file and then parse, e.g.:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">stdout</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">mafft_cline</span><span class="p">()</span>
<span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s2">&quot;aligned.fasta&quot;</span><span class="p">,</span> <span class="s2">&quot;w&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">handle</span><span class="p">:</span>
    <span class="n">handle</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">stdout</span><span class="p">)</span>
<span class="kn">from</span> <span class="nn">Bio</span> <span class="k">import</span> <span class="n">AlignIO</span>
<span class="n">align</span> <span class="o">=</span> <span class="n">AlignIO</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="s2">&quot;aligned.fasta&quot;</span><span class="p">,</span> <span class="s2">&quot;fasta&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p>Alternatively, to parse the output with AlignIO directly you can
use StringIO to turn the string into a handle:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">stdout</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">mafft_cline</span><span class="p">()</span>
<span class="kn">from</span> <span class="nn">StringIO</span> <span class="k">import</span> <span class="n">StringIO</span>
<span class="kn">from</span> <span class="nn">Bio</span> <span class="k">import</span> <span class="n">AlignIO</span>
<span class="n">align</span> <span class="o">=</span> <span class="n">AlignIO</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="n">StringIO</span><span class="p">(</span><span class="n">stdout</span><span class="p">),</span> <span class="s2">&quot;fasta&quot;</span><span class="p">)</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Align.Applications.DialignCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Align.Applications.</code><code class="sig-name descname">DialignCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='dialign2-2'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Align.Applications.DialignCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for the multiple alignment program DIALIGN2-2.</p>
<p><a class="reference external" href="http://bibiserv.techfak.uni-bielefeld.de/dialign/welcome.html">http://bibiserv.techfak.uni-bielefeld.de/dialign/welcome.html</a></p>
<p class="rubric">Notes</p>
<p>Last checked against version: 2.2</p>
<p class="rubric">References</p>
<p>B. Morgenstern (2004). DIALIGN: Multiple DNA and Protein Sequence
Alignment at BiBiServ. Nucleic Acids Research 32, W33-W36.</p>
<p class="rubric">Examples</p>
<p>To align a FASTA file (unaligned.fasta) with the output files names
aligned.* including a FASTA output file (aligned.fa), use:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Align.Applications</span> <span class="k">import</span> <span class="n">DialignCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">dialign_cline</span> <span class="o">=</span> <span class="n">DialignCommandline</span><span class="p">(</span><span class="nb">input</span><span class="o">=</span><span class="s2">&quot;unaligned.fasta&quot;</span><span class="p">,</span>
<span class="gp">... </span>                                   <span class="n">fn</span><span class="o">=</span><span class="s2">&quot;aligned&quot;</span><span class="p">,</span> <span class="n">fa</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">dialign_cline</span><span class="p">)</span>
<span class="go">dialign2-2 -fa -fn aligned unaligned.fasta</span>
</pre></div>
</div>
<p>You would typically run the command line with dialign_cline() or via
the Python subprocess module, as described in the Biopython tutorial.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Align.Applications.ProbconsCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Align.Applications.</code><code class="sig-name descname">ProbconsCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='probcons'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Align.Applications.ProbconsCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for the multiple alignment program PROBCONS.</p>
<p><a class="reference external" href="http://probcons.stanford.edu/">http://probcons.stanford.edu/</a></p>
<p class="rubric">Notes</p>
<p>Last checked against version: 1.12</p>
<p class="rubric">References</p>
<p>Do, C.B., Mahabhashyam, M.S.P., Brudno, M., and Batzoglou, S. 2005.
PROBCONS: Probabilistic Consistency-based Multiple Sequence Alignment.
Genome Research 15: 330-340.</p>
<p class="rubric">Examples</p>
<p>To align a FASTA file (unaligned.fasta) with the output in ClustalW
format, and otherwise default settings, use:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Align.Applications</span> <span class="k">import</span> <span class="n">ProbconsCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">probcons_cline</span> <span class="o">=</span> <span class="n">ProbconsCommandline</span><span class="p">(</span><span class="nb">input</span><span class="o">=</span><span class="s2">&quot;unaligned.fasta&quot;</span><span class="p">,</span>
<span class="gp">... </span>                                     <span class="n">clustalw</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">probcons_cline</span><span class="p">)</span>
<span class="go">probcons -clustalw unaligned.fasta</span>
</pre></div>
</div>
<p>You would typically run the command line with probcons_cline() or via
the Python subprocess module, as described in the Biopython tutorial.</p>
<p>Note that PROBCONS will write the alignment to stdout, which you may
want to save to a file and then parse, e.g.:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">stdout</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">probcons_cline</span><span class="p">()</span>
<span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s2">&quot;aligned.aln&quot;</span><span class="p">,</span> <span class="s2">&quot;w&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">handle</span><span class="p">:</span>
    <span class="n">handle</span><span class="o">.</span><span class="n">write</span><span class="p">(</span><span class="n">stdout</span><span class="p">)</span>
<span class="kn">from</span> <span class="nn">Bio</span> <span class="k">import</span> <span class="n">AlignIO</span>
<span class="n">align</span> <span class="o">=</span> <span class="n">AlignIO</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="s2">&quot;aligned.fasta&quot;</span><span class="p">,</span> <span class="s2">&quot;clustalw&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p>Alternatively, to parse the output with AlignIO directly you can
use StringIO to turn the string into a handle:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">stdout</span><span class="p">,</span> <span class="n">stderr</span> <span class="o">=</span> <span class="n">probcons_cline</span><span class="p">()</span>
<span class="kn">from</span> <span class="nn">StringIO</span> <span class="k">import</span> <span class="n">StringIO</span>
<span class="kn">from</span> <span class="nn">Bio</span> <span class="k">import</span> <span class="n">AlignIO</span>
<span class="n">align</span> <span class="o">=</span> <span class="n">AlignIO</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="n">StringIO</span><span class="p">(</span><span class="n">stdout</span><span class="p">),</span> <span class="s2">&quot;clustalw&quot;</span><span class="p">)</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Align.Applications.TCoffeeCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Align.Applications.</code><code class="sig-name descname">TCoffeeCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='t_coffee'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Align.Applications.TCoffeeCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Commandline object for the TCoffee alignment program.</p>
<p><a class="reference external" href="http://www.tcoffee.org/Projects_home_page/t_coffee_home_page.html">http://www.tcoffee.org/Projects_home_page/t_coffee_home_page.html</a></p>
<p>The T-Coffee command line tool has a lot of switches and options.
This wrapper implements a VERY limited number of options - if you
would like to help improve it please get in touch.</p>
<p class="rubric">Notes</p>
<p>Last checked against: Version_6.92</p>
<p class="rubric">References</p>
<p>T-Coffee: A novel method for multiple sequence alignments.
Notredame, Higgins, Heringa, JMB,302(205-217) 2000</p>
<p class="rubric">Examples</p>
<p>To align a FASTA file (unaligned.fasta) with the output in ClustalW
format (file aligned.aln), and otherwise default settings, use:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Align.Applications</span> <span class="k">import</span> <span class="n">TCoffeeCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">tcoffee_cline</span> <span class="o">=</span> <span class="n">TCoffeeCommandline</span><span class="p">(</span><span class="n">infile</span><span class="o">=</span><span class="s2">&quot;unaligned.fasta&quot;</span><span class="p">,</span>
<span class="gp">... </span>                                   <span class="n">output</span><span class="o">=</span><span class="s2">&quot;clustalw&quot;</span><span class="p">,</span>
<span class="gp">... </span>                                   <span class="n">outfile</span><span class="o">=</span><span class="s2">&quot;aligned.aln&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">tcoffee_cline</span><span class="p">)</span>
<span class="go">t_coffee -output clustalw -infile unaligned.fasta -outfile aligned.aln</span>
</pre></div>
</div>
<p>You would typically run the command line with tcoffee_cline() or via
the Python subprocess module, as described in the Biopython tutorial.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="Bio.Align.Applications.TCoffeeCommandline.SEQ_TYPES">
<code class="sig-name descname">SEQ_TYPES</code><em class="property"> = ['dna', 'protein', 'dna_protein']</em><a class="headerlink" href="#Bio.Align.Applications.TCoffeeCommandline.SEQ_TYPES" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="class">
<dt id="Bio.Align.Applications.MSAProbsCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Align.Applications.</code><code class="sig-name descname">MSAProbsCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='msaprobs'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Align.Applications.MSAProbsCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for MSAProbs.</p>
<p><a class="reference external" href="http://msaprobs.sourceforge.net">http://msaprobs.sourceforge.net</a></p>
<p class="rubric">Notes</p>
<p>Last checked against version: 0.9.7</p>
<p class="rubric">References</p>
<p>Yongchao Liu, Bertil Schmidt, Douglas L. Maskell: “MSAProbs: multiple
sequence alignment based on pair hidden Markov models and partition
function posterior probabilities”. Bioinformatics, 2010, 26(16): 1958 -1964</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Align.Applications</span> <span class="k">import</span> <span class="n">MSAProbsCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">in_file</span> <span class="o">=</span> <span class="s2">&quot;unaligned.fasta&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">out_file</span> <span class="o">=</span> <span class="s2">&quot;aligned.cla&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span> <span class="o">=</span> <span class="n">MSAProbsCommandline</span><span class="p">(</span><span class="n">infile</span><span class="o">=</span><span class="n">in_file</span><span class="p">,</span> <span class="n">outfile</span><span class="o">=</span><span class="n">out_file</span><span class="p">,</span> <span class="n">clustalw</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">cline</span><span class="p">)</span>
<span class="go">msaprobs -o aligned.cla -clustalw unaligned.fasta</span>
</pre></div>
</div>
<p>You would typically run the command line with cline() or via
the Python subprocess module, as described in the Biopython tutorial.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="Bio.Align.AlignInfo.html" class="btn btn-neutral float-right" title="Bio.Align.AlignInfo module" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="Bio.Align.html" class="btn btn-neutral float-left" title="Bio.Align package" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 1999-2017, The Biopython Contributors

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>