

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Bio.SeqIO.PirIO module &mdash; Biopython 1.74 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Bio.SeqIO.QualityIO module" href="Bio.SeqIO.QualityIO.html" />
    <link rel="prev" title="Bio.SeqIO.PhdIO module" href="Bio.SeqIO.PhdIO.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search"  style="background: #10100F" >
          

          
            <a href="index.html" class="icon icon-home"> Biopython
          

          
            
            <img src="_static/biopython_logo.svg" class="logo" alt="Logo"/>
          
          </a>

          
            
            
              <div class="version">
                1.74
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">API Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="Bio.html">Bio package</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="Bio.html#subpackages">Subpackages</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="Bio.Affy.html">Bio.Affy package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Align.html">Bio.Align package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.AlignIO.html">Bio.AlignIO package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Alphabet.html">Bio.Alphabet package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Application.html">Bio.Application package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Blast.html">Bio.Blast package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.CAPS.html">Bio.CAPS package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Cluster.html">Bio.Cluster package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Compass.html">Bio.Compass package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Crystal.html">Bio.Crystal package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Data.html">Bio.Data package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Emboss.html">Bio.Emboss package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Entrez.html">Bio.Entrez package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.ExPASy.html">Bio.ExPASy package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.FSSP.html">Bio.FSSP package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.GenBank.html">Bio.GenBank package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Geo.html">Bio.Geo package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Graphics.html">Bio.Graphics package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.HMM.html">Bio.HMM package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.KDTree.html">Bio.KDTree package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.KEGG.html">Bio.KEGG package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Medline.html">Bio.Medline package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.NMR.html">Bio.NMR package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Nexus.html">Bio.Nexus package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.PDB.html">Bio.PDB package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Pathway.html">Bio.Pathway package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Phylo.html">Bio.Phylo package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.PopGen.html">Bio.PopGen package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Restriction.html">Bio.Restriction package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SCOP.html">Bio.SCOP package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SVDSuperimposer.html">Bio.SVDSuperimposer package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SearchIO.html">Bio.SearchIO package</a></li>
<li class="toctree-l3 current"><a class="reference internal" href="Bio.SeqIO.html">Bio.SeqIO package</a><ul class="current">
<li class="toctree-l4 current"><a class="reference internal" href="Bio.SeqIO.html#submodules">Submodules</a></li>
<li class="toctree-l4"><a class="reference internal" href="Bio.SeqIO.html#module-Bio.SeqIO">Module contents</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SeqUtils.html">Bio.SeqUtils package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Sequencing.html">Bio.Sequencing package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Statistics.html">Bio.Statistics package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SubsMat.html">Bio.SubsMat package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SwissProt.html">Bio.SwissProt package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.TogoWS.html">Bio.TogoWS package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.UniGene.html">Bio.UniGene package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.UniProt.html">Bio.UniProt package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Wise.html">Bio.Wise package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.codonalign.html">Bio.codonalign package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.motifs.html">Bio.motifs package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.phenotype.html">Bio.phenotype package</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="Bio.html#submodules">Submodules</a></li>
<li class="toctree-l2"><a class="reference internal" href="Bio.html#module-Bio">Module contents</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="BioSQL.html">BioSQL package</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Biopython</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
          <li><a href="Bio.html">Bio package</a> &raquo;</li>
        
          <li><a href="Bio.SeqIO.html">Bio.SeqIO package</a> &raquo;</li>
        
      <li>Bio.SeqIO.PirIO module</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
        
      </li>
    
  </ul>

  
  <div class="rst-breadcrumbs-buttons" role="navigation" aria-label="breadcrumb navigation">
      
        <a href="Bio.SeqIO.QualityIO.html" class="btn btn-neutral float-right" title="Bio.SeqIO.QualityIO module" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="Bio.SeqIO.PhdIO.html" class="btn btn-neutral float-left" title="Bio.SeqIO.PhdIO module" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
  </div>
  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="module-Bio.SeqIO.PirIO">
<span id="bio-seqio-pirio-module"></span><h1>Bio.SeqIO.PirIO module<a class="headerlink" href="#module-Bio.SeqIO.PirIO" title="Permalink to this headline">¶</a></h1>
<p>Bio.SeqIO support for the “pir” (aka PIR or NBRF) file format.</p>
<p>This module is for reading and writing PIR or NBRF format files as
SeqRecord objects.</p>
<p>You are expected to use this module via the Bio.SeqIO functions, or if
the file contains a sequence alignment, optionally via Bio.AlignIO instead.</p>
<p>This format was introduced for the Protein Information Resource (PIR), a
project of the National Biomedical Research Foundation (NBRF).  The PIR
database itself is now part of UniProt.</p>
<p>The file format is described online at:
<a class="reference external" href="http://www.ebi.ac.uk/help/pir_frame.html">http://www.ebi.ac.uk/help/pir_frame.html</a>
<a class="reference external" href="http://www.cmbi.kun.nl/bioinf/tools/crab_pir.html">http://www.cmbi.kun.nl/bioinf/tools/crab_pir.html</a> (currently down)</p>
<p>An example file in this format would be:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span><span class="n">P1</span><span class="p">;</span><span class="n">CRAB_ANAPL</span>
<span class="n">ALPHA</span> <span class="n">CRYSTALLIN</span> <span class="n">B</span> <span class="n">CHAIN</span> <span class="p">(</span><span class="n">ALPHA</span><span class="p">(</span><span class="n">B</span><span class="p">)</span><span class="o">-</span><span class="n">CRYSTALLIN</span><span class="p">)</span><span class="o">.</span>
  <span class="n">MDITIHNPLI</span> <span class="n">RRPLFSWLAP</span> <span class="n">SRIFDQIFGE</span> <span class="n">HLQESELLPA</span> <span class="n">SPSLSPFLMR</span>
  <span class="n">SPIFRMPSWL</span> <span class="n">ETGLSEMRLE</span> <span class="n">KDKFSVNLDV</span> <span class="n">KHFSPEELKV</span> <span class="n">KVLGDMVEIH</span>
  <span class="n">GKHEERQDEH</span> <span class="n">GFIAREFNRK</span> <span class="n">YRIPADVDPL</span> <span class="n">TITSSLSLDG</span> <span class="n">VLTVSAPRKQ</span>
  <span class="n">SDVPERSIPI</span> <span class="n">TREEKPAIAG</span> <span class="n">AQRK</span><span class="o">*</span>

<span class="o">&gt;</span><span class="n">P1</span><span class="p">;</span><span class="n">CRAB_BOVIN</span>
<span class="n">ALPHA</span> <span class="n">CRYSTALLIN</span> <span class="n">B</span> <span class="n">CHAIN</span> <span class="p">(</span><span class="n">ALPHA</span><span class="p">(</span><span class="n">B</span><span class="p">)</span><span class="o">-</span><span class="n">CRYSTALLIN</span><span class="p">)</span><span class="o">.</span>
  <span class="n">MDIAIHHPWI</span> <span class="n">RRPFFPFHSP</span> <span class="n">SRLFDQFFGE</span> <span class="n">HLLESDLFPA</span> <span class="n">STSLSPFYLR</span>
  <span class="n">PPSFLRAPSW</span> <span class="n">IDTGLSEMRL</span> <span class="n">EKDRFSVNLD</span> <span class="n">VKHFSPEELK</span> <span class="n">VKVLGDVIEV</span>
  <span class="n">HGKHEERQDE</span> <span class="n">HGFISREFHR</span> <span class="n">KYRIPADVDP</span> <span class="n">LAITSSLSSD</span> <span class="n">GVLTVNGPRK</span>
  <span class="n">QASGPERTIP</span> <span class="n">ITREEKPAVT</span> <span class="n">AAPKK</span><span class="o">*</span>
</pre></div>
</div>
<p>Or, an example of a multiple sequence alignment:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="o">&gt;</span><span class="n">P1</span><span class="p">;</span><span class="n">S27231</span>
<span class="n">rhodopsin</span> <span class="o">-</span> <span class="n">northern</span> <span class="n">leopard</span> <span class="n">frog</span>
<span class="n">MNGTEGPNFY</span> <span class="n">IPMSNKTGVV</span> <span class="n">RSPFDYPQYY</span> <span class="n">LAEPWKYSVL</span> <span class="n">AAYMFLLILL</span> <span class="n">GLPINFMTLY</span>
<span class="n">VTIQHKKLRT</span> <span class="n">PLNYILLNLG</span> <span class="n">VCNHFMVLCG</span> <span class="n">FTITMYTSLH</span> <span class="n">GYFVFGQTGC</span> <span class="n">YFEGFFATLG</span>
<span class="n">GEIALWSLVV</span> <span class="n">LAIERYIVVC</span> <span class="n">KPMSNFRFGE</span> <span class="n">NHAMMGVAFT</span> <span class="n">WIMALACAVP</span> <span class="n">PLFGWSRYIP</span>
<span class="n">EGMQCSCGVD</span> <span class="n">YYTLKPEVNN</span> <span class="n">ESFVIYMFVV</span> <span class="n">HFLIPLIIIS</span> <span class="n">FCYGRLVCTV</span> <span class="n">KEAAAQQQES</span>
<span class="n">ATTQKAEKEV</span> <span class="n">TRMVIIMVIF</span> <span class="n">FLICWVPYAY</span> <span class="n">VAFYIFTHQG</span> <span class="n">SEFGPIFMTV</span> <span class="n">PAFFAKSSAI</span>
<span class="n">YNPVIYIMLN</span> <span class="n">KQFRNCMITT</span> <span class="n">LCCGKNPFGD</span> <span class="n">DDASSAATSK</span> <span class="n">TEATSVSTSQ</span> <span class="n">VSPA</span><span class="o">*</span>

<span class="o">&gt;</span><span class="n">P1</span><span class="p">;</span><span class="n">I51200</span>
<span class="n">rhodopsin</span> <span class="o">-</span> <span class="n">African</span> <span class="n">clawed</span> <span class="n">frog</span>
<span class="n">MNGTEGPNFY</span> <span class="n">VPMSNKTGVV</span> <span class="n">RSPFDYPQYY</span> <span class="n">LAEPWQYSAL</span> <span class="n">AAYMFLLILL</span> <span class="n">GLPINFMTLF</span>
<span class="n">VTIQHKKLRT</span> <span class="n">PLNYILLNLV</span> <span class="n">FANHFMVLCG</span> <span class="n">FTVTMYTSMH</span> <span class="n">GYFIFGPTGC</span> <span class="n">YIEGFFATLG</span>
<span class="n">GEVALWSLVV</span> <span class="n">LAVERYIVVC</span> <span class="n">KPMANFRFGE</span> <span class="n">NHAIMGVAFT</span> <span class="n">WIMALSCAAP</span> <span class="n">PLFGWSRYIP</span>
<span class="n">EGMQCSCGVD</span> <span class="n">YYTLKPEVNN</span> <span class="n">ESFVIYMFIV</span> <span class="n">HFTIPLIVIF</span> <span class="n">FCYGRLLCTV</span> <span class="n">KEAAAQQQES</span>
<span class="n">LTTQKAEKEV</span> <span class="n">TRMVVIMVVF</span> <span class="n">FLICWVPYAY</span> <span class="n">VAFYIFTHQG</span> <span class="n">SNFGPVFMTV</span> <span class="n">PAFFAKSSAI</span>
<span class="n">YNPVIYIVLN</span> <span class="n">KQFRNCLITT</span> <span class="n">LCCGKNPFGD</span> <span class="n">EDGSSAATSK</span> <span class="n">TEASSVSSSQ</span> <span class="n">VSPA</span><span class="o">*</span>

<span class="o">&gt;</span><span class="n">P1</span><span class="p">;</span><span class="n">JN0120</span>
<span class="n">rhodopsin</span> <span class="o">-</span> <span class="n">Japanese</span> <span class="n">lamprey</span>
<span class="n">MNGTEGDNFY</span> <span class="n">VPFSNKTGLA</span> <span class="n">RSPYEYPQYY</span> <span class="n">LAEPWKYSAL</span> <span class="n">AAYMFFLILV</span> <span class="n">GFPVNFLTLF</span>
<span class="n">VTVQHKKLRT</span> <span class="n">PLNYILLNLA</span> <span class="n">MANLFMVLFG</span> <span class="n">FTVTMYTSMN</span> <span class="n">GYFVFGPTMC</span> <span class="n">SIEGFFATLG</span>
<span class="n">GEVALWSLVV</span> <span class="n">LAIERYIVIC</span> <span class="n">KPMGNFRFGN</span> <span class="n">THAIMGVAFT</span> <span class="n">WIMALACAAP</span> <span class="n">PLVGWSRYIP</span>
<span class="n">EGMQCSCGPD</span> <span class="n">YYTLNPNFNN</span> <span class="n">ESYVVYMFVV</span> <span class="n">HFLVPFVIIF</span> <span class="n">FCYGRLLCTV</span> <span class="n">KEAAAAQQES</span>
<span class="n">ASTQKAEKEV</span> <span class="n">TRMVVLMVIG</span> <span class="n">FLVCWVPYAS</span> <span class="n">VAFYIFTHQG</span> <span class="n">SDFGATFMTL</span> <span class="n">PAFFAKSSAL</span>
<span class="n">YNPVIYILMN</span> <span class="n">KQFRNCMITT</span> <span class="n">LCCGKNPLGD</span> <span class="n">DE</span><span class="o">-</span><span class="n">SGASTSKT</span> <span class="n">EVSSVSTSPV</span> <span class="n">SPA</span><span class="o">*</span>
</pre></div>
</div>
<p>As with the FASTA format, each record starts with a line beginning with “&gt;”
character.  There is then a two letter sequence type (P1, F1, DL, DC, RL,
RC, or XX), a semi colon, and the identification code.  The second like is
free text description.  The remaining lines contain the sequence itself,
terminating in an asterisk.  Space separated blocks of ten letters as shown
above are typical.</p>
<dl class="simple">
<dt>Sequence codes and their meanings:</dt><dd><ul class="simple">
<li><p>P1 - Protein (complete)</p></li>
<li><p>F1 - Protein (fragment)</p></li>
<li><p>D1 - DNA (e.g. EMBOSS seqret output)</p></li>
<li><p>DL - DNA (linear)</p></li>
<li><p>DC - DNA (circular)</p></li>
<li><p>RL - RNA (linear)</p></li>
<li><p>RC - RNA (circular)</p></li>
<li><p>N3 - tRNA</p></li>
<li><p>N1 - Other functional RNA</p></li>
<li><p>XX - Unknown</p></li>
</ul>
</dd>
</dl>
<dl class="function">
<dt id="Bio.SeqIO.PirIO.PirIterator">
<code class="sig-prename descclassname">Bio.SeqIO.PirIO.</code><code class="sig-name descname">PirIterator</code><span class="sig-paren">(</span><em class="sig-param">handle</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.SeqIO.PirIO.PirIterator" title="Permalink to this definition">¶</a></dt>
<dd><p>Iterate over Fasta records as SeqRecord objects.</p>
<p>handle - input file
alphabet - optional alphabet
title2ids - A function that, when given the title of the FASTA
file (without the beginning &gt;), will return the id, name and
description (in that order) for the record as a tuple of strings.</p>
<p>If this is not given, then the entire title line will be used
as the description, and the first word as the id and name.</p>
<p>Note that use of title2ids matches that of Bio.Fasta.SequenceParser
but the defaults are slightly different.</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s2">&quot;NBRF/DMB_prot.pir&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">handle</span><span class="p">:</span>
<span class="gp">... </span>   <span class="k">for</span> <span class="n">record</span> <span class="ow">in</span> <span class="n">PirIterator</span><span class="p">(</span><span class="n">handle</span><span class="p">):</span>
<span class="gp">... </span>       <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;</span><span class="si">%s</span><span class="s2"> length </span><span class="si">%i</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">record</span><span class="o">.</span><span class="n">id</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">record</span><span class="p">)))</span>
<span class="go">HLA:HLA00489 length 263</span>
<span class="go">HLA:HLA00490 length 94</span>
<span class="go">HLA:HLA00491 length 94</span>
<span class="go">HLA:HLA00492 length 80</span>
<span class="go">HLA:HLA00493 length 175</span>
<span class="go">HLA:HLA01083 length 188</span>
</pre></div>
</div>
</dd></dl>

<dl class="class">
<dt id="Bio.SeqIO.PirIO.PirWriter">
<em class="property">class </em><code class="sig-prename descclassname">Bio.SeqIO.PirIO.</code><code class="sig-name descname">PirWriter</code><span class="sig-paren">(</span><em class="sig-param">handle</em>, <em class="sig-param">wrap=60</em>, <em class="sig-param">record2title=None</em>, <em class="sig-param">code=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.SeqIO.PirIO.PirWriter" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.SeqIO.Interfaces.html#Bio.SeqIO.Interfaces.SequentialSequenceWriter" title="Bio.SeqIO.Interfaces.SequentialSequenceWriter"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.SeqIO.Interfaces.SequentialSequenceWriter</span></code></a></p>
<p>Class to write PIR format files.</p>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">clean</span></code>(self, text)</p></td>
<td><p>Use this to avoid getting newlines in the output.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">write_file</span></code>(self, records)</p></td>
<td><p>Use this to write an entire file containing the given records.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">write_footer</span></code>(self)</p></td>
<td><p>Write the file footer.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">write_header</span></code>(self)</p></td>
<td><p>Write the file header.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.SeqIO.PirIO.PirWriter.write_record" title="Bio.SeqIO.PirIO.PirWriter.write_record"><code class="xref py py-obj docutils literal notranslate"><span class="pre">write_record</span></code></a>(self, record)</p></td>
<td><p>Write a single PIR record to the file.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">write_records</span></code>(self, records)</p></td>
<td><p>Write multiple record to the output file.</p></td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="Bio.SeqIO.PirIO.PirWriter.write_record">
<code class="sig-name descname">write_record</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">record</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.SeqIO.PirIO.PirWriter.write_record" title="Permalink to this definition">¶</a></dt>
<dd><p>Write a single PIR record to the file.</p>
</dd></dl>

</dd></dl>

</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="Bio.SeqIO.QualityIO.html" class="btn btn-neutral float-right" title="Bio.SeqIO.QualityIO module" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="Bio.SeqIO.PhdIO.html" class="btn btn-neutral float-left" title="Bio.SeqIO.PhdIO module" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 1999-2017, The Biopython Contributors

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>