

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Bio.Emboss.Applications module &mdash; Biopython 1.74 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Bio.Emboss.Primer3 module" href="Bio.Emboss.Primer3.html" />
    <link rel="prev" title="Bio.Emboss package" href="Bio.Emboss.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search"  style="background: #10100F" >
          

          
            <a href="index.html" class="icon icon-home"> Biopython
          

          
            
            <img src="_static/biopython_logo.svg" class="logo" alt="Logo"/>
          
          </a>

          
            
            
              <div class="version">
                1.74
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">API Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="Bio.html">Bio package</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="Bio.html#subpackages">Subpackages</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="Bio.Affy.html">Bio.Affy package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Align.html">Bio.Align package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.AlignIO.html">Bio.AlignIO package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Alphabet.html">Bio.Alphabet package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Application.html">Bio.Application package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Blast.html">Bio.Blast package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.CAPS.html">Bio.CAPS package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Cluster.html">Bio.Cluster package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Compass.html">Bio.Compass package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Crystal.html">Bio.Crystal package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Data.html">Bio.Data package</a></li>
<li class="toctree-l3 current"><a class="reference internal" href="Bio.Emboss.html">Bio.Emboss package</a><ul class="current">
<li class="toctree-l4 current"><a class="reference internal" href="Bio.Emboss.html#submodules">Submodules</a></li>
<li class="toctree-l4"><a class="reference internal" href="Bio.Emboss.html#module-Bio.Emboss">Module contents</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Entrez.html">Bio.Entrez package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.ExPASy.html">Bio.ExPASy package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.FSSP.html">Bio.FSSP package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.GenBank.html">Bio.GenBank package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Geo.html">Bio.Geo package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Graphics.html">Bio.Graphics package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.HMM.html">Bio.HMM package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.KDTree.html">Bio.KDTree package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.KEGG.html">Bio.KEGG package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Medline.html">Bio.Medline package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.NMR.html">Bio.NMR package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Nexus.html">Bio.Nexus package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.PDB.html">Bio.PDB package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Pathway.html">Bio.Pathway package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Phylo.html">Bio.Phylo package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.PopGen.html">Bio.PopGen package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Restriction.html">Bio.Restriction package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SCOP.html">Bio.SCOP package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SVDSuperimposer.html">Bio.SVDSuperimposer package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SearchIO.html">Bio.SearchIO package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SeqIO.html">Bio.SeqIO package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SeqUtils.html">Bio.SeqUtils package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Sequencing.html">Bio.Sequencing package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Statistics.html">Bio.Statistics package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SubsMat.html">Bio.SubsMat package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SwissProt.html">Bio.SwissProt package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.TogoWS.html">Bio.TogoWS package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.UniGene.html">Bio.UniGene package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.UniProt.html">Bio.UniProt package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Wise.html">Bio.Wise package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.codonalign.html">Bio.codonalign package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.motifs.html">Bio.motifs package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.phenotype.html">Bio.phenotype package</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="Bio.html#submodules">Submodules</a></li>
<li class="toctree-l2"><a class="reference internal" href="Bio.html#module-Bio">Module contents</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="BioSQL.html">BioSQL package</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Biopython</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
          <li><a href="Bio.html">Bio package</a> &raquo;</li>
        
          <li><a href="Bio.Emboss.html">Bio.Emboss package</a> &raquo;</li>
        
      <li>Bio.Emboss.Applications module</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
        
      </li>
    
  </ul>

  
  <div class="rst-breadcrumbs-buttons" role="navigation" aria-label="breadcrumb navigation">
      
        <a href="Bio.Emboss.Primer3.html" class="btn btn-neutral float-right" title="Bio.Emboss.Primer3 module" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="Bio.Emboss.html" class="btn btn-neutral float-left" title="Bio.Emboss package" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
  </div>
  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="module-Bio.Emboss.Applications">
<span id="bio-emboss-applications-module"></span><h1>Bio.Emboss.Applications module<a class="headerlink" href="#module-Bio.Emboss.Applications" title="Permalink to this headline">¶</a></h1>
<p>Code to interact with and run various EMBOSS programs.</p>
<p>These classes follow the AbstractCommandline interfaces for running
programs.</p>
<dl class="class">
<dt id="Bio.Emboss.Applications.DiffseqCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Emboss.Applications.</code><code class="sig-name descname">DiffseqCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='diffseq'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Emboss.Applications.DiffseqCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Emboss.Applications._EmbossCommandLine</span></code></p>
<p>Commandline object for the diffseq program from EMBOSS.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Emboss.Applications.EInvertedCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Emboss.Applications.</code><code class="sig-name descname">EInvertedCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='einverted'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Emboss.Applications.EInvertedCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Emboss.Applications._EmbossCommandLine</span></code></p>
<p>Commandline object for the einverted program from EMBOSS.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Emboss.Applications.ETandemCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Emboss.Applications.</code><code class="sig-name descname">ETandemCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='etandem'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Emboss.Applications.ETandemCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Emboss.Applications._EmbossCommandLine</span></code></p>
<p>Commandline object for the etandem program from EMBOSS.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Emboss.Applications.Est2GenomeCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Emboss.Applications.</code><code class="sig-name descname">Est2GenomeCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='est2genome'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Emboss.Applications.Est2GenomeCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Emboss.Applications._EmbossCommandLine</span></code></p>
<p>Commandline object for the est2genome program from EMBOSS.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Emboss.Applications.FConsenseCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Emboss.Applications.</code><code class="sig-name descname">FConsenseCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='fconsense'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Emboss.Applications.FConsenseCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Emboss.Applications._EmbossCommandLine</span></code></p>
<p>Commandline object for the fconsense program from EMBOSS.</p>
<p>fconsense is an EMBOSS wrapper for the PHYLIP program consense used to
calculate consensus trees.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Emboss.Applications.FDNADistCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Emboss.Applications.</code><code class="sig-name descname">FDNADistCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='fdnadist'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Emboss.Applications.FDNADistCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Emboss.Applications._EmbossCommandLine</span></code></p>
<p>Commandline object for the fdnadist program from EMBOSS.</p>
<p>fdnadist is an EMBOSS wrapper for the PHYLIP program dnadist for
calulating distance matrices from DNA sequence files.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Emboss.Applications.FDNAParsCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Emboss.Applications.</code><code class="sig-name descname">FDNAParsCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='fdnapars'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Emboss.Applications.FDNAParsCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Emboss.Applications._EmbossCommandLine</span></code></p>
<p>Commandline object for the fdnapars program from EMBOSS.</p>
<p>fdnapars is an EMBOSS version of the PHYLIP program dnapars, for
estimating trees from DNA sequences using parsiomny. Calling this command
without providing a value for the option “-intreefile” will invoke
“interactive mode” (and as a result fail if called with subprocess) if
“-auto” is not set to true.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Emboss.Applications.FNeighborCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Emboss.Applications.</code><code class="sig-name descname">FNeighborCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='fneighbor'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Emboss.Applications.FNeighborCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Emboss.Applications._EmbossCommandLine</span></code></p>
<p>Commandline object for the fneighbor program from EMBOSS.</p>
<p>fneighbor is an EMBOSS wrapper for the PHYLIP program neighbor used for
calulating neighbor-joining or UPGMA trees from distance matrices.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Emboss.Applications.FProtDistCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Emboss.Applications.</code><code class="sig-name descname">FProtDistCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='fprotdist'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Emboss.Applications.FProtDistCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Emboss.Applications._EmbossCommandLine</span></code></p>
<p>Commandline object for the fprotdist program from EMBOSS.</p>
<p>fprotdist is an EMBOSS wrapper for the PHYLIP program protdist used to
estimate trees from protein sequences using parsimony</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Emboss.Applications.FProtParsCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Emboss.Applications.</code><code class="sig-name descname">FProtParsCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='fprotpars'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Emboss.Applications.FProtParsCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Emboss.Applications._EmbossCommandLine</span></code></p>
<p>Commandline object for the fdnapars program from EMBOSS.</p>
<p>fprotpars is an EMBOSS version of the PHYLIP program protpars, for
estimating trees from protein  sequences using parsiomny. Calling this
command without providing a value for the option “-intreefile” will invoke
“interactive mode” (and as a result fail if called with subprocess) if
“-auto” is not set to true.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Emboss.Applications.FSeqBootCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Emboss.Applications.</code><code class="sig-name descname">FSeqBootCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='fseqboot'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Emboss.Applications.FSeqBootCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Emboss.Applications._EmbossCommandLine</span></code></p>
<p>Commandline object for the fseqboot program from EMBOSS.</p>
<p>fseqboot is an EMBOSS wrapper for the PHYLIP program seqboot used to
pseudo-sample alignment files.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Emboss.Applications.FTreeDistCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Emboss.Applications.</code><code class="sig-name descname">FTreeDistCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='ftreedist'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Emboss.Applications.FTreeDistCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Emboss.Applications._EmbossCommandLine</span></code></p>
<p>Commandline object for the ftreedist program from EMBOSS.</p>
<p>ftreedist is an EMBOSS wrapper for the PHYLIP program treedist used for
calulating distance measures between phylogentic trees.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Emboss.Applications.FuzznucCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Emboss.Applications.</code><code class="sig-name descname">FuzznucCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='fuzznuc'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Emboss.Applications.FuzznucCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Emboss.Applications._EmbossCommandLine</span></code></p>
<p>Commandline object for the fuzznuc program from EMBOSS.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Emboss.Applications.FuzzproCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Emboss.Applications.</code><code class="sig-name descname">FuzzproCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='fuzzpro'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Emboss.Applications.FuzzproCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Emboss.Applications._EmbossCommandLine</span></code></p>
<p>Commandline object for the fuzzpro program from EMBOSS.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Emboss.Applications.IepCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Emboss.Applications.</code><code class="sig-name descname">IepCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='iep'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Emboss.Applications.IepCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Emboss.Applications._EmbossCommandLine</span></code></p>
<p>Commandline for EMBOSS iep: calculated isoelectric point and charge.</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Emboss.Applications</span> <span class="k">import</span> <span class="n">IepCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">iep_cline</span> <span class="o">=</span> <span class="n">IepCommandline</span><span class="p">(</span><span class="n">sequence</span><span class="o">=</span><span class="s2">&quot;proteins.faa&quot;</span><span class="p">,</span>
<span class="gp">... </span>                           <span class="n">outfile</span><span class="o">=</span><span class="s2">&quot;proteins.txt&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">iep_cline</span><span class="p">)</span>
<span class="go">iep -outfile=proteins.txt -sequence=proteins.faa</span>
</pre></div>
</div>
<p>You would typically run the command line with iep_cline() or via the
Python subprocess module, as described in the Biopython tutorial.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Emboss.Applications.NeedleCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Emboss.Applications.</code><code class="sig-name descname">NeedleCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='needle'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Emboss.Applications.NeedleCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Emboss.Applications._EmbossCommandLine</span></code></p>
<p>Commandline object for the needle program from EMBOSS.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Emboss.Applications.NeedleallCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Emboss.Applications.</code><code class="sig-name descname">NeedleallCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='needleall'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Emboss.Applications.NeedleallCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Emboss.Applications._EmbossCommandLine</span></code></p>
<p>Commandline object for the needleall program from EMBOSS.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Emboss.Applications.PalindromeCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Emboss.Applications.</code><code class="sig-name descname">PalindromeCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='palindrome'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Emboss.Applications.PalindromeCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Emboss.Applications._EmbossCommandLine</span></code></p>
<p>Commandline object for the palindrome program from EMBOSS.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Emboss.Applications.Primer3Commandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Emboss.Applications.</code><code class="sig-name descname">Primer3Commandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='eprimer3'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Emboss.Applications.Primer3Commandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Emboss.Applications._EmbossCommandLine</span></code></p>
<p>Commandline object for the Primer3 interface from EMBOSS.</p>
<p>The precise set of supported arguments depends on your version of EMBOSS.
This version accepts arguments current at EMBOSS 6.1.0, but in order to
remain backwards compatible also support the old argument names as well.</p>
<p>e.g. Using EMBOSS 6.1.0 or later,</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span> <span class="o">=</span> <span class="n">Primer3Commandline</span><span class="p">(</span><span class="n">sequence</span><span class="o">=</span><span class="s2">&quot;mysequence.fas&quot;</span><span class="p">,</span> <span class="n">auto</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">hybridprobe</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span><span class="o">.</span><span class="n">explainflag</span> <span class="o">=</span> <span class="kc">True</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span><span class="o">.</span><span class="n">osizeopt</span><span class="o">=</span><span class="mi">20</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span><span class="o">.</span><span class="n">psizeopt</span><span class="o">=</span><span class="mi">200</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span><span class="o">.</span><span class="n">outfile</span> <span class="o">=</span> <span class="s2">&quot;myresults.out&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span><span class="o">.</span><span class="n">bogusparameter</span> <span class="o">=</span> <span class="mi">1967</span>     <span class="c1"># Invalid parameter</span>
<span class="gt">Traceback (most recent call last):</span>
    <span class="o">...</span>
<span class="gr">ValueError</span>: <span class="n">Option name bogusparameter was not found.</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">cline</span><span class="p">)</span>
<span class="go">eprimer3 -auto -outfile=myresults.out -sequence=mysequence.fas -hybridprobe=True -psizeopt=200 -osizeopt=20 -explainflag=True</span>
</pre></div>
</div>
<p>The equivalent for anyone still using an older version of EMBOSS would be:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span> <span class="o">=</span> <span class="n">Primer3Commandline</span><span class="p">(</span><span class="n">sequence</span><span class="o">=</span><span class="s2">&quot;mysequence.fas&quot;</span><span class="p">,</span> <span class="n">auto</span><span class="o">=</span><span class="kc">True</span><span class="p">,</span> <span class="n">hybridprobe</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span><span class="o">.</span><span class="n">explainflag</span> <span class="o">=</span> <span class="kc">True</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span><span class="o">.</span><span class="n">oligosize</span><span class="o">=</span><span class="mi">20</span>              <span class="c1"># Old EMBOSS, instead of osizeopt</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span><span class="o">.</span><span class="n">productosize</span><span class="o">=</span><span class="mi">200</span>          <span class="c1"># Old EMBOSS, instead of psizeopt</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">cline</span><span class="o">.</span><span class="n">outfile</span> <span class="o">=</span> <span class="s2">&quot;myresults.out&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">cline</span><span class="p">)</span>
<span class="go">eprimer3 -auto -outfile=myresults.out -sequence=mysequence.fas -hybridprobe=True -productosize=200 -oligosize=20 -explainflag=True</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Emboss.Applications.PrimerSearchCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Emboss.Applications.</code><code class="sig-name descname">PrimerSearchCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='primersearch'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Emboss.Applications.PrimerSearchCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Emboss.Applications._EmbossCommandLine</span></code></p>
<p>Commandline object for the primersearch program from EMBOSS.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Emboss.Applications.SeqmatchallCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Emboss.Applications.</code><code class="sig-name descname">SeqmatchallCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='seqmatchall'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Emboss.Applications.SeqmatchallCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Emboss.Applications._EmbossCommandLine</span></code></p>
<p>Commandline object for the seqmatchall program from EMBOSS.</p>
<p>e.g.
&gt;&gt;&gt; cline = SeqmatchallCommandline(sequence=”opuntia.fasta”, outfile=”opuntia.txt”)
&gt;&gt;&gt; cline.auto = True
&gt;&gt;&gt; cline.wordsize = 18
&gt;&gt;&gt; cline.aformat = “pair”
&gt;&gt;&gt; print(cline)
seqmatchall -auto -outfile=opuntia.txt -sequence=opuntia.fasta -wordsize=18 -aformat=pair</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Emboss.Applications.SeqretCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Emboss.Applications.</code><code class="sig-name descname">SeqretCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='seqret'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Emboss.Applications.SeqretCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Emboss.Applications._EmbossMinimalCommandLine</span></code></p>
<p>Commandline object for the seqret program from EMBOSS.</p>
<p>This tool allows you to interconvert between different sequence file
formats (e.g. GenBank to FASTA). Combining Biopython’s Bio.SeqIO module
with seqret using a suitable intermediate file format can allow you to
read/write to an even wider range of file formats.</p>
<p>This wrapper currently only supports the core functionality, things like
feature tables (in EMBOSS 6.1.0 onwards) are not yet included.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Emboss.Applications.StretcherCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Emboss.Applications.</code><code class="sig-name descname">StretcherCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='stretcher'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Emboss.Applications.StretcherCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Emboss.Applications._EmbossCommandLine</span></code></p>
<p>Commandline object for the stretcher program from EMBOSS.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Emboss.Applications.TranalignCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Emboss.Applications.</code><code class="sig-name descname">TranalignCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='tranalign'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Emboss.Applications.TranalignCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Emboss.Applications._EmbossCommandLine</span></code></p>
<p>Commandline object for the tranalign program from EMBOSS.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Emboss.Applications.WaterCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Emboss.Applications.</code><code class="sig-name descname">WaterCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='water'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Emboss.Applications.WaterCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Emboss.Applications._EmbossCommandLine</span></code></p>
<p>Commandline object for the water program from EMBOSS.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="Bio.Emboss.Primer3.html" class="btn btn-neutral float-right" title="Bio.Emboss.Primer3 module" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="Bio.Emboss.html" class="btn btn-neutral float-left" title="Bio.Emboss package" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 1999-2017, The Biopython Contributors

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>