

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Bio.motifs.meme module &mdash; Biopython 1.76.dev0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Bio.motifs.minimal module" href="Bio.motifs.minimal.html" />
    <link rel="prev" title="Bio.motifs.matrix module" href="Bio.motifs.matrix.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search"  style="background: #10100F" >
          

          
            <a href="index.html">
          

          
            
            <img src="_static/biopython_logo.svg" class="logo" alt="Logo"/>
          
          </a>

          
            
            
              <div class="version">
                1.76.dev0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">API Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="Bio.html">Bio package</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="Bio.html#subpackages">Subpackages</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="Bio.Affy.html">Bio.Affy package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Align.html">Bio.Align package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.AlignIO.html">Bio.AlignIO package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Alphabet.html">Bio.Alphabet package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Application.html">Bio.Application package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Blast.html">Bio.Blast package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.CAPS.html">Bio.CAPS package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Cluster.html">Bio.Cluster package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Compass.html">Bio.Compass package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Crystal.html">Bio.Crystal package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Data.html">Bio.Data package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Emboss.html">Bio.Emboss package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Entrez.html">Bio.Entrez package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.ExPASy.html">Bio.ExPASy package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.FSSP.html">Bio.FSSP package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.GenBank.html">Bio.GenBank package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Geo.html">Bio.Geo package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Graphics.html">Bio.Graphics package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.HMM.html">Bio.HMM package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.KDTree.html">Bio.KDTree package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.KEGG.html">Bio.KEGG package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Medline.html">Bio.Medline package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.NMR.html">Bio.NMR package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Nexus.html">Bio.Nexus package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.PDB.html">Bio.PDB package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Pathway.html">Bio.Pathway package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Phylo.html">Bio.Phylo package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.PopGen.html">Bio.PopGen package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Restriction.html">Bio.Restriction package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SCOP.html">Bio.SCOP package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SVDSuperimposer.html">Bio.SVDSuperimposer package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SearchIO.html">Bio.SearchIO package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SeqIO.html">Bio.SeqIO package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SeqUtils.html">Bio.SeqUtils package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Sequencing.html">Bio.Sequencing package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Statistics.html">Bio.Statistics package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SubsMat.html">Bio.SubsMat package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SwissProt.html">Bio.SwissProt package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.TogoWS.html">Bio.TogoWS package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.UniGene.html">Bio.UniGene package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.UniProt.html">Bio.UniProt package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Wise.html">Bio.Wise package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.codonalign.html">Bio.codonalign package</a></li>
<li class="toctree-l3 current"><a class="reference internal" href="Bio.motifs.html">Bio.motifs package</a><ul class="current">
<li class="toctree-l4"><a class="reference internal" href="Bio.motifs.html#subpackages">Subpackages</a></li>
<li class="toctree-l4 current"><a class="reference internal" href="Bio.motifs.html#submodules">Submodules</a></li>
<li class="toctree-l4"><a class="reference internal" href="Bio.motifs.html#module-Bio.motifs">Module contents</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="Bio.phenotype.html">Bio.phenotype package</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="Bio.html#submodules">Submodules</a></li>
<li class="toctree-l2"><a class="reference internal" href="Bio.html#module-Bio">Module contents</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="BioSQL.html">BioSQL package</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Biopython</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
          <li><a href="Bio.html">Bio package</a> &raquo;</li>
        
          <li><a href="Bio.motifs.html">Bio.motifs package</a> &raquo;</li>
        
      <li>Bio.motifs.meme module</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            
              <!-- User defined GitHub URL -->
              <a href="https://github.com/biopython/biopython/blob/master/Bio/motifs/meme.py" class="fa fa-github"> Edit on GitHub</a>
            
          
        
      </li>
    
  </ul>

  
  <div class="rst-breadcrumbs-buttons" role="navigation" aria-label="breadcrumb navigation">
      
        <a href="Bio.motifs.minimal.html" class="btn btn-neutral float-right" title="Bio.motifs.minimal module" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="Bio.motifs.matrix.html" class="btn btn-neutral float-left" title="Bio.motifs.matrix module" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
  </div>
  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="module-Bio.motifs.meme">
<span id="bio-motifs-meme-module"></span><h1>Bio.motifs.meme module<a class="headerlink" href="#module-Bio.motifs.meme" title="Permalink to this headline">¶</a></h1>
<p>Module for the support of MEME motif format.</p>
<dl class="class">
<dt id="Bio.motifs.meme.Instance">
<em class="property">class </em><code class="sig-prename descclassname">Bio.motifs.meme.</code><code class="sig-name descname">Instance</code><span class="sig-paren">(</span><em class="sig-param">*args</em>, <em class="sig-param">**kwds</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.motifs.meme.Instance" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Seq.html#Bio.Seq.Seq" title="Bio.Seq.Seq"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Seq.Seq</span></code></a></p>
<p>A class describing the instances of a MEME motif, and the data thereof.</p>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">back_transcribe</span></code>(self)</p></td>
<td><p>Return the DNA sequence from an RNA sequence by creating a new Seq object.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">complement</span></code>(self)</p></td>
<td><p>Return the complement sequence by creating a new Seq object.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">count</span></code>(self, sub[, start, end])</p></td>
<td><p>Return a non-overlapping count, like that of a python string.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">count_overlap</span></code>(self, sub[, start, end])</p></td>
<td><p>Return an overlapping count.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">endswith</span></code>(self, suffix[, start, end])</p></td>
<td><p>Return True if the Seq ends with the given suffix, False otherwise.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">find</span></code>(self, sub[, start, end])</p></td>
<td><p>Find method, like that of a python string.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">index</span></code>(self, sub[, start, end])</p></td>
<td><p>Like find() but raise ValueError when the substring is not found.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">join</span></code>(self, other)</p></td>
<td><p>Return a merge of the sequences in other, spaced by the sequence from self.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">lower</span></code>(self)</p></td>
<td><p>Return a lower case copy of the sequence.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">lstrip</span></code>(self[, chars])</p></td>
<td><p>Return a new Seq object with leading (left) end stripped.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">reverse_complement</span></code>(self)</p></td>
<td><p>Return the reverse complement sequence by creating a new Seq object.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">rfind</span></code>(self, sub[, start, end])</p></td>
<td><p>Find from right method, like that of a python string.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">rindex</span></code>(self, sub[, start, end])</p></td>
<td><p>Like rfind() but raise ValueError when the substring is not found.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">rsplit</span></code>(self[, sep, maxsplit])</p></td>
<td><p>Do a right split method, like that of a python string.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">rstrip</span></code>(self[, chars])</p></td>
<td><p>Return a new Seq object with trailing (right) end stripped.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">split</span></code>(self[, sep, maxsplit])</p></td>
<td><p>Split method, like that of a python string.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">startswith</span></code>(self, prefix[, start, end])</p></td>
<td><p>Return True if the Seq starts with the given prefix, False otherwise.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">strip</span></code>(self[, chars])</p></td>
<td><p>Return a new Seq object with leading and trailing ends stripped.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">tomutable</span></code>(self)</p></td>
<td><p>Return the full sequence as a MutableSeq object.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">transcribe</span></code>(self)</p></td>
<td><p>Return the RNA sequence from a DNA sequence by creating a new Seq object.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">translate</span></code>(self[, table, stop_symbol, …])</p></td>
<td><p>Turn a nucleotide sequence into a protein sequence by creating a new Seq object.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">ungap</span></code>(self[, gap])</p></td>
<td><p>Return a copy of the sequence without the gap character(s).</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">upper</span></code>(self)</p></td>
<td><p>Return an upper case copy of the sequence.</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.motifs.meme.Motif">
<em class="property">class </em><code class="sig-prename descclassname">Bio.motifs.meme.</code><code class="sig-name descname">Motif</code><span class="sig-paren">(</span><em class="sig-param">alphabet=None</em>, <em class="sig-param">instances=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.motifs.meme.Motif" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.motifs.html#Bio.motifs.Motif" title="Bio.motifs.Motif"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.motifs.Motif</span></code></a></p>
<p>A subclass of Motif used in parsing MEME (and MAST) output.</p>
<p>This subclass defines functions and data specific to MEME motifs.
This includes the motif name, the evalue for a motif, and its number
of occurrences.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">anticonsensus</span></code></dt><dd><p>Return the least probable pattern to be generated from this motif.</p>
</dd>
<dt><strong>background</strong></dt><dd></dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">consensus</span></code></dt><dd><p>Return the consensus sequence.</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">degenerate_consensus</span></code></dt><dd><p>Generate degenerate consesnsus sequence.</p>
</dd>
<dt><strong>mask</strong></dt><dd></dd>
<dt><strong>pseudocounts</strong></dt><dd></dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">pssm</span></code></dt><dd><p>Compute position specific scoring matrices.</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">pwm</span></code></dt><dd><p>Compute position weight matrices.</p>
</dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">format</span></code>(self, format)</p></td>
<td><p>Return a string representation of the Motif in the given format.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">reverse_complement</span></code>(self)</p></td>
<td><p>Return the reverse complement of the motif as a new motif.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">weblogo</span></code>(self, fname[, format, version])</p></td>
<td><p>Download and save a weblogo using the Berkeley weblogo service.</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.motifs.meme.Record">
<em class="property">class </em><code class="sig-prename descclassname">Bio.motifs.meme.</code><code class="sig-name descname">Record</code><a class="headerlink" href="#Bio.motifs.meme.Record" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">list</span></code></p>
<p>A class for holding the results of a MEME run.</p>
<p>A meme.Record is an object that holds the results from running
MEME. It implements no methods of its own.</p>
<p>The meme.Record class inherits from list, so you can access individual
motifs in the record by their index. Alternatively, you can find a motif
by its name:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio</span> <span class="k">import</span> <span class="n">motifs</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s2">&quot;motifs/meme.out&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
<span class="gp">... </span>    <span class="n">record</span> <span class="o">=</span> <span class="n">motifs</span><span class="o">.</span><span class="n">parse</span><span class="p">(</span><span class="n">f</span><span class="p">,</span> <span class="s1">&#39;MEME&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">motif</span> <span class="o">=</span> <span class="n">record</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">motif</span><span class="o">.</span><span class="n">name</span><span class="p">)</span>
<span class="go">Motif 1</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">motif</span> <span class="o">=</span> <span class="n">record</span><span class="p">[</span><span class="s1">&#39;Motif 1&#39;</span><span class="p">]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">motif</span><span class="o">.</span><span class="n">name</span><span class="p">)</span>
<span class="go">Motif 1</span>
</pre></div>
</div>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">append</span></code>(self, object, /)</p></td>
<td><p>Append object to the end of the list.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">clear</span></code>(self, /)</p></td>
<td><p>Remove all items from list.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">copy</span></code>(self, /)</p></td>
<td><p>Return a shallow copy of the list.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">count</span></code>(self, value, /)</p></td>
<td><p>Return number of occurrences of value.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">extend</span></code>(self, iterable, /)</p></td>
<td><p>Extend list by appending elements from the iterable.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">index</span></code>(self, value[, start, stop])</p></td>
<td><p>Return first index of value.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">insert</span></code>(self, index, object, /)</p></td>
<td><p>Insert object before index.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">pop</span></code>(self[, index])</p></td>
<td><p>Remove and return item at index (default last).</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">remove</span></code>(self, value, /)</p></td>
<td><p>Remove first occurrence of value.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">reverse</span></code>(self, /)</p></td>
<td><p>Reverse <em>IN PLACE</em>.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">sort</span></code>(self, /, \*[, key, reverse])</p></td>
<td><p>Stable sort <em>IN PLACE</em>.</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="function">
<dt id="Bio.motifs.meme.read">
<code class="sig-prename descclassname">Bio.motifs.meme.</code><code class="sig-name descname">read</code><span class="sig-paren">(</span><em class="sig-param">handle</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.motifs.meme.read" title="Permalink to this definition">¶</a></dt>
<dd><p>Parse the text output of the MEME program into a meme.Record object.</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.motifs</span> <span class="k">import</span> <span class="n">meme</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s2">&quot;motifs/meme.out&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">f</span><span class="p">:</span>
<span class="gp">... </span>    <span class="n">record</span> <span class="o">=</span> <span class="n">meme</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="n">f</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">for</span> <span class="n">motif</span> <span class="ow">in</span> <span class="n">record</span><span class="p">:</span>
<span class="gp">... </span>    <span class="k">for</span> <span class="n">instance</span> <span class="ow">in</span> <span class="n">motif</span><span class="o">.</span><span class="n">instances</span><span class="p">:</span>
<span class="gp">... </span>        <span class="nb">print</span><span class="p">(</span><span class="n">instance</span><span class="o">.</span><span class="n">motif_name</span><span class="p">,</span> <span class="n">instance</span><span class="o">.</span><span class="n">sequence_name</span><span class="p">,</span> <span class="n">instance</span><span class="o">.</span><span class="n">strand</span><span class="p">,</span> <span class="n">instance</span><span class="o">.</span><span class="n">pvalue</span><span class="p">)</span>
<span class="go">Motif 1 SEQ10; + 8.71e-07</span>
<span class="go">Motif 1 SEQ9; + 8.71e-07</span>
<span class="go">Motif 1 SEQ8; + 8.71e-07</span>
<span class="go">Motif 1 SEQ7; + 8.71e-07</span>
<span class="go">Motif 1 SEQ6; + 8.71e-07</span>
<span class="go">Motif 1 SEQ5; + 8.71e-07</span>
<span class="go">Motif 1 SEQ4; + 8.71e-07</span>
<span class="go">Motif 1 SEQ3; + 8.71e-07</span>
<span class="go">Motif 1 SEQ2; + 8.71e-07</span>
<span class="go">Motif 1 SEQ1; + 8.71e-07</span>
</pre></div>
</div>
</dd></dl>

</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="Bio.motifs.minimal.html" class="btn btn-neutral float-right" title="Bio.motifs.minimal module" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="Bio.motifs.matrix.html" class="btn btn-neutral float-left" title="Bio.motifs.matrix module" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 1999-2017, The Biopython Contributors

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>