

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Bio.SeqUtils.MeltingTemp module &mdash; Biopython 1.74.dev0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Bio.SeqUtils.ProtParam module" href="Bio.SeqUtils.ProtParam.html" />
    <link rel="prev" title="Bio.SeqUtils.IsoelectricPoint module" href="Bio.SeqUtils.IsoelectricPoint.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search"  style="background: #10100F" >
          

          
            <a href="index.html" class="icon icon-home"> Biopython
          

          
            
            <img src="_static/biopython_logo.svg" class="logo" alt="Logo"/>
          
          </a>

          
            
            
              <div class="version">
                1.74.dev0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">API Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="Bio.html">Bio package</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="Bio.html#subpackages">Subpackages</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="Bio.Affy.html">Bio.Affy package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Align.html">Bio.Align package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.AlignIO.html">Bio.AlignIO package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Alphabet.html">Bio.Alphabet package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Application.html">Bio.Application package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Blast.html">Bio.Blast package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.CAPS.html">Bio.CAPS package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Cluster.html">Bio.Cluster package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Compass.html">Bio.Compass package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Crystal.html">Bio.Crystal package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Data.html">Bio.Data package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Emboss.html">Bio.Emboss package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Entrez.html">Bio.Entrez package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.ExPASy.html">Bio.ExPASy package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.FSSP.html">Bio.FSSP package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.GenBank.html">Bio.GenBank package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Geo.html">Bio.Geo package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Graphics.html">Bio.Graphics package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.HMM.html">Bio.HMM package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.KDTree.html">Bio.KDTree package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.KEGG.html">Bio.KEGG package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Medline.html">Bio.Medline package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.NMR.html">Bio.NMR package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Nexus.html">Bio.Nexus package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.PDB.html">Bio.PDB package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Pathway.html">Bio.Pathway package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Phylo.html">Bio.Phylo package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.PopGen.html">Bio.PopGen package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Restriction.html">Bio.Restriction package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SCOP.html">Bio.SCOP package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SVDSuperimposer.html">Bio.SVDSuperimposer package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SearchIO.html">Bio.SearchIO package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SeqIO.html">Bio.SeqIO package</a></li>
<li class="toctree-l3 current"><a class="reference internal" href="Bio.SeqUtils.html">Bio.SeqUtils package</a><ul class="current">
<li class="toctree-l4 current"><a class="reference internal" href="Bio.SeqUtils.html#submodules">Submodules</a></li>
<li class="toctree-l4"><a class="reference internal" href="Bio.SeqUtils.html#module-Bio.SeqUtils">Module contents</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Sequencing.html">Bio.Sequencing package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Statistics.html">Bio.Statistics package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SubsMat.html">Bio.SubsMat package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SwissProt.html">Bio.SwissProt package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.TogoWS.html">Bio.TogoWS package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.UniGene.html">Bio.UniGene package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.UniProt.html">Bio.UniProt package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Wise.html">Bio.Wise package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.codonalign.html">Bio.codonalign package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.motifs.html">Bio.motifs package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.phenotype.html">Bio.phenotype package</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="Bio.html#submodules">Submodules</a></li>
<li class="toctree-l2"><a class="reference internal" href="Bio.html#module-Bio">Module contents</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="BioSQL.html">BioSQL package</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Biopython</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
          <li><a href="Bio.html">Bio package</a> &raquo;</li>
        
          <li><a href="Bio.SeqUtils.html">Bio.SeqUtils package</a> &raquo;</li>
        
      <li>Bio.SeqUtils.MeltingTemp module</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
        
      </li>
    
  </ul>

  
  <div class="rst-breadcrumbs-buttons" role="navigation" aria-label="breadcrumb navigation">
      
        <a href="Bio.SeqUtils.ProtParam.html" class="btn btn-neutral float-right" title="Bio.SeqUtils.ProtParam module" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="Bio.SeqUtils.IsoelectricPoint.html" class="btn btn-neutral float-left" title="Bio.SeqUtils.IsoelectricPoint module" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
  </div>
  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="module-Bio.SeqUtils.MeltingTemp">
<span id="bio-sequtils-meltingtemp-module"></span><h1>Bio.SeqUtils.MeltingTemp module<a class="headerlink" href="#module-Bio.SeqUtils.MeltingTemp" title="Permalink to this headline">¶</a></h1>
<p>Calculate the melting temperature of nucleotide sequences.</p>
<p>This module contains three different methods to calculate the melting
temperature of oligonucleotides:</p>
<ol class="arabic simple">
<li><p>Tm_Wallace: ‘Rule of thumb’</p></li>
<li><p>Tm_GC: Empirical formulas based on GC content. Salt and mismatch corrections
can be included.</p></li>
<li><p>Tm_NN: Calculation based on nearest neighbor thermodynamics. Several tables
for DNA/DNA, DNA/RNA and RNA/RNA hybridizations are included.
Correction for mismatches, dangling ends, salt concentration and other
additives are available.</p></li>
</ol>
<p>Tm_staluc is the ‘old’ NN calculation and is kept for compatibility. It is,
however, recommended to use Tm_NN instead, since Tm_staluc may be depreceated
in the future. Also, Tm_NN has much more options. Using Tm_staluc and Tm_NN
with default parameters gives (essentially) the same results.</p>
<dl class="simple">
<dt>General parameters for most Tm methods:</dt><dd><ul class="simple">
<li><p>seq – A Biopython sequence object or a string.</p></li>
<li><p>check – Checks if the sequence is valid for the given method (default=
True). In general, whitespaces and non-base characters are removed and
characters are converted to uppercase. RNA will be backtranscribed.</p></li>
<li><p>strict – Do not allow base characters or neighbor duplex keys (e.g.
‘AT/NA’) that could not or not unambigiously be evaluated for the respective
method (default=True). Note that W (= A or T) and S (= C or G) are not
ambiguous for Tm_Wallace and Tm_GC. If ‘False’, average values (if
applicable) will be used.</p></li>
</ul>
</dd>
</dl>
<p>This module is not able to detect self-complementary and it will not use
alignment tools to align an oligonucleotide sequence to its target sequence.
Thus it can not detect dangling-ends and mismatches by itself (don’t even think
about bulbs and loops). These parameters have to be handed over to the
respective method.</p>
<dl class="simple">
<dt>Other public methods of this module:</dt><dd><ul class="simple">
<li><p>make_table     : To create a table with thermodynamic data.</p></li>
<li><p>salt_correction: To adjust Tm to a given salt concentration by different
formulas. This method is called from Tm_GC and Tm_NN but may
also be accessed ‘manually’. It returns a correction term, not
a corrected Tm!</p></li>
<li><p>chem_correction: To adjust Tm regarding the chemical additives DMSO and
formaldehyde. The method returns a corrected Tm. Chemical
correction is not an integral part of the Tm methods and must
be called additionally.</p></li>
</ul>
</dd>
</dl>
<p>For example:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.SeqUtils</span> <span class="k">import</span> <span class="n">MeltingTemp</span> <span class="k">as</span> <span class="n">mt</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mystring</span> <span class="o">=</span> <span class="s1">&#39;CGTTCCAAAGATGTGGGCATGAGCTTAC&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">myseq</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="n">mystring</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">Tm_Wallace</span><span class="p">(</span><span class="n">mystring</span><span class="p">))</span>
<span class="go">84.00</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">Tm_Wallace</span><span class="p">(</span><span class="n">myseq</span><span class="p">))</span>
<span class="go">84.00</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">Tm_GC</span><span class="p">(</span><span class="n">myseq</span><span class="p">))</span>
<span class="go">58.73</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">Tm_NN</span><span class="p">(</span><span class="n">myseq</span><span class="p">))</span>
<span class="go">60.32</span>
</pre></div>
</div>
<p>Tm_NN with default values gives same result as ‘old’ Tm_staluc. However, values
differ for RNA, since Tm_staluc had some errors for RNA calculation. These
errors have been fixed in this version.</p>
<p>New Tm_NN can do slightly more:
Using different thermodynamic tables, e.g. from Breslauer ‘86 or Sugimoto ‘96:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">Tm_NN</span><span class="p">(</span><span class="n">myseq</span><span class="p">,</span> <span class="n">nn_table</span><span class="o">=</span><span class="n">mt</span><span class="o">.</span><span class="n">DNA_NN1</span><span class="p">))</span>  <span class="c1"># Breslauer &#39;86</span>
<span class="go">72.19</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">Tm_NN</span><span class="p">(</span><span class="n">myseq</span><span class="p">,</span> <span class="n">nn_table</span><span class="o">=</span><span class="n">mt</span><span class="o">.</span><span class="n">DNA_NN2</span><span class="p">))</span>  <span class="c1"># Sugimoto &#39;96</span>
<span class="go">65.47</span>
</pre></div>
</div>
<p>Tables for RNA and RNA/DNA hybrids are included:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">Tm_NN</span><span class="p">(</span><span class="n">myseq</span><span class="p">,</span> <span class="n">nn_table</span><span class="o">=</span><span class="n">mt</span><span class="o">.</span><span class="n">RNA_NN1</span><span class="p">))</span>  <span class="c1"># Freier &#39;86</span>
<span class="go">73.35</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">Tm_NN</span><span class="p">(</span><span class="n">myseq</span><span class="p">,</span> <span class="n">nn_table</span><span class="o">=</span><span class="n">mt</span><span class="o">.</span><span class="n">R_DNA_NN1</span><span class="p">))</span>  <span class="c1"># Sugimoto &#39;95</span>
<span class="go">57.17</span>
</pre></div>
</div>
<p>Several types of salc correction (for Tm_NN and Tm_GC):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="k">for</span> <span class="n">i</span> <span class="ow">in</span> <span class="nb">range</span><span class="p">(</span><span class="mi">1</span><span class="p">,</span> <span class="mi">8</span><span class="p">):</span>
<span class="gp">... </span>    <span class="nb">print</span><span class="p">(</span><span class="s2">&quot;Type: </span><span class="si">%d</span><span class="s2">, Tm: </span><span class="si">%0.2f</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">i</span><span class="p">,</span> <span class="n">Tm_NN</span><span class="p">(</span><span class="n">myseq</span><span class="p">,</span> <span class="n">saltcorr</span><span class="o">=</span><span class="n">i</span><span class="p">)))</span>
<span class="gp">...</span>
<span class="go">Type: 1, Tm: 54.27</span>
<span class="go">Type: 2, Tm: 54.02</span>
<span class="go">Type: 3, Tm: 59.60</span>
<span class="go">Type: 4, Tm: 60.64</span>
<span class="go">Type: 5, Tm: 60.32</span>
<span class="go">Type: 6, Tm: 59.78</span>
<span class="go">Type: 7, Tm: 59.78</span>
</pre></div>
</div>
<p>Correction for other monovalent cations (K+, Tris), Mg2+ and dNTPs according
to von Ahsen et al. (2001) or Owczarzy et al. (2008) (for Tm_NN and Tm_GC):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">Tm_NN</span><span class="p">(</span><span class="n">myseq</span><span class="p">,</span> <span class="n">Na</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span> <span class="n">Tris</span><span class="o">=</span><span class="mi">10</span><span class="p">))</span>
<span class="go">60.79</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">Tm_NN</span><span class="p">(</span><span class="n">myseq</span><span class="p">,</span> <span class="n">Na</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span> <span class="n">Tris</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> <span class="n">Mg</span><span class="o">=</span><span class="mf">1.5</span><span class="p">))</span>
<span class="go">67.39</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">Tm_NN</span><span class="p">(</span><span class="n">myseq</span><span class="p">,</span> <span class="n">Na</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span> <span class="n">Tris</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> <span class="n">Mg</span><span class="o">=</span><span class="mf">1.5</span><span class="p">,</span> <span class="n">saltcorr</span><span class="o">=</span><span class="mi">7</span><span class="p">))</span>
<span class="go">66.81</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">Tm_NN</span><span class="p">(</span><span class="n">myseq</span><span class="p">,</span> <span class="n">Na</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span> <span class="n">Tris</span><span class="o">=</span><span class="mi">10</span><span class="p">,</span> <span class="n">Mg</span><span class="o">=</span><span class="mf">1.5</span><span class="p">,</span> <span class="n">dNTPs</span><span class="o">=</span><span class="mf">0.6</span><span class="p">,</span>
<span class="gp">... </span>                         <span class="n">saltcorr</span><span class="o">=</span><span class="mi">7</span><span class="p">))</span>
<span class="go">66.04</span>
</pre></div>
</div>
<p>Dangling ends and mismatches, e.g.:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">Oligo</span><span class="p">:</span>     <span class="n">CGTTCCaAAGATGTGGGCATGAGCTTAC</span>       <span class="n">CGTTCCaAAGATGTGGGCATGAGCTTAC</span>
           <span class="p">::::::</span><span class="n">X</span><span class="p">:::::::::::::::::::::</span>  <span class="ow">or</span>   <span class="p">::::::</span><span class="n">X</span><span class="p">:::::::::::::::::::::</span>
<span class="n">Template</span><span class="p">:</span>  <span class="n">GCAAGGcTTCTACACCCGTACTCGAATG</span>      <span class="n">TGCAAGGcTTCTACACCCGTACTCGAATGC</span>
</pre></div>
</div>
<p>Here:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">Tm_NN</span><span class="p">(</span><span class="s1">&#39;CGTTCCAAAGATGTGGGCATGAGCTTAC&#39;</span><span class="p">))</span>
<span class="go">60.32</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">Tm_NN</span><span class="p">(</span><span class="s1">&#39;CGTTCCAAAGATGTGGGCATGAGCTTAC&#39;</span><span class="p">,</span>
<span class="gp">... </span>                   <span class="n">c_seq</span><span class="o">=</span><span class="s1">&#39;GCAAGGcTTCTACACCCGTACTCGAATG&#39;</span><span class="p">))</span>
<span class="go">55.39</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">Tm_NN</span><span class="p">(</span><span class="s1">&#39;CGTTCCAAAGATGTGGGCATGAGCTTAC&#39;</span><span class="p">,</span> <span class="n">shift</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span>
<span class="gp">... </span>                  <span class="n">c_seq</span><span class="o">=</span><span class="s1">&#39;TGCAAGGcTTCTACACCCGTACTCGAATGC&#39;</span><span class="p">))</span>
<span class="go">55.69</span>
</pre></div>
</div>
<p>The same for RNA:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">Tm_NN</span><span class="p">(</span><span class="s1">&#39;CGUUCCAAAGAUGUGGGCAUGAGCUUAC&#39;</span><span class="p">,</span>
<span class="gp">... </span>                  <span class="n">c_seq</span><span class="o">=</span><span class="s1">&#39;UGCAAGGcUUCUACACCCGUACUCGAAUGC&#39;</span><span class="p">,</span>
<span class="gp">... </span>                  <span class="n">shift</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">nn_table</span><span class="o">=</span><span class="n">mt</span><span class="o">.</span><span class="n">RNA_NN3</span><span class="p">,</span>
<span class="gp">... </span>                  <span class="n">de_table</span><span class="o">=</span><span class="n">mt</span><span class="o">.</span><span class="n">RNA_DE1</span><span class="p">))</span>
<span class="go">73.00</span>
</pre></div>
</div>
<p>Note, that thermodynamic data are not available for all kind of mismatches,
e.g. most double mismatches or terminal mismaches combined with danglind ends:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">Tm_NN</span><span class="p">(</span><span class="s1">&#39;CGTTCCAAAGATGTGGGCATGAGCTTAC&#39;</span><span class="p">,</span>
<span class="gp">... </span>                  <span class="n">c_seq</span><span class="o">=</span><span class="s1">&#39;TtCAAGGcTTCTACACCCGTACTCGAATGC&#39;</span><span class="p">,</span>
<span class="gp">... </span>                  <span class="n">shift</span><span class="o">=</span><span class="mi">1</span><span class="p">))</span>
<span class="gt">Traceback (most recent call last):</span>
<span class="gr">ValueError</span>: <span class="n">no thermodynamic data for neighbors &#39;.C/TT&#39; available</span>
</pre></div>
</div>
<p>Make your own tables, or update/extend existing tables. E.g., add values for
locked nucleotides. Here, ‘locked A’ (and its complement) should be represented
by ‘1’:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">mytable</span> <span class="o">=</span> <span class="n">mt</span><span class="o">.</span><span class="n">make_table</span><span class="p">(</span><span class="n">oldtable</span><span class="o">=</span><span class="n">mt</span><span class="o">.</span><span class="n">DNA_NN3</span><span class="p">,</span>
<span class="gp">... </span>                        <span class="n">values</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;A1/T1&#39;</span><span class="p">:(</span><span class="o">-</span><span class="mf">6.608</span><span class="p">,</span> <span class="o">-</span><span class="mf">17.235</span><span class="p">),</span>
<span class="gp">... </span>                        <span class="s1">&#39;1A/1T&#39;</span><span class="p">:(</span><span class="o">-</span><span class="mf">6.893</span><span class="p">,</span> <span class="o">-</span><span class="mf">15.923</span><span class="p">)})</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">Tm_NN</span><span class="p">(</span><span class="s1">&#39;CGTTCCAAAGATGTGGGCATGAGCTTAC&#39;</span><span class="p">))</span>
<span class="go">60.32</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">Tm_NN</span><span class="p">(</span><span class="s1">&#39;CGTTCCA1AGATGTGGGCATGAGCTTAC&#39;</span><span class="p">,</span>
<span class="gp">... </span>                          <span class="n">nn_table</span><span class="o">=</span><span class="n">mytable</span><span class="p">,</span> <span class="n">check</span><span class="o">=</span><span class="kc">False</span><span class="p">))</span>
<span class="gp">... </span><span class="c1"># &#39;check&#39; must be False, otherwise &#39;1&#39; would be discarded</span>
<span class="go">62.53</span>
</pre></div>
</div>
<dl class="function">
<dt id="Bio.SeqUtils.MeltingTemp.Tm_GC">
<code class="sig-prename descclassname">Bio.SeqUtils.MeltingTemp.</code><code class="sig-name descname">Tm_GC</code><span class="sig-paren">(</span><em class="sig-param">seq</em>, <em class="sig-param">check=True</em>, <em class="sig-param">strict=True</em>, <em class="sig-param">valueset=7</em>, <em class="sig-param">userset=None</em>, <em class="sig-param">Na=50</em>, <em class="sig-param">K=0</em>, <em class="sig-param">Tris=0</em>, <em class="sig-param">Mg=0</em>, <em class="sig-param">dNTPs=0</em>, <em class="sig-param">saltcorr=0</em>, <em class="sig-param">mismatch=True</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.SeqUtils.MeltingTemp.Tm_GC" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the Tm using empirical formulas based on GC content.</p>
<p>General format: Tm = A + B(%GC) - C/N + salt correction - D(%mismatch)</p>
<p>A, B, C, D: empirical constants, N: primer length
D (amount of decrease in Tm per % mismatch) is often 1, but sometimes other
values have been used (0.6-1.5). Use ‘X’ to indicate the mismatch position
in the sequence. Note that this mismatch correction is a rough estimate.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.SeqUtils</span> <span class="k">import</span> <span class="n">MeltingTemp</span> <span class="k">as</span> <span class="n">mt</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s2">&quot;</span><span class="si">%0.2f</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">Tm_GC</span><span class="p">(</span><span class="s1">&#39;CTGCTGATXGCACGAGGTTATGG&#39;</span><span class="p">,</span> <span class="n">valueset</span><span class="o">=</span><span class="mi">2</span><span class="p">))</span>
<span class="go">69.20</span>
</pre></div>
</div>
<dl>
<dt>Arguments:</dt><dd><ul>
<li><p>valueset: A few often cited variants are included:</p>
<blockquote>
<div><ol class="arabic simple">
<li><p>Tm = 69.3 + 0.41(%GC) - 650/N
(Marmur &amp; Doty 1962, J Mol Biol 5: 109-118; Chester &amp; Marshak 1993),
Anal Biochem 209: 284-290)</p></li>
<li><p>Tm = 81.5 + 0.41(%GC) - 675/N - %mismatch
‘QuikChange’ formula. Recommended (by the manufacturer) for the
design of primers for QuikChange mutagenesis.</p></li>
<li><p>Tm = 81.5 + 0.41(%GC) - 675/N + 16.6 x log[Na+]
(Marmur &amp; Doty 1962, J Mol Biol 5: 109-118; Schildkraut &amp; Lifson
1965, Biopolymers 3: 195-208)</p></li>
<li><p>Tm = 81.5 + 0.41(%GC) - 500/N + 16.6 x log([Na+]/(1.0 + 0.7 x
[Na+])) - %mismatch
(Wetmur 1991, Crit Rev Biochem Mol Biol 126: 227-259). This is the
standard formula in approximative mode of MELTING 4.3.</p></li>
<li><p>Tm = 78 + 0.7(%GC) - 500/N + 16.6 x log([Na+]/(1.0 + 0.7 x [Na+]))
- %mismatch
(Wetmur 1991, Crit Rev Biochem Mol Biol 126: 227-259). For RNA.</p></li>
<li><p>Tm = 67 + 0.8(%GC) - 500/N + 16.6 x log([Na+]/(1.0 + 0.7 x [Na+]))
- %mismatch
(Wetmur 1991, Crit Rev Biochem Mol Biol 126: 227-259). For RNA/DNA
hybrids.</p></li>
<li><p>Tm = 81.5 + 0.41(%GC) - 600/N + 16.6 x log[Na+]
Used by Primer3Plus to calculate the product Tm. Default set.</p></li>
<li><p>Tm = 77.1 + 0.41(%GC) - 528/N + 11.7 x log[Na+]
(von Ahsen et al. 2001, Clin Chem 47: 1956-1961). Recommended ‘as a
tradeoff between accuracy and ease of use’.</p></li>
</ol>
</div></blockquote>
</li>
<li><p>userset: Tuple of four values for A, B, C, and D. Usersets override
valuesets.</p></li>
<li><p>Na, K, Tris, Mg, dNTPs: Concentration of the respective ions [mM]. If
any of K, Tris, Mg and dNTPS is non-zero, a ‘sodium-equivalent’
concentration is calculated and used for salt correction (von Ahsen et
al., 2001).</p></li>
<li><p>saltcorr: Type of salt correction (see method salt_correction).
Default=5. 0 or None means no salt correction.</p></li>
<li><p>mismatch: If ‘True’ (default) every ‘X’ in the sequence is counted as
mismatch.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="Bio.SeqUtils.MeltingTemp.Tm_NN">
<code class="sig-prename descclassname">Bio.SeqUtils.MeltingTemp.</code><code class="sig-name descname">Tm_NN</code><span class="sig-paren">(</span><em class="sig-param">seq</em>, <em class="sig-param">check=True</em>, <em class="sig-param">strict=True</em>, <em class="sig-param">c_seq=None</em>, <em class="sig-param">shift=0</em>, <em class="sig-param">nn_table={'init': (0</em>, <em class="sig-param">0)</em>, <em class="sig-param">'init_A/T': (2.3</em>, <em class="sig-param">4.1)</em>, <em class="sig-param">'init_G/C': (0.1</em>, <em class="sig-param">-2.8)</em>, <em class="sig-param">'init_oneG/C': (0</em>, <em class="sig-param">0)</em>, <em class="sig-param">'init_allA/T': (0</em>, <em class="sig-param">0)</em>, <em class="sig-param">'init_5T/A': (0</em>, <em class="sig-param">0)</em>, <em class="sig-param">'sym': (0</em>, <em class="sig-param">-1.4)</em>, <em class="sig-param">'AA/TT': (-7.9</em>, <em class="sig-param">-22.2)</em>, <em class="sig-param">'AT/TA': (-7.2</em>, <em class="sig-param">-20.4)</em>, <em class="sig-param">'TA/AT': (-7.2</em>, <em class="sig-param">-21.3)</em>, <em class="sig-param">'CA/GT': (-8.5</em>, <em class="sig-param">-22.7)</em>, <em class="sig-param">'GT/CA': (-8.4</em>, <em class="sig-param">-22.4)</em>, <em class="sig-param">'CT/GA': (-7.8</em>, <em class="sig-param">-21.0)</em>, <em class="sig-param">'GA/CT': (-8.2</em>, <em class="sig-param">-22.2)</em>, <em class="sig-param">'CG/GC': (-10.6</em>, <em class="sig-param">-27.2)</em>, <em class="sig-param">'GC/CG': (-9.8</em>, <em class="sig-param">-24.4)</em>, <em class="sig-param">'GG/CC': (-8.0</em>, <em class="sig-param">-19.9)}</em>, <em class="sig-param">tmm_table={'AA/TA': (-3.1</em>, <em class="sig-param">-7.8)</em>, <em class="sig-param">'TA/AA': (-2.5</em>, <em class="sig-param">-6.3)</em>, <em class="sig-param">'CA/GA': (-4.3</em>, <em class="sig-param">-10.7)</em>, <em class="sig-param">'GA/CA': (-8.0</em>, <em class="sig-param">-22.5)</em>, <em class="sig-param">'AC/TC': (-0.1</em>, <em class="sig-param">0.5)</em>, <em class="sig-param">'TC/AC': (-0.7</em>, <em class="sig-param">-1.3)</em>, <em class="sig-param">' CC/GC': (-2.1</em>, <em class="sig-param">-5.1)</em>, <em class="sig-param">'GC/CC': (-3.9</em>, <em class="sig-param">-10.6)</em>, <em class="sig-param">'AG/TG': (-1.1</em>, <em class="sig-param">-2.1)</em>, <em class="sig-param">'TG/AG': (-1.1</em>, <em class="sig-param">-2.7)</em>, <em class="sig-param">'CG/GG': (-3.8</em>, <em class="sig-param">-9.5)</em>, <em class="sig-param">'GG/CG': (-0.7</em>, <em class="sig-param">-19.2)</em>, <em class="sig-param">'AT/TT': (-2.4</em>, <em class="sig-param">-6.5)</em>, <em class="sig-param">'TT/AT': (-3.2</em>, <em class="sig-param">-8.9)</em>, <em class="sig-param">'CT/GT': (-6.1</em>, <em class="sig-param">-16.9)</em>, <em class="sig-param">'GT/CT': (-7.4</em>, <em class="sig-param">-21.2)</em>, <em class="sig-param">'AA/TC': (-1.6</em>, <em class="sig-param">-4.0)</em>, <em class="sig-param">'AC/TA': (-1.8</em>, <em class="sig-param">-3.8)</em>, <em class="sig-param">'CA/GC': (-2.6</em>, <em class="sig-param">-5.9)</em>, <em class="sig-param">'CC/GA': (-2.7</em>, <em class="sig-param">-6.0)</em>, <em class="sig-param">'GA/CC': (-5.0</em>, <em class="sig-param">-13.8)</em>, <em class="sig-param">'GC/CA': (-3.2</em>, <em class="sig-param">-7.1)</em>, <em class="sig-param">'TA/AC': (-2.3</em>, <em class="sig-param">-5.9)</em>, <em class="sig-param">'TC/AA': (-2.7</em>, <em class="sig-param">-7.0)</em>, <em class="sig-param">'AC/TT': (-0.9</em>, <em class="sig-param">-1.7)</em>, <em class="sig-param">'AT/TC': (-2.3</em>, <em class="sig-param">-6.3)</em>, <em class="sig-param">'CC/GT': (-3.2</em>, <em class="sig-param">-8.0)</em>, <em class="sig-param">'CT/GC': (-3.9</em>, <em class="sig-param">-10.6)</em>, <em class="sig-param">'GC/CT': (-4.9</em>, <em class="sig-param">-13.5)</em>, <em class="sig-param">'GT/CC': (-3.0</em>, <em class="sig-param">-7.8)</em>, <em class="sig-param">'TC/AT': (-2.5</em>, <em class="sig-param">-6.3)</em>, <em class="sig-param">'TT/AC': (-0.7</em>, <em class="sig-param">-1.2)</em>, <em class="sig-param">'AA/TG': (-1.9</em>, <em class="sig-param">-4.4)</em>, <em class="sig-param">'AG/TA': (-2.5</em>, <em class="sig-param">-5.9)</em>, <em class="sig-param">'CA/GG': (-3.9</em>, <em class="sig-param">-9.6)</em>, <em class="sig-param">'CG/GA': (-6.0</em>, <em class="sig-param">-15.5)</em>, <em class="sig-param">'GA/CG': (-4.3</em>, <em class="sig-param">-11.1)</em>, <em class="sig-param">' GG/CA': (-4.6</em>, <em class="sig-param">-11.4)</em>, <em class="sig-param">'TA/AG': (-2.0</em>, <em class="sig-param">-4.7)</em>, <em class="sig-param">'TG/AA': (-2.4</em>, <em class="sig-param">-5.8)</em>, <em class="sig-param">'AG/TT': (-3.2</em>, <em class="sig-param">-8.7)</em>, <em class="sig-param">'AT/TG': (-3.5</em>, <em class="sig-param">-9.4)</em>, <em class="sig-param">'CG/GT': (-3.8</em>, <em class="sig-param">-9.0)</em>, <em class="sig-param">'CT/GG': (-6.6</em>, <em class="sig-param">-18.7)</em>, <em class="sig-param">'GG/CT': (-5.7</em>, <em class="sig-param">-15.9)</em>, <em class="sig-param">'GT/CG': (-5.9</em>, <em class="sig-param">-16.1)</em>, <em class="sig-param">'TG/AT': (-3.9</em>, <em class="sig-param">-10.5)</em>, <em class="sig-param">'TT/AG': (-3.6</em>, <em class="sig-param">-9.8)}</em>, <em class="sig-param">imm_table={'AG/TT': (1.0</em>, <em class="sig-param">0.9)</em>, <em class="sig-param">'AT/TG': (-2.5</em>, <em class="sig-param">-8.3)</em>, <em class="sig-param">'CG/GT': (-4.1</em>, <em class="sig-param">-11.7)</em>, <em class="sig-param">'CT/GG': (-2.8</em>, <em class="sig-param">-8.0)</em>, <em class="sig-param">'GG/CT': (3.3</em>, <em class="sig-param">10.4)</em>, <em class="sig-param">'GG/TT': (5.8</em>, <em class="sig-param">16.3)</em>, <em class="sig-param">'GT/CG': (-4.4</em>, <em class="sig-param">-12.3)</em>, <em class="sig-param">'GT/TG': (4.1</em>, <em class="sig-param">9.5)</em>, <em class="sig-param">'TG/AT': (-0.1</em>, <em class="sig-param">-1.7)</em>, <em class="sig-param">'TG/GT': (-1.4</em>, <em class="sig-param">-6.2)</em>, <em class="sig-param">'TT/AG': (-1.3</em>, <em class="sig-param">-5.3)</em>, <em class="sig-param">'AA/TG': (-0.6</em>, <em class="sig-param">-2.3)</em>, <em class="sig-param">'AG/TA': (-0.7</em>, <em class="sig-param">-2.3)</em>, <em class="sig-param">'CA/GG': (-0.7</em>, <em class="sig-param">-2.3)</em>, <em class="sig-param">'CG/GA': (-4.0</em>, <em class="sig-param">-13.2)</em>, <em class="sig-param">'GA/CG': (-0.6</em>, <em class="sig-param">-1.0)</em>, <em class="sig-param">'GG/CA': (0.5</em>, <em class="sig-param">3.2)</em>, <em class="sig-param">'TA/AG': (0.7</em>, <em class="sig-param">0.7)</em>, <em class="sig-param">'TG/AA': (3.0</em>, <em class="sig-param">7.4)</em>, <em class="sig-param">'AC/TT': (0.7</em>, <em class="sig-param">0.2)</em>, <em class="sig-param">'AT/TC': (-1.2</em>, <em class="sig-param">-6.2)</em>, <em class="sig-param">'CC/GT': (-0.8</em>, <em class="sig-param">-4.5)</em>, <em class="sig-param">'CT/GC': (-1.5</em>, <em class="sig-param">-6.1)</em>, <em class="sig-param">'GC/CT': (2.3</em>, <em class="sig-param">5.4)</em>, <em class="sig-param">'GT/CC': (5.2</em>, <em class="sig-param">13.5)</em>, <em class="sig-param">'TC/AT': (1.2</em>, <em class="sig-param">0.7)</em>, <em class="sig-param">'TT/AC': (1.0</em>, <em class="sig-param">0.7)</em>, <em class="sig-param">'AA/TC': (2.3</em>, <em class="sig-param">4.6)</em>, <em class="sig-param">'AC/TA': (5.3</em>, <em class="sig-param">14.6)</em>, <em class="sig-param">'CA/GC': (1.9</em>, <em class="sig-param">3.7)</em>, <em class="sig-param">'CC/GA': (0.6</em>, <em class="sig-param">-0.6)</em>, <em class="sig-param">'GA/CC': (5.2</em>, <em class="sig-param">14.2)</em>, <em class="sig-param">'GC/CA': (-0.7</em>, <em class="sig-param">-3.8)</em>, <em class="sig-param">'TA/AC': (3.4</em>, <em class="sig-param">8.0)</em>, <em class="sig-param">'TC/AA': (7.6</em>, <em class="sig-param">20.2)</em>, <em class="sig-param">'AA/TA': (1.2</em>, <em class="sig-param">1.7)</em>, <em class="sig-param">'CA/GA': (-0.9</em>, <em class="sig-param">-4.2)</em>, <em class="sig-param">'GA/CA': (-2.9</em>, <em class="sig-param">-9.8)</em>, <em class="sig-param">'TA/AA': (4.7</em>, <em class="sig-param">12.9)</em>, <em class="sig-param">'AC/TC': (0.0</em>, <em class="sig-param">-4.4)</em>, <em class="sig-param">'CC/GC': (-1.5</em>, <em class="sig-param">-7.2)</em>, <em class="sig-param">'GC/CC': (3.6</em>, <em class="sig-param">8.9)</em>, <em class="sig-param">'TC/AC': (6.1</em>, <em class="sig-param">16.4)</em>, <em class="sig-param">'AG/TG': (-3.1</em>, <em class="sig-param">-9.5)</em>, <em class="sig-param">'CG/GG': (-4.9</em>, <em class="sig-param">-15.3)</em>, <em class="sig-param">'GG/CG': (-6.0</em>, <em class="sig-param">-15.8)</em>, <em class="sig-param">'TG/AG': (1.6</em>, <em class="sig-param">3.6)</em>, <em class="sig-param">'AT/TT': (-2.7</em>, <em class="sig-param">-10.8)</em>, <em class="sig-param">'CT/GT': (-5.0</em>, <em class="sig-param">-15.8)</em>, <em class="sig-param">'GT/CT': (-2.2</em>, <em class="sig-param">-8.4)</em>, <em class="sig-param">'TT/AT': (0.2</em>, <em class="sig-param">-1.5)</em>, <em class="sig-param">'AI/TC': (-8.9</em>, <em class="sig-param">-25.5)</em>, <em class="sig-param">'TI/AC': (-5.9</em>, <em class="sig-param">-17.4)</em>, <em class="sig-param">'AC/TI': (-8.8</em>, <em class="sig-param">-25.4)</em>, <em class="sig-param">'TC/AI': (-4.9</em>, <em class="sig-param">-13.9)</em>, <em class="sig-param">'CI/GC': (-5.4</em>, <em class="sig-param">-13.7)</em>, <em class="sig-param">'GI/CC': (-6.8</em>, <em class="sig-param">-19.1)</em>, <em class="sig-param">'CC/GI': (-8.3</em>, <em class="sig-param">-23.8)</em>, <em class="sig-param">'GC/CI': (-5.0</em>, <em class="sig-param">-12.6)</em>, <em class="sig-param">'AI/TA': (-8.3</em>, <em class="sig-param">-25.0)</em>, <em class="sig-param">'TI/AA': (-3.4</em>, <em class="sig-param">-11.2)</em>, <em class="sig-param">'AA/TI': (-0.7</em>, <em class="sig-param">-2.6)</em>, <em class="sig-param">'TA/AI': (-1.3</em>, <em class="sig-param">-4.6)</em>, <em class="sig-param">'CI/GA': (2.6</em>, <em class="sig-param">8.9)</em>, <em class="sig-param">'GI/CA': (-7.8</em>, <em class="sig-param">-21.1)</em>, <em class="sig-param">'CA/GI': (-7.0</em>, <em class="sig-param">-20.0)</em>, <em class="sig-param">'GA/CI': (-7.6</em>, <em class="sig-param">-20.2)</em>, <em class="sig-param">'AI/TT': (0.49</em>, <em class="sig-param">-0.7)</em>, <em class="sig-param">'TI/AT': (-6.5</em>, <em class="sig-param">-22.0)</em>, <em class="sig-param">'AT/TI': (-5.6</em>, <em class="sig-param">-18.7)</em>, <em class="sig-param">'TT/AI': (-0.8</em>, <em class="sig-param">-4.3)</em>, <em class="sig-param">'CI/GT': (-1.0</em>, <em class="sig-param">-2.4)</em>, <em class="sig-param">'GI/CT': (-3.5</em>, <em class="sig-param">-10.6)</em>, <em class="sig-param">'CT/GI': (0.1</em>, <em class="sig-param">-1.0)</em>, <em class="sig-param">'GT/CI': (-4.3</em>, <em class="sig-param">-12.1)</em>, <em class="sig-param">'AI/TG': (-4.9</em>, <em class="sig-param">-15.8)</em>, <em class="sig-param">'TI/AG': (-1.9</em>, <em class="sig-param">-8.5)</em>, <em class="sig-param">'AG/TI': (0.1</em>, <em class="sig-param">-1.8)</em>, <em class="sig-param">'TG/AI': (1.0</em>, <em class="sig-param">1.0)</em>, <em class="sig-param">'CI/GG': (7.1</em>, <em class="sig-param">21.3)</em>, <em class="sig-param">'GI/CG': (-1.1</em>, <em class="sig-param">-3.2)</em>, <em class="sig-param">'CG/GI': (5.8</em>, <em class="sig-param">16.9)</em>, <em class="sig-param">'GG/CI': (-7.6</em>, <em class="sig-param">-22.0)</em>, <em class="sig-param">'AI/TI': (-3.3</em>, <em class="sig-param">-11.9)</em>, <em class="sig-param">'TI/AI': (0.1</em>, <em class="sig-param">-2.3)</em>, <em class="sig-param">'CI/GI': (1.3</em>, <em class="sig-param">3.0)</em>, <em class="sig-param">'GI/CI': (-0.5</em>, <em class="sig-param">-1.3)}</em>, <em class="sig-param">de_table={'AA/.T': (0.2</em>, <em class="sig-param">2.3)</em>, <em class="sig-param">'AC/.G': (-6.3</em>, <em class="sig-param">-17.1)</em>, <em class="sig-param">'AG/.C': (-3.7</em>, <em class="sig-param">-10.0)</em>, <em class="sig-param">'AT/.A': (-2.9</em>, <em class="sig-param">-7.6)</em>, <em class="sig-param">'CA/.T': (0.6</em>, <em class="sig-param">3.3)</em>, <em class="sig-param">'CC/.G': (-4.4</em>, <em class="sig-param">-12.6)</em>, <em class="sig-param">'CG/.C': (-4.0</em>, <em class="sig-param">-11.9)</em>, <em class="sig-param">'CT/.A': (-4.1</em>, <em class="sig-param">-13.0)</em>, <em class="sig-param">'GA/.T': (-1.1</em>, <em class="sig-param">-1.6)</em>, <em class="sig-param">'GC/.G': (-5.1</em>, <em class="sig-param">-14.0)</em>, <em class="sig-param">'GG/.C': (-3.9</em>, <em class="sig-param">-10.9)</em>, <em class="sig-param">'GT/.A': (-4.2</em>, <em class="sig-param">-15.0)</em>, <em class="sig-param">'TA/.T': (-6.9</em>, <em class="sig-param">-20.0)</em>, <em class="sig-param">'TC/.G': (-4.0</em>, <em class="sig-param">-10.9)</em>, <em class="sig-param">'TG/.C': (-4.9</em>, <em class="sig-param">-13.8)</em>, <em class="sig-param">'TT/.A': (-0.2</em>, <em class="sig-param">-0.5)</em>, <em class="sig-param">'.A/AT': (-0.7</em>, <em class="sig-param">-0.8)</em>, <em class="sig-param">'.C/AG': (-2.1</em>, <em class="sig-param">-3.9)</em>, <em class="sig-param">'.G/AC': (-5.9</em>, <em class="sig-param">-16.5)</em>, <em class="sig-param">'.T/AA': (-0.5</em>, <em class="sig-param">-1.1)</em>, <em class="sig-param">'.A/CT': (4.4</em>, <em class="sig-param">14.9)</em>, <em class="sig-param">'.C/CG': (-0.2</em>, <em class="sig-param">-0.1)</em>, <em class="sig-param">'.G/CC': (-2.6</em>, <em class="sig-param">-7.4)</em>, <em class="sig-param">'.T/CA': (4.7</em>, <em class="sig-param">14.2)</em>, <em class="sig-param">'.A/GT': (-1.6</em>, <em class="sig-param">-3.6)</em>, <em class="sig-param">'.C/GG': (-3.9</em>, <em class="sig-param">-11.2)</em>, <em class="sig-param">'.G/GC': (-3.2</em>, <em class="sig-param">-10.4)</em>, <em class="sig-param">'.T/GA': (-4.1</em>, <em class="sig-param">-13.1)</em>, <em class="sig-param">'.A/TT': (2.9</em>, <em class="sig-param">10.4)</em>, <em class="sig-param">'.C/TG': (-4.4</em>, <em class="sig-param">-13.1)</em>, <em class="sig-param">'.G/TC': (-5.2</em>, <em class="sig-param">-15.0)</em>, <em class="sig-param">'.T/TA': (-3.8</em>, <em class="sig-param">-12.6)}</em>, <em class="sig-param">dnac1=25</em>, <em class="sig-param">dnac2=25</em>, <em class="sig-param">selfcomp=False</em>, <em class="sig-param">Na=50</em>, <em class="sig-param">K=0</em>, <em class="sig-param">Tris=0</em>, <em class="sig-param">Mg=0</em>, <em class="sig-param">dNTPs=0</em>, <em class="sig-param">saltcorr=5</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.SeqUtils.MeltingTemp.Tm_NN" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the Tm using nearest neighbor thermodynamics.</p>
<dl>
<dt>Arguments:</dt><dd><ul>
<li><p>seq: The primer/probe sequence as string or Biopython sequence object.
For RNA/DNA hybridizations seq must be the RNA sequence.</p></li>
<li><p>c_seq: Complementary sequence. The sequence of the template/target in
3’-&gt;5’ direction. c_seq is necessary for mismatch correction and
dangling-ends correction. Both corrections will automatically be
applied if mismatches or dangling ends are present. Default=None.</p></li>
<li><p>shift: Shift of the primer/probe sequence on the template/target
sequence, e.g.:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>                   <span class="n">shift</span><span class="o">=</span><span class="mi">0</span>       <span class="n">shift</span><span class="o">=</span><span class="mi">1</span>        <span class="n">shift</span><span class="o">=</span> <span class="o">-</span><span class="mi">1</span>
<span class="n">Primer</span> <span class="p">(</span><span class="n">seq</span><span class="p">):</span>      <span class="mi">5</span><span class="s1">&#39; ATGC...    5&#39;</span>  <span class="n">ATGC</span><span class="o">...</span>    <span class="mi">5</span><span class="s1">&#39; ATGC...</span>
<span class="n">Template</span> <span class="p">(</span><span class="n">c_seq</span><span class="p">):</span>  <span class="mi">3</span><span class="s1">&#39; TACG...    3&#39;</span> <span class="n">CTACG</span><span class="o">...</span>    <span class="mi">3</span><span class="s1">&#39;  ACG...</span>
</pre></div>
</div>
<p>The shift parameter is necessary to align seq and c_seq if they have
different lengths or if they should have dangling ends. Default=0</p>
</li>
<li><p>table: Thermodynamic NN values, eight tables are implemented:
For DNA/DNA hybridizations:</p>
<blockquote>
<div><ul class="simple">
<li><p>DNA_NN1: values from Breslauer et al. (1986)</p></li>
<li><p>DNA_NN2: values from Sugimoto et al. (1996)</p></li>
<li><p>DNA_NN3: values from Allawi &amp; SantaLucia (1997) (default)</p></li>
<li><p>DNA_NN4: values from SantaLucia &amp; Hicks (2004)</p></li>
</ul>
</div></blockquote>
<p>For RNA/RNA hybridizations:</p>
<blockquote>
<div><ul class="simple">
<li><p>RNA_NN1: values from Freier et al. (1986)</p></li>
<li><p>RNA_NN2: values from Xia et al. (1998)</p></li>
<li><p>RNA_NN3: valuse from Chen et al. (2012)</p></li>
</ul>
</div></blockquote>
<p>For RNA/DNA hybridizations:</p>
<blockquote>
<div><ul class="simple">
<li><p>R_DNA_NN1: values from Sugimoto et al. (1995)
Note that <code class="docutils literal notranslate"><span class="pre">seq</span></code> must be the RNA sequence.</p></li>
</ul>
</div></blockquote>
<p>Use the module’s maketable method to make a new table or to update one
one of the implemented tables.</p>
</li>
<li><p>tmm_table: Thermodynamic values for terminal mismatches.
Default: DNA_TMM1 (SantaLucia &amp; Peyret, 2001)</p></li>
<li><p>imm_table: Thermodynamic values for internal mismatches, may include
insosine mismatches. Default: DNA_IMM1 (Allawi &amp; SantaLucia, 1997-1998;
Peyret et al., 1999; Watkins &amp; SantaLucia, 2005)</p></li>
<li><p>de_table: Thermodynamic values for dangling ends:</p>
<blockquote>
<div><ul class="simple">
<li><p>DNA_DE1: for DNA. Values from Bommarito et al. (2000). Default</p></li>
<li><p>RNA_DE1: for RNA. Values from Turner &amp; Mathews (2010)</p></li>
</ul>
</div></blockquote>
</li>
<li><p>dnac1: Concentration of the higher concentrated strand [nM]. Typically
this will be the primer (for PCR) or the probe. Default=25.</p></li>
<li><p>dnac2: Concentration of the lower concentrated strand [nM]. In PCR this
is the template strand which concentration is typically very low and may
be ignored (dnac2=0). In oligo/oligo hybridization experiments, dnac1
equals dnac1. Default=25.
MELTING and Primer3Plus use k = [Oligo(Total)]/4 by default. To mimic
this behaviour, you have to divide [Oligo(Total)] by 2 and assign this
concentration to dnac1 and dnac2. E.g., Total oligo concentration of
50 nM in Primer3Plus means dnac1=25, dnac2=25.</p></li>
<li><p>selfcomp: Is the sequence self-complementary? Default=False. If ‘True’
the primer is thought binding to itself, thus dnac2 is not considered.</p></li>
<li><p>Na, K, Tris, Mg, dNTPs: See method ‘Tm_GC’ for details. Defaults: Na=50,
K=0, Tris=0, Mg=0, dNTPs=0.</p></li>
<li><p>saltcorr: See method ‘Tm_GC’. Default=5. 0 means no salt correction.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="Bio.SeqUtils.MeltingTemp.Tm_Wallace">
<code class="sig-prename descclassname">Bio.SeqUtils.MeltingTemp.</code><code class="sig-name descname">Tm_Wallace</code><span class="sig-paren">(</span><em class="sig-param">seq</em>, <em class="sig-param">check=True</em>, <em class="sig-param">strict=True</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.SeqUtils.MeltingTemp.Tm_Wallace" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate and return the Tm using the ‘Wallace rule’.</p>
<p>Tm = 4 degC * (G + C) + 2 degC * (A+T)</p>
<p>The Wallace rule (Thein &amp; Wallace 1986, in Human genetic diseases: a
practical approach, 33-50) is often used as rule of thumb for approximate
Tm calculations for primers of 14 to 20 nt length.</p>
<p>Non-DNA characters (e.g., E, F, J, !, 1, etc) are ignored by this method.</p>
<dl>
<dt>Examples:</dt><dd><div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.SeqUtils</span> <span class="k">import</span> <span class="n">MeltingTemp</span> <span class="k">as</span> <span class="n">mt</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mt</span><span class="o">.</span><span class="n">Tm_Wallace</span><span class="p">(</span><span class="s1">&#39;ACGTTGCAATGCCGTA&#39;</span><span class="p">)</span>
<span class="go">48.0</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mt</span><span class="o">.</span><span class="n">Tm_Wallace</span><span class="p">(</span><span class="s1">&#39;ACGT TGCA ATGC CGTA&#39;</span><span class="p">)</span>
<span class="go">48.0</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mt</span><span class="o">.</span><span class="n">Tm_Wallace</span><span class="p">(</span><span class="s1">&#39;1ACGT2TGCA3ATGC4CGTA&#39;</span><span class="p">)</span>
<span class="go">48.0</span>
</pre></div>
</div>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="Bio.SeqUtils.MeltingTemp.Tm_staluc">
<code class="sig-prename descclassname">Bio.SeqUtils.MeltingTemp.</code><code class="sig-name descname">Tm_staluc</code><span class="sig-paren">(</span><em class="sig-param">s</em>, <em class="sig-param">dnac=50</em>, <em class="sig-param">saltc=50</em>, <em class="sig-param">rna=0</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.SeqUtils.MeltingTemp.Tm_staluc" title="Permalink to this definition">¶</a></dt>
<dd><p>Return DNA/DNA Tm using nearest neighbor thermodynamics (OBSOLETE).</p>
<p>This method may be depreceated in the future. Use Tm_NN instead. Tm_NN
with default values gives the same result as Tm_staluc.</p>
<p>s is the sequence as string or Seq object
dnac is DNA concentration [nM]
saltc is salt concentration [mM].
rna=0 is for DNA/DNA (default), use 1 for RNA/RNA hybridisation.</p>
<p>For DNA/DNA, see Allawi &amp; SantaLucia (1997), Biochemistry 36: 10581-10594
For RNA/RNA, see Xia et al (1998), Biochemistry 37: 14719-14735</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s2">&quot;</span><span class="si">%0.2f</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="n">Tm_staluc</span><span class="p">(</span><span class="s1">&#39;CAGTCAGTACGTACGTGTACTGCCGTA&#39;</span><span class="p">))</span>
<span class="go">59.87</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s2">&quot;</span><span class="si">%0.2f</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="n">Tm_staluc</span><span class="p">(</span><span class="s1">&#39;CAGTCAGTACGTACGTGTACTGCCGTA&#39;</span><span class="p">,</span> <span class="n">rna</span><span class="o">=</span><span class="kc">True</span><span class="p">))</span>
<span class="go">77.90</span>
</pre></div>
</div>
<p>You can also use a Seq object instead of a string,</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="k">import</span> <span class="n">generic_nucleotide</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">s</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s1">&#39;CAGTCAGTACGTACGTGTACTGCCGTA&#39;</span><span class="p">,</span> <span class="n">generic_nucleotide</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s2">&quot;</span><span class="si">%0.2f</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="n">Tm_staluc</span><span class="p">(</span><span class="n">s</span><span class="p">))</span>
<span class="go">59.87</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s2">&quot;</span><span class="si">%0.2f</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="n">Tm_staluc</span><span class="p">(</span><span class="n">s</span><span class="p">,</span> <span class="n">rna</span><span class="o">=</span><span class="kc">True</span><span class="p">))</span>
<span class="go">77.90</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Bio.SeqUtils.MeltingTemp.chem_correction">
<code class="sig-prename descclassname">Bio.SeqUtils.MeltingTemp.</code><code class="sig-name descname">chem_correction</code><span class="sig-paren">(</span><em class="sig-param">melting_temp</em>, <em class="sig-param">DMSO=0</em>, <em class="sig-param">fmd=0</em>, <em class="sig-param">DMSOfactor=0.75</em>, <em class="sig-param">fmdfactor=0.65</em>, <em class="sig-param">fmdmethod=1</em>, <em class="sig-param">GC=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.SeqUtils.MeltingTemp.chem_correction" title="Permalink to this definition">¶</a></dt>
<dd><p>Correct a given Tm for DMSO and formamide.</p>
<p>Please note that these corrections are +/- rough approximations.</p>
<dl>
<dt>Arguments:</dt><dd><ul>
<li><p>melting_temp: Melting temperature.</p></li>
<li><p>DMSO: Percent DMSO.</p></li>
<li><p>fmd: Formamide concentration in %(fmdmethod=1) or molar (fmdmethod=2).</p></li>
<li><p>DMSOfactor: How much should Tm decreases per percent DMSO. Default=0.65
(von Ahsen et al. 2001). Other published values are 0.5, 0.6 and 0.675.</p></li>
<li><p>fmdfactor: How much should Tm decrease per percent formamide.
Default=0.65. Several papers report factors between 0.6 and 0.72.</p></li>
<li><p>fmdmethod:</p>
<blockquote>
<div><ol class="arabic simple">
<li><p>Tm = Tm - factor(%formamide) (Default)</p></li>
<li><p>Tm = Tm + (0.453(f(GC)) - 2.88) x [formamide]</p></li>
</ol>
</div></blockquote>
<p>Here f(GC) is fraction of GC.
Note (again) that in fmdmethod=1 formamide concentration is given in %,
while in fmdmethod=2 it is given in molar.</p>
</li>
<li><p>GC: GC content in percent.</p></li>
</ul>
</dd>
<dt>Examples:</dt><dd><div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.SeqUtils</span> <span class="k">import</span> <span class="n">MeltingTemp</span> <span class="k">as</span> <span class="n">mt</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">mt</span><span class="o">.</span><span class="n">chem_correction</span><span class="p">(</span><span class="mi">70</span><span class="p">)</span>
<span class="go">70</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">chem_correction</span><span class="p">(</span><span class="mi">70</span><span class="p">,</span> <span class="n">DMSO</span><span class="o">=</span><span class="mi">3</span><span class="p">))</span>
<span class="go">67.75</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">chem_correction</span><span class="p">(</span><span class="mi">70</span><span class="p">,</span> <span class="n">fmd</span><span class="o">=</span><span class="mi">5</span><span class="p">))</span>
<span class="go">66.75</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">chem_correction</span><span class="p">(</span><span class="mi">70</span><span class="p">,</span> <span class="n">fmdmethod</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">fmd</span><span class="o">=</span><span class="mf">1.25</span><span class="p">,</span>
<span class="gp">... </span>                                   <span class="n">GC</span><span class="o">=</span><span class="mi">50</span><span class="p">))</span>
<span class="go">66.68</span>
</pre></div>
</div>
</dd>
</dl>
</dd></dl>

<dl class="function">
<dt id="Bio.SeqUtils.MeltingTemp.make_table">
<code class="sig-prename descclassname">Bio.SeqUtils.MeltingTemp.</code><code class="sig-name descname">make_table</code><span class="sig-paren">(</span><em class="sig-param">oldtable=None</em>, <em class="sig-param">values=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.SeqUtils.MeltingTemp.make_table" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a table with thermodynamic parameters (as dictionary).</p>
<p>Parameters:
oldtable: An existing dictionary with thermodynamic parameters.
values: A dictionary with new or updated values.</p>
<p>E.g., to replace the initiation parameters in the Sugimoto ‘96 dataset with
the initiation parameters from Allawi &amp; SantaLucia ‘97:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.SeqUtils.MeltingTemp</span> <span class="k">import</span> <span class="n">make_table</span><span class="p">,</span> <span class="n">DNA_NN2</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">table</span> <span class="o">=</span> <span class="n">DNA_NN2</span>                               <span class="c1"># Sugimoto &#39;96</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">table</span><span class="p">[</span><span class="s1">&#39;init_A/T&#39;</span><span class="p">]</span>
<span class="go">(0, 0)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">newtable</span> <span class="o">=</span> <span class="n">make_table</span><span class="p">(</span><span class="n">oldtable</span><span class="o">=</span><span class="n">DNA_NN2</span><span class="p">,</span> <span class="n">values</span><span class="o">=</span><span class="p">{</span><span class="s1">&#39;init&#39;</span><span class="p">:</span> <span class="p">(</span><span class="mi">0</span><span class="p">,</span> <span class="mi">0</span><span class="p">),</span>
<span class="gp">... </span>                      <span class="s1">&#39;init_A/T&#39;</span><span class="p">:</span> <span class="p">(</span><span class="mf">2.3</span><span class="p">,</span> <span class="mf">4.1</span><span class="p">),</span>
<span class="gp">... </span>                      <span class="s1">&#39;init_G/C&#39;</span><span class="p">:</span> <span class="p">(</span><span class="mf">0.1</span><span class="p">,</span> <span class="o">-</span><span class="mf">2.8</span><span class="p">)})</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s2">&quot;</span><span class="si">%0.1f</span><span class="s2">, </span><span class="si">%0.1f</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="n">newtable</span><span class="p">[</span><span class="s1">&#39;init_A/T&#39;</span><span class="p">])</span>
<span class="go">2.3, 4.1</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Bio.SeqUtils.MeltingTemp.salt_correction">
<code class="sig-prename descclassname">Bio.SeqUtils.MeltingTemp.</code><code class="sig-name descname">salt_correction</code><span class="sig-paren">(</span><em class="sig-param">Na=0</em>, <em class="sig-param">K=0</em>, <em class="sig-param">Tris=0</em>, <em class="sig-param">Mg=0</em>, <em class="sig-param">dNTPs=0</em>, <em class="sig-param">method=1</em>, <em class="sig-param">seq=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.SeqUtils.MeltingTemp.salt_correction" title="Permalink to this definition">¶</a></dt>
<dd><p>Calculate a term to correct Tm for salt ions.</p>
<p>Depending on the Tm calculation, the term will correct Tm or entropy. To
calculate corrected Tm values, different operations need to be applied:</p>
<blockquote>
<div><ul class="simple">
<li><p>methods 1-4: Tm(new) = Tm(old) + corr</p></li>
<li><p>method 5: deltaS(new) = deltaS(old) + corr</p></li>
<li><p>methods 6+7: Tm(new) = 1/(1/Tm(old) + corr)</p></li>
</ul>
</div></blockquote>
<dl class="simple">
<dt>Parameters:</dt><dd><ul class="simple">
<li><p>Na, K, Tris, Mg, dNTPS: Millimolar concentration of respective ion. To
have a simple ‘salt correction’, just pass Na. If any of K, Tris, Mg and
dNTPS is non-zero, a ‘sodium-equivalent’ concentration is calculated
according to von Ahsen et al. (2001, Clin Chem 47: 1956-1961):
[Na_eq] = [Na+] + [K+] + [Tris]/2 + 120*([Mg2+] - [dNTPs])^0.5
If [dNTPs] &gt;= [Mg2+]: [Na_eq] = [Na+] + [K+] + [Tris]/2</p></li>
<li><p>method: Which method to be applied. Methods 1-4 correct Tm, method 5
corrects deltaS, methods 6 and 7 correct 1/Tm. The methods are:</p>
<ol class="arabic simple">
<li><p>16.6 x log[Na+]
(Schildkraut &amp; Lifson (1965), Biopolymers 3: 195-208)</p></li>
<li><p>16.6 x log([Na+]/(1.0 + 0.7*[Na+]))
(Wetmur (1991), Crit Rev Biochem Mol Biol 126: 227-259)</p></li>
<li><p>12.5 x log(Na+]
(SantaLucia et al. (1996), Biochemistry 35: 3555-3562</p></li>
<li><p>11.7 x log[Na+]
(SantaLucia (1998), Proc Natl Acad Sci USA 95: 1460-1465</p></li>
<li><p>Correction for deltaS: 0.368 x (N-1) x ln[Na+]
(SantaLucia (1998), Proc Natl Acad Sci USA 95: 1460-1465)</p></li>
<li><p>(4.29(%GC)-3.95)x1e-5 x ln[Na+] + 9.40e-6 x ln[Na+]^2
(Owczarzy et al. (2004), Biochemistry 43: 3537-3554)</p></li>
<li><p>Complex formula with decision tree and 7 empirical constants.
Mg2+ is corrected for dNTPs binding (if present)
(Owczarzy et al. (2008), Biochemistry 47: 5336-5353)</p></li>
</ol>
</li>
</ul>
</dd>
</dl>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.SeqUtils</span> <span class="k">import</span> <span class="n">MeltingTemp</span> <span class="k">as</span> <span class="n">mt</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">salt_correction</span><span class="p">(</span><span class="n">Na</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span> <span class="n">method</span><span class="o">=</span><span class="mi">1</span><span class="p">))</span>
<span class="go">-21.60</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">salt_correction</span><span class="p">(</span><span class="n">Na</span><span class="o">=</span><span class="mi">50</span><span class="p">,</span> <span class="n">method</span><span class="o">=</span><span class="mi">2</span><span class="p">))</span>
<span class="go">-21.85</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">salt_correction</span><span class="p">(</span><span class="n">Na</span><span class="o">=</span><span class="mi">100</span><span class="p">,</span> <span class="n">Tris</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">method</span><span class="o">=</span><span class="mi">2</span><span class="p">))</span>
<span class="go">-16.45</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s1">&#39;</span><span class="si">%0.2f</span><span class="s1">&#39;</span> <span class="o">%</span> <span class="n">mt</span><span class="o">.</span><span class="n">salt_correction</span><span class="p">(</span><span class="n">Na</span><span class="o">=</span><span class="mi">100</span><span class="p">,</span> <span class="n">Tris</span><span class="o">=</span><span class="mi">20</span><span class="p">,</span> <span class="n">Mg</span><span class="o">=</span><span class="mf">1.5</span><span class="p">,</span> <span class="n">method</span><span class="o">=</span><span class="mi">2</span><span class="p">))</span>
<span class="go">-10.99</span>
</pre></div>
</div>
</dd></dl>

</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="Bio.SeqUtils.ProtParam.html" class="btn btn-neutral float-right" title="Bio.SeqUtils.ProtParam module" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="Bio.SeqUtils.IsoelectricPoint.html" class="btn btn-neutral float-left" title="Bio.SeqUtils.IsoelectricPoint module" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 1999-2017, The Biopython Contributors

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>