

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Bio.Graphics.GenomeDiagram package &mdash; Biopython 1.75.dev0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Bio.Graphics.BasicChromosome module" href="Bio.Graphics.BasicChromosome.html" />
    <link rel="prev" title="Bio.Graphics package" href="Bio.Graphics.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search"  style="background: #10100F" >
          

          
            <a href="index.html" class="icon icon-home"> Biopython
          

          
            
            <img src="_static/biopython_logo.svg" class="logo" alt="Logo"/>
          
          </a>

          
            
            
              <div class="version">
                1.75.dev0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">API Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="Bio.html">Bio package</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="Bio.html#subpackages">Subpackages</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="Bio.Affy.html">Bio.Affy package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Align.html">Bio.Align package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.AlignIO.html">Bio.AlignIO package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Alphabet.html">Bio.Alphabet package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Application.html">Bio.Application package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Blast.html">Bio.Blast package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.CAPS.html">Bio.CAPS package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Cluster.html">Bio.Cluster package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Compass.html">Bio.Compass package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Crystal.html">Bio.Crystal package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Data.html">Bio.Data package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Emboss.html">Bio.Emboss package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Entrez.html">Bio.Entrez package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.ExPASy.html">Bio.ExPASy package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.FSSP.html">Bio.FSSP package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.GenBank.html">Bio.GenBank package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Geo.html">Bio.Geo package</a></li>
<li class="toctree-l3 current"><a class="reference internal" href="Bio.Graphics.html">Bio.Graphics package</a><ul class="current">
<li class="toctree-l4 current"><a class="reference internal" href="Bio.Graphics.html#subpackages">Subpackages</a></li>
<li class="toctree-l4"><a class="reference internal" href="Bio.Graphics.html#submodules">Submodules</a></li>
<li class="toctree-l4"><a class="reference internal" href="Bio.Graphics.html#module-Bio.Graphics">Module contents</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="Bio.HMM.html">Bio.HMM package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.KDTree.html">Bio.KDTree package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.KEGG.html">Bio.KEGG package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Medline.html">Bio.Medline package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.NMR.html">Bio.NMR package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Nexus.html">Bio.Nexus package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.PDB.html">Bio.PDB package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Pathway.html">Bio.Pathway package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Phylo.html">Bio.Phylo package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.PopGen.html">Bio.PopGen package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Restriction.html">Bio.Restriction package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SCOP.html">Bio.SCOP package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SVDSuperimposer.html">Bio.SVDSuperimposer package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SearchIO.html">Bio.SearchIO package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SeqIO.html">Bio.SeqIO package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SeqUtils.html">Bio.SeqUtils package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Sequencing.html">Bio.Sequencing package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Statistics.html">Bio.Statistics package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SubsMat.html">Bio.SubsMat package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SwissProt.html">Bio.SwissProt package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.TogoWS.html">Bio.TogoWS package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.UniGene.html">Bio.UniGene package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.UniProt.html">Bio.UniProt package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Wise.html">Bio.Wise package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.codonalign.html">Bio.codonalign package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.motifs.html">Bio.motifs package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.phenotype.html">Bio.phenotype package</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="Bio.html#submodules">Submodules</a></li>
<li class="toctree-l2"><a class="reference internal" href="Bio.html#module-Bio">Module contents</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="BioSQL.html">BioSQL package</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Biopython</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
          <li><a href="Bio.html">Bio package</a> &raquo;</li>
        
          <li><a href="Bio.Graphics.html">Bio.Graphics package</a> &raquo;</li>
        
      <li>Bio.Graphics.GenomeDiagram package</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
        
      </li>
    
  </ul>

  
  <div class="rst-breadcrumbs-buttons" role="navigation" aria-label="breadcrumb navigation">
      
        <a href="Bio.Graphics.BasicChromosome.html" class="btn btn-neutral float-right" title="Bio.Graphics.BasicChromosome module" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="Bio.Graphics.html" class="btn btn-neutral float-left" title="Bio.Graphics package" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
  </div>
  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="bio-graphics-genomediagram-package">
<h1>Bio.Graphics.GenomeDiagram package<a class="headerlink" href="#bio-graphics-genomediagram-package" title="Permalink to this headline">¶</a></h1>
<div class="section" id="module-Bio.Graphics.GenomeDiagram">
<span id="module-contents"></span><h2>Module contents<a class="headerlink" href="#module-Bio.Graphics.GenomeDiagram" title="Permalink to this headline">¶</a></h2>
<p>GenomeDiagram module integrated into Biopython.</p>
<dl class="class">
<dt id="Bio.Graphics.GenomeDiagram.Diagram">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Graphics.GenomeDiagram.</code><code class="sig-name descname">Diagram</code><span class="sig-paren">(</span><em class="sig-param">name=None</em>, <em class="sig-param">format='circular'</em>, <em class="sig-param">pagesize='A3'</em>, <em class="sig-param">orientation='landscape'</em>, <em class="sig-param">x=0.05</em>, <em class="sig-param">y=0.05</em>, <em class="sig-param">xl=None</em>, <em class="sig-param">xr=None</em>, <em class="sig-param">yt=None</em>, <em class="sig-param">yb=None</em>, <em class="sig-param">start=None</em>, <em class="sig-param">end=None</em>, <em class="sig-param">tracklines=False</em>, <em class="sig-param">fragments=10</em>, <em class="sig-param">fragment_size=None</em>, <em class="sig-param">track_size=0.75</em>, <em class="sig-param">circular=True</em>, <em class="sig-param">circle_core=0.0</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Diagram" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Diagram container.</p>
<p>Arguments:</p>
<ul class="simple">
<li><p>name           - a string, identifier for the diagram.</p></li>
<li><p>tracks         - a list of Track objects comprising the diagram.</p></li>
<li><p>format         - a string, format of the diagram ‘circular’ or
‘linear’, depending on the sort of diagram required.</p></li>
<li><p>pagesize       - a string, the pagesize of output describing the ISO
size of the image, or a tuple of pixels.</p></li>
<li><p>orientation    - a string describing the required orientation of the
final drawing (‘landscape’ or ‘portrait’).</p></li>
<li><p>x              - a float (0-&gt;1), the proportion of the page to take
up with even X margins t the page.</p></li>
<li><p>y              - a float (0-&gt;1), the proportion of the page to take
up with even Y margins to the page.</p></li>
<li><p>xl             - a float (0-&gt;1), the proportion of the page to take
up with the left X margin to the page (overrides x).</p></li>
<li><p>xr             - a float (0-&gt;1), the proportion of the page to take
up with the right X margin to the page (overrides x).</p></li>
<li><p>yt             - a float (0-&gt;1), the proportion of the page to take
up with the top Y margin to the page (overrides y).</p></li>
<li><p>yb             - a float (0-&gt;1), the proportion of the page to take
up with the bottom Y margin to the page (overrides y).</p></li>
<li><p>circle_core    - a float, the proportion of the available radius to
leave empty at the center of a circular diagram (0 to 1).</p></li>
<li><p>start          - an integer, the base/aa position to start the diagram at.</p></li>
<li><p>end            - an integer, the base/aa position to end the diagram at.</p></li>
<li><p>tracklines     - a boolean, True if track guidelines are to be drawn.</p></li>
<li><p>fragments      - and integer, for a linear diagram, the number of equal
divisions into which the sequence is divided.</p></li>
<li><p>fragment_size  - a float (0-&gt;1), the proportion of the space
available to each fragment that should be used in drawing.</p></li>
<li><p>track_size     - a float (0-&gt;1), the proportion of the space
available to each track that should be used in drawing with sigils.</p></li>
<li><p>circular       - a boolean, True if the genome/sequence to be drawn
is, in reality, circular.</p></li>
</ul>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.Diagram.add_track" title="Bio.Graphics.GenomeDiagram.Diagram.add_track"><code class="xref py py-obj docutils literal notranslate"><span class="pre">add_track</span></code></a>(self, track, track_level)</p></td>
<td><p>Add a Track object to the diagram.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.Diagram.del_track" title="Bio.Graphics.GenomeDiagram.Diagram.del_track"><code class="xref py py-obj docutils literal notranslate"><span class="pre">del_track</span></code></a>(self, track_level)</p></td>
<td><p>Remove the track to be drawn at a particular level on the diagram.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.Diagram.draw" title="Bio.Graphics.GenomeDiagram.Diagram.draw"><code class="xref py py-obj docutils literal notranslate"><span class="pre">draw</span></code></a>(self[, format, pagesize, orientation, …])</p></td>
<td><p>Draw the diagram, with passed parameters overriding existing attributes.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.Diagram.get_drawn_levels" title="Bio.Graphics.GenomeDiagram.Diagram.get_drawn_levels"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_drawn_levels</span></code></a>(self)</p></td>
<td><p>Return a sorted list of levels occupied by tracks.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.Diagram.get_levels" title="Bio.Graphics.GenomeDiagram.Diagram.get_levels"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_levels</span></code></a>(self)</p></td>
<td><p>Return a sorted list of levels occupied by tracks in the diagram.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.Diagram.get_tracks" title="Bio.Graphics.GenomeDiagram.Diagram.get_tracks"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_tracks</span></code></a>(self)</p></td>
<td><p>Return a list of the tracks contained in the diagram.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.Diagram.move_track" title="Bio.Graphics.GenomeDiagram.Diagram.move_track"><code class="xref py py-obj docutils literal notranslate"><span class="pre">move_track</span></code></a>(self, from_level, to_level)</p></td>
<td><p>Move a track from one level on the diagram to another.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.Diagram.new_track" title="Bio.Graphics.GenomeDiagram.Diagram.new_track"><code class="xref py py-obj docutils literal notranslate"><span class="pre">new_track</span></code></a>(self, track_level, \*\*args)</p></td>
<td><p>Add a new Track to the diagram at a given level.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.Diagram.range" title="Bio.Graphics.GenomeDiagram.Diagram.range"><code class="xref py py-obj docutils literal notranslate"><span class="pre">range</span></code></a>(self)</p></td>
<td><p>Return lowest and highest base numbers from track features.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.Diagram.renumber_tracks" title="Bio.Graphics.GenomeDiagram.Diagram.renumber_tracks"><code class="xref py py-obj docutils literal notranslate"><span class="pre">renumber_tracks</span></code></a>(self[, low, step])</p></td>
<td><p>Renumber all tracks consecutively.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.Diagram.set_all_tracks" title="Bio.Graphics.GenomeDiagram.Diagram.set_all_tracks"><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_all_tracks</span></code></a>(self, attr, value)</p></td>
<td><p>Set the passed attribute of all tracks in the set to the passed value.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.Diagram.write" title="Bio.Graphics.GenomeDiagram.Diagram.write"><code class="xref py py-obj docutils literal notranslate"><span class="pre">write</span></code></a>(self[, filename, output, dpi])</p></td>
<td><p>Write the drawn diagram to a specified file, in a specified format.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.Diagram.write_to_string" title="Bio.Graphics.GenomeDiagram.Diagram.write_to_string"><code class="xref py py-obj docutils literal notranslate"><span class="pre">write_to_string</span></code></a>(self[, output, dpi])</p></td>
<td><p>Return a byte string containing the diagram in the requested format.</p></td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.Diagram.add_track">
<code class="sig-name descname">add_track</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">track</em>, <em class="sig-param">track_level</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Diagram.add_track" title="Permalink to this definition">¶</a></dt>
<dd><p>Add a Track object to the diagram.</p>
<p>It also accepts instructions to place it at a particular level on the
diagram.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>track          - Track object to draw.</p></li>
<li><p>track_level    - an integer. The level at which the track will be
drawn (above an arbitrary baseline).</p></li>
</ul>
</dd>
</dl>
<p>add_track(self, track, track_level)</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.Diagram.del_track">
<code class="sig-name descname">del_track</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">track_level</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Diagram.del_track" title="Permalink to this definition">¶</a></dt>
<dd><p>Remove the track to be drawn at a particular level on the diagram.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>track_level   - an integer. The level of the track on the diagram
to delete.</p></li>
</ul>
</dd>
</dl>
<p>del_track(self, track_level)</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.Diagram.draw">
<code class="sig-name descname">draw</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">format=None</em>, <em class="sig-param">pagesize=None</em>, <em class="sig-param">orientation=None</em>, <em class="sig-param">x=None</em>, <em class="sig-param">y=None</em>, <em class="sig-param">xl=None</em>, <em class="sig-param">xr=None</em>, <em class="sig-param">yt=None</em>, <em class="sig-param">yb=None</em>, <em class="sig-param">start=None</em>, <em class="sig-param">end=None</em>, <em class="sig-param">tracklines=None</em>, <em class="sig-param">fragments=None</em>, <em class="sig-param">fragment_size=None</em>, <em class="sig-param">track_size=None</em>, <em class="sig-param">circular=None</em>, <em class="sig-param">circle_core=None</em>, <em class="sig-param">cross_track_links=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Diagram.draw" title="Permalink to this definition">¶</a></dt>
<dd><p>Draw the diagram, with passed parameters overriding existing attributes.</p>
<p>gdd.draw(format=’circular’)</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.Diagram.get_drawn_levels">
<code class="sig-name descname">get_drawn_levels</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Diagram.get_drawn_levels" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a sorted list of levels occupied by tracks.</p>
<p>These tracks are not explicitly hidden.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.Diagram.get_levels">
<code class="sig-name descname">get_levels</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Diagram.get_levels" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a sorted list of levels occupied by tracks in the diagram.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.Diagram.get_tracks">
<code class="sig-name descname">get_tracks</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Diagram.get_tracks" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a list of the tracks contained in the diagram.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.Diagram.move_track">
<code class="sig-name descname">move_track</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">from_level</em>, <em class="sig-param">to_level</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Diagram.move_track" title="Permalink to this definition">¶</a></dt>
<dd><p>Move a track from one level on the diagram to another.</p>
<p>Arguments:</p>
<ul class="simple">
<li><p>from_level   - an integer. The level at which the track to be
moved is found.</p></li>
<li><p>to_level     - an integer. The level to move the track to.</p></li>
</ul>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.Diagram.new_track">
<code class="sig-name descname">new_track</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">track_level</em>, <em class="sig-param">**args</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Diagram.new_track" title="Permalink to this definition">¶</a></dt>
<dd><p>Add a new Track to the diagram at a given level.</p>
<p>The track is returned for further user manipulation.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>track_level   - an integer. The level at which the track will be
drawn (above an arbitrary baseline).</p></li>
</ul>
</dd>
</dl>
<p>new_track(self, track_level)</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.Diagram.range">
<code class="sig-name descname">range</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Diagram.range" title="Permalink to this definition">¶</a></dt>
<dd><p>Return lowest and highest base numbers from track features.</p>
<p>Returned type is a tuple.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.Diagram.renumber_tracks">
<code class="sig-name descname">renumber_tracks</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">low=1</em>, <em class="sig-param">step=1</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Diagram.renumber_tracks" title="Permalink to this definition">¶</a></dt>
<dd><p>Renumber all tracks consecutively.</p>
<p>Optionally from a passed lowest number.</p>
<p>Arguments:</p>
<ul class="simple">
<li><p>low     - an integer. The track number to start from.</p></li>
<li><p>step    - an integer. The track interval for separation of
tracks.</p></li>
</ul>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.Diagram.set_all_tracks">
<code class="sig-name descname">set_all_tracks</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">attr</em>, <em class="sig-param">value</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Diagram.set_all_tracks" title="Permalink to this definition">¶</a></dt>
<dd><p>Set the passed attribute of all tracks in the set to the passed value.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>attr    - An attribute of the Track class.</p></li>
<li><p>value   - The value to set that attribute.</p></li>
</ul>
</dd>
</dl>
<p>set_all_tracks(self, attr, value)</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.Diagram.write">
<code class="sig-name descname">write</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">filename='test1.ps'</em>, <em class="sig-param">output='PS'</em>, <em class="sig-param">dpi=72</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Diagram.write" title="Permalink to this definition">¶</a></dt>
<dd><p>Write the drawn diagram to a specified file, in a specified format.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>filename   - a string indicating the name of the output file,
or a handle to write to.</p></li>
<li><p>output     - a string indicating output format, one of PS, PDF,
SVG, or provided the ReportLab renderPM module is installed, one
of the bitmap formats JPG, BMP, GIF, PNG, TIFF or TIFF.  The
format can be given in upper or lower case.</p></li>
<li><p>dpi        - an integer. Resolution (dots per inch) for bitmap formats.</p></li>
</ul>
</dd>
<dt>Returns:</dt><dd><p>No return value.</p>
</dd>
</dl>
<p>write(self, filename=’test1.ps’, output=’PS’, dpi=72)</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.Diagram.write_to_string">
<code class="sig-name descname">write_to_string</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">output='PS'</em>, <em class="sig-param">dpi=72</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Diagram.write_to_string" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a byte string containing the diagram in the requested format.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>output    - a string indicating output format, one of PS, PDF,
SVG, JPG, BMP, GIF, PNG, TIFF or TIFF (as specified for the write
method).</p></li>
<li><p>dpi       - Resolution (dots per inch) for bitmap formats.</p></li>
</ul>
</dd>
<dt>Returns:</dt><dd><p>Return the completed drawing as a bytes string in a prescribed
format.</p>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="Bio.Graphics.GenomeDiagram.Track">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Graphics.GenomeDiagram.</code><code class="sig-name descname">Track</code><span class="sig-paren">(</span><em class="sig-param">name=None</em>, <em class="sig-param">height=1</em>, <em class="sig-param">hide=0</em>, <em class="sig-param">greytrack=0</em>, <em class="sig-param">greytrack_labels=5</em>, <em class="sig-param">greytrack_fontsize=8</em>, <em class="sig-param">greytrack_font='Helvetica'</em>, <em class="sig-param">greytrack_font_rotation=0</em>, <em class="sig-param">greytrack_font_color=Color(.6</em>, <em class="sig-param">.6</em>, <em class="sig-param">.6</em>, <em class="sig-param">1)</em>, <em class="sig-param">scale=1</em>, <em class="sig-param">scale_format=None</em>, <em class="sig-param">scale_color=Color(0</em>, <em class="sig-param">0</em>, <em class="sig-param">0</em>, <em class="sig-param">1)</em>, <em class="sig-param">scale_font='Helvetica'</em>, <em class="sig-param">scale_fontsize=6</em>, <em class="sig-param">scale_fontangle=45</em>, <em class="sig-param">scale_largeticks=0.5</em>, <em class="sig-param">scale_ticks=1</em>, <em class="sig-param">scale_smallticks=0.3</em>, <em class="sig-param">scale_largetick_interval=1000000.0</em>, <em class="sig-param">scale_smalltick_interval=10000.0</em>, <em class="sig-param">scale_largetick_labels=1</em>, <em class="sig-param">scale_smalltick_labels=0</em>, <em class="sig-param">axis_labels=1</em>, <em class="sig-param">start=None</em>, <em class="sig-param">end=None</em>, <em class="sig-param">greytrack_font_colour=None</em>, <em class="sig-param">scale_colour=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Track" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Track.</p>
<dl class="simple">
<dt>Attributes:</dt><dd><ul class="simple">
<li><p>height    Int describing the relative height to other trackscale_fontsizes
in the diagram</p></li>
<li><p>name      String describing the track</p></li>
<li><p>hide      Boolean, 0 if the track is not to be drawn</p></li>
<li><p>start, end    Integers (or None) specifying start/end to draw just
a partial track.</p></li>
<li><p>greytrack     Boolean, 1 if a grey background to the track is to be
drawn</p></li>
<li><p>greytrack_labels  Int describing how many track-identifying labels
should be placed on the track at regular intervals</p></li>
<li><p>greytrack_font    String describing the font to use for the greytrack
labels</p></li>
<li><p>greytrack_fontsize    Int describing the font size to display the
labels on the grey track</p></li>
<li><p>greytrack_font_rotation   Int describing the angle through which to
rotate the grey track labels (Linear only)</p></li>
<li><p>greytrack_font_color     colors.Color describing the color to draw
the grey track labels</p></li>
<li><p>scale     Boolean, 1 if a scale is to be drawn on the track</p></li>
<li><p>scale_format  String, defaults to None, when scale values are written
as numerals.  Setting this to ‘SInt’ invokes SI
unit-like multiples, such as Mbp, Kbp and so on.</p></li>
<li><p>scale_color  colors.Color to draw the elements of the scale</p></li>
<li><p>scale_font    String describing the font to use for the scale labels</p></li>
<li><p>scale_fontsize    Int describing the size of the scale label font</p></li>
<li><p>scale_fontangle   Int describing the angle at which to draw the scale
labels (linear only)</p></li>
<li><p>scale_ticks       Boolean, 1 if ticks should be drawn at all on the
scale</p></li>
<li><p>scale_largeticks  Float (0-&gt;1) describing the height of large
scale ticks relative to the track height.</p></li>
<li><p>scale_smallticks  Float (0-&gt;1) describing the height of large
scale ticks relative to the track height.</p></li>
<li><p>scale_largetick_interval  Int, describing the number of bases that
should separate large ticks</p></li>
<li><p>scale_smalltick_interval  Int, describing the number of bases that
should separate small ticks</p></li>
<li><p>scale_largetick_labels    Boolean describing whether position labels
should be written over large ticks</p></li>
<li><p>scale_smalltick_labels    Boolean describing whether position labels
should be written over small ticks</p></li>
<li><p>axis_labels       Boolean describing whether the value labels should
be placed on the Y axes</p></li>
</ul>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.Track.add_set" title="Bio.Graphics.GenomeDiagram.Track.add_set"><code class="xref py py-obj docutils literal notranslate"><span class="pre">add_set</span></code></a>(self, set)</p></td>
<td><p>Add a preexisting FeatureSet or GraphSet object to the track.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.Track.del_set" title="Bio.Graphics.GenomeDiagram.Track.del_set"><code class="xref py py-obj docutils literal notranslate"><span class="pre">del_set</span></code></a>(self, set_id)</p></td>
<td><p>Remove the set with the passed id from the track.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.Track.get_ids" title="Bio.Graphics.GenomeDiagram.Track.get_ids"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_ids</span></code></a>(self)</p></td>
<td><p>Return the ids of all sets contained in this track.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.Track.get_sets" title="Bio.Graphics.GenomeDiagram.Track.get_sets"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_sets</span></code></a>(self)</p></td>
<td><p>Return the sets contained in this track.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.Track.new_set" title="Bio.Graphics.GenomeDiagram.Track.new_set"><code class="xref py py-obj docutils literal notranslate"><span class="pre">new_set</span></code></a>(self[, type])</p></td>
<td><p>Create a new FeatureSet or GraphSet object.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.Track.range" title="Bio.Graphics.GenomeDiagram.Track.range"><code class="xref py py-obj docutils literal notranslate"><span class="pre">range</span></code></a>(self)</p></td>
<td><p>Return the lowest and highest base (or mark) numbers as a tuple.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.Track.to_string" title="Bio.Graphics.GenomeDiagram.Track.to_string"><code class="xref py py-obj docutils literal notranslate"><span class="pre">to_string</span></code></a>(self[, verbose])</p></td>
<td><p>Return a formatted string with information about the track.</p></td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.Track.add_set">
<code class="sig-name descname">add_set</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">set</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Track.add_set" title="Permalink to this definition">¶</a></dt>
<dd><p>Add a preexisting FeatureSet or GraphSet object to the track.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.Track.del_set">
<code class="sig-name descname">del_set</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">set_id</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Track.del_set" title="Permalink to this definition">¶</a></dt>
<dd><p>Remove the set with the passed id from the track.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.Track.get_ids">
<code class="sig-name descname">get_ids</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Track.get_ids" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the ids of all sets contained in this track.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.Track.get_sets">
<code class="sig-name descname">get_sets</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Track.get_sets" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the sets contained in this track.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.Track.new_set">
<code class="sig-name descname">new_set</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">type='feature'</em>, <em class="sig-param">**args</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Track.new_set" title="Permalink to this definition">¶</a></dt>
<dd><p>Create a new FeatureSet or GraphSet object.</p>
<p>Create a new FeatureSet or GraphSet object, add it to the
track, and return for user manipulation</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.Track.range">
<code class="sig-name descname">range</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Track.range" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the lowest and highest base (or mark) numbers as a tuple.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.Track.to_string">
<code class="sig-name descname">to_string</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">verbose=0</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Track.to_string" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a formatted string with information about the track.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>verbose - Boolean indicating whether a short or complete
account of the track is required</p></li>
</ul>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="Bio.Graphics.GenomeDiagram.FeatureSet">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Graphics.GenomeDiagram.</code><code class="sig-name descname">FeatureSet</code><span class="sig-paren">(</span><em class="sig-param">set_id=None</em>, <em class="sig-param">name=None</em>, <em class="sig-param">parent=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.FeatureSet" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>FeatureSet object.</p>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.FeatureSet.add_feature" title="Bio.Graphics.GenomeDiagram.FeatureSet.add_feature"><code class="xref py py-obj docutils literal notranslate"><span class="pre">add_feature</span></code></a>(self, feature, \*\*kwargs)</p></td>
<td><p>Add a new feature.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.FeatureSet.del_feature" title="Bio.Graphics.GenomeDiagram.FeatureSet.del_feature"><code class="xref py py-obj docutils literal notranslate"><span class="pre">del_feature</span></code></a>(self, feature_id)</p></td>
<td><p>Delete a feature.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.FeatureSet.get_features" title="Bio.Graphics.GenomeDiagram.FeatureSet.get_features"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_features</span></code></a>(self[, attribute, value, …])</p></td>
<td><p>Retrieve features.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.FeatureSet.get_ids" title="Bio.Graphics.GenomeDiagram.FeatureSet.get_ids"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_ids</span></code></a>(self)</p></td>
<td><p>Return a list of all ids for the feature set.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.FeatureSet.range" title="Bio.Graphics.GenomeDiagram.FeatureSet.range"><code class="xref py py-obj docutils literal notranslate"><span class="pre">range</span></code></a>(self)</p></td>
<td><p>Return the lowest and highest base (or mark) numbers as a tuple.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.FeatureSet.set_all_features" title="Bio.Graphics.GenomeDiagram.FeatureSet.set_all_features"><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_all_features</span></code></a>(self, attr, value)</p></td>
<td><p>Set an attribute of all the features.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.FeatureSet.to_string" title="Bio.Graphics.GenomeDiagram.FeatureSet.to_string"><code class="xref py py-obj docutils literal notranslate"><span class="pre">to_string</span></code></a>(self[, verbose])</p></td>
<td><p>Return a formatted string with information about the set.</p></td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.FeatureSet.add_feature">
<code class="sig-name descname">add_feature</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">feature</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.FeatureSet.add_feature" title="Permalink to this definition">¶</a></dt>
<dd><p>Add a new feature.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>feature: Bio.SeqFeature object</p></li>
<li><p>kwargs: Keyword arguments for Feature.  Named attributes
of the Feature</p></li>
</ul>
</dd>
</dl>
<p>Add a Bio.SeqFeature object to the diagram (will be stored
internally in a Feature wrapper).</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.FeatureSet.del_feature">
<code class="sig-name descname">del_feature</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">feature_id</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.FeatureSet.del_feature" title="Permalink to this definition">¶</a></dt>
<dd><p>Delete a feature.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>feature_id: Unique id of the feature to delete</p></li>
</ul>
</dd>
</dl>
<p>Remove a feature from the set, indicated by its id.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.FeatureSet.get_features">
<code class="sig-name descname">get_features</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">attribute=None</em>, <em class="sig-param">value=None</em>, <em class="sig-param">comparator=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.FeatureSet.get_features" title="Permalink to this definition">¶</a></dt>
<dd><p>Retrieve features.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>attribute: String, attribute of a Feature object</p></li>
<li><p>value: The value desired of the attribute</p></li>
<li><p>comparator: String, how to compare the Feature attribute to the
passed value</p></li>
</ul>
</dd>
</dl>
<p>If no attribute or value is given, return a list of all features in the
feature set.  If both an attribute and value are given, then depending
on the comparator, then a list of all features in the FeatureSet
matching (or not) the passed value will be returned.  Allowed comparators
are: ‘startswith’, ‘not’, ‘like’.</p>
<p>The user is expected to make a responsible decision about which feature
attributes to use with which passed values and comparator settings.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.FeatureSet.get_ids">
<code class="sig-name descname">get_ids</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.FeatureSet.get_ids" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a list of all ids for the feature set.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.FeatureSet.range">
<code class="sig-name descname">range</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.FeatureSet.range" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the lowest and highest base (or mark) numbers as a tuple.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.FeatureSet.set_all_features">
<code class="sig-name descname">set_all_features</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">attr</em>, <em class="sig-param">value</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.FeatureSet.set_all_features" title="Permalink to this definition">¶</a></dt>
<dd><p>Set an attribute of all the features.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>attr: An attribute of the Feature class</p></li>
<li><p>value: The value to set that attribute to</p></li>
</ul>
</dd>
</dl>
<p>Set the passed attribute of all features in the set to the
passed value.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.FeatureSet.to_string">
<code class="sig-name descname">to_string</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">verbose=0</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.FeatureSet.to_string" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a formatted string with information about the set.</p>
<p>Arguments:</p>
<ul class="simple">
<li><p>verbose: Boolean indicating whether a short (default) or
complete account of the set is required</p></li>
</ul>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="Bio.Graphics.GenomeDiagram.Feature">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Graphics.GenomeDiagram.</code><code class="sig-name descname">Feature</code><span class="sig-paren">(</span><em class="sig-param">parent=None</em>, <em class="sig-param">feature_id=None</em>, <em class="sig-param">feature=None</em>, <em class="sig-param">color=Color(.564706</em>, <em class="sig-param">.933333</em>, <em class="sig-param">.564706</em>, <em class="sig-param">1)</em>, <em class="sig-param">label=0</em>, <em class="sig-param">border=None</em>, <em class="sig-param">colour=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Feature" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Class to wrap Bio.SeqFeature objects for GenomeDiagram.</p>
<dl class="simple">
<dt>Attributes:</dt><dd><ul class="simple">
<li><p>parent    FeatureSet, container for the object</p></li>
<li><p>id        Unique id</p></li>
<li><p>color    color.Color, color to draw the feature</p></li>
<li><p>hide      Boolean for whether the feature will be drawn or not</p></li>
<li><p>sigil     String denoting the type of sigil to use for the feature.
Currently either “BOX” or “ARROW” are supported.</p></li>
<li><p>arrowhead_length  Float denoting length of the arrow head to be drawn,
relative to the bounding box height.  The arrow shaft
takes up the remainder of the bounding box’s length.</p></li>
<li><p>arrowshaft_height  Float denoting length of the representative arrow
shaft to be drawn, relative to the bounding box height.
The arrow head takes the full height of the bound box.</p></li>
<li><p>name_qualifiers   List of Strings, describes the qualifiers that may
contain feature names in the wrapped Bio.SeqFeature object</p></li>
<li><p>label     Boolean, 1 if the label should be shown</p></li>
<li><p>label_font    String describing the font to use for the feature label</p></li>
<li><p>label_size    Int describing the feature label font size</p></li>
<li><p>label_color  color.Color describing the feature label color</p></li>
<li><p>label_angle   Float describing the angle through which to rotate the
feature label in degrees (default = 45, linear only)</p></li>
<li><p>label_position    String, ‘start’, ‘end’ or ‘middle’ denoting where
to place the feature label. Leave as None for the default
which is ‘start’ for linear diagrams, and at the bottom of
the feature as drawn on circular diagrams.</p></li>
<li><p>label_strand  Integer -1 or +1 to explicitly place the label on the
forward or reverse strand. Default (None) follows th
feature’s strand. Use -1 to put labels under (linear) or
inside (circular) the track, +1 to put them above (linear)
or outside (circular) the track.</p></li>
<li><p>locations     List of tuples of (start, end) ints describing where the
feature and any subfeatures start and end</p></li>
<li><p>type      String denoting the feature type</p></li>
<li><p>name      String denoting the feature name</p></li>
<li><p>strand    Int describing the strand on which the feature is found</p></li>
</ul>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.Feature.get_feature" title="Bio.Graphics.GenomeDiagram.Feature.get_feature"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_feature</span></code></a>(self)</p></td>
<td><p>Return the unwrapped Bio.SeqFeature object.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.Feature.set_color" title="Bio.Graphics.GenomeDiagram.Feature.set_color"><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_color</span></code></a>(self, color)</p></td>
<td><p>Set the color in which the feature will be drawn.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.Feature.set_colour" title="Bio.Graphics.GenomeDiagram.Feature.set_colour"><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_colour</span></code></a>(self, colour)</p></td>
<td><p>Backwards compatible variant of set_color(self, color) using UK spelling.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.Feature.set_feature" title="Bio.Graphics.GenomeDiagram.Feature.set_feature"><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_feature</span></code></a>(self, feature)</p></td>
<td><p>Define the Bio.SeqFeature object to be wrapped.</p></td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.Feature.get_feature">
<code class="sig-name descname">get_feature</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Feature.get_feature" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the unwrapped Bio.SeqFeature object.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.Feature.set_color">
<code class="sig-name descname">set_color</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">color</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Feature.set_color" title="Permalink to this definition">¶</a></dt>
<dd><p>Set the color in which the feature will be drawn.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>color    The color to draw the feature - either a colors.Color
object, an RGB tuple of floats, or an integer corresponding a
colors in colors.txt</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.Feature.set_colour">
<code class="sig-name descname">set_colour</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">colour</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Feature.set_colour" title="Permalink to this definition">¶</a></dt>
<dd><p>Backwards compatible variant of set_color(self, color) using UK spelling.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.Feature.set_feature">
<code class="sig-name descname">set_feature</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">feature</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.Feature.set_feature" title="Permalink to this definition">¶</a></dt>
<dd><p>Define the Bio.SeqFeature object to be wrapped.</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="Bio.Graphics.GenomeDiagram.GraphSet">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Graphics.GenomeDiagram.</code><code class="sig-name descname">GraphSet</code><span class="sig-paren">(</span><em class="sig-param">name=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.GraphSet" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Graph Set.</p>
<dl class="simple">
<dt>Attributes:</dt><dd><ul class="simple">
<li><p>id        Unique identifier for the set</p></li>
<li><p>name      String describing the set</p></li>
</ul>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.GraphSet.data_quartiles" title="Bio.Graphics.GenomeDiagram.GraphSet.data_quartiles"><code class="xref py py-obj docutils literal notranslate"><span class="pre">data_quartiles</span></code></a>(self)</p></td>
<td><p>Return (minimum, lowerQ, medianQ, upperQ, maximum) values as a tuple.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.GraphSet.del_graph" title="Bio.Graphics.GenomeDiagram.GraphSet.del_graph"><code class="xref py py-obj docutils literal notranslate"><span class="pre">del_graph</span></code></a>(self, graph_id)</p></td>
<td><p>Remove a graph from the set, indicated by its id.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.GraphSet.get_graphs" title="Bio.Graphics.GenomeDiagram.GraphSet.get_graphs"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_graphs</span></code></a>(self)</p></td>
<td><p>Return list of all graphs in the graph set, sorted by id.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.GraphSet.get_ids" title="Bio.Graphics.GenomeDiagram.GraphSet.get_ids"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_ids</span></code></a>(self)</p></td>
<td><p>Return a list of all ids for the graph set.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.GraphSet.new_graph" title="Bio.Graphics.GenomeDiagram.GraphSet.new_graph"><code class="xref py py-obj docutils literal notranslate"><span class="pre">new_graph</span></code></a>(self, data[, name, style, color, …])</p></td>
<td><p>Add a GraphData object to the diagram.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.GraphSet.range" title="Bio.Graphics.GenomeDiagram.GraphSet.range"><code class="xref py py-obj docutils literal notranslate"><span class="pre">range</span></code></a>(self)</p></td>
<td><p>Return the lowest and highest base (or mark) numbers as a tuple.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.GraphSet.to_string" title="Bio.Graphics.GenomeDiagram.GraphSet.to_string"><code class="xref py py-obj docutils literal notranslate"><span class="pre">to_string</span></code></a>(self[, verbose])</p></td>
<td><p>Return a formatted string with information about the set.</p></td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.GraphSet.data_quartiles">
<code class="sig-name descname">data_quartiles</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.GraphSet.data_quartiles" title="Permalink to this definition">¶</a></dt>
<dd><p>Return (minimum, lowerQ, medianQ, upperQ, maximum) values as a tuple.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.GraphSet.del_graph">
<code class="sig-name descname">del_graph</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">graph_id</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.GraphSet.del_graph" title="Permalink to this definition">¶</a></dt>
<dd><p>Remove a graph from the set, indicated by its id.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.GraphSet.get_graphs">
<code class="sig-name descname">get_graphs</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.GraphSet.get_graphs" title="Permalink to this definition">¶</a></dt>
<dd><p>Return list of all graphs in the graph set, sorted by id.</p>
<p>Sorting is to ensure reliable stacking.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.GraphSet.get_ids">
<code class="sig-name descname">get_ids</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.GraphSet.get_ids" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a list of all ids for the graph set.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.GraphSet.new_graph">
<code class="sig-name descname">new_graph</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">data</em>, <em class="sig-param">name=None</em>, <em class="sig-param">style='bar'</em>, <em class="sig-param">color=Color(.564706</em>, <em class="sig-param">.933333</em>, <em class="sig-param">.564706</em>, <em class="sig-param">1)</em>, <em class="sig-param">altcolor=Color(.560784</em>, <em class="sig-param">.737255</em>, <em class="sig-param">.545098</em>, <em class="sig-param">1)</em>, <em class="sig-param">linewidth=1</em>, <em class="sig-param">center=None</em>, <em class="sig-param">colour=None</em>, <em class="sig-param">altcolour=None</em>, <em class="sig-param">centre=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.GraphSet.new_graph" title="Permalink to this definition">¶</a></dt>
<dd><p>Add a GraphData object to the diagram.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>data      List of (position, value) int tuples</p></li>
<li><p>name      String, description of the graph</p></li>
<li><p>style     String (‘bar’, ‘heat’, ‘line’) describing how the graph
will be drawn</p></li>
<li><p>color    colors.Color describing the color to draw all or ‘high’
(some styles) data (overridden by backwards compatible
argument with UK spelling, colour).</p></li>
<li><p>altcolor  colors.Color describing the color to draw ‘low’ (some
styles) data (overridden by backwards compatible argument
with UK spelling, colour).</p></li>
<li><p>linewidth     Float describing linewidth for graph</p></li>
<li><p>center        Float setting the value at which the x-axis
crosses the y-axis (overridden by backwards
compatible argument with UK spelling, centre)</p></li>
</ul>
</dd>
</dl>
<p>Add a GraphData object to the diagram (will be stored internally).</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.GraphSet.range">
<code class="sig-name descname">range</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.GraphSet.range" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the lowest and highest base (or mark) numbers as a tuple.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.GraphSet.to_string">
<code class="sig-name descname">to_string</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">verbose=0</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.GraphSet.to_string" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a formatted string with information about the set.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>verbose - Flag indicating whether a short or complete account
of the set is required</p></li>
</ul>
</dd>
</dl>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="Bio.Graphics.GenomeDiagram.GraphData">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Graphics.GenomeDiagram.</code><code class="sig-name descname">GraphData</code><span class="sig-paren">(</span><em class="sig-param">id=None</em>, <em class="sig-param">data=None</em>, <em class="sig-param">name=None</em>, <em class="sig-param">style='bar'</em>, <em class="sig-param">color=Color(.564706</em>, <em class="sig-param">.933333</em>, <em class="sig-param">.564706</em>, <em class="sig-param">1)</em>, <em class="sig-param">altcolor=Color(.560784</em>, <em class="sig-param">.737255</em>, <em class="sig-param">.545098</em>, <em class="sig-param">1)</em>, <em class="sig-param">center=None</em>, <em class="sig-param">colour=None</em>, <em class="sig-param">altcolour=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.GraphData" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Graph Data.</p>
<dl class="simple">
<dt>Attributes:</dt><dd><ul class="simple">
<li><p>id    Unique identifier for the data</p></li>
<li><p>data  Dictionary of describing the data, keyed by position</p></li>
<li><p>name  String describing the data</p></li>
<li><p>style String (‘bar’, ‘heat’, ‘line’) describing how to draw the data</p></li>
<li><p>poscolor     colors.Color for drawing high (some styles) or all
values</p></li>
<li><p>negcolor     colors.Color for drawing low values (some styles)</p></li>
<li><p>linewidth     Int, thickness to draw the line in ‘line’ styles</p></li>
</ul>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.GraphData.add_point" title="Bio.Graphics.GenomeDiagram.GraphData.add_point"><code class="xref py py-obj docutils literal notranslate"><span class="pre">add_point</span></code></a>(self, point)</p></td>
<td><p>Add a single point to the set of data as a (position, value) tuple.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.GraphData.get_data" title="Bio.Graphics.GenomeDiagram.GraphData.get_data"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_data</span></code></a>(self)</p></td>
<td><p>Return data as a list of sorted (position, value) tuples.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.GraphData.mean" title="Bio.Graphics.GenomeDiagram.GraphData.mean"><code class="xref py py-obj docutils literal notranslate"><span class="pre">mean</span></code></a>(self)</p></td>
<td><p>Return the mean value for the data points (float).</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.GraphData.quartiles" title="Bio.Graphics.GenomeDiagram.GraphData.quartiles"><code class="xref py py-obj docutils literal notranslate"><span class="pre">quartiles</span></code></a>(self)</p></td>
<td><p>Return (minimum, lowerQ, medianQ, upperQ, maximum) values as tuple.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.GraphData.range" title="Bio.Graphics.GenomeDiagram.GraphData.range"><code class="xref py py-obj docutils literal notranslate"><span class="pre">range</span></code></a>(self)</p></td>
<td><p>Return range of data as (start, end) tuple.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.GraphData.set_data" title="Bio.Graphics.GenomeDiagram.GraphData.set_data"><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_data</span></code></a>(self, data)</p></td>
<td><p>Add data as a list of (position, value) tuples.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.GraphData.stdev" title="Bio.Graphics.GenomeDiagram.GraphData.stdev"><code class="xref py py-obj docutils literal notranslate"><span class="pre">stdev</span></code></a>(self)</p></td>
<td><p>Return the sample standard deviation for the data (float).</p></td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.GraphData.add_point">
<code class="sig-name descname">add_point</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">point</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.GraphData.add_point" title="Permalink to this definition">¶</a></dt>
<dd><p>Add a single point to the set of data as a (position, value) tuple.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.GraphData.get_data">
<code class="sig-name descname">get_data</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.GraphData.get_data" title="Permalink to this definition">¶</a></dt>
<dd><p>Return data as a list of sorted (position, value) tuples.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.GraphData.mean">
<code class="sig-name descname">mean</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.GraphData.mean" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the mean value for the data points (float).</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.GraphData.quartiles">
<code class="sig-name descname">quartiles</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.GraphData.quartiles" title="Permalink to this definition">¶</a></dt>
<dd><p>Return (minimum, lowerQ, medianQ, upperQ, maximum) values as tuple.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.GraphData.range">
<code class="sig-name descname">range</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.GraphData.range" title="Permalink to this definition">¶</a></dt>
<dd><p>Return range of data as (start, end) tuple.</p>
<p>Returns the range of the data, i.e. its start and end points on
the genome as a (start, end) tuple.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.GraphData.set_data">
<code class="sig-name descname">set_data</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">data</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.GraphData.set_data" title="Permalink to this definition">¶</a></dt>
<dd><p>Add data as a list of (position, value) tuples.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.GraphData.stdev">
<code class="sig-name descname">stdev</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.GraphData.stdev" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the sample standard deviation for the data (float).</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="Bio.Graphics.GenomeDiagram.CrossLink">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Graphics.GenomeDiagram.</code><code class="sig-name descname">CrossLink</code><span class="sig-paren">(</span><em class="sig-param">featureA</em>, <em class="sig-param">featureB</em>, <em class="sig-param">color=Color(.564706</em>, <em class="sig-param">.933333</em>, <em class="sig-param">.564706</em>, <em class="sig-param">1)</em>, <em class="sig-param">border=None</em>, <em class="sig-param">flip=False</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.CrossLink" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Hold information for drawing a cross link between features.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.CrossLink.endA" title="Bio.Graphics.GenomeDiagram.CrossLink.endA"><code class="xref py py-obj docutils literal notranslate"><span class="pre">endA</span></code></a></dt><dd><p>End position of Feature A.</p>
</dd>
<dt><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.CrossLink.endB" title="Bio.Graphics.GenomeDiagram.CrossLink.endB"><code class="xref py py-obj docutils literal notranslate"><span class="pre">endB</span></code></a></dt><dd><p>End position of Feature B.</p>
</dd>
<dt><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.CrossLink.startA" title="Bio.Graphics.GenomeDiagram.CrossLink.startA"><code class="xref py py-obj docutils literal notranslate"><span class="pre">startA</span></code></a></dt><dd><p>Start position of Feature A.</p>
</dd>
<dt><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.CrossLink.startB" title="Bio.Graphics.GenomeDiagram.CrossLink.startB"><code class="xref py py-obj docutils literal notranslate"><span class="pre">startB</span></code></a></dt><dd><p>Start position of Feature B.</p>
</dd>
</dl>
</dd>
</dl>
<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.CrossLink.endA">
<em class="property">property </em><code class="sig-name descname">endA</code><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.CrossLink.endA" title="Permalink to this definition">¶</a></dt>
<dd><p>End position of Feature A.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.CrossLink.endB">
<em class="property">property </em><code class="sig-name descname">endB</code><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.CrossLink.endB" title="Permalink to this definition">¶</a></dt>
<dd><p>End position of Feature B.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.CrossLink.startA">
<em class="property">property </em><code class="sig-name descname">startA</code><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.CrossLink.startA" title="Permalink to this definition">¶</a></dt>
<dd><p>Start position of Feature A.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.CrossLink.startB">
<em class="property">property </em><code class="sig-name descname">startB</code><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.CrossLink.startB" title="Permalink to this definition">¶</a></dt>
<dd><p>Start position of Feature B.</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="Bio.Graphics.GenomeDiagram.ColorTranslator">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Graphics.GenomeDiagram.</code><code class="sig-name descname">ColorTranslator</code><span class="sig-paren">(</span><em class="sig-param">filename=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.ColorTranslator" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Class providing methods for translating representations of color into.</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Graphics</span> <span class="k">import</span> <span class="n">GenomeDiagram</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gdct</span><span class="o">=</span><span class="n">GenomeDiagram</span><span class="o">.</span><span class="n">_Colors</span><span class="o">.</span><span class="n">ColorTranslator</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">gdct</span><span class="o">.</span><span class="n">float1_color</span><span class="p">((</span><span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">)))</span>
<span class="go">Color(.5,.5,.5,1)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">gdct</span><span class="o">.</span><span class="n">int255_color</span><span class="p">((</span><span class="mi">1</span><span class="p">,</span> <span class="mi">75</span><span class="p">,</span> <span class="mi">240</span><span class="p">)))</span>
<span class="go">Color(.003922,.294118,.941176,1)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">gdct</span><span class="o">.</span><span class="n">artemis_color</span><span class="p">(</span><span class="mi">7</span><span class="p">))</span>
<span class="go">Color(1,1,0,1)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">gdct</span><span class="o">.</span><span class="n">scheme_color</span><span class="p">(</span><span class="mi">2</span><span class="p">))</span>
<span class="go">Color(1,0,0,1)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gdct</span><span class="o">.</span><span class="n">get_artemis_colorscheme</span><span class="p">()</span>
<span class="go">{0: (Color(1,1,1,1), &#39;pathogenicity, adaptation, chaperones&#39;), 1: (Color(.39,.39,.39,1), &#39;energy metabolism&#39;), 2: (Color(1,0,0,1), &#39;information transfer&#39;), 3: (Color(0,1,0,1), &#39;surface&#39;), 4: (Color(0,0,1,1), &#39;stable RNA&#39;), 5: (Color(0,1,1,1), &#39;degradation of large molecules&#39;), 6: (Color(1,0,1,1), &#39;degradation of small molecules&#39;), 7: (Color(1,1,0,1), &#39;central/intermediary/miscellaneous metabolism&#39;), 8: (Color(.6,.98,.6,1), &#39;unknown&#39;), 9: (Color(.53,.81,.98,1), &#39;regulators&#39;), 10: (Color(1,.65,0,1), &#39;conserved hypotheticals&#39;), 11: (Color(.78,.59,.39,1), &#39;pseudogenes and partial genes&#39;), 12: (Color(1,.78,.78,1), &#39;phage/IS elements&#39;), 13: (Color(.7,.7,.7,1), &#39;some miscellaneous information&#39;), 14: (Color(0,0,0,1), &#39;&#39;), 15: (Color(1,.25,.25,1), &#39;secondary metabolism&#39;), 16: (Color(1,.5,.5,1), &#39;&#39;), 17: (Color(1,.75,.75,1), &#39;&#39;)}</span>
</pre></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">gdct</span><span class="o">.</span><span class="n">translate</span><span class="p">((</span><span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">,</span> <span class="mf">0.5</span><span class="p">)))</span>
<span class="go">Color(.5,.5,.5,1)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">gdct</span><span class="o">.</span><span class="n">translate</span><span class="p">((</span><span class="mi">1</span><span class="p">,</span> <span class="mi">75</span><span class="p">,</span> <span class="mi">240</span><span class="p">)))</span>
<span class="go">Color(.003922,.294118,.941176,1)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">gdct</span><span class="o">.</span><span class="n">translate</span><span class="p">(</span><span class="mi">7</span><span class="p">))</span>
<span class="go">Color(1,1,0,1)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">gdct</span><span class="o">.</span><span class="n">translate</span><span class="p">(</span><span class="mi">2</span><span class="p">))</span>
<span class="go">Color(1,0,0,1)</span>
</pre></div>
</div>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.ColorTranslator.artemis_color" title="Bio.Graphics.GenomeDiagram.ColorTranslator.artemis_color"><code class="xref py py-obj docutils literal notranslate"><span class="pre">artemis_color</span></code></a>(self, value)</p></td>
<td><p>Artemis color (integer) to ReportLab Color object.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.ColorTranslator.float1_color" title="Bio.Graphics.GenomeDiagram.ColorTranslator.float1_color"><code class="xref py py-obj docutils literal notranslate"><span class="pre">float1_color</span></code></a>(self, values)</p></td>
<td><p>Map float (red, green, blue) tuple to a ReportLab Color object.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.ColorTranslator.get_artemis_colorscheme" title="Bio.Graphics.GenomeDiagram.ColorTranslator.get_artemis_colorscheme"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_artemis_colorscheme</span></code></a>(self)</p></td>
<td><p>Return the Artemis color scheme as a dictionary.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.ColorTranslator.get_colorscheme" title="Bio.Graphics.GenomeDiagram.ColorTranslator.get_colorscheme"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_colorscheme</span></code></a>(self)</p></td>
<td><p>Return the user-defined color scheme as a dictionary.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.ColorTranslator.int255_color" title="Bio.Graphics.GenomeDiagram.ColorTranslator.int255_color"><code class="xref py py-obj docutils literal notranslate"><span class="pre">int255_color</span></code></a>(self, values)</p></td>
<td><p>Map integer (red, green, blue) tuple to a ReportLab Color object.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.ColorTranslator.read_colorscheme" title="Bio.Graphics.GenomeDiagram.ColorTranslator.read_colorscheme"><code class="xref py py-obj docutils literal notranslate"><span class="pre">read_colorscheme</span></code></a>(self, filename)</p></td>
<td><p>Load colour scheme from file.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.ColorTranslator.scheme_color" title="Bio.Graphics.GenomeDiagram.ColorTranslator.scheme_color"><code class="xref py py-obj docutils literal notranslate"><span class="pre">scheme_color</span></code></a>(self, value)</p></td>
<td><p>Map a user-defined color integer to a ReportLab Color object.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Graphics.GenomeDiagram.ColorTranslator.translate" title="Bio.Graphics.GenomeDiagram.ColorTranslator.translate"><code class="xref py py-obj docutils literal notranslate"><span class="pre">translate</span></code></a>(self[, color, colour])</p></td>
<td><p>Translate a color into a ReportLab Color object.</p></td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.ColorTranslator.artemis_color">
<code class="sig-name descname">artemis_color</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">value</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.ColorTranslator.artemis_color" title="Permalink to this definition">¶</a></dt>
<dd><p>Artemis color (integer) to ReportLab Color object.</p>
<p>Arguments:</p>
<ul class="simple">
<li><p>value: An int representing a functional class in the Artemis
color scheme (see www.sanger.ac.uk for a description),
or a string from a GenBank feature annotation for the
color which may be dot delimited (in which case the
first value is used).</p></li>
</ul>
<p>Takes an int representing a functional class in the Artemis color
scheme, and returns the appropriate colors.Color object</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.ColorTranslator.float1_color">
<code class="sig-name descname">float1_color</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">values</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.ColorTranslator.float1_color" title="Permalink to this definition">¶</a></dt>
<dd><p>Map float (red, green, blue) tuple to a ReportLab Color object.</p>
<ul class="simple">
<li><p>values: A tuple of (red, green, blue) intensities as floats
in the range 0 -&gt; 1</p></li>
</ul>
<p>Takes a tuple of (red, green, blue) intensity values in the range
0 -&gt; 1 and returns an appropriate colors.Color object.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.ColorTranslator.get_artemis_colorscheme">
<code class="sig-name descname">get_artemis_colorscheme</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.ColorTranslator.get_artemis_colorscheme" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the Artemis color scheme as a dictionary.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.ColorTranslator.get_colorscheme">
<code class="sig-name descname">get_colorscheme</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.ColorTranslator.get_colorscheme" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the user-defined color scheme as a dictionary.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.ColorTranslator.int255_color">
<code class="sig-name descname">int255_color</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">values</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.ColorTranslator.int255_color" title="Permalink to this definition">¶</a></dt>
<dd><p>Map integer (red, green, blue) tuple to a ReportLab Color object.</p>
<ul class="simple">
<li><p>values: A tuple of (red, green, blue) intensities as
integers in the range 0-&gt;255</p></li>
</ul>
<p>Takes a tuple of (red, green, blue) intensity values in the range
0 -&gt; 255 and returns an appropriate colors.Color object.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.ColorTranslator.read_colorscheme">
<code class="sig-name descname">read_colorscheme</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">filename</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.ColorTranslator.read_colorscheme" title="Permalink to this definition">¶</a></dt>
<dd><p>Load colour scheme from file.</p>
<p>Reads information from a file containing color information and stores
it internally.</p>
<p>Argument filename is the location of a file defining colors in
tab-separated format plaintext as:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="n">INT</span> \<span class="n">t</span> <span class="n">RED</span> \<span class="n">t</span> <span class="n">GREEN</span> \<span class="n">t</span> <span class="n">BLUE</span> \<span class="n">t</span> <span class="n">Comment</span>
</pre></div>
</div>
<p>Where RED, GREEN and BLUE are intensities in the range 0 -&gt; 255, e.g.:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="mi">2</span> \<span class="n">t</span> <span class="mi">255</span> \<span class="n">t</span> <span class="mi">0</span> \<span class="n">t</span> <span class="mi">0</span> \<span class="n">t</span> <span class="n">Red</span><span class="p">:</span> <span class="n">Information</span> <span class="n">transfer</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.ColorTranslator.scheme_color">
<code class="sig-name descname">scheme_color</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">value</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.ColorTranslator.scheme_color" title="Permalink to this definition">¶</a></dt>
<dd><p>Map a user-defined color integer to a ReportLab Color object.</p>
<ul class="simple">
<li><p>value: An int representing a single color in the user-defined
color scheme</p></li>
</ul>
<p>Takes an int representing a user-defined color and returns the
appropriate colors.Color object.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Graphics.GenomeDiagram.ColorTranslator.translate">
<code class="sig-name descname">translate</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">color=None</em>, <em class="sig-param">colour=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Graphics.GenomeDiagram.ColorTranslator.translate" title="Permalink to this definition">¶</a></dt>
<dd><p>Translate a color into a ReportLab Color object.</p>
<p>Arguments:</p>
<ul class="simple">
<li><p>color - Color defined as an int, a tuple of three ints 0-&gt;255
or a tuple of three floats 0 -&gt; 1, or a string giving
one of the named colors defined by ReportLab, or a
ReportLab color object (returned as is).</p></li>
<li><p>colour - Backards compatible alias using UK spelling (which
will over-ride any color argument).</p></li>
</ul>
<p>Returns a colors.Color object, determined semi-intelligently
depending on the input values</p>
</dd></dl>

</dd></dl>

</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="Bio.Graphics.BasicChromosome.html" class="btn btn-neutral float-right" title="Bio.Graphics.BasicChromosome module" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="Bio.Graphics.html" class="btn btn-neutral float-left" title="Bio.Graphics package" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 1999-2017, The Biopython Contributors

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>