

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Bio.SeqRecord module &mdash; Biopython 1.77.dev0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Bio.bgzf module" href="Bio.bgzf.html" />
    <link rel="prev" title="Bio.SeqFeature module" href="Bio.SeqFeature.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search"  style="background: #10100F" >
          

          
            <a href="index.html">
          

          
            
            <img src="_static/biopython_logo.svg" class="logo" alt="Logo"/>
          
          </a>

          
            
            
              <div class="version">
                1.77.dev0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">API Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="Bio.html">Bio package</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="Bio.html#subpackages">Subpackages</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="Bio.html#submodules">Submodules</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="Bio.File.html">Bio.File module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.LogisticRegression.html">Bio.LogisticRegression module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.MarkovModel.html">Bio.MarkovModel module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.MaxEntropy.html">Bio.MaxEntropy module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.NaiveBayes.html">Bio.NaiveBayes module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Seq.html">Bio.Seq module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SeqFeature.html">Bio.SeqFeature module</a></li>
<li class="toctree-l3 current"><a class="current reference internal" href="#">Bio.SeqRecord module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.bgzf.html">Bio.bgzf module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.kNN.html">Bio.kNN module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.pairwise2.html">Bio.pairwise2 module</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="Bio.html#module-Bio">Module contents</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="BioSQL.html">BioSQL package</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Biopython</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
          <li><a href="Bio.html">Bio package</a> &raquo;</li>
        
      <li>Bio.SeqRecord module</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            
              <!-- User defined GitHub URL -->
              <a href="https://github.com/biopython/biopython/blob/master/Bio/SeqRecord.py" class="fa fa-github"> Edit on GitHub</a>
            
          
        
      </li>
    
  </ul>

  
  <div class="rst-breadcrumbs-buttons" role="navigation" aria-label="breadcrumb navigation">
      
        <a href="Bio.bgzf.html" class="btn btn-neutral float-right" title="Bio.bgzf module" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="Bio.SeqFeature.html" class="btn btn-neutral float-left" title="Bio.SeqFeature module" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
  </div>
  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="module-Bio.SeqRecord">
<span id="bio-seqrecord-module"></span><h1>Bio.SeqRecord module<a class="headerlink" href="#module-Bio.SeqRecord" title="Permalink to this headline">¶</a></h1>
<p>Represent a Sequence Record, a sequence with annotation.</p>
<dl class="class">
<dt id="Bio.SeqRecord.SeqRecord">
<em class="property">class </em><code class="sig-prename descclassname">Bio.SeqRecord.</code><code class="sig-name descname">SeqRecord</code><span class="sig-paren">(</span><em class="sig-param">seq</em>, <em class="sig-param">id='&lt;unknown id&gt;'</em>, <em class="sig-param">name='&lt;unknown name&gt;'</em>, <em class="sig-param">description='&lt;unknown description&gt;'</em>, <em class="sig-param">dbxrefs=None</em>, <em class="sig-param">features=None</em>, <em class="sig-param">annotations=None</em>, <em class="sig-param">letter_annotations=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.SeqRecord.SeqRecord" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>A SeqRecord object holds a sequence and information about it.</p>
<dl class="simple">
<dt>Main attributes:</dt><dd><ul class="simple">
<li><p>id          - Identifier such as a locus tag (string)</p></li>
<li><p>seq         - The sequence itself (Seq object or similar)</p></li>
</ul>
</dd>
<dt>Additional attributes:</dt><dd><ul class="simple">
<li><p>name        - Sequence name, e.g. gene name (string)</p></li>
<li><p>description - Additional text (string)</p></li>
<li><p>dbxrefs     - List of database cross references (list of strings)</p></li>
<li><p>features    - Any (sub)features defined (list of SeqFeature objects)</p></li>
<li><p>annotations - Further information about the whole sequence (dictionary).
Most entries are strings, or lists of strings.</p></li>
<li><p>letter_annotations - Per letter/symbol annotation (restricted
dictionary). This holds Python sequences (lists, strings
or tuples) whose length matches that of the sequence.
A typical use would be to hold a list of integers
representing sequencing quality scores, or a string
representing the secondary structure.</p></li>
</ul>
</dd>
</dl>
<p>You will typically use Bio.SeqIO to read in sequences from files as
SeqRecord objects.  However, you may want to create your own SeqRecord
objects directly (see the __init__ method for further details):</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.SeqRecord</span> <span class="kn">import</span> <span class="n">SeqRecord</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="kn">import</span> <span class="n">IUPAC</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">record</span> <span class="o">=</span> <span class="n">SeqRecord</span><span class="p">(</span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF&quot;</span><span class="p">,</span>
<span class="gp">... </span>                        <span class="n">IUPAC</span><span class="o">.</span><span class="n">protein</span><span class="p">),</span>
<span class="gp">... </span>                   <span class="nb">id</span><span class="o">=</span><span class="s2">&quot;YP_025292.1&quot;</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s2">&quot;HokC&quot;</span><span class="p">,</span>
<span class="gp">... </span>                   <span class="n">description</span><span class="o">=</span><span class="s2">&quot;toxic membrane protein&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">record</span><span class="p">)</span>
<span class="go">ID: YP_025292.1</span>
<span class="go">Name: HokC</span>
<span class="go">Description: toxic membrane protein</span>
<span class="go">Number of features: 0</span>
<span class="go">Seq(&#39;MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF&#39;, IUPACProtein())</span>
</pre></div>
</div>
<p>If you want to save SeqRecord objects to a sequence file, use Bio.SeqIO
for this.  For the special case where you want the SeqRecord turned into
a string in a particular file format there is a format method which uses
Bio.SeqIO internally:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">record</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="s2">&quot;fasta&quot;</span><span class="p">))</span>
<span class="go">&gt;YP_025292.1 toxic membrane protein</span>
<span class="go">MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF</span>
</pre></div>
</div>
<p>You can also do things like slicing a SeqRecord, checking its length, etc</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">record</span><span class="p">)</span>
<span class="go">44</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">edited</span> <span class="o">=</span> <span class="n">record</span><span class="p">[:</span><span class="mi">10</span><span class="p">]</span> <span class="o">+</span> <span class="n">record</span><span class="p">[</span><span class="mi">11</span><span class="p">:]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">edited</span><span class="o">.</span><span class="n">seq</span><span class="p">)</span>
<span class="go">MKQHKAMIVAIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">record</span><span class="o">.</span><span class="n">seq</span><span class="p">)</span>
<span class="go">MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF</span>
</pre></div>
</div>
<dl class="method">
<dt id="Bio.SeqRecord.SeqRecord.format">
<code class="sig-name descname">format</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">format</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.SeqRecord.SeqRecord.format" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the record as a string in the specified file format.</p>
<p>The format should be a lower case string supported as an output
format by Bio.SeqIO, which is used to turn the SeqRecord into a
string.  e.g.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.SeqRecord</span> <span class="kn">import</span> <span class="n">SeqRecord</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="kn">import</span> <span class="n">IUPAC</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">record</span> <span class="o">=</span> <span class="n">SeqRecord</span><span class="p">(</span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF&quot;</span><span class="p">,</span>
<span class="gp">... </span>                        <span class="n">IUPAC</span><span class="o">.</span><span class="n">protein</span><span class="p">),</span>
<span class="gp">... </span>                   <span class="nb">id</span><span class="o">=</span><span class="s2">&quot;YP_025292.1&quot;</span><span class="p">,</span> <span class="n">name</span><span class="o">=</span><span class="s2">&quot;HokC&quot;</span><span class="p">,</span>
<span class="gp">... </span>                   <span class="n">description</span><span class="o">=</span><span class="s2">&quot;toxic membrane protein&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">record</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="s2">&quot;fasta&quot;</span><span class="p">)</span>
<span class="go">&#39;&gt;YP_025292.1 toxic membrane protein\nMKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF\n&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">record</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="s2">&quot;fasta&quot;</span><span class="p">))</span>
<span class="go">&gt;YP_025292.1 toxic membrane protein</span>
<span class="go">MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF</span>
</pre></div>
</div>
<p>The Python print function automatically appends a new line, meaning
in this example a blank line is shown.  If you look at the string
representation you can see there is a trailing new line (shown as
slash n) which is important when writing to a file or if
concatenating multiple sequence strings together.</p>
<p>Note that this method will NOT work on every possible file format
supported by Bio.SeqIO (e.g. some are for multiple sequences only,
and binary formats are not supported).</p>
</dd></dl>

<dl class="method">
<dt id="Bio.SeqRecord.SeqRecord.letter_annotations">
<em class="property">property </em><code class="sig-name descname">letter_annotations</code><a class="headerlink" href="#Bio.SeqRecord.SeqRecord.letter_annotations" title="Permalink to this definition">¶</a></dt>
<dd><p>Dictionary of per-letter-annotation for the sequence.</p>
<p>For example, this can hold quality scores used in FASTQ or QUAL files.
Consider this example using Bio.SeqIO to read in an example Solexa
variant FASTQ file as a SeqRecord:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio</span> <span class="kn">import</span> <span class="n">SeqIO</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">record</span> <span class="o">=</span> <span class="n">SeqIO</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="s2">&quot;Quality/solexa_faked.fastq&quot;</span><span class="p">,</span> <span class="s2">&quot;fastq-solexa&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s2">&quot;</span><span class="si">%s</span><span class="s2"> </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">record</span><span class="o">.</span><span class="n">id</span><span class="p">,</span> <span class="n">record</span><span class="o">.</span><span class="n">seq</span><span class="p">))</span>
<span class="go">slxa_0001_1_0001_01 ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTNNNNNN</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">record</span><span class="o">.</span><span class="n">letter_annotations</span><span class="p">))</span>
<span class="go">[&#39;solexa_quality&#39;]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">record</span><span class="o">.</span><span class="n">letter_annotations</span><span class="p">[</span><span class="s2">&quot;solexa_quality&quot;</span><span class="p">])</span>
<span class="go">[40, 39, 38, 37, 36, 35, 34, 33, 32, 31, 30, 29, 28, 27, 26, 25, 24, 23, 22, 21, 20, 19, 18, 17, 16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0, -1, -2, -3, -4, -5]</span>
</pre></div>
</div>
<p>The letter_annotations get sliced automatically if you slice the
parent SeqRecord, for example taking the last ten bases:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">sub_record</span> <span class="o">=</span> <span class="n">record</span><span class="p">[</span><span class="o">-</span><span class="mi">10</span><span class="p">:]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s2">&quot;</span><span class="si">%s</span><span class="s2"> </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">sub_record</span><span class="o">.</span><span class="n">id</span><span class="p">,</span> <span class="n">sub_record</span><span class="o">.</span><span class="n">seq</span><span class="p">))</span>
<span class="go">slxa_0001_1_0001_01 ACGTNNNNNN</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">sub_record</span><span class="o">.</span><span class="n">letter_annotations</span><span class="p">[</span><span class="s2">&quot;solexa_quality&quot;</span><span class="p">])</span>
<span class="go">[4, 3, 2, 1, 0, -1, -2, -3, -4, -5]</span>
</pre></div>
</div>
<p>Any python sequence (i.e. list, tuple or string) can be recorded in
the SeqRecord’s letter_annotations dictionary as long as the length
matches that of the SeqRecord’s sequence.  e.g.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">sub_record</span><span class="o">.</span><span class="n">letter_annotations</span><span class="p">)</span>
<span class="go">1</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">sub_record</span><span class="o">.</span><span class="n">letter_annotations</span><span class="p">[</span><span class="s2">&quot;dummy&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;abcdefghij&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">sub_record</span><span class="o">.</span><span class="n">letter_annotations</span><span class="p">)</span>
<span class="go">2</span>
</pre></div>
</div>
<p>You can delete entries from the letter_annotations dictionary as usual:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="k">del</span> <span class="n">sub_record</span><span class="o">.</span><span class="n">letter_annotations</span><span class="p">[</span><span class="s2">&quot;solexa_quality&quot;</span><span class="p">]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">sub_record</span><span class="o">.</span><span class="n">letter_annotations</span>
<span class="go">{&#39;dummy&#39;: &#39;abcdefghij&#39;}</span>
</pre></div>
</div>
<p>You can completely clear the dictionary easily as follows:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">sub_record</span><span class="o">.</span><span class="n">letter_annotations</span> <span class="o">=</span> <span class="p">{}</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">sub_record</span><span class="o">.</span><span class="n">letter_annotations</span>
<span class="go">{}</span>
</pre></div>
</div>
<p>Note that if replacing the record’s sequence with a sequence of a
different length you must first clear the letter_annotations dict.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.SeqRecord.SeqRecord.lower">
<code class="sig-name descname">lower</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.SeqRecord.SeqRecord.lower" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a copy of the record with a lower case sequence.</p>
<p>All the annotation is preserved unchanged. e.g.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio</span> <span class="kn">import</span> <span class="n">SeqIO</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">record</span> <span class="o">=</span> <span class="n">SeqIO</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="s2">&quot;Fasta/aster.pro&quot;</span><span class="p">,</span> <span class="s2">&quot;fasta&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">record</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="s2">&quot;fasta&quot;</span><span class="p">))</span>
<span class="go">&gt;gi|3298468|dbj|BAA31520.1| SAMIPF</span>
<span class="go">GGHVNPAVTFGAFVGGNITLLRGIVYIIAQLLGSTVACLLLKFVTNDMAVGVFSLSAGVG</span>
<span class="go">VTNALVFEIVMTFGLVYTVYATAIDPKKGSLGTIAPIAIGFIVGANI</span>

<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">record</span><span class="o">.</span><span class="n">lower</span><span class="p">()</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="s2">&quot;fasta&quot;</span><span class="p">))</span>
<span class="go">&gt;gi|3298468|dbj|BAA31520.1| SAMIPF</span>
<span class="go">gghvnpavtfgafvggnitllrgivyiiaqllgstvaclllkfvtndmavgvfslsagvg</span>
<span class="go">vtnalvfeivmtfglvytvyataidpkkgslgtiapiaigfivgani</span>
</pre></div>
</div>
<p>To take a more annotation rich example,</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio</span> <span class="kn">import</span> <span class="n">SeqIO</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">old</span> <span class="o">=</span> <span class="n">SeqIO</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="s2">&quot;EMBL/TRBG361.embl&quot;</span><span class="p">,</span> <span class="s2">&quot;embl&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">old</span><span class="o">.</span><span class="n">features</span><span class="p">)</span>
<span class="go">3</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">new</span> <span class="o">=</span> <span class="n">old</span><span class="o">.</span><span class="n">lower</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">old</span><span class="o">.</span><span class="n">features</span><span class="p">)</span> <span class="o">==</span> <span class="nb">len</span><span class="p">(</span><span class="n">new</span><span class="o">.</span><span class="n">features</span><span class="p">)</span>
<span class="go">True</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">old</span><span class="o">.</span><span class="n">annotations</span><span class="p">[</span><span class="s2">&quot;organism&quot;</span><span class="p">]</span> <span class="o">==</span> <span class="n">new</span><span class="o">.</span><span class="n">annotations</span><span class="p">[</span><span class="s2">&quot;organism&quot;</span><span class="p">]</span>
<span class="go">True</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">old</span><span class="o">.</span><span class="n">dbxrefs</span> <span class="o">==</span> <span class="n">new</span><span class="o">.</span><span class="n">dbxrefs</span>
<span class="go">True</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="Bio.SeqRecord.SeqRecord.reverse_complement">
<code class="sig-name descname">reverse_complement</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">id=False</em>, <em class="sig-param">name=False</em>, <em class="sig-param">description=False</em>, <em class="sig-param">features=True</em>, <em class="sig-param">annotations=False</em>, <em class="sig-param">letter_annotations=True</em>, <em class="sig-param">dbxrefs=False</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.SeqRecord.SeqRecord.reverse_complement" title="Permalink to this definition">¶</a></dt>
<dd><p>Return new SeqRecord with reverse complement sequence.</p>
<p>By default the new record does NOT preserve the sequence identifier,
name, description, general annotation or database cross-references -
these are unlikely to apply to the reversed sequence.</p>
<p>You can specify the returned record’s id, name and description as
strings, or True to keep that of the parent, or False for a default.</p>
<p>You can specify the returned record’s features with a list of
SeqFeature objects, or True to keep that of the parent, or False to
omit them. The default is to keep the original features (with the
strand and locations adjusted).</p>
<p>You can also specify both the returned record’s annotations and
letter_annotations as dictionaries, True to keep that of the parent,
or False to omit them. The default is to keep the original
annotations (with the letter annotations reversed).</p>
<p>To show what happens to the pre-letter annotations, consider an
example Solexa variant FASTQ file with a single entry, which we’ll
read in as a SeqRecord:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio</span> <span class="kn">import</span> <span class="n">SeqIO</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">record</span> <span class="o">=</span> <span class="n">SeqIO</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="s2">&quot;Quality/solexa_faked.fastq&quot;</span><span class="p">,</span> <span class="s2">&quot;fastq-solexa&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s2">&quot;</span><span class="si">%s</span><span class="s2"> </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">record</span><span class="o">.</span><span class="n">id</span><span class="p">,</span> <span class="n">record</span><span class="o">.</span><span class="n">seq</span><span class="p">))</span>
<span class="go">slxa_0001_1_0001_01 ACGTACGTACGTACGTACGTACGTACGTACGTACGTACGTNNNNNN</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="nb">list</span><span class="p">(</span><span class="n">record</span><span class="o">.</span><span class="n">letter_annotations</span><span class="p">))</span>
<span class="go">[&#39;solexa_quality&#39;]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">record</span><span class="o">.</span><span class="n">letter_annotations</span><span class="p">[</span><span class="s2">&quot;solexa_quality&quot;</span><span class="p">])</span>
<span class="go">[40, 39, 38, 37, 36, 35, 34, 33, 32, 31, 30, 29, 28, 27, 26, 25, 24, 23, 22, 21, 20, 19, 18, 17, 16, 15, 14, 13, 12, 11, 10, 9, 8, 7, 6, 5, 4, 3, 2, 1, 0, -1, -2, -3, -4, -5]</span>
</pre></div>
</div>
<p>Now take the reverse complement, here we explicitly give a new
identifier (the old identifier with a suffix):</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">rc_record</span> <span class="o">=</span> <span class="n">record</span><span class="o">.</span><span class="n">reverse_complement</span><span class="p">(</span><span class="nb">id</span><span class="o">=</span><span class="n">record</span><span class="o">.</span><span class="n">id</span> <span class="o">+</span> <span class="s2">&quot;_rc&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s2">&quot;</span><span class="si">%s</span><span class="s2"> </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">rc_record</span><span class="o">.</span><span class="n">id</span><span class="p">,</span> <span class="n">rc_record</span><span class="o">.</span><span class="n">seq</span><span class="p">))</span>
<span class="go">slxa_0001_1_0001_01_rc NNNNNNACGTACGTACGTACGTACGTACGTACGTACGTACGTACGT</span>
</pre></div>
</div>
<p>Notice that the per-letter-annotations have also been reversed,
although this may not be appropriate for all cases.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">rc_record</span><span class="o">.</span><span class="n">letter_annotations</span><span class="p">[</span><span class="s2">&quot;solexa_quality&quot;</span><span class="p">])</span>
<span class="go">[-5, -4, -3, -2, -1, 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24, 25, 26, 27, 28, 29, 30, 31, 32, 33, 34, 35, 36, 37, 38, 39, 40]</span>
</pre></div>
</div>
<p>Now for the features, we need a different example. Parsing a GenBank
file is probably the easiest way to get an nice example with features
in it…</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio</span> <span class="kn">import</span> <span class="n">SeqIO</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">with</span> <span class="nb">open</span><span class="p">(</span><span class="s2">&quot;GenBank/pBAD30.gb&quot;</span><span class="p">)</span> <span class="k">as</span> <span class="n">handle</span><span class="p">:</span>
<span class="gp">... </span>    <span class="n">plasmid</span> <span class="o">=</span> <span class="n">SeqIO</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="n">handle</span><span class="p">,</span> <span class="s2">&quot;gb&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s2">&quot;</span><span class="si">%s</span><span class="s2"> </span><span class="si">%i</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">plasmid</span><span class="o">.</span><span class="n">id</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">plasmid</span><span class="p">)))</span>
<span class="go">pBAD30 4923</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">plasmid</span><span class="o">.</span><span class="n">seq</span>
<span class="go">Seq(&#39;GCTAGCGGAGTGTATACTGGCTTACTATGTTGGCACTGATGAGGGTGTCAGTGA...ATG&#39;, IUPACAmbiguousDNA())</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">plasmid</span><span class="o">.</span><span class="n">features</span><span class="p">)</span>
<span class="go">13</span>
</pre></div>
</div>
<p>Now, let’s take the reverse complement of this whole plasmid:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">rc_plasmid</span> <span class="o">=</span> <span class="n">plasmid</span><span class="o">.</span><span class="n">reverse_complement</span><span class="p">(</span><span class="nb">id</span><span class="o">=</span><span class="n">plasmid</span><span class="o">.</span><span class="n">id</span><span class="o">+</span><span class="s2">&quot;_rc&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s2">&quot;</span><span class="si">%s</span><span class="s2"> </span><span class="si">%i</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">rc_plasmid</span><span class="o">.</span><span class="n">id</span><span class="p">,</span> <span class="nb">len</span><span class="p">(</span><span class="n">rc_plasmid</span><span class="p">)))</span>
<span class="go">pBAD30_rc 4923</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">rc_plasmid</span><span class="o">.</span><span class="n">seq</span>
<span class="go">Seq(&#39;CATGGGCAAATATTATACGCAAGGCGACAAGGTGCTGATGCCGCTGGCGATTCA...AGC&#39;, IUPACAmbiguousDNA())</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">rc_plasmid</span><span class="o">.</span><span class="n">features</span><span class="p">)</span>
<span class="go">13</span>
</pre></div>
</div>
<p>Let’s compare the first CDS feature - it has gone from being the
second feature (index 1) to the second last feature (index -2), its
strand has changed, and the location switched round.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">plasmid</span><span class="o">.</span><span class="n">features</span><span class="p">[</span><span class="mi">1</span><span class="p">])</span>
<span class="go">type: CDS</span>
<span class="go">location: [1081:1960](-)</span>
<span class="go">qualifiers:</span>
<span class="go">    Key: label, Value: [&#39;araC&#39;]</span>
<span class="go">    Key: note, Value: [&#39;araC regulator of the arabinose BAD promoter&#39;]</span>
<span class="go">    Key: vntifkey, Value: [&#39;4&#39;]</span>

<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">rc_plasmid</span><span class="o">.</span><span class="n">features</span><span class="p">[</span><span class="o">-</span><span class="mi">2</span><span class="p">])</span>
<span class="go">type: CDS</span>
<span class="go">location: [2963:3842](+)</span>
<span class="go">qualifiers:</span>
<span class="go">    Key: label, Value: [&#39;araC&#39;]</span>
<span class="go">    Key: note, Value: [&#39;araC regulator of the arabinose BAD promoter&#39;]</span>
<span class="go">    Key: vntifkey, Value: [&#39;4&#39;]</span>
</pre></div>
</div>
<p>You can check this new location, based on the length of the plasmid:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">plasmid</span><span class="p">)</span> <span class="o">-</span> <span class="mi">1081</span>
<span class="go">3842</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">plasmid</span><span class="p">)</span> <span class="o">-</span> <span class="mi">1960</span>
<span class="go">2963</span>
</pre></div>
</div>
<p>Note that if the SeqFeature annotation includes any strand specific
information (e.g. base changes for a SNP), this information is not
amended, and would need correction after the reverse complement.</p>
<p>Note trying to reverse complement a protein SeqRecord raises an
exception:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.SeqRecord</span> <span class="kn">import</span> <span class="n">SeqRecord</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="kn">import</span> <span class="n">IUPAC</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">protein_rec</span> <span class="o">=</span> <span class="n">SeqRecord</span><span class="p">(</span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;MAIVMGR&quot;</span><span class="p">,</span> <span class="n">IUPAC</span><span class="o">.</span><span class="n">protein</span><span class="p">),</span> <span class="nb">id</span><span class="o">=</span><span class="s2">&quot;Test&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">protein_rec</span><span class="o">.</span><span class="n">reverse_complement</span><span class="p">()</span>
<span class="gt">Traceback (most recent call last):</span>
   <span class="c">...</span>
<span class="gr">ValueError</span>: <span class="n">Proteins do not have complements!</span>
</pre></div>
</div>
<p>Also note you can reverse complement a SeqRecord using a MutableSeq:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.SeqRecord</span> <span class="kn">import</span> <span class="n">SeqRecord</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">MutableSeq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="kn">import</span> <span class="n">generic_dna</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">rec</span> <span class="o">=</span> <span class="n">SeqRecord</span><span class="p">(</span><span class="n">MutableSeq</span><span class="p">(</span><span class="s2">&quot;ACGT&quot;</span><span class="p">,</span> <span class="n">generic_dna</span><span class="p">),</span> <span class="nb">id</span><span class="o">=</span><span class="s2">&quot;Test&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">rec</span><span class="o">.</span><span class="n">seq</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;T&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s2">&quot;</span><span class="si">%s</span><span class="s2"> </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">rec</span><span class="o">.</span><span class="n">id</span><span class="p">,</span> <span class="n">rec</span><span class="o">.</span><span class="n">seq</span><span class="p">))</span>
<span class="go">Test TCGT</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">rc</span> <span class="o">=</span> <span class="n">rec</span><span class="o">.</span><span class="n">reverse_complement</span><span class="p">(</span><span class="nb">id</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="s2">&quot;</span><span class="si">%s</span><span class="s2"> </span><span class="si">%s</span><span class="s2">&quot;</span> <span class="o">%</span> <span class="p">(</span><span class="n">rc</span><span class="o">.</span><span class="n">id</span><span class="p">,</span> <span class="n">rc</span><span class="o">.</span><span class="n">seq</span><span class="p">))</span>
<span class="go">Test ACGA</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="Bio.SeqRecord.SeqRecord.seq">
<em class="property">property </em><code class="sig-name descname">seq</code><a class="headerlink" href="#Bio.SeqRecord.SeqRecord.seq" title="Permalink to this definition">¶</a></dt>
<dd><p>The sequence itself, as a Seq or MutableSeq object.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.SeqRecord.SeqRecord.translate">
<code class="sig-name descname">translate</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">table='Standard'</em>, <em class="sig-param">stop_symbol='*'</em>, <em class="sig-param">to_stop=False</em>, <em class="sig-param">cds=False</em>, <em class="sig-param">gap=None</em>, <em class="sig-param">id=False</em>, <em class="sig-param">name=False</em>, <em class="sig-param">description=False</em>, <em class="sig-param">features=False</em>, <em class="sig-param">annotations=False</em>, <em class="sig-param">letter_annotations=False</em>, <em class="sig-param">dbxrefs=False</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.SeqRecord.SeqRecord.translate" title="Permalink to this definition">¶</a></dt>
<dd><p>Return new SeqRecord with translated sequence.</p>
<p>This calls the record’s .seq.translate() method (which describes
the translation related arguments, like table for the genetic code),</p>
<p>By default the new record does NOT preserve the sequence identifier,
name, description, general annotation or database cross-references -
these are unlikely to apply to the translated sequence.</p>
<p>You can specify the returned record’s id, name and description as
strings, or True to keep that of the parent, or False for a default.</p>
<p>You can specify the returned record’s features with a list of
SeqFeature objects, or False (default) to omit them.</p>
<p>You can also specify both the returned record’s annotations and
letter_annotations as dictionaries, True to keep that of the parent
(annotations only), or False (default) to omit them.</p>
<p>e.g. Loading a FASTA gene and translating it,</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio</span> <span class="kn">import</span> <span class="n">SeqIO</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">gene_record</span> <span class="o">=</span> <span class="n">SeqIO</span><span class="o">.</span><span class="n">read</span><span class="p">(</span><span class="s2">&quot;Fasta/sweetpea.nu&quot;</span><span class="p">,</span> <span class="s2">&quot;fasta&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">gene_record</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="s2">&quot;fasta&quot;</span><span class="p">))</span>
<span class="go">&gt;gi|3176602|gb|U78617.1|LOU78617 Lathyrus odoratus phytochrome A (PHYA) gene, partial cds</span>
<span class="go">CAGGCTGCGCGGTTTCTATTTATGAAGAACAAGGTCCGTATGATAGTTGATTGTCATGCA</span>
<span class="go">AAACATGTGAAGGTTCTTCAAGACGAAAAACTCCCATTTGATTTGACTCTGTGCGGTTCG</span>
<span class="go">ACCTTAAGAGCTCCACATAGTTGCCATTTGCAGTACATGGCTAACATGGATTCAATTGCT</span>
<span class="go">TCATTGGTTATGGCAGTGGTCGTCAATGACAGCGATGAAGATGGAGATAGCCGTGACGCA</span>
<span class="go">GTTCTACCACAAAAGAAAAAGAGACTTTGGGGTTTGGTAGTTTGTCATAACACTACTCCG</span>
<span class="go">AGGTTTGTT</span>
</pre></div>
</div>
<p>And now translating the record, specifying the new ID and description:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">protein_record</span> <span class="o">=</span> <span class="n">gene_record</span><span class="o">.</span><span class="n">translate</span><span class="p">(</span><span class="n">table</span><span class="o">=</span><span class="mi">11</span><span class="p">,</span>
<span class="gp">... </span>                                       <span class="nb">id</span><span class="o">=</span><span class="s2">&quot;phya&quot;</span><span class="p">,</span>
<span class="gp">... </span>                                       <span class="n">description</span><span class="o">=</span><span class="s2">&quot;translation&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">protein_record</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="s2">&quot;fasta&quot;</span><span class="p">))</span>
<span class="go">&gt;phya translation</span>
<span class="go">QAARFLFMKNKVRMIVDCHAKHVKVLQDEKLPFDLTLCGSTLRAPHSCHLQYMANMDSIA</span>
<span class="go">SLVMAVVVNDSDEDGDSRDAVLPQKKKRLWGLVVCHNTTPRFV</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="Bio.SeqRecord.SeqRecord.upper">
<code class="sig-name descname">upper</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.SeqRecord.SeqRecord.upper" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a copy of the record with an upper case sequence.</p>
<p>All the annotation is preserved unchanged. e.g.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="kn">import</span> <span class="n">generic_dna</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.SeqRecord</span> <span class="kn">import</span> <span class="n">SeqRecord</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">record</span> <span class="o">=</span> <span class="n">SeqRecord</span><span class="p">(</span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;acgtACGT&quot;</span><span class="p">,</span> <span class="n">generic_dna</span><span class="p">),</span> <span class="nb">id</span><span class="o">=</span><span class="s2">&quot;Test&quot;</span><span class="p">,</span>
<span class="gp">... </span>                   <span class="n">description</span> <span class="o">=</span> <span class="s2">&quot;Made up for this example&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">record</span><span class="o">.</span><span class="n">letter_annotations</span><span class="p">[</span><span class="s2">&quot;phred_quality&quot;</span><span class="p">]</span> <span class="o">=</span> <span class="p">[</span><span class="mi">1</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="mi">4</span><span class="p">,</span> <span class="mi">5</span><span class="p">,</span> <span class="mi">6</span><span class="p">,</span> <span class="mi">7</span><span class="p">,</span> <span class="mi">8</span><span class="p">]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">record</span><span class="o">.</span><span class="n">upper</span><span class="p">()</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="s2">&quot;fastq&quot;</span><span class="p">))</span>
<span class="go">@Test Made up for this example</span>
<span class="go">ACGTACGT</span>
<span class="go">+</span>
<span class="go">&quot;#$%&amp;&#39;()</span>
</pre></div>
</div>
<p>Naturally, there is a matching lower method:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">record</span><span class="o">.</span><span class="n">lower</span><span class="p">()</span><span class="o">.</span><span class="n">format</span><span class="p">(</span><span class="s2">&quot;fastq&quot;</span><span class="p">))</span>
<span class="go">@Test Made up for this example</span>
<span class="go">acgtacgt</span>
<span class="go">+</span>
<span class="go">&quot;#$%&amp;&#39;()</span>
</pre></div>
</div>
</dd></dl>

</dd></dl>

</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="Bio.bgzf.html" class="btn btn-neutral float-right" title="Bio.bgzf module" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="Bio.SeqFeature.html" class="btn btn-neutral float-left" title="Bio.SeqFeature module" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 1999-2017, The Biopython Contributors

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>