

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Bio.Phylo.PhyloXML module &mdash; Biopython 1.76.dev0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Bio.Phylo.PhyloXMLIO module" href="Bio.Phylo.PhyloXMLIO.html" />
    <link rel="prev" title="Bio.Phylo.NexusIO module" href="Bio.Phylo.NexusIO.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search"  style="background: #10100F" >
          

          
            <a href="index.html">
          

          
            
            <img src="_static/biopython_logo.svg" class="logo" alt="Logo"/>
          
          </a>

          
            
            
              <div class="version">
                1.76.dev0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">API Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="Bio.html">Bio package</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="Bio.html#subpackages">Subpackages</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="Bio.Affy.html">Bio.Affy package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Align.html">Bio.Align package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.AlignIO.html">Bio.AlignIO package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Alphabet.html">Bio.Alphabet package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Application.html">Bio.Application package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Blast.html">Bio.Blast package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.CAPS.html">Bio.CAPS package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Cluster.html">Bio.Cluster package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Compass.html">Bio.Compass package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Crystal.html">Bio.Crystal package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Data.html">Bio.Data package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Emboss.html">Bio.Emboss package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Entrez.html">Bio.Entrez package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.ExPASy.html">Bio.ExPASy package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.FSSP.html">Bio.FSSP package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.GenBank.html">Bio.GenBank package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Geo.html">Bio.Geo package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Graphics.html">Bio.Graphics package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.HMM.html">Bio.HMM package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.KDTree.html">Bio.KDTree package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.KEGG.html">Bio.KEGG package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Medline.html">Bio.Medline package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.NMR.html">Bio.NMR package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Nexus.html">Bio.Nexus package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.PDB.html">Bio.PDB package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Pathway.html">Bio.Pathway package</a></li>
<li class="toctree-l3 current"><a class="reference internal" href="Bio.Phylo.html">Bio.Phylo package</a><ul class="current">
<li class="toctree-l4"><a class="reference internal" href="Bio.Phylo.html#subpackages">Subpackages</a></li>
<li class="toctree-l4 current"><a class="reference internal" href="Bio.Phylo.html#submodules">Submodules</a></li>
<li class="toctree-l4"><a class="reference internal" href="Bio.Phylo.html#module-Bio.Phylo">Module contents</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="Bio.PopGen.html">Bio.PopGen package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Restriction.html">Bio.Restriction package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SCOP.html">Bio.SCOP package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SVDSuperimposer.html">Bio.SVDSuperimposer package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SearchIO.html">Bio.SearchIO package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SeqIO.html">Bio.SeqIO package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SeqUtils.html">Bio.SeqUtils package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Sequencing.html">Bio.Sequencing package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Statistics.html">Bio.Statistics package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SubsMat.html">Bio.SubsMat package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SwissProt.html">Bio.SwissProt package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.TogoWS.html">Bio.TogoWS package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.UniGene.html">Bio.UniGene package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.UniProt.html">Bio.UniProt package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Wise.html">Bio.Wise package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.codonalign.html">Bio.codonalign package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.motifs.html">Bio.motifs package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.phenotype.html">Bio.phenotype package</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="Bio.html#submodules">Submodules</a></li>
<li class="toctree-l2"><a class="reference internal" href="Bio.html#module-Bio">Module contents</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="BioSQL.html">BioSQL package</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Biopython</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
          <li><a href="Bio.html">Bio package</a> &raquo;</li>
        
          <li><a href="Bio.Phylo.html">Bio.Phylo package</a> &raquo;</li>
        
      <li>Bio.Phylo.PhyloXML module</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            
              <!-- User defined GitHub URL -->
              <a href="https://github.com/biopython/biopython/blob/master/Bio/Phylo/PhyloXML.py" class="fa fa-github"> Edit on GitHub</a>
            
          
        
      </li>
    
  </ul>

  
  <div class="rst-breadcrumbs-buttons" role="navigation" aria-label="breadcrumb navigation">
      
        <a href="Bio.Phylo.PhyloXMLIO.html" class="btn btn-neutral float-right" title="Bio.Phylo.PhyloXMLIO module" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="Bio.Phylo.NexusIO.html" class="btn btn-neutral float-left" title="Bio.Phylo.NexusIO module" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
  </div>
  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="module-Bio.Phylo.PhyloXML">
<span id="bio-phylo-phyloxml-module"></span><h1>Bio.Phylo.PhyloXML module<a class="headerlink" href="#module-Bio.Phylo.PhyloXML" title="Permalink to this headline">¶</a></h1>
<p>Classes corresponding to phyloXML elements.</p>
<div class="section" id="see-also">
<h2>See Also<a class="headerlink" href="#see-also" title="Permalink to this headline">¶</a></h2>
<dl class="simple">
<dt>Official specification:</dt><dd><p><a class="reference external" href="http://phyloxml.org/">http://phyloxml.org/</a></p>
</dd>
<dt>Journal article:</dt><dd><p>Han and Zmasek (2009), <a class="reference external" href="https://doi.org/10.1186/1471-2105-10-356">https://doi.org/10.1186/1471-2105-10-356</a></p>
</dd>
</dl>
<dl class="class">
<dt id="Bio.Phylo.PhyloXML.Accession">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">Accession</code><span class="sig-paren">(</span><em class="sig-param">value</em>, <em class="sig-param">source</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Accession" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Bio.Phylo.PhyloXML.PhyloElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.PhyloXML.PhyloElement</span></code></a></p>
<p>Captures the local part in a sequence identifier.</p>
<p>Example: In <code class="docutils literal notranslate"><span class="pre">UniProtKB:P17304</span></code>, the Accession instance attribute <code class="docutils literal notranslate"><span class="pre">value</span></code>
is ‘P17304’ and the <code class="docutils literal notranslate"><span class="pre">source</span></code> attribute is ‘UniProtKB’.</p>
</dd></dl>

<dl class="class">
<dt id="Bio.Phylo.PhyloXML.Annotation">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">Annotation</code><span class="sig-paren">(</span><em class="sig-param">ref=None</em>, <em class="sig-param">source=None</em>, <em class="sig-param">evidence=None</em>, <em class="sig-param">type=None</em>, <em class="sig-param">desc=None</em>, <em class="sig-param">confidence=None</em>, <em class="sig-param">uri=None</em>, <em class="sig-param">properties=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Annotation" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Bio.Phylo.PhyloXML.PhyloElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.PhyloXML.PhyloElement</span></code></a></p>
<p>The annotation of a molecular sequence.</p>
<p>It is recommended to annotate by using the optional ‘ref’ attribute.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt>ref<span class="classifier">string</span></dt><dd><p>reference string, e.g. ‘<a class="reference external" href="GO:0008270">GO:0008270</a>’,
‘KEGG:Tetrachloroethene degradation’, ‘EC:1.1.1.1’</p>
</dd>
<dt>source<span class="classifier">string</span></dt><dd><p>plain-text source for this annotation</p>
</dd>
<dt>evidence<span class="classifier">str</span></dt><dd><p>describe evidence as free text (e.g. ‘experimental’)</p>
</dd>
<dt>desc<span class="classifier">string</span></dt><dd><p>free text description</p>
</dd>
<dt>confidence<span class="classifier">Confidence</span></dt><dd><p>state the type and value of support (type Confidence)</p>
</dd>
<dt>properties<span class="classifier">list</span></dt><dd><p>typed and referenced annotations from external resources</p>
</dd>
<dt>uri<span class="classifier">Uri</span></dt><dd><p>link</p>
</dd>
</dl>
</dd>
</dl>
<dl class="attribute">
<dt id="Bio.Phylo.PhyloXML.Annotation.re_ref">
<code class="sig-name descname">re_ref</code><em class="property"> = re.compile('[a-zA-Z0-9_]+:[a-zA-Z0-9_\\.\\-\\s]+')</em><a class="headerlink" href="#Bio.Phylo.PhyloXML.Annotation.re_ref" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="class">
<dt id="Bio.Phylo.PhyloXML.BinaryCharacters">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">BinaryCharacters</code><span class="sig-paren">(</span><em class="sig-param">type=None</em>, <em class="sig-param">gained_count=None</em>, <em class="sig-param">lost_count=None</em>, <em class="sig-param">present_count=None</em>, <em class="sig-param">absent_count=None</em>, <em class="sig-param">gained=None</em>, <em class="sig-param">lost=None</em>, <em class="sig-param">present=None</em>, <em class="sig-param">absent=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.BinaryCharacters" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Bio.Phylo.PhyloXML.PhyloElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.PhyloXML.PhyloElement</span></code></a></p>
<p>Binary characters at the root of a clade.</p>
<p>The names and/or counts of binary characters present, gained, and lost
at the root of a clade.</p>
</dd></dl>

<dl class="class">
<dt id="Bio.Phylo.PhyloXML.BranchColor">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">BranchColor</code><span class="sig-paren">(</span><em class="sig-param">*args</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.BranchColor" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Bio.Phylo.PhyloXML.PhyloElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.PhyloXML.PhyloElement</span></code></a>, <a class="reference internal" href="Bio.Phylo.BaseTree.html#Bio.Phylo.BaseTree.BranchColor" title="Bio.Phylo.BaseTree.BranchColor"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.BaseTree.BranchColor</span></code></a></p>
<p>Manage Tree branch’s color.</p>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">from_hex</span></code>(hexstr)</p></td>
<td><p>Construct a BranchColor object from a hexadecimal string.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">from_name</span></code>(colorname)</p></td>
<td><p>Construct a BranchColor object by the color’s name.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">to_hex</span></code>(self)</p></td>
<td><p>Return a 24-bit hexadecimal RGB representation of this color.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">to_rgb</span></code>(self)</p></td>
<td><p>Return a tuple of RGB values (0 to 255) representing this color.</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Phylo.PhyloXML.Clade">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">Clade</code><span class="sig-paren">(</span><em class="sig-param">branch_length=None</em>, <em class="sig-param">id_source=None</em>, <em class="sig-param">name=None</em>, <em class="sig-param">width=None</em>, <em class="sig-param">color=None</em>, <em class="sig-param">node_id=None</em>, <em class="sig-param">events=None</em>, <em class="sig-param">binary_characters=None</em>, <em class="sig-param">date=None</em>, <em class="sig-param">confidences=None</em>, <em class="sig-param">taxonomies=None</em>, <em class="sig-param">sequences=None</em>, <em class="sig-param">distributions=None</em>, <em class="sig-param">references=None</em>, <em class="sig-param">properties=None</em>, <em class="sig-param">clades=None</em>, <em class="sig-param">other=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Clade" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Bio.Phylo.PhyloXML.PhyloElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.PhyloXML.PhyloElement</span></code></a>, <a class="reference internal" href="Bio.Phylo.BaseTree.html#Bio.Phylo.BaseTree.Clade" title="Bio.Phylo.BaseTree.Clade"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.BaseTree.Clade</span></code></a></p>
<p>Describes a branch of the current phylogenetic tree.</p>
<p>Used recursively, describes the topology of a phylogenetic tree.</p>
<p>Both <code class="docutils literal notranslate"><span class="pre">color</span></code> and <code class="docutils literal notranslate"><span class="pre">width</span></code> elements should be interpreted by client code
as applying to the whole clade, including all descendents, unless
overwritten in-sub clades. This module doesn’t automatically assign these
attributes to sub-clades to achieve this cascade – and neither should you.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt>branch_length</dt><dd><p>parent branch length of this clade</p>
</dd>
<dt>id_source</dt><dd><p>link other elements to a clade (on the xml-level)</p>
</dd>
<dt>name<span class="classifier">string</span></dt><dd><p>short label for this clade</p>
</dd>
<dt>confidences<span class="classifier">list of Confidence objects</span></dt><dd><p>used to indicate the support for a clade/parent branch.</p>
</dd>
<dt>width<span class="classifier">float</span></dt><dd><p>branch width for this clade (including branch from parent)</p>
</dd>
<dt>color<span class="classifier">BranchColor</span></dt><dd><p>color used for graphical display of this clade</p>
</dd>
<dt>node_id</dt><dd><p>unique identifier for the root node of this clade</p>
</dd>
<dt>taxonomies<span class="classifier">list</span></dt><dd><p>Taxonomy objects</p>
</dd>
<dt>sequences<span class="classifier">list</span></dt><dd><p>Sequence objects</p>
</dd>
<dt>events<span class="classifier">Events</span></dt><dd><p>describe such events as gene-duplications at the root node/parent
branch of this clade</p>
</dd>
<dt>binary_characters<span class="classifier">BinaryCharacters</span></dt><dd><p>binary characters</p>
</dd>
<dt>distributions<span class="classifier">list of Distribution objects</span></dt><dd><p>distribution(s) of this clade</p>
</dd>
<dt>date<span class="classifier">Date</span></dt><dd><p>a date for the root node of this clade</p>
</dd>
<dt>references<span class="classifier">list</span></dt><dd><p>Reference objects</p>
</dd>
<dt>properties<span class="classifier">list</span></dt><dd><p>Property objects</p>
</dd>
<dt>clades<span class="classifier">list Clade objects</span></dt><dd><p>Sub-clades</p>
</dd>
<dt>other<span class="classifier">list of Other objects</span></dt><dd><p>non-phyloXML objects</p>
</dd>
</dl>
</dd>
<dt class="field-even">Attributes</dt>
<dd class="field-even"><dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">color</span></code></dt><dd><p>Branch color.</p>
</dd>
<dt><a class="reference internal" href="#Bio.Phylo.PhyloXML.Clade.confidence" title="Bio.Phylo.PhyloXML.Clade.confidence"><code class="xref py py-obj docutils literal notranslate"><span class="pre">confidence</span></code></a></dt><dd><p>Return confidence values (PRIVATE).</p>
</dd>
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">root</span></code></dt><dd><p>Allow TreeMixin methods to traverse clades properly.</p>
</dd>
<dt><a class="reference internal" href="#Bio.Phylo.PhyloXML.Clade.taxonomy" title="Bio.Phylo.PhyloXML.Clade.taxonomy"><code class="xref py py-obj docutils literal notranslate"><span class="pre">taxonomy</span></code></a></dt><dd><p>Get taxonomy list for the clade (PRIVATE).</p>
</dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">collapse</span></code>(self[, target])</p></td>
<td><p>Delete target from the tree, relinking its children to its parent.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">collapse_all</span></code>(self[, target])</p></td>
<td><p>Collapse all the descendents of this tree, leaving only terminals.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">common_ancestor</span></code>(self, targets, \*more_targets)</p></td>
<td><p>Most recent common ancestor (clade) of all the given targets.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">count_terminals</span></code>(self)</p></td>
<td><p>Count the number of terminal (leaf) nodes within this tree.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">depths</span></code>(self[, unit_branch_lengths])</p></td>
<td><p>Create a mapping of tree clades to depths (by branch length).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">distance</span></code>(self, target1[, target2])</p></td>
<td><p>Calculate the sum of the branch lengths between two targets.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">find_any</span></code>(self, \*args, \*\*kwargs)</p></td>
<td><p>Return the first element found by find_elements(), or None.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">find_clades</span></code>(self[, target, terminal, order])</p></td>
<td><p>Find each clade containing a matching element.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">find_elements</span></code>(self[, target, terminal, order])</p></td>
<td><p>Find all tree elements matching the given attributes.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Phylo.PhyloXML.Clade.from_clade" title="Bio.Phylo.PhyloXML.Clade.from_clade"><code class="xref py py-obj docutils literal notranslate"><span class="pre">from_clade</span></code></a>(clade, \*\*kwargs)</p></td>
<td><p>Create a new PhyloXML Clade from a Newick or BaseTree Clade object.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_nonterminals</span></code>(self[, order])</p></td>
<td><p>Get a list of all of this tree’s nonterminal (internal) nodes.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_path</span></code>(self[, target])</p></td>
<td><p>List the clades directly between this root and the given target.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_terminals</span></code>(self[, order])</p></td>
<td><p>Get a list of all of this tree’s terminal (leaf) nodes.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">is_bifurcating</span></code>(self)</p></td>
<td><p>Return True if tree downstream of node is strictly bifurcating.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">is_monophyletic</span></code>(self, terminals, …)</p></td>
<td><p>MRCA of terminals if they comprise a complete subclade, or False.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">is_parent_of</span></code>(self[, target])</p></td>
<td><p>Check if target is a descendent of this tree.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">is_preterminal</span></code>(self)</p></td>
<td><p>Check if all direct descendents are terminal.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">is_terminal</span></code>(self)</p></td>
<td><p>Check if this is a terminal (leaf) node.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">ladderize</span></code>(self[, reverse])</p></td>
<td><p>Sort clades in-place according to the number of terminal nodes.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">prune</span></code>(self[, target])</p></td>
<td><p>Prunes a terminal clade from the tree.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">split</span></code>(self[, n, branch_length])</p></td>
<td><p>Generate n (default 2) new descendants.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Phylo.PhyloXML.Clade.to_phylogeny" title="Bio.Phylo.PhyloXML.Clade.to_phylogeny"><code class="xref py py-obj docutils literal notranslate"><span class="pre">to_phylogeny</span></code></a>(self, \*\*kwargs)</p></td>
<td><p>Create a new phylogeny containing just this clade.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">total_branch_length</span></code>(self)</p></td>
<td><p>Calculate the sum of all the branch lengths in this tree.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">trace</span></code>(self, start, finish)</p></td>
<td><p>List of all clade object between two targets in this tree.</p></td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="Bio.Phylo.PhyloXML.Clade.confidence">
<em class="property">property </em><code class="sig-name descname">confidence</code><a class="headerlink" href="#Bio.Phylo.PhyloXML.Clade.confidence" title="Permalink to this definition">¶</a></dt>
<dd><p>Return confidence values (PRIVATE).</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Phylo.PhyloXML.Clade.from_clade">
<em class="property">classmethod </em><code class="sig-name descname">from_clade</code><span class="sig-paren">(</span><em class="sig-param">clade</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Clade.from_clade" title="Permalink to this definition">¶</a></dt>
<dd><p>Create a new PhyloXML Clade from a Newick or BaseTree Clade object.</p>
<p>Keyword arguments are the usual PhyloXML Clade constructor parameters.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Phylo.PhyloXML.Clade.taxonomy">
<em class="property">property </em><code class="sig-name descname">taxonomy</code><a class="headerlink" href="#Bio.Phylo.PhyloXML.Clade.taxonomy" title="Permalink to this definition">¶</a></dt>
<dd><p>Get taxonomy list for the clade (PRIVATE).</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Phylo.PhyloXML.Clade.to_phylogeny">
<code class="sig-name descname">to_phylogeny</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Clade.to_phylogeny" title="Permalink to this definition">¶</a></dt>
<dd><p>Create a new phylogeny containing just this clade.</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="Bio.Phylo.PhyloXML.CladeRelation">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">CladeRelation</code><span class="sig-paren">(</span><em class="sig-param">type</em>, <em class="sig-param">id_ref_0</em>, <em class="sig-param">id_ref_1</em>, <em class="sig-param">distance=None</em>, <em class="sig-param">confidence=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.CladeRelation" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Bio.Phylo.PhyloXML.PhyloElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.PhyloXML.PhyloElement</span></code></a></p>
<p>Expresses a typed relationship between two clades.</p>
<p>For example, this could be used to describe multiple parents of a clade.</p>
<dl class="field-list simple">
</dl>
</dd></dl>

<dl class="class">
<dt id="Bio.Phylo.PhyloXML.Confidence">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">Confidence</code><span class="sig-paren">(</span><em class="sig-param">value</em>, <em class="sig-param">type='unknown'</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Confidence" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Bio.Phylo.PhyloXML.PhyloElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.PhyloXML.PhyloElement</span></code></a></p>
<p>A general purpose confidence element.</p>
<p>For example, this can be used to express the bootstrap support value of a
clade (in which case the <code class="docutils literal notranslate"><span class="pre">type</span></code> attribute is ‘bootstrap’).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt>value<span class="classifier">float</span></dt><dd><p>confidence value</p>
</dd>
<dt>type<span class="classifier">string</span></dt><dd><p>label for the type of confidence, e.g. ‘bootstrap’</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="class">
<dt id="Bio.Phylo.PhyloXML.Date">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">Date</code><span class="sig-paren">(</span><em class="sig-param">value=None</em>, <em class="sig-param">unit=None</em>, <em class="sig-param">desc=None</em>, <em class="sig-param">minimum=None</em>, <em class="sig-param">maximum=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Date" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Bio.Phylo.PhyloXML.PhyloElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.PhyloXML.PhyloElement</span></code></a></p>
<p>A date associated with a clade/node.</p>
<p>Its value can be numerical by using the ‘value’ element and/or free text
with the ‘desc’ element’ (e.g. ‘Silurian’). If a numerical value is used, it
is recommended to employ the ‘unit’ attribute.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt>unit<span class="classifier">string</span></dt><dd><p>type of numerical value (e.g. ‘mya’ for ‘million years ago’)</p>
</dd>
<dt>value<span class="classifier">float</span></dt><dd><p>the date value</p>
</dd>
<dt>desc<span class="classifier">string</span></dt><dd><p>plain-text description of the date</p>
</dd>
<dt>minimum<span class="classifier">float</span></dt><dd><p>lower bound on the date value</p>
</dd>
<dt>maximum<span class="classifier">float</span></dt><dd><p>upper bound on the date value</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="class">
<dt id="Bio.Phylo.PhyloXML.Distribution">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">Distribution</code><span class="sig-paren">(</span><em class="sig-param">desc=None</em>, <em class="sig-param">points=None</em>, <em class="sig-param">polygons=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Distribution" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Bio.Phylo.PhyloXML.PhyloElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.PhyloXML.PhyloElement</span></code></a></p>
<p>Geographic distribution of the items of a clade (species, sequences).</p>
<p>Intended for phylogeographic applications.</p>
<dl class="field-list">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl>
<dt>desc<span class="classifier">string</span></dt><dd><p>free-text description of the location</p>
</dd>
<dt>points<span class="classifier">list of <code class="docutils literal notranslate"><span class="pre">Point</span></code> objects</span></dt><dd><p>coordinates (similar to the ‘Point’ element in Google’s KML format)</p>
</dd>
<dt>polygons<span class="classifier">list of <code class="docutils literal notranslate"><span class="pre">Polygon</span></code> objects</span></dt><dd><p>coordinate sets defining geographic regions</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="class">
<dt id="Bio.Phylo.PhyloXML.DomainArchitecture">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">DomainArchitecture</code><span class="sig-paren">(</span><em class="sig-param">length=None</em>, <em class="sig-param">domains=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.DomainArchitecture" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Bio.Phylo.PhyloXML.PhyloElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.PhyloXML.PhyloElement</span></code></a></p>
<p>Domain architecture of a protein.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt>length<span class="classifier">int</span></dt><dd><p>total length of the protein sequence</p>
</dd>
<dt>domains<span class="classifier">list ProteinDomain objects</span></dt><dd><p>the domains within this protein</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="class">
<dt id="Bio.Phylo.PhyloXML.Events">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">Events</code><span class="sig-paren">(</span><em class="sig-param">type=None</em>, <em class="sig-param">duplications=None</em>, <em class="sig-param">speciations=None</em>, <em class="sig-param">losses=None</em>, <em class="sig-param">confidence=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Events" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Bio.Phylo.PhyloXML.PhyloElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.PhyloXML.PhyloElement</span></code></a></p>
<p>Events at the root node of a clade (e.g. one gene duplication).</p>
<p>All attributes are set to None by default, but this object can also be
treated as a dictionary, in which case None values are treated as missing
keys and deleting a key resets that attribute’s value back to None.</p>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Phylo.PhyloXML.Events.items" title="Bio.Phylo.PhyloXML.Events.items"><code class="xref py py-obj docutils literal notranslate"><span class="pre">items</span></code></a>(self)</p></td>
<td><p>Return Event’s items.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Phylo.PhyloXML.Events.keys" title="Bio.Phylo.PhyloXML.Events.keys"><code class="xref py py-obj docutils literal notranslate"><span class="pre">keys</span></code></a>(self)</p></td>
<td><p>Return Event’s keys.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Phylo.PhyloXML.Events.values" title="Bio.Phylo.PhyloXML.Events.values"><code class="xref py py-obj docutils literal notranslate"><span class="pre">values</span></code></a>(self)</p></td>
<td><p>Return values from a key-value pair in an Events dict.</p></td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="Bio.Phylo.PhyloXML.Events.items">
<code class="sig-name descname">items</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Events.items" title="Permalink to this definition">¶</a></dt>
<dd><p>Return Event’s items.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Phylo.PhyloXML.Events.keys">
<code class="sig-name descname">keys</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Events.keys" title="Permalink to this definition">¶</a></dt>
<dd><p>Return Event’s keys.</p>
</dd></dl>

<dl class="attribute">
<dt id="Bio.Phylo.PhyloXML.Events.ok_type">
<code class="sig-name descname">ok_type</code><em class="property"> = {'fusion', 'mixed', 'other', 'speciation_or_duplication', 'transfer', 'unassigned'}</em><a class="headerlink" href="#Bio.Phylo.PhyloXML.Events.ok_type" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="Bio.Phylo.PhyloXML.Events.values">
<code class="sig-name descname">values</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Events.values" title="Permalink to this definition">¶</a></dt>
<dd><p>Return values from a key-value pair in an Events dict.</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="Bio.Phylo.PhyloXML.Id">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">Id</code><span class="sig-paren">(</span><em class="sig-param">value</em>, <em class="sig-param">provider=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Id" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Bio.Phylo.PhyloXML.PhyloElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.PhyloXML.PhyloElement</span></code></a></p>
<p>A general-purpose identifier element.</p>
<p>Allows to indicate the provider (or authority) of an identifier, e.g. NCBI,
along with the value itself.</p>
</dd></dl>

<dl class="class">
<dt id="Bio.Phylo.PhyloXML.MolSeq">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">MolSeq</code><span class="sig-paren">(</span><em class="sig-param">value</em>, <em class="sig-param">is_aligned=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.MolSeq" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Bio.Phylo.PhyloXML.PhyloElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.PhyloXML.PhyloElement</span></code></a></p>
<p>Store a molecular sequence.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt>value<span class="classifier">string</span></dt><dd><p>the sequence itself</p>
</dd>
<dt>is_aligned<span class="classifier">bool</span></dt><dd><p>True if this sequence is aligned with the others (usually meaning
all aligned seqs are the same length and gaps may be present)</p>
</dd>
</dl>
</dd>
</dl>
<dl class="attribute">
<dt id="Bio.Phylo.PhyloXML.MolSeq.re_value">
<code class="sig-name descname">re_value</code><em class="property"> = re.compile('[a-zA-Z\\.\\-\\?\\*_]+')</em><a class="headerlink" href="#Bio.Phylo.PhyloXML.MolSeq.re_value" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="class">
<dt id="Bio.Phylo.PhyloXML.Other">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">Other</code><span class="sig-paren">(</span><em class="sig-param">tag</em>, <em class="sig-param">namespace=None</em>, <em class="sig-param">attributes=None</em>, <em class="sig-param">value=None</em>, <em class="sig-param">children=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Other" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Bio.Phylo.PhyloXML.PhyloElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.PhyloXML.PhyloElement</span></code></a></p>
<p>Container for non-phyloXML elements in the tree.</p>
<p>Usually, an Other object will have either a ‘value’ or a non-empty list
of ‘children’, but not both. This is not enforced here, though.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt>tag<span class="classifier">string</span></dt><dd><p>local tag for the XML node</p>
</dd>
<dt>namespace<span class="classifier">string</span></dt><dd><p>XML namespace for the node – should not be the default phyloXML
namespace.</p>
</dd>
<dt>attributes<span class="classifier">dict of strings</span></dt><dd><p>attributes on the XML node</p>
</dd>
<dt>value<span class="classifier">string</span></dt><dd><p>text contained directly within this XML node</p>
</dd>
<dt>children<span class="classifier">list</span></dt><dd><p>child nodes, if any (also <code class="docutils literal notranslate"><span class="pre">Other</span></code> instances)</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="class">
<dt id="Bio.Phylo.PhyloXML.PhyloElement">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">PhyloElement</code><a class="headerlink" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Phylo.BaseTree.html#Bio.Phylo.BaseTree.TreeElement" title="Bio.Phylo.BaseTree.TreeElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.BaseTree.TreeElement</span></code></a></p>
<p>Base class for all PhyloXML objects.</p>
</dd></dl>

<dl class="exception">
<dt id="Bio.Phylo.PhyloXML.PhyloXMLWarning">
<em class="property">exception </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">PhyloXMLWarning</code><a class="headerlink" href="#Bio.Phylo.PhyloXML.PhyloXMLWarning" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.html#Bio.BiopythonWarning" title="Bio.BiopythonWarning"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.BiopythonWarning</span></code></a></p>
<p>Warning for non-compliance with the phyloXML specification.</p>
</dd></dl>

<dl class="class">
<dt id="Bio.Phylo.PhyloXML.Phylogeny">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">Phylogeny</code><span class="sig-paren">(</span><em class="sig-param">root=None</em>, <em class="sig-param">rooted=True</em>, <em class="sig-param">rerootable=None</em>, <em class="sig-param">branch_length_unit=None</em>, <em class="sig-param">type=None</em>, <em class="sig-param">name=None</em>, <em class="sig-param">id=None</em>, <em class="sig-param">description=None</em>, <em class="sig-param">date=None</em>, <em class="sig-param">confidences=None</em>, <em class="sig-param">clade_relations=None</em>, <em class="sig-param">sequence_relations=None</em>, <em class="sig-param">properties=None</em>, <em class="sig-param">other=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Phylogeny" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Bio.Phylo.PhyloXML.PhyloElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.PhyloXML.PhyloElement</span></code></a>, <a class="reference internal" href="Bio.Phylo.BaseTree.html#Bio.Phylo.BaseTree.Tree" title="Bio.Phylo.BaseTree.Tree"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.BaseTree.Tree</span></code></a></p>
<p>A phylogenetic tree.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt>root<span class="classifier">Clade</span></dt><dd><p>the root node/clade of this tree</p>
</dd>
<dt>rooted<span class="classifier">bool</span></dt><dd><p>True if this tree is rooted</p>
</dd>
<dt>rerootable<span class="classifier">bool</span></dt><dd><p>True if this tree is rerootable</p>
</dd>
<dt>branch_length_unit<span class="classifier">string</span></dt><dd><p>unit for branch_length values on clades</p>
</dd>
<dt>name<span class="classifier">string</span></dt><dd><p>identifier for this tree, not required to be unique</p>
</dd>
<dt>id<span class="classifier">Id</span></dt><dd><p>unique identifier for this tree</p>
</dd>
<dt>description<span class="classifier">string</span></dt><dd><p>plain-text description</p>
</dd>
<dt>date<span class="classifier">Date</span></dt><dd><p>date for the root node of this tree</p>
</dd>
<dt>confidences<span class="classifier">list</span></dt><dd><p>Confidence objects for this tree</p>
</dd>
<dt>clade_relations<span class="classifier">list</span></dt><dd><p>CladeRelation objects</p>
</dd>
<dt>sequence_relations<span class="classifier">list</span></dt><dd><p>SequenceRelation objects</p>
</dd>
<dt>properties<span class="classifier">list</span></dt><dd><p>Property objects</p>
</dd>
<dt>other<span class="classifier">list</span></dt><dd><p>non-phyloXML elements (type <code class="docutils literal notranslate"><span class="pre">Other</span></code>)</p>
</dd>
</dl>
</dd>
<dt class="field-even">Attributes</dt>
<dd class="field-even"><dl class="simple">
<dt><code class="xref py py-obj docutils literal notranslate"><span class="pre">clade</span></code></dt><dd><p>Return first clade in this tree (not itself).</p>
</dd>
<dt><a class="reference internal" href="#Bio.Phylo.PhyloXML.Phylogeny.confidence" title="Bio.Phylo.PhyloXML.Phylogeny.confidence"><code class="xref py py-obj docutils literal notranslate"><span class="pre">confidence</span></code></a></dt><dd><p>Equivalent to self.confidences[0] if there is only 1 value (PRIVATE).</p>
</dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Phylo.PhyloXML.Phylogeny.as_phyloxml" title="Bio.Phylo.PhyloXML.Phylogeny.as_phyloxml"><code class="xref py py-obj docutils literal notranslate"><span class="pre">as_phyloxml</span></code></a>(self)</p></td>
<td><p>Return this tree, a PhyloXML-compatible Phylogeny object.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">collapse</span></code>(self[, target])</p></td>
<td><p>Delete target from the tree, relinking its children to its parent.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">collapse_all</span></code>(self[, target])</p></td>
<td><p>Collapse all the descendents of this tree, leaving only terminals.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">common_ancestor</span></code>(self, targets, \*more_targets)</p></td>
<td><p>Most recent common ancestor (clade) of all the given targets.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">count_terminals</span></code>(self)</p></td>
<td><p>Count the number of terminal (leaf) nodes within this tree.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">depths</span></code>(self[, unit_branch_lengths])</p></td>
<td><p>Create a mapping of tree clades to depths (by branch length).</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">distance</span></code>(self, target1[, target2])</p></td>
<td><p>Calculate the sum of the branch lengths between two targets.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">find_any</span></code>(self, \*args, \*\*kwargs)</p></td>
<td><p>Return the first element found by find_elements(), or None.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">find_clades</span></code>(self[, target, terminal, order])</p></td>
<td><p>Find each clade containing a matching element.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">find_elements</span></code>(self[, target, terminal, order])</p></td>
<td><p>Find all tree elements matching the given attributes.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">format</span></code>(self, format)</p></td>
<td><p>Serialize the tree as a string in the specified file format.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Phylo.PhyloXML.Phylogeny.from_clade" title="Bio.Phylo.PhyloXML.Phylogeny.from_clade"><code class="xref py py-obj docutils literal notranslate"><span class="pre">from_clade</span></code></a>(clade, \*\*kwargs)</p></td>
<td><p>Create a new Phylogeny given a Newick or BaseTree Clade object.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Phylo.PhyloXML.Phylogeny.from_tree" title="Bio.Phylo.PhyloXML.Phylogeny.from_tree"><code class="xref py py-obj docutils literal notranslate"><span class="pre">from_tree</span></code></a>(tree, \*\*kwargs)</p></td>
<td><p>Create a new Phylogeny given a Tree (from Newick/Nexus or BaseTree).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_nonterminals</span></code>(self[, order])</p></td>
<td><p>Get a list of all of this tree’s nonterminal (internal) nodes.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_path</span></code>(self[, target])</p></td>
<td><p>List the clades directly between this root and the given target.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_terminals</span></code>(self[, order])</p></td>
<td><p>Get a list of all of this tree’s terminal (leaf) nodes.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">is_bifurcating</span></code>(self)</p></td>
<td><p>Return True if tree downstream of node is strictly bifurcating.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">is_monophyletic</span></code>(self, terminals, …)</p></td>
<td><p>MRCA of terminals if they comprise a complete subclade, or False.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">is_parent_of</span></code>(self[, target])</p></td>
<td><p>Check if target is a descendent of this tree.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">is_preterminal</span></code>(self)</p></td>
<td><p>Check if all direct descendents are terminal.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">is_terminal</span></code>(self)</p></td>
<td><p>Check if the root of this tree is terminal.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">ladderize</span></code>(self[, reverse])</p></td>
<td><p>Sort clades in-place according to the number of terminal nodes.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">prune</span></code>(self[, target])</p></td>
<td><p>Prunes a terminal clade from the tree.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">randomized</span></code>(taxa[, branch_length, branch_stdev])</p></td>
<td><p>Create a randomized bifurcating tree given a list of taxa.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">root_at_midpoint</span></code>(self)</p></td>
<td><p>Root the tree at the midpoint of the two most distant taxa.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">root_with_outgroup</span></code>(self, outgroup_targets, …)</p></td>
<td><p>Reroot this tree with the outgroup clade containing outgroup_targets.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">split</span></code>(self[, n, branch_length])</p></td>
<td><p>Generate n (default 2) new descendants.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Phylo.PhyloXML.Phylogeny.to_alignment" title="Bio.Phylo.PhyloXML.Phylogeny.to_alignment"><code class="xref py py-obj docutils literal notranslate"><span class="pre">to_alignment</span></code></a>(self)</p></td>
<td><p>Construct an alignment from the aligned sequences in this tree.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Phylo.PhyloXML.Phylogeny.to_phyloxml_container" title="Bio.Phylo.PhyloXML.Phylogeny.to_phyloxml_container"><code class="xref py py-obj docutils literal notranslate"><span class="pre">to_phyloxml_container</span></code></a>(self, \*\*kwargs)</p></td>
<td><p>Create a new Phyloxml object containing just this phylogeny.</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">total_branch_length</span></code>(self)</p></td>
<td><p>Calculate the sum of all the branch lengths in this tree.</p></td>
</tr>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">trace</span></code>(self, start, finish)</p></td>
<td><p>List of all clade object between two targets in this tree.</p></td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="Bio.Phylo.PhyloXML.Phylogeny.as_phyloxml">
<code class="sig-name descname">as_phyloxml</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Phylogeny.as_phyloxml" title="Permalink to this definition">¶</a></dt>
<dd><p>Return this tree, a PhyloXML-compatible Phylogeny object.</p>
<p>Overrides the <code class="docutils literal notranslate"><span class="pre">BaseTree</span></code> method.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Phylo.PhyloXML.Phylogeny.confidence">
<em class="property">property </em><code class="sig-name descname">confidence</code><a class="headerlink" href="#Bio.Phylo.PhyloXML.Phylogeny.confidence" title="Permalink to this definition">¶</a></dt>
<dd><p>Equivalent to self.confidences[0] if there is only 1 value (PRIVATE).</p>
<p>See Also: <code class="docutils literal notranslate"><span class="pre">Clade.confidence</span></code>, <code class="docutils literal notranslate"><span class="pre">Clade.taxonomy</span></code></p>
</dd></dl>

<dl class="method">
<dt id="Bio.Phylo.PhyloXML.Phylogeny.from_clade">
<em class="property">classmethod </em><code class="sig-name descname">from_clade</code><span class="sig-paren">(</span><em class="sig-param">clade</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Phylogeny.from_clade" title="Permalink to this definition">¶</a></dt>
<dd><p>Create a new Phylogeny given a Newick or BaseTree Clade object.</p>
<p>Keyword arguments are the usual <code class="docutils literal notranslate"><span class="pre">PhyloXML.Clade</span></code> constructor parameters.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Phylo.PhyloXML.Phylogeny.from_tree">
<em class="property">classmethod </em><code class="sig-name descname">from_tree</code><span class="sig-paren">(</span><em class="sig-param">tree</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Phylogeny.from_tree" title="Permalink to this definition">¶</a></dt>
<dd><p>Create a new Phylogeny given a Tree (from Newick/Nexus or BaseTree).</p>
<p>Keyword arguments are the usual <code class="docutils literal notranslate"><span class="pre">Phylogeny</span></code> constructor parameters.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Phylo.PhyloXML.Phylogeny.to_alignment">
<code class="sig-name descname">to_alignment</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Phylogeny.to_alignment" title="Permalink to this definition">¶</a></dt>
<dd><p>Construct an alignment from the aligned sequences in this tree.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Phylo.PhyloXML.Phylogeny.to_phyloxml_container">
<code class="sig-name descname">to_phyloxml_container</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Phylogeny.to_phyloxml_container" title="Permalink to this definition">¶</a></dt>
<dd><p>Create a new Phyloxml object containing just this phylogeny.</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="Bio.Phylo.PhyloXML.Phyloxml">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">Phyloxml</code><span class="sig-paren">(</span><em class="sig-param">attributes</em>, <em class="sig-param">phylogenies=None</em>, <em class="sig-param">other=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Phyloxml" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Bio.Phylo.PhyloXML.PhyloElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.PhyloXML.PhyloElement</span></code></a></p>
<p>Root node of the PhyloXML document.</p>
<p>Contains an arbitrary number of Phylogeny elements, possibly followed by
elements from other namespaces.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt>attributes<span class="classifier">dict</span></dt><dd><p>(XML namespace definitions)</p>
</dd>
<dt>phylogenies<span class="classifier">list</span></dt><dd><p>The phylogenetic trees</p>
</dd>
<dt>other<span class="classifier">list</span></dt><dd><p>Arbitrary non-phyloXML elements, if any</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="class">
<dt id="Bio.Phylo.PhyloXML.Point">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">Point</code><span class="sig-paren">(</span><em class="sig-param">geodetic_datum</em>, <em class="sig-param">lat</em>, <em class="sig-param">long</em>, <em class="sig-param">alt=None</em>, <em class="sig-param">alt_unit=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Point" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Bio.Phylo.PhyloXML.PhyloElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.PhyloXML.PhyloElement</span></code></a></p>
<p>Geographic coordinates of a point, with an optional altitude.</p>
<p>Used by element ‘Distribution’.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt>geodetic_datum<span class="classifier">string, required</span></dt><dd><p>the geodetic datum (also called ‘map datum’). For example, Google’s
KML uses ‘WGS84’.</p>
</dd>
<dt>lat<span class="classifier">numeric</span></dt><dd><p>latitude</p>
</dd>
<dt>long<span class="classifier">numeric</span></dt><dd><p>longitude</p>
</dd>
<dt>alt<span class="classifier">numeric</span></dt><dd><p>altitude</p>
</dd>
<dt>alt_unit<span class="classifier">string</span></dt><dd><p>unit for the altitude (e.g. ‘meter’)</p>
</dd>
</dl>
</dd>
</dl>
</dd></dl>

<dl class="class">
<dt id="Bio.Phylo.PhyloXML.Polygon">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">Polygon</code><span class="sig-paren">(</span><em class="sig-param">points=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Polygon" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Bio.Phylo.PhyloXML.PhyloElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.PhyloXML.PhyloElement</span></code></a></p>
<p>A polygon defined by a list of ‘Points’ (used by element ‘Distribution’).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><p><strong>points</strong> – list of 3 or more points representing vertices.</p>
</dd>
</dl>
</dd></dl>

<dl class="class">
<dt id="Bio.Phylo.PhyloXML.Property">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">Property</code><span class="sig-paren">(</span><em class="sig-param">value</em>, <em class="sig-param">ref</em>, <em class="sig-param">applies_to</em>, <em class="sig-param">datatype</em>, <em class="sig-param">unit=None</em>, <em class="sig-param">id_ref=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Property" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Bio.Phylo.PhyloXML.PhyloElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.PhyloXML.PhyloElement</span></code></a></p>
<p>A typed and referenced property from an external resources.</p>
<p>Can be attached to <code class="docutils literal notranslate"><span class="pre">Phylogeny</span></code>, <code class="docutils literal notranslate"><span class="pre">Clade</span></code>, and <code class="docutils literal notranslate"><span class="pre">Annotation</span></code> objects.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt>value<span class="classifier">string</span></dt><dd><p>the value of the property</p>
</dd>
<dt>ref<span class="classifier">string</span></dt><dd><p>reference to an external resource, e.g. “NOAA:depth”</p>
</dd>
<dt>applies_to<span class="classifier">string</span></dt><dd><p>indicates the item to which a property applies to (e.g.  ‘node’ for
the parent node of a clade, ‘parent_branch’ for the parent branch of
a clade, or just ‘clade’).</p>
</dd>
<dt>datatype<span class="classifier">string</span></dt><dd><p>the type of a property; limited to xsd-datatypes
(e.g. ‘xsd:string’, ‘xsd:boolean’, ‘xsd:integer’, ‘xsd:decimal’,
‘xsd:float’, ‘xsd:double’, ‘xsd:date’, ‘xsd:anyURI’).</p>
</dd>
<dt>unit<span class="classifier">string (optional)</span></dt><dd><p>the unit of the property, e.g. “METRIC:m”</p>
</dd>
<dt>id_ref<span class="classifier">Id (optional)</span></dt><dd><p>allows to attached a property specifically to one element (on the
xml-level)</p>
</dd>
</dl>
</dd>
</dl>
<dl class="attribute">
<dt id="Bio.Phylo.PhyloXML.Property.ok_applies_to">
<code class="sig-name descname">ok_applies_to</code><em class="property"> = {'annotation', 'clade', 'node', 'other', 'parent_branch', 'phylogeny'}</em><a class="headerlink" href="#Bio.Phylo.PhyloXML.Property.ok_applies_to" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="Bio.Phylo.PhyloXML.Property.ok_datatype">
<code class="sig-name descname">ok_datatype</code><em class="property"> = {'xsd:anyURI', 'xsd:base64Binary', 'xsd:boolean', 'xsd:byte', 'xsd:date', 'xsd:dateTime', 'xsd:decimal', 'xsd:double', 'xsd:duration', 'xsd:float', 'xsd:gDay', 'xsd:gMonth', 'xsd:gMonthDay', 'xsd:gYear', 'xsd:gYearMonth', 'xsd:hexBinary', 'xsd:int', 'xsd:integer', 'xsd:long', 'xsd:negativeInteger', 'xsd:nonNegativeInteger', 'xsd:nonPositiveInteger', 'xsd:normalizedString', 'xsd:positiveInteger', 'xsd:short', 'xsd:string', 'xsd:time', 'xsd:token', 'xsd:unsignedByte', 'xsd:unsignedInt', 'xsd:unsignedLong', 'xsd:unsignedShort'}</em><a class="headerlink" href="#Bio.Phylo.PhyloXML.Property.ok_datatype" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="Bio.Phylo.PhyloXML.Property.re_ref">
<code class="sig-name descname">re_ref</code><em class="property"> = re.compile('[a-zA-Z0-9_]+:[a-zA-Z0-9_\\.\\-\\s]+')</em><a class="headerlink" href="#Bio.Phylo.PhyloXML.Property.re_ref" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="class">
<dt id="Bio.Phylo.PhyloXML.ProteinDomain">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">ProteinDomain</code><span class="sig-paren">(</span><em class="sig-param">value</em>, <em class="sig-param">start</em>, <em class="sig-param">end</em>, <em class="sig-param">confidence=None</em>, <em class="sig-param">id=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.ProteinDomain" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Bio.Phylo.PhyloXML.PhyloElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.PhyloXML.PhyloElement</span></code></a></p>
<p>Represents an individual domain in a domain architecture.</p>
<p>The locations use 0-based indexing, as most Python objects including
SeqFeature do, rather than the usual biological convention starting at 1.
This means the start and end attributes can be used directly as slice
indexes on Seq objects.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt>start<span class="classifier">non-negative integer</span></dt><dd><p>start of the domain on the sequence, using 0-based indexing</p>
</dd>
<dt>end<span class="classifier">non-negative integer</span></dt><dd><p>end of the domain on the sequence</p>
</dd>
<dt>confidence<span class="classifier">float</span></dt><dd><p>can be used to store e.g. E-values</p>
</dd>
<dt>id<span class="classifier">string</span></dt><dd><p>unique identifier/name</p>
</dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Phylo.PhyloXML.ProteinDomain.from_seqfeature" title="Bio.Phylo.PhyloXML.ProteinDomain.from_seqfeature"><code class="xref py py-obj docutils literal notranslate"><span class="pre">from_seqfeature</span></code></a>(feat)</p></td>
<td><p>Create ProteinDomain object from SeqFeature.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Phylo.PhyloXML.ProteinDomain.to_seqfeature" title="Bio.Phylo.PhyloXML.ProteinDomain.to_seqfeature"><code class="xref py py-obj docutils literal notranslate"><span class="pre">to_seqfeature</span></code></a>(self)</p></td>
<td><p>Create a SeqFeature from the ProteinDomain Object.</p></td>
</tr>
</tbody>
</table>
<dl class="method">
<dt id="Bio.Phylo.PhyloXML.ProteinDomain.from_seqfeature">
<em class="property">classmethod </em><code class="sig-name descname">from_seqfeature</code><span class="sig-paren">(</span><em class="sig-param">feat</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.ProteinDomain.from_seqfeature" title="Permalink to this definition">¶</a></dt>
<dd><p>Create ProteinDomain object from SeqFeature.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Phylo.PhyloXML.ProteinDomain.to_seqfeature">
<code class="sig-name descname">to_seqfeature</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.ProteinDomain.to_seqfeature" title="Permalink to this definition">¶</a></dt>
<dd><p>Create a SeqFeature from the ProteinDomain Object.</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="Bio.Phylo.PhyloXML.Reference">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">Reference</code><span class="sig-paren">(</span><em class="sig-param">doi=None</em>, <em class="sig-param">desc=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Reference" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Bio.Phylo.PhyloXML.PhyloElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.PhyloXML.PhyloElement</span></code></a></p>
<p>Literature reference for a clade.</p>
<p>NB: Whenever possible, use the <code class="docutils literal notranslate"><span class="pre">doi</span></code> attribute instead of the free-text
<code class="docutils literal notranslate"><span class="pre">desc</span></code> element.</p>
<dl class="attribute">
<dt id="Bio.Phylo.PhyloXML.Reference.re_doi">
<code class="sig-name descname">re_doi</code><em class="property"> = re.compile('[a-zA-Z0-9_\\.]+/[a-zA-Z0-9_\\.]+')</em><a class="headerlink" href="#Bio.Phylo.PhyloXML.Reference.re_doi" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="class">
<dt id="Bio.Phylo.PhyloXML.Sequence">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">Sequence</code><span class="sig-paren">(</span><em class="sig-param">type=None</em>, <em class="sig-param">id_ref=None</em>, <em class="sig-param">id_source=None</em>, <em class="sig-param">symbol=None</em>, <em class="sig-param">accession=None</em>, <em class="sig-param">name=None</em>, <em class="sig-param">location=None</em>, <em class="sig-param">mol_seq=None</em>, <em class="sig-param">uri=None</em>, <em class="sig-param">domain_architecture=None</em>, <em class="sig-param">annotations=None</em>, <em class="sig-param">other=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Sequence" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Bio.Phylo.PhyloXML.PhyloElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.PhyloXML.PhyloElement</span></code></a></p>
<p>A molecular sequence (Protein, DNA, RNA) associated with a node.</p>
<p>One intended use for <code class="docutils literal notranslate"><span class="pre">id_ref</span></code> is to link a sequence to a taxonomy (via the
taxonomy’s <code class="docutils literal notranslate"><span class="pre">id_source</span></code>) in case of multiple sequences and taxonomies per
node.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt>type<span class="classifier">{‘dna’, ‘rna’, ‘protein’}</span></dt><dd><p>type of molecule this sequence represents</p>
</dd>
<dt>id_ref<span class="classifier">string</span></dt><dd><p>reference to another resource</p>
</dd>
<dt>id_source<span class="classifier">string</span></dt><dd><p>source for the reference</p>
</dd>
<dt>symbol<span class="classifier">string</span></dt><dd><p>short symbol of the sequence, e.g. ‘ACTM’ (max. 10 chars)</p>
</dd>
<dt>accession<span class="classifier">Accession</span></dt><dd><p>accession code for this sequence.</p>
</dd>
<dt>name<span class="classifier">string</span></dt><dd><p>full name of the sequence, e.g. ‘muscle Actin’</p>
</dd>
<dt>location</dt><dd><p>location of a sequence on a genome/chromosome.</p>
</dd>
<dt>mol_seq<span class="classifier">MolSeq</span></dt><dd><p>the molecular sequence itself</p>
</dd>
<dt>uri<span class="classifier">Uri</span></dt><dd><p>link</p>
</dd>
<dt>annotations<span class="classifier">list of Annotation objects</span></dt><dd><p>annotations on this sequence</p>
</dd>
<dt>domain_architecture<span class="classifier">DomainArchitecture</span></dt><dd><p>protein domains on this sequence</p>
</dd>
<dt>other<span class="classifier">list of Other objects</span></dt><dd><p>non-phyloXML elements</p>
</dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Phylo.PhyloXML.Sequence.from_seqrecord" title="Bio.Phylo.PhyloXML.Sequence.from_seqrecord"><code class="xref py py-obj docutils literal notranslate"><span class="pre">from_seqrecord</span></code></a>(record[, is_aligned])</p></td>
<td><p>Create a new PhyloXML Sequence from a SeqRecord object.</p></td>
</tr>
<tr class="row-even"><td><p><a class="reference internal" href="#Bio.Phylo.PhyloXML.Sequence.get_alphabet" title="Bio.Phylo.PhyloXML.Sequence.get_alphabet"><code class="xref py py-obj docutils literal notranslate"><span class="pre">get_alphabet</span></code></a>(self)</p></td>
<td><p>Get the alphabet for the sequence.</p></td>
</tr>
<tr class="row-odd"><td><p><a class="reference internal" href="#Bio.Phylo.PhyloXML.Sequence.to_seqrecord" title="Bio.Phylo.PhyloXML.Sequence.to_seqrecord"><code class="xref py py-obj docutils literal notranslate"><span class="pre">to_seqrecord</span></code></a>(self)</p></td>
<td><p>Create a SeqRecord object from this Sequence instance.</p></td>
</tr>
</tbody>
</table>
<dl class="attribute">
<dt id="Bio.Phylo.PhyloXML.Sequence.alphabets">
<code class="sig-name descname">alphabets</code><em class="property"> = {'dna': DNAAlphabet(), 'protein': ProteinAlphabet(), 'rna': RNAAlphabet()}</em><a class="headerlink" href="#Bio.Phylo.PhyloXML.Sequence.alphabets" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="Bio.Phylo.PhyloXML.Sequence.from_seqrecord">
<em class="property">classmethod </em><code class="sig-name descname">from_seqrecord</code><span class="sig-paren">(</span><em class="sig-param">record</em>, <em class="sig-param">is_aligned=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Sequence.from_seqrecord" title="Permalink to this definition">¶</a></dt>
<dd><p>Create a new PhyloXML Sequence from a SeqRecord object.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Phylo.PhyloXML.Sequence.get_alphabet">
<code class="sig-name descname">get_alphabet</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Sequence.get_alphabet" title="Permalink to this definition">¶</a></dt>
<dd><p>Get the alphabet for the sequence.</p>
</dd></dl>

<dl class="attribute">
<dt id="Bio.Phylo.PhyloXML.Sequence.re_symbol">
<code class="sig-name descname">re_symbol</code><em class="property"> = re.compile('\\S{1,10}')</em><a class="headerlink" href="#Bio.Phylo.PhyloXML.Sequence.re_symbol" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="method">
<dt id="Bio.Phylo.PhyloXML.Sequence.to_seqrecord">
<code class="sig-name descname">to_seqrecord</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Sequence.to_seqrecord" title="Permalink to this definition">¶</a></dt>
<dd><p>Create a SeqRecord object from this Sequence instance.</p>
<p>The seqrecord.annotations dictionary is packed like so:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="p">{</span> <span class="c1"># Sequence attributes with no SeqRecord equivalent:</span>
  <span class="s1">&#39;id_ref&#39;</span><span class="p">:</span> <span class="bp">self</span><span class="o">.</span><span class="n">id_ref</span><span class="p">,</span>
  <span class="s1">&#39;id_source&#39;</span><span class="p">:</span> <span class="bp">self</span><span class="o">.</span><span class="n">id_source</span><span class="p">,</span>
  <span class="s1">&#39;location&#39;</span><span class="p">:</span> <span class="bp">self</span><span class="o">.</span><span class="n">location</span><span class="p">,</span>
  <span class="s1">&#39;uri&#39;</span><span class="p">:</span> <span class="p">{</span> <span class="s1">&#39;value&#39;</span><span class="p">:</span> <span class="bp">self</span><span class="o">.</span><span class="n">uri</span><span class="o">.</span><span class="n">value</span><span class="p">,</span>
                  <span class="s1">&#39;desc&#39;</span><span class="p">:</span> <span class="bp">self</span><span class="o">.</span><span class="n">uri</span><span class="o">.</span><span class="n">desc</span><span class="p">,</span>
                  <span class="s1">&#39;type&#39;</span><span class="p">:</span> <span class="bp">self</span><span class="o">.</span><span class="n">uri</span><span class="o">.</span><span class="n">type</span> <span class="p">},</span>
  <span class="c1"># Sequence.annotations attribute (list of Annotations)</span>
  <span class="s1">&#39;annotations&#39;</span><span class="p">:</span> <span class="p">[{</span><span class="s1">&#39;ref&#39;</span><span class="p">:</span> <span class="n">ann</span><span class="o">.</span><span class="n">ref</span><span class="p">,</span>
                   <span class="s1">&#39;source&#39;</span><span class="p">:</span> <span class="n">ann</span><span class="o">.</span><span class="n">source</span><span class="p">,</span>
                   <span class="s1">&#39;evidence&#39;</span><span class="p">:</span> <span class="n">ann</span><span class="o">.</span><span class="n">evidence</span><span class="p">,</span>
                   <span class="s1">&#39;type&#39;</span><span class="p">:</span> <span class="n">ann</span><span class="o">.</span><span class="n">type</span><span class="p">,</span>
                   <span class="s1">&#39;confidence&#39;</span><span class="p">:</span> <span class="p">[</span><span class="n">ann</span><span class="o">.</span><span class="n">confidence</span><span class="o">.</span><span class="n">value</span><span class="p">,</span>
                                  <span class="n">ann</span><span class="o">.</span><span class="n">confidence</span><span class="o">.</span><span class="n">type</span><span class="p">],</span>
                   <span class="s1">&#39;properties&#39;</span><span class="p">:</span> <span class="p">[{</span><span class="s1">&#39;value&#39;</span><span class="p">:</span> <span class="n">prop</span><span class="o">.</span><span class="n">value</span><span class="p">,</span>
                                    <span class="s1">&#39;ref&#39;</span><span class="p">:</span> <span class="n">prop</span><span class="o">.</span><span class="n">ref</span><span class="p">,</span>
                                    <span class="s1">&#39;applies_to&#39;</span><span class="p">:</span> <span class="n">prop</span><span class="o">.</span><span class="n">applies_to</span><span class="p">,</span>
                                    <span class="s1">&#39;datatype&#39;</span><span class="p">:</span> <span class="n">prop</span><span class="o">.</span><span class="n">datatype</span><span class="p">,</span>
                                    <span class="s1">&#39;unit&#39;</span><span class="p">:</span> <span class="n">prop</span><span class="o">.</span><span class="n">unit</span><span class="p">,</span>
                                    <span class="s1">&#39;id_ref&#39;</span><span class="p">:</span> <span class="n">prop</span><span class="o">.</span><span class="n">id_ref</span><span class="p">}</span>
                                   <span class="k">for</span> <span class="n">prop</span> <span class="ow">in</span> <span class="n">ann</span><span class="o">.</span><span class="n">properties</span><span class="p">],</span>
                  <span class="p">}</span> <span class="k">for</span> <span class="n">ann</span> <span class="ow">in</span> <span class="bp">self</span><span class="o">.</span><span class="n">annotations</span><span class="p">],</span>
<span class="p">}</span>
</pre></div>
</div>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="Bio.Phylo.PhyloXML.SequenceRelation">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">SequenceRelation</code><span class="sig-paren">(</span><em class="sig-param">type</em>, <em class="sig-param">id_ref_0</em>, <em class="sig-param">id_ref_1</em>, <em class="sig-param">distance=None</em>, <em class="sig-param">confidence=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.SequenceRelation" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Bio.Phylo.PhyloXML.PhyloElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.PhyloXML.PhyloElement</span></code></a></p>
<p>Express a typed relationship between two sequences.</p>
<p>For example, this could be used to describe an orthology (in which case
attribute ‘type’ is ‘orthology’).</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt>id_ref_0<span class="classifier">Id</span></dt><dd><p>first sequence reference identifier</p>
</dd>
<dt>id_ref_1<span class="classifier">Id</span></dt><dd><p>second sequence reference identifier</p>
</dd>
<dt>distance<span class="classifier">float</span></dt><dd><p>distance between the two sequences</p>
</dd>
<dt>type<span class="classifier">restricted string</span></dt><dd><p>describe the type of relationship</p>
</dd>
<dt>confidence<span class="classifier">Confidence</span></dt><dd><p>confidence value for this relation</p>
</dd>
</dl>
</dd>
</dl>
<dl class="attribute">
<dt id="Bio.Phylo.PhyloXML.SequenceRelation.ok_type">
<code class="sig-name descname">ok_type</code><em class="property"> = {'one_to_one_orthology', 'orthology', 'other', 'paralogy', 'super_orthology', 'ultra_paralogy', 'unknown', 'xenology'}</em><a class="headerlink" href="#Bio.Phylo.PhyloXML.SequenceRelation.ok_type" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="class">
<dt id="Bio.Phylo.PhyloXML.Taxonomy">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">Taxonomy</code><span class="sig-paren">(</span><em class="sig-param">id_source=None</em>, <em class="sig-param">id=None</em>, <em class="sig-param">code=None</em>, <em class="sig-param">scientific_name=None</em>, <em class="sig-param">authority=None</em>, <em class="sig-param">rank=None</em>, <em class="sig-param">uri=None</em>, <em class="sig-param">common_names=None</em>, <em class="sig-param">synonyms=None</em>, <em class="sig-param">other=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Taxonomy" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Bio.Phylo.PhyloXML.PhyloElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.PhyloXML.PhyloElement</span></code></a></p>
<p>Describe taxonomic information for a clade.</p>
<dl class="field-list simple">
<dt class="field-odd">Parameters</dt>
<dd class="field-odd"><dl class="simple">
<dt>id_source<span class="classifier">Id</span></dt><dd><p>link other elements to a taxonomy (on the XML level)</p>
</dd>
<dt>id<span class="classifier">Id</span></dt><dd><p>unique identifier of a taxon, e.g. Id(‘6500’,
provider=’ncbi_taxonomy’) for the California sea hare</p>
</dd>
<dt>code<span class="classifier">restricted string</span></dt><dd><p>store UniProt/Swiss-Prot style organism codes, e.g. ‘APLCA’ for the
California sea hare ‘Aplysia californica’</p>
</dd>
<dt>scientific_name<span class="classifier">string</span></dt><dd><p>the standard scientific name for this organism, e.g. ‘Aplysia
californica’ for the California sea hare</p>
</dd>
<dt>authority<span class="classifier">string</span></dt><dd><p>keep the authority, such as ‘J. G. Cooper, 1863’, associated with
the ‘scientific_name’</p>
</dd>
<dt>common_names<span class="classifier">list of strings</span></dt><dd><p>common names for this organism</p>
</dd>
<dt>synonyms<span class="classifier">list of strings</span></dt><dd><p>synonyms for this taxon?</p>
</dd>
<dt>rank<span class="classifier">restricted string</span></dt><dd><p>taxonomic rank</p>
</dd>
<dt>uri<span class="classifier">Uri</span></dt><dd><p>link</p>
</dd>
<dt>other<span class="classifier">list of Other objects</span></dt><dd><p>non-phyloXML elements</p>
</dd>
</dl>
</dd>
</dl>
<dl class="attribute">
<dt id="Bio.Phylo.PhyloXML.Taxonomy.ok_rank">
<code class="sig-name descname">ok_rank</code><em class="property"> = {'branch', 'class', 'cohort', 'cultivar', 'division', 'domain', 'family', 'form', 'genus', 'infraclass', 'infracohort', 'infradivision', 'infrakingdom', 'infralegion', 'infraphylum', 'infratribe', 'kingdom', 'legion', 'microphylum', 'order', 'other', 'phylum', 'species', 'subclass', 'subcohort', 'subdivision', 'subfamily', 'subform', 'subgenus', 'subkingdom', 'sublegion', 'suborder', 'subphylum', 'subspecies', 'subtribe', 'subvariety', 'superclass', 'supercohort', 'superdivision', 'superfamily', 'superlegion', 'superorder', 'superphylum', 'superspecies', 'supertribe', 'tribe', 'unknown', 'variety'}</em><a class="headerlink" href="#Bio.Phylo.PhyloXML.Taxonomy.ok_rank" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="attribute">
<dt id="Bio.Phylo.PhyloXML.Taxonomy.re_code">
<code class="sig-name descname">re_code</code><em class="property"> = re.compile('[a-zA-Z0-9_]{2,10}')</em><a class="headerlink" href="#Bio.Phylo.PhyloXML.Taxonomy.re_code" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="class">
<dt id="Bio.Phylo.PhyloXML.Uri">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Phylo.PhyloXML.</code><code class="sig-name descname">Uri</code><span class="sig-paren">(</span><em class="sig-param">value</em>, <em class="sig-param">desc=None</em>, <em class="sig-param">type=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Phylo.PhyloXML.Uri" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Phylo.PhyloXML.PhyloElement" title="Bio.Phylo.PhyloXML.PhyloElement"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Phylo.PhyloXML.PhyloElement</span></code></a></p>
<p>A uniform resource identifier.</p>
<p>In general, this is expected to be an URL (for example, to link to an image
on a website, in which case the <code class="docutils literal notranslate"><span class="pre">type</span></code> attribute might be ‘image’ and
<code class="docutils literal notranslate"><span class="pre">desc</span></code> might be ‘image of a California sea hare’).</p>
</dd></dl>

</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="Bio.Phylo.PhyloXMLIO.html" class="btn btn-neutral float-right" title="Bio.Phylo.PhyloXMLIO module" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="Bio.Phylo.NexusIO.html" class="btn btn-neutral float-left" title="Bio.Phylo.NexusIO module" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 1999-2017, The Biopython Contributors

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>