

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Bio.Sequencing.Applications package &mdash; Biopython 1.74.dev0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Bio.Sequencing.Ace module" href="Bio.Sequencing.Ace.html" />
    <link rel="prev" title="Bio.Sequencing package" href="Bio.Sequencing.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search"  style="background: #10100F" >
          

          
            <a href="index.html" class="icon icon-home"> Biopython
          

          
            
            <img src="_static/biopython_logo.svg" class="logo" alt="Logo"/>
          
          </a>

          
            
            
              <div class="version">
                1.74.dev0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">API Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="Bio.html">Bio package</a><ul class="current">
<li class="toctree-l2 current"><a class="reference internal" href="Bio.html#subpackages">Subpackages</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="Bio.Affy.html">Bio.Affy package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Align.html">Bio.Align package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.AlignIO.html">Bio.AlignIO package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Alphabet.html">Bio.Alphabet package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Application.html">Bio.Application package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Blast.html">Bio.Blast package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.CAPS.html">Bio.CAPS package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Cluster.html">Bio.Cluster package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Compass.html">Bio.Compass package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Crystal.html">Bio.Crystal package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Data.html">Bio.Data package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Emboss.html">Bio.Emboss package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Entrez.html">Bio.Entrez package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.ExPASy.html">Bio.ExPASy package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.FSSP.html">Bio.FSSP package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.GenBank.html">Bio.GenBank package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Geo.html">Bio.Geo package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Graphics.html">Bio.Graphics package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.HMM.html">Bio.HMM package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.KDTree.html">Bio.KDTree package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.KEGG.html">Bio.KEGG package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Medline.html">Bio.Medline package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.NMR.html">Bio.NMR package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Nexus.html">Bio.Nexus package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.PDB.html">Bio.PDB package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Pathway.html">Bio.Pathway package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Phylo.html">Bio.Phylo package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.PopGen.html">Bio.PopGen package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Restriction.html">Bio.Restriction package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SCOP.html">Bio.SCOP package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SVDSuperimposer.html">Bio.SVDSuperimposer package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SearchIO.html">Bio.SearchIO package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SeqIO.html">Bio.SeqIO package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SeqUtils.html">Bio.SeqUtils package</a></li>
<li class="toctree-l3 current"><a class="reference internal" href="Bio.Sequencing.html">Bio.Sequencing package</a><ul class="current">
<li class="toctree-l4 current"><a class="reference internal" href="Bio.Sequencing.html#subpackages">Subpackages</a></li>
<li class="toctree-l4"><a class="reference internal" href="Bio.Sequencing.html#submodules">Submodules</a></li>
<li class="toctree-l4"><a class="reference internal" href="Bio.Sequencing.html#module-Bio.Sequencing">Module contents</a></li>
</ul>
</li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Statistics.html">Bio.Statistics package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SubsMat.html">Bio.SubsMat package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SwissProt.html">Bio.SwissProt package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.TogoWS.html">Bio.TogoWS package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.UniGene.html">Bio.UniGene package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.UniProt.html">Bio.UniProt package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Wise.html">Bio.Wise package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.codonalign.html">Bio.codonalign package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.motifs.html">Bio.motifs package</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.phenotype.html">Bio.phenotype package</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="Bio.html#submodules">Submodules</a></li>
<li class="toctree-l2"><a class="reference internal" href="Bio.html#module-Bio">Module contents</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="BioSQL.html">BioSQL package</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Biopython</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
          <li><a href="Bio.html">Bio package</a> &raquo;</li>
        
          <li><a href="Bio.Sequencing.html">Bio.Sequencing package</a> &raquo;</li>
        
      <li>Bio.Sequencing.Applications package</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
        
      </li>
    
  </ul>

  
  <div class="rst-breadcrumbs-buttons" role="navigation" aria-label="breadcrumb navigation">
      
        <a href="Bio.Sequencing.Ace.html" class="btn btn-neutral float-right" title="Bio.Sequencing.Ace module" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="Bio.Sequencing.html" class="btn btn-neutral float-left" title="Bio.Sequencing package" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
  </div>
  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="bio-sequencing-applications-package">
<h1>Bio.Sequencing.Applications package<a class="headerlink" href="#bio-sequencing-applications-package" title="Permalink to this headline">¶</a></h1>
<div class="section" id="module-Bio.Sequencing.Applications">
<span id="module-contents"></span><h2>Module contents<a class="headerlink" href="#module-Bio.Sequencing.Applications" title="Permalink to this headline">¶</a></h2>
<p>Main entry point for sequencing related command line application wrappers.</p>
<dl class="class">
<dt id="Bio.Sequencing.Applications.BwaIndexCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Sequencing.Applications.</code><code class="sig-name descname">BwaIndexCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='bwa'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Sequencing.Applications.BwaIndexCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for Burrows Wheeler Aligner (BWA) index.</p>
<p>Index database sequences in the FASTA format, equivalent to:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ bwa index [-p prefix] [-a algoType] [-c] &lt;in.db.fasta&gt;
</pre></div>
</div>
<p>See <a class="reference external" href="http://bio-bwa.sourceforge.net/bwa.shtml">http://bio-bwa.sourceforge.net/bwa.shtml</a> for details.</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Sequencing.Applications</span> <span class="k">import</span> <span class="n">BwaIndexCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">reference_genome</span> <span class="o">=</span> <span class="s2">&quot;/path/to/reference_genome.fasta&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">index_cmd</span> <span class="o">=</span> <span class="n">BwaIndexCommandline</span><span class="p">(</span><span class="n">infile</span><span class="o">=</span><span class="n">reference_genome</span><span class="p">,</span> <span class="n">algorithm</span><span class="o">=</span><span class="s2">&quot;bwtsw&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">index_cmd</span><span class="p">)</span>
<span class="go">bwa index -a bwtsw /path/to/reference_genome.fasta</span>
</pre></div>
</div>
<p>You would typically run the command using index_cmd() or via the
Python subprocess module, as described in the Biopython tutorial.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Sequencing.Applications.BwaAlignCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Sequencing.Applications.</code><code class="sig-name descname">BwaAlignCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='bwa'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Sequencing.Applications.BwaAlignCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for Burrows Wheeler Aligner (BWA) aln.</p>
<p>Run a BWA alignment, equivalent to:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ bwa aln [...] &lt;in.db.fasta&gt; &lt;in.query.fq&gt; &gt; &lt;out.sai&gt;
</pre></div>
</div>
<p>See <a class="reference external" href="http://bio-bwa.sourceforge.net/bwa.shtml">http://bio-bwa.sourceforge.net/bwa.shtml</a> for details.</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Sequencing.Applications</span> <span class="k">import</span> <span class="n">BwaAlignCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">reference_genome</span> <span class="o">=</span> <span class="s2">&quot;/path/to/reference_genome.fasta&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">read_file</span> <span class="o">=</span> <span class="s2">&quot;/path/to/read_1.fq&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">output_sai_file</span> <span class="o">=</span> <span class="s2">&quot;/path/to/read_1.sai&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">align_cmd</span> <span class="o">=</span> <span class="n">BwaAlignCommandline</span><span class="p">(</span><span class="n">reference</span><span class="o">=</span><span class="n">reference_genome</span><span class="p">,</span> <span class="n">read_file</span><span class="o">=</span><span class="n">read_file</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">align_cmd</span><span class="p">)</span>
<span class="go">bwa aln /path/to/reference_genome.fasta /path/to/read_1.fq</span>
</pre></div>
</div>
<p>You would typically run the command line using align_cmd(stdout=output_sai_file)
or via the Python subprocess module, as described in the Biopython tutorial.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Sequencing.Applications.BwaSamseCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Sequencing.Applications.</code><code class="sig-name descname">BwaSamseCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='bwa'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Sequencing.Applications.BwaSamseCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for Burrows Wheeler Aligner (BWA) samse.</p>
<p>Generate alignments in the SAM format given single-end reads.
Equvialent to:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ bwa samse [-n maxOcc] &lt;in.db.fasta&gt; &lt;in.sai&gt; &lt;in.fq&gt; &gt; &lt;out.sam&gt;
</pre></div>
</div>
<p>See <a class="reference external" href="http://bio-bwa.sourceforge.net/bwa.shtml">http://bio-bwa.sourceforge.net/bwa.shtml</a> for details.</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Sequencing.Applications</span> <span class="k">import</span> <span class="n">BwaSamseCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">reference_genome</span> <span class="o">=</span> <span class="s2">&quot;/path/to/reference_genome.fasta&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">read_file</span> <span class="o">=</span> <span class="s2">&quot;/path/to/read_1.fq&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">sai_file</span> <span class="o">=</span> <span class="s2">&quot;/path/to/read_1.sai&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">output_sam_file</span> <span class="o">=</span> <span class="s2">&quot;/path/to/read_1.sam&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">samse_cmd</span> <span class="o">=</span> <span class="n">BwaSamseCommandline</span><span class="p">(</span><span class="n">reference</span><span class="o">=</span><span class="n">reference_genome</span><span class="p">,</span>
<span class="gp">... </span>                                <span class="n">read_file</span><span class="o">=</span><span class="n">read_file</span><span class="p">,</span> <span class="n">sai_file</span><span class="o">=</span><span class="n">sai_file</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">samse_cmd</span><span class="p">)</span>
<span class="go">bwa samse /path/to/reference_genome.fasta /path/to/read_1.sai /path/to/read_1.fq</span>
</pre></div>
</div>
<p>You would typically run the command line using samse_cmd(stdout=output_sam_file)
or via the Python subprocess module, as described in the Biopython tutorial.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Sequencing.Applications.BwaSampeCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Sequencing.Applications.</code><code class="sig-name descname">BwaSampeCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='bwa'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Sequencing.Applications.BwaSampeCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for Burrows Wheeler Aligner (BWA) sampe.</p>
<p>Generate alignments in the SAM format given paired-end reads.
Equivalent to:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ bwa sampe [...] &lt;in.db.fasta&gt; &lt;in1.sai&gt; &lt;in2.sai&gt; &lt;in1.fq&gt; &lt;in2.fq&gt; &gt; &lt;out.sam&gt;
</pre></div>
</div>
<p>See <a class="reference external" href="http://bio-bwa.sourceforge.net/bwa.shtml">http://bio-bwa.sourceforge.net/bwa.shtml</a> for details.</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Sequencing.Applications</span> <span class="k">import</span> <span class="n">BwaSampeCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">reference_genome</span> <span class="o">=</span> <span class="s2">&quot;/path/to/reference_genome.fasta&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">read_file1</span> <span class="o">=</span> <span class="s2">&quot;/path/to/read_1.fq&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">read_file2</span> <span class="o">=</span> <span class="s2">&quot;/path/to/read_2.fq&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">sai_file1</span> <span class="o">=</span> <span class="s2">&quot;/path/to/read_1.sai&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">sai_file2</span> <span class="o">=</span> <span class="s2">&quot;/path/to/read_2.sai&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">output_sam_file</span> <span class="o">=</span> <span class="s2">&quot;/path/to/output.sam&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">read_group</span> <span class="o">=</span> <span class="sa">r</span><span class="s2">&quot;@RG\tID:foo\tSM:bar&quot;</span>  <span class="c1"># BWA will turn backslash-t into tab</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">sampe_cmd</span> <span class="o">=</span> <span class="n">BwaSampeCommandline</span><span class="p">(</span><span class="n">reference</span><span class="o">=</span><span class="n">reference_genome</span><span class="p">,</span>
<span class="gp">... </span>                                <span class="n">sai_file1</span><span class="o">=</span><span class="n">sai_file1</span><span class="p">,</span> <span class="n">sai_file2</span><span class="o">=</span><span class="n">sai_file2</span><span class="p">,</span>
<span class="gp">... </span>                                <span class="n">read_file1</span><span class="o">=</span><span class="n">read_file1</span><span class="p">,</span> <span class="n">read_file2</span><span class="o">=</span><span class="n">read_file2</span><span class="p">,</span>
<span class="gp">... </span>                                <span class="n">r</span><span class="o">=</span><span class="n">read_group</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">sampe_cmd</span><span class="p">)</span>
<span class="go">bwa sampe /path/to/reference_genome.fasta /path/to/read_1.sai /path/to/read_2.sai /path/to/read_1.fq /path/to/read_2.fq -r @RG\tID:foo\tSM:bar</span>
</pre></div>
</div>
<p>You would typically run the command line using sampe_cmd(stdout=output_sam_file)
or via the Python subprocess module, as described in the Biopython tutorial.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Sequencing.Applications.BwaBwaswCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Sequencing.Applications.</code><code class="sig-name descname">BwaBwaswCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='bwa'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Sequencing.Applications.BwaBwaswCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for Burrows Wheeler Aligner (BWA) bwasw.</p>
<p>Align query sequences from FASTQ files. Equivalent to:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ bwa bwasw [...] &lt;in.db.fasta&gt; &lt;in.fq&gt;
</pre></div>
</div>
<p>See <a class="reference external" href="http://bio-bwa.sourceforge.net/bwa.shtml">http://bio-bwa.sourceforge.net/bwa.shtml</a> for details.</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Sequencing.Applications</span> <span class="k">import</span> <span class="n">BwaBwaswCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">reference_genome</span> <span class="o">=</span> <span class="s2">&quot;/path/to/reference_genome.fasta&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">read_file</span> <span class="o">=</span> <span class="s2">&quot;/path/to/read_1.fq&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">bwasw_cmd</span> <span class="o">=</span> <span class="n">BwaBwaswCommandline</span><span class="p">(</span><span class="n">reference</span><span class="o">=</span><span class="n">reference_genome</span><span class="p">,</span> <span class="n">read_file</span><span class="o">=</span><span class="n">read_file</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">bwasw_cmd</span><span class="p">)</span>
<span class="go">bwa bwasw /path/to/reference_genome.fasta /path/to/read_1.fq</span>
</pre></div>
</div>
<p>You would typically run the command line using bwasw_cmd() or via the
Python subprocess module, as described in the Biopython tutorial.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Sequencing.Applications.BwaMemCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Sequencing.Applications.</code><code class="sig-name descname">BwaMemCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='bwa'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Sequencing.Applications.BwaMemCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for Burrows Wheeler Aligner (BWA) mem.</p>
<p>Run a BWA-MEM alignment, with single- or paired-end reads, equivalent to:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ bwa mem [...] &lt;in.db.fasta&gt; &lt;in1.fq&gt; &lt;in2.fq&gt; &gt; &lt;out.sam&gt;
</pre></div>
</div>
<p>See <a class="reference external" href="http://bio-bwa.sourceforge.net/bwa.shtml">http://bio-bwa.sourceforge.net/bwa.shtml</a> for details.</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Sequencing.Applications</span> <span class="k">import</span> <span class="n">BwaMemCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">reference_genome</span> <span class="o">=</span> <span class="s2">&quot;/path/to/reference_genome.fasta&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">read_file</span> <span class="o">=</span> <span class="s2">&quot;/path/to/read_1.fq&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">output_sam_file</span> <span class="o">=</span> <span class="s2">&quot;/path/to/output.sam&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">align_cmd</span> <span class="o">=</span> <span class="n">BwaMemCommandline</span><span class="p">(</span><span class="n">reference</span><span class="o">=</span><span class="n">reference_genome</span><span class="p">,</span> <span class="n">read_file1</span><span class="o">=</span><span class="n">read_file</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">align_cmd</span><span class="p">)</span>
<span class="go">bwa mem /path/to/reference_genome.fasta /path/to/read_1.fq</span>
</pre></div>
</div>
<p>You would typically run the command line using align_cmd(stdout=output_sam_file)
or via the Python subprocess module, as described in the Biopython tutorial.</p>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Sequencing.Applications.SamtoolsViewCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Sequencing.Applications.</code><code class="sig-name descname">SamtoolsViewCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='samtools'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Sequencing.Applications.SamtoolsViewCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for samtools view.</p>
<p>Extract/print all or sub alignments in SAM or BAM format, equivalent to:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ samtools view [-bchuHS] [-t in.refList] [-o output] [-f reqFlag]
                [-F skipFlag] [-q minMapQ] [-l library] [-r readGroup]
                [-R rgFile] &lt;in.bam&gt;|&lt;in.sam&gt; [region1 [...]]
</pre></div>
</div>
<p>See <a class="reference external" href="http://samtools.sourceforge.net/samtools.shtml">http://samtools.sourceforge.net/samtools.shtml</a> for more details</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Sequencing.Applications</span> <span class="k">import</span> <span class="n">SamtoolsViewCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">input_file</span> <span class="o">=</span> <span class="s2">&quot;/path/to/sam_or_bam_file&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">samtools_view_cmd</span> <span class="o">=</span> <span class="n">SamtoolsViewCommandline</span><span class="p">(</span><span class="n">input_file</span><span class="o">=</span><span class="n">input_file</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">samtools_view_cmd</span><span class="p">)</span>
<span class="go">samtools view /path/to/sam_or_bam_file</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Sequencing.Applications.SamtoolsCalmdCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Sequencing.Applications.</code><code class="sig-name descname">SamtoolsCalmdCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='samtools'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Sequencing.Applications.SamtoolsCalmdCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for samtools calmd.</p>
<p>Generate the MD tag, equivalent to:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ samtools calmd [-EeubSr] [-C capQcoef] &lt;aln.bam&gt; &lt;ref.fasta&gt;
</pre></div>
</div>
<p>See <a class="reference external" href="http://samtools.sourceforge.net/samtools.shtml">http://samtools.sourceforge.net/samtools.shtml</a> for more details</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Sequencing.Applications</span> <span class="k">import</span> <span class="n">SamtoolsCalmdCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">input_bam</span> <span class="o">=</span> <span class="s2">&quot;/path/to/aln.bam&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">reference_fasta</span> <span class="o">=</span> <span class="s2">&quot;/path/to/reference.fasta&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">calmd_cmd</span> <span class="o">=</span> <span class="n">SamtoolsCalmdCommandline</span><span class="p">(</span><span class="n">input_bam</span><span class="o">=</span><span class="n">input_bam</span><span class="p">,</span>
<span class="gp">... </span>                                     <span class="n">reference</span><span class="o">=</span><span class="n">reference_fasta</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">calmd_cmd</span><span class="p">)</span>
<span class="go">samtools calmd /path/to/aln.bam /path/to/reference.fasta</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Sequencing.Applications.SamtoolsCatCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Sequencing.Applications.</code><code class="sig-name descname">SamtoolsCatCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='samtools'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Sequencing.Applications.SamtoolsCatCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for samtools cat.</p>
<p>Concatenate BAMs, equivalent to:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ samtools cat [-h header.sam] [-o out.bam] &lt;in1.bam&gt; &lt;in2.bam&gt; [ ... ]
</pre></div>
</div>
<p>See <a class="reference external" href="http://samtools.sourceforge.net/samtools.shtml">http://samtools.sourceforge.net/samtools.shtml</a> for more details</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Sequencing.Applications</span> <span class="k">import</span> <span class="n">SamtoolsCatCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">input_bam1</span> <span class="o">=</span> <span class="s2">&quot;/path/to/input_bam1&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">input_bam2</span> <span class="o">=</span> <span class="s2">&quot;/path/to/input_bam2&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">input_bams</span> <span class="o">=</span> <span class="p">[</span><span class="n">input_bam1</span><span class="p">,</span> <span class="n">input_bam2</span><span class="p">]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">samtools_cat_cmd</span> <span class="o">=</span> <span class="n">SamtoolsCatCommandline</span><span class="p">(</span><span class="n">input_bam</span><span class="o">=</span><span class="n">input_bams</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">samtools_cat_cmd</span><span class="p">)</span>
<span class="go">samtools cat /path/to/input_bam1 /path/to/input_bam2</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Sequencing.Applications.SamtoolsFaidxCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Sequencing.Applications.</code><code class="sig-name descname">SamtoolsFaidxCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='samtools'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Sequencing.Applications.SamtoolsFaidxCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for samtools faidx.</p>
<p>Retrieve and print stats in the index file, equivalent to:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ samtools faidx &lt;ref.fasta&gt; [region1 [...]]
</pre></div>
</div>
<p>See <a class="reference external" href="http://samtools.sourceforge.net/samtools.shtml">http://samtools.sourceforge.net/samtools.shtml</a> for more details</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Sequencing.Applications</span> <span class="k">import</span> <span class="n">SamtoolsFaidxCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">reference</span> <span class="o">=</span> <span class="s2">&quot;/path/to/reference.fasta&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">samtools_faidx_cmd</span> <span class="o">=</span> <span class="n">SamtoolsFaidxCommandline</span><span class="p">(</span><span class="n">reference</span><span class="o">=</span><span class="n">reference</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">samtools_faidx_cmd</span><span class="p">)</span>
<span class="go">samtools faidx /path/to/reference.fasta</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Sequencing.Applications.SamtoolsFixmateCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Sequencing.Applications.</code><code class="sig-name descname">SamtoolsFixmateCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='samtools'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Sequencing.Applications.SamtoolsFixmateCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for samtools fixmate.</p>
<p>Fill in mate coordinates, ISIZE and mate related
flags from a name-sorted alignment, equivalent to:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ samtools fixmate &lt;in.nameSrt.bam&gt; &lt;out.bam&gt;
</pre></div>
</div>
<p>See <a class="reference external" href="http://samtools.sourceforge.net/samtools.shtml">http://samtools.sourceforge.net/samtools.shtml</a> for more details</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Sequencing.Applications</span> <span class="k">import</span> <span class="n">SamtoolsFixmateCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">in_bam</span> <span class="o">=</span> <span class="s2">&quot;/path/to/in.nameSrt.bam&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">out_bam</span> <span class="o">=</span> <span class="s2">&quot;/path/to/out.bam&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">fixmate_cmd</span> <span class="o">=</span> <span class="n">SamtoolsFixmateCommandline</span><span class="p">(</span><span class="n">input_bam</span><span class="o">=</span><span class="n">in_bam</span><span class="p">,</span>
<span class="gp">... </span>                                         <span class="n">out_bam</span><span class="o">=</span><span class="n">out_bam</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">fixmate_cmd</span><span class="p">)</span>
<span class="go">samtools fixmate /path/to/in.nameSrt.bam /path/to/out.bam</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Sequencing.Applications.SamtoolsIdxstatsCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Sequencing.Applications.</code><code class="sig-name descname">SamtoolsIdxstatsCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='samtools'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Sequencing.Applications.SamtoolsIdxstatsCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for samtools idxstats.</p>
<p>Retrieve and print stats in the index file, equivalent to:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ samtools idxstats &lt;aln.bam&gt;
</pre></div>
</div>
<p>See <a class="reference external" href="http://samtools.sourceforge.net/samtools.shtml">http://samtools.sourceforge.net/samtools.shtml</a> for more details</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Sequencing.Applications</span> <span class="k">import</span> <span class="n">SamtoolsIdxstatsCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">input</span> <span class="o">=</span> <span class="s2">&quot;/path/to/aln_bam&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">samtools_idxstats_cmd</span> <span class="o">=</span> <span class="n">SamtoolsIdxstatsCommandline</span><span class="p">(</span><span class="n">input_bam</span><span class="o">=</span><span class="nb">input</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">samtools_idxstats_cmd</span><span class="p">)</span>
<span class="go">samtools idxstats /path/to/aln_bam</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Sequencing.Applications.SamtoolsIndexCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Sequencing.Applications.</code><code class="sig-name descname">SamtoolsIndexCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='samtools'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Sequencing.Applications.SamtoolsIndexCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for samtools index.</p>
<p>Index sorted alignment for fast random access, equivalent to:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ samtools index &lt;aln.bam&gt;
</pre></div>
</div>
<p>See <a class="reference external" href="http://samtools.sourceforge.net/samtools.shtml">http://samtools.sourceforge.net/samtools.shtml</a> for more details</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Sequencing.Applications</span> <span class="k">import</span> <span class="n">SamtoolsIndexCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">input</span> <span class="o">=</span> <span class="s2">&quot;/path/to/aln_bam&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">samtools_index_cmd</span> <span class="o">=</span> <span class="n">SamtoolsIndexCommandline</span><span class="p">(</span><span class="n">input_bam</span><span class="o">=</span><span class="nb">input</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">samtools_index_cmd</span><span class="p">)</span>
<span class="go">samtools index /path/to/aln_bam</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Sequencing.Applications.SamtoolsMergeCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Sequencing.Applications.</code><code class="sig-name descname">SamtoolsMergeCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='samtools'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Sequencing.Applications.SamtoolsMergeCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for samtools merge.</p>
<p>Merge multiple sorted alignments, equivalent to:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ samtools merge [-nur1f] [-h inh.sam] [-R reg]
                 &lt;out.bam&gt; &lt;in1.bam&gt; &lt;in2.bam&gt; [...]
</pre></div>
</div>
<p>See <a class="reference external" href="http://samtools.sourceforge.net/samtools.shtml">http://samtools.sourceforge.net/samtools.shtml</a> for more details</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Sequencing.Applications</span> <span class="k">import</span> <span class="n">SamtoolsMergeCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">out_bam</span> <span class="o">=</span> <span class="s2">&quot;/path/to/out_bam&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">in_bam</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;/path/to/input_bam1&quot;</span><span class="p">,</span> <span class="s2">&quot;/path/to/input_bam2&quot;</span><span class="p">]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">merge_cmd</span> <span class="o">=</span> <span class="n">SamtoolsMergeCommandline</span><span class="p">(</span><span class="n">out_bam</span><span class="o">=</span><span class="n">out_bam</span><span class="p">,</span>
<span class="gp">... </span>                                     <span class="n">input_bam</span><span class="o">=</span><span class="n">in_bam</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">merge_cmd</span><span class="p">)</span>
<span class="go">samtools merge /path/to/out_bam /path/to/input_bam1 /path/to/input_bam2</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Sequencing.Applications.SamtoolsMpileupCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Sequencing.Applications.</code><code class="sig-name descname">SamtoolsMpileupCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='samtools'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Sequencing.Applications.SamtoolsMpileupCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for samtools mpileup.</p>
<p>Generate BCF or pileup for one or multiple BAM files, equivalent to:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ samtools mpileup [-EBug] [-C capQcoef] [-r reg] [-f in.fa]
                   [-l list] [-M capMapQ] [-Q minBaseQ]
                   [-q minMapQ] in.bam [in2.bam [...]]
</pre></div>
</div>
<p>See <a class="reference external" href="http://samtools.sourceforge.net/samtools.shtml">http://samtools.sourceforge.net/samtools.shtml</a> for more details</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Sequencing.Applications</span> <span class="k">import</span> <span class="n">SamtoolsMpileupCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">input</span> <span class="o">=</span> <span class="p">[</span><span class="s2">&quot;/path/to/sam_or_bam_file&quot;</span><span class="p">]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">samtools_mpileup_cmd</span> <span class="o">=</span> <span class="n">SamtoolsMpileupCommandline</span><span class="p">(</span><span class="n">input_file</span><span class="o">=</span><span class="nb">input</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">samtools_mpileup_cmd</span><span class="p">)</span>
<span class="go">samtools mpileup /path/to/sam_or_bam_file</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Sequencing.Applications.SamtoolsPhaseCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Sequencing.Applications.</code><code class="sig-name descname">SamtoolsPhaseCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='samtools'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Sequencing.Applications.SamtoolsPhaseCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for samtools phase.</p>
<p>Call and phase heterozygous SNPs, equivalent to:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ samtools phase [-AF] [-k len] [-b prefix]
                 [-q minLOD] [-Q minBaseQ] &lt;in.bam&gt;
</pre></div>
</div>
<p>See <a class="reference external" href="http://samtools.sourceforge.net/samtools.shtml">http://samtools.sourceforge.net/samtools.shtml</a> for more details</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Sequencing.Applications</span> <span class="k">import</span> <span class="n">SamtoolsPhaseCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">input_bam</span> <span class="o">=</span> <span class="s2">&quot;/path/to/in.bam&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">samtools_phase_cmd</span> <span class="o">=</span> <span class="n">SamtoolsPhaseCommandline</span><span class="p">(</span><span class="n">input_bam</span><span class="o">=</span><span class="n">input_bam</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">samtools_phase_cmd</span><span class="p">)</span>
<span class="go">samtools phase /path/to/in.bam</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Sequencing.Applications.SamtoolsReheaderCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Sequencing.Applications.</code><code class="sig-name descname">SamtoolsReheaderCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='samtools'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Sequencing.Applications.SamtoolsReheaderCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for samtools reheader.</p>
<p>Replace the header in in.bam with the header
in in.header.sam, equivalent to:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ samtools reheader &lt;in.header.sam&gt; &lt;in.bam&gt;
</pre></div>
</div>
<p>See <a class="reference external" href="http://samtools.sourceforge.net/samtools.shtml">http://samtools.sourceforge.net/samtools.shtml</a> for more details</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Sequencing.Applications</span> <span class="k">import</span> <span class="n">SamtoolsReheaderCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">input_header</span> <span class="o">=</span> <span class="s2">&quot;/path/to/header_sam_file&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">input_bam</span> <span class="o">=</span> <span class="s2">&quot;/path/to/input_bam_file&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">reheader_cmd</span> <span class="o">=</span> <span class="n">SamtoolsReheaderCommandline</span><span class="p">(</span><span class="n">input_header</span><span class="o">=</span><span class="n">input_header</span><span class="p">,</span>
<span class="gp">... </span>                                           <span class="n">input_bam</span><span class="o">=</span><span class="n">input_bam</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">reheader_cmd</span><span class="p">)</span>
<span class="go">samtools reheader /path/to/header_sam_file /path/to/input_bam_file</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Sequencing.Applications.SamtoolsRmdupCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Sequencing.Applications.</code><code class="sig-name descname">SamtoolsRmdupCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='samtools'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Sequencing.Applications.SamtoolsRmdupCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for samtools rmdup.</p>
<p>Remove potential PCR duplicates, equivalent to:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ samtools rmdup [-sS] &lt;input.srt.bam&gt; &lt;out.bam&gt;
</pre></div>
</div>
<p>See <a class="reference external" href="http://samtools.sourceforge.net/samtools.shtml">http://samtools.sourceforge.net/samtools.shtml</a> for more details</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Sequencing.Applications</span> <span class="k">import</span> <span class="n">SamtoolsRmdupCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">input_sorted_bam</span> <span class="o">=</span> <span class="s2">&quot;/path/to/input.srt.bam&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">out_bam</span> <span class="o">=</span> <span class="s2">&quot;/path/to/out.bam&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">rmdup_cmd</span> <span class="o">=</span> <span class="n">SamtoolsRmdupCommandline</span><span class="p">(</span><span class="n">input_bam</span><span class="o">=</span><span class="n">input_sorted_bam</span><span class="p">,</span>
<span class="gp">... </span>                                     <span class="n">out_bam</span><span class="o">=</span><span class="n">out_bam</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">rmdup_cmd</span><span class="p">)</span>
<span class="go">samtools rmdup /path/to/input.srt.bam /path/to/out.bam</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="attribute">
<dt id="Bio.Sequencing.Applications.SamtoolsSortCommandline">
<code class="sig-prename descclassname">Bio.Sequencing.Applications.</code><code class="sig-name descname">SamtoolsSortCommandline</code><a class="headerlink" href="#Bio.Sequencing.Applications.SamtoolsSortCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>alias of <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Sequencing.Applications._samtools.SamtoolsVersion0xSortCommandline</span></code></p>
</dd></dl>

<dl class="class">
<dt id="Bio.Sequencing.Applications.SamtoolsVersion0xSortCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Sequencing.Applications.</code><code class="sig-name descname">SamtoolsVersion0xSortCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='samtools'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Sequencing.Applications.SamtoolsVersion0xSortCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for samtools version 0.1.x sort.</p>
<p>Concatenate BAMs, equivalent to:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ samtools sort [-no] [-m maxMem] &lt;in.bam&gt; &lt;out.prefix&gt;
</pre></div>
</div>
<p>See <a class="reference external" href="http://samtools.sourceforge.net/samtools.shtml">http://samtools.sourceforge.net/samtools.shtml</a> for more details</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Sequencing.Applications</span> <span class="k">import</span> <span class="n">SamtoolsVersion0xSortCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">input_bam</span> <span class="o">=</span> <span class="s2">&quot;/path/to/input_bam&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">out_prefix</span> <span class="o">=</span> <span class="s2">&quot;/path/to/out_prefix&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">samtools_sort_cmd</span> <span class="o">=</span> <span class="n">SamtoolsVersion0xSortCommandline</span><span class="p">(</span><span class="nb">input</span><span class="o">=</span><span class="n">input_bam</span><span class="p">,</span> <span class="n">out_prefix</span><span class="o">=</span><span class="n">out_prefix</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">samtools_sort_cmd</span><span class="p">)</span>
<span class="go">samtools sort /path/to/input_bam /path/to/out_prefix</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Sequencing.Applications.SamtoolsVersion1xSortCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Sequencing.Applications.</code><code class="sig-name descname">SamtoolsVersion1xSortCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='samtools'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Sequencing.Applications.SamtoolsVersion1xSortCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for samtools version 1.3.x sort.</p>
<p>Concatenate BAMs, equivalent to:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ samtools sort [-n] [-T FREFIX] [-o file] [-I INT] [-m maxMem] &lt;in.bam&gt;
</pre></div>
</div>
<p>See <a class="reference external" href="http://samtools.sourceforge.net/samtools.shtml">http://samtools.sourceforge.net/samtools.shtml</a> for more details</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Sequencing.Applications</span> <span class="k">import</span> <span class="n">SamtoolsVersion1xSortCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">input_bam</span> <span class="o">=</span> <span class="s2">&quot;/path/to/input_bam&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">FREFIX</span> <span class="o">=</span> <span class="s2">&quot;/path/to/out_prefix&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">file_name</span> <span class="o">=</span> <span class="s2">&quot;/path/to/out_file&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">samtools_sort_cmd</span> <span class="o">=</span> <span class="n">SamtoolsVersion1xSortCommandline</span><span class="p">(</span><span class="nb">input</span><span class="o">=</span><span class="n">input_bam</span><span class="p">,</span> <span class="n">T</span><span class="o">=</span><span class="n">FREFIX</span><span class="p">,</span> <span class="n">o</span><span class="o">=</span><span class="n">file_name</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">samtools_sort_cmd</span><span class="p">)</span>
<span class="go">samtools sort -o /path/to/out_file -T /path/to/out_prefix /path/to/input_bam</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

<dl class="class">
<dt id="Bio.Sequencing.Applications.SamtoolsTargetcutCommandline">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Sequencing.Applications.</code><code class="sig-name descname">SamtoolsTargetcutCommandline</code><span class="sig-paren">(</span><em class="sig-param">cmd='samtools'</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Sequencing.Applications.SamtoolsTargetcutCommandline" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="Bio.Application.html#Bio.Application.AbstractCommandline" title="Bio.Application.AbstractCommandline"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Application.AbstractCommandline</span></code></a></p>
<p>Command line wrapper for samtools targetcut.</p>
<p>This command identifies target regions by examining the continuity
of read depth, computes haploid consensus sequences of targets
and outputs a SAM with each sequence corresponding to a target,
equivalent to:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span>$ samtools targetcut [-Q minBaseQ] [-i inPenalty] [-0 em0]
                     [-1 em1] [-2 em2] [-f ref] &lt;in.bam&gt;
</pre></div>
</div>
<p>See <a class="reference external" href="http://samtools.sourceforge.net/samtools.shtml">http://samtools.sourceforge.net/samtools.shtml</a> for more details</p>
<p class="rubric">Examples</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Sequencing.Applications</span> <span class="k">import</span> <span class="n">SamtoolsTargetcutCommandline</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">input_bam</span> <span class="o">=</span> <span class="s2">&quot;/path/to/aln.bam&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">samtools_targetcut_cmd</span> <span class="o">=</span> <span class="n">SamtoolsTargetcutCommandline</span><span class="p">(</span><span class="n">input_bam</span><span class="o">=</span><span class="n">input_bam</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">samtools_targetcut_cmd</span><span class="p">)</span>
<span class="go">samtools targetcut /path/to/aln.bam</span>
</pre></div>
</div>
<dl class="field-list simple">
<dt class="field-odd">Attributes</dt>
<dd class="field-odd"><dl class="simple">
<dt><strong>parameters</strong></dt><dd></dd>
</dl>
</dd>
</dl>
<p class="rubric">Methods</p>
<table class="longtable docutils align-default">
<colgroup>
<col style="width: 10%" />
<col style="width: 90%" />
</colgroup>
<tbody>
<tr class="row-odd"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">__call__</span></code>(self[, stdin, stdout, stderr, cwd, env])</p></td>
<td><p>Execute command, wait for it to finish, return (stdout, stderr).</p></td>
</tr>
<tr class="row-even"><td><p><code class="xref py py-obj docutils literal notranslate"><span class="pre">set_parameter</span></code>(self, name[, value])</p></td>
<td><p>Set a commandline option for a program (OBSOLETE).</p></td>
</tr>
</tbody>
</table>
</dd></dl>

</div>
</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="Bio.Sequencing.Ace.html" class="btn btn-neutral float-right" title="Bio.Sequencing.Ace module" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="Bio.Sequencing.html" class="btn btn-neutral float-left" title="Bio.Sequencing package" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 1999-2017, The Biopython Contributors

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>