

<!DOCTYPE html>
<html class="writer-html5" lang="en" >
<head>
  <meta charset="utf-8" />
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0" />
  
  <title>Bio.Seq module &mdash; Biopython 1.79.dev0 documentation</title>
  

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
  <link rel="stylesheet" href="_static/biopython.css" type="text/css" />

  
  

  
  

  

  
  <!--[if lt IE 9]>
    <script src="_static/js/html5shiv.min.js"></script>
  <![endif]-->
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script src="_static/jquery.js"></script>
        <script src="_static/underscore.js"></script>
        <script src="_static/doctools.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Bio.SeqFeature module" href="Bio.SeqFeature.html" />
    <link rel="prev" title="Bio.NaiveBayes module" href="Bio.NaiveBayes.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search"  style="background: #10100F" >
          

          
            <a href="index.html">
          

          
            
            <img src="_static/biopython_logo_white.png" class="logo" alt="Logo"/>
          
          </a>

          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        
        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">API Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="Bio.html">Bio package</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="Bio.html#subpackages">Subpackages</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="Bio.html#submodules">Submodules</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="Bio.File.html">Bio.File module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.LogisticRegression.html">Bio.LogisticRegression module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.MarkovModel.html">Bio.MarkovModel module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.MaxEntropy.html">Bio.MaxEntropy module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.NaiveBayes.html">Bio.NaiveBayes module</a></li>
<li class="toctree-l3 current"><a class="current reference internal" href="#">Bio.Seq module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SeqFeature.html">Bio.SeqFeature module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SeqRecord.html">Bio.SeqRecord module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.bgzf.html">Bio.bgzf module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.kNN.html">Bio.kNN module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.pairwise2.html">Bio.pairwise2 module</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="Bio.html#module-Bio">Module contents</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="BioSQL.html">BioSQL package</a></li>
</ul>

            
          
        </div>
        
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Biopython</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          



















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html" class="icon icon-home"></a> &raquo;</li>
        
          <li><a href="Bio.html">Bio package</a> &raquo;</li>
        
      <li>Bio.Seq module</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
          
            
              <!-- User defined GitHub URL -->
              <a href="https://github.com/biopython/biopython/blob/master/Bio/Seq.py" class="fa fa-github"> Edit on GitHub</a>
            
          
        
      </li>
    
  </ul>

  
  <div class="rst-breadcrumbs-buttons" role="navigation" aria-label="breadcrumb navigation">
      
        <a href="Bio.SeqFeature.html" class="btn btn-neutral float-right" title="Bio.SeqFeature module" accesskey="n">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
      
      
        <a href="Bio.NaiveBayes.html" class="btn btn-neutral float-left" title="Bio.NaiveBayes module" accesskey="p"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
      
  </div>
  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="module-Bio.Seq">
<span id="bio-seq-module"></span><h1>Bio.Seq module<a class="headerlink" href="#module-Bio.Seq" title="Permalink to this headline">¶</a></h1>
<p>Provide objects to represent biological sequences.</p>
<dl class="simple">
<dt>See also the <a class="reference external" href="http://biopython.org/wiki/Seq">Seq</a> wiki and the chapter in our tutorial:</dt><dd><ul class="simple">
<li><p><a class="reference external" href="http://biopython.org/DIST/docs/tutorial/Tutorial.html">HTML Tutorial</a></p></li>
<li><p><a class="reference external" href="http://biopython.org/DIST/docs/tutorial/Tutorial.pdf">PDF Tutorial</a></p></li>
</ul>
</dd>
</dl>
<dl class="py class">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Seq.</code><code class="sig-name descname">SequenceDataAbstractBaseClass</code><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">abc.ABC</span></code></p>
<p>Abstract base class for sequence content providers.</p>
<p>Most users will not need to use this class. It is used internally as a base
class for sequence content provider classes such as _UndefinedSequenceData
defined in this module, and _TwoBitSequenceData in Bio.SeqIO.TwoBitIO.
Instances of these classes can be used instead of a <code class="docutils literal notranslate"><span class="pre">bytes</span></code> object as the
data argument when creating a Seq object, and provide the sequence content
only when requested via <code class="docutils literal notranslate"><span class="pre">__getitem__</span></code>. This allows lazy parsers to load
and parse sequence data from a file only for the requested sequence regions,
and _UndefinedSequenceData instances to raise an exception when undefined
sequence data are requested.</p>
<p>Future implementations of lazy parsers that similarly provide on-demand
parsing of sequence data should use a subclass of this abstract class and
implement the abstract methods <code class="docutils literal notranslate"><span class="pre">__len__</span></code> and <code class="docutils literal notranslate"><span class="pre">__getitem__</span></code>:</p>
<ul>
<li><p><code class="docutils literal notranslate"><span class="pre">__len__</span></code> must return the sequence length;</p></li>
<li><p><code class="docutils literal notranslate"><span class="pre">__getitem__</span></code> must return</p>
<ul class="simple">
<li><p>a <code class="docutils literal notranslate"><span class="pre">bytes</span></code> object for the requested region; or</p></li>
<li><p>a new instance of the subclass for the requested region; or</p></li>
<li><p>raise an <code class="docutils literal notranslate"><span class="pre">UndefinedSequenceError</span></code>.</p></li>
</ul>
<p>Calling <code class="docutils literal notranslate"><span class="pre">__getitem__</span></code> for a sequence region of size zero should always
return an empty <code class="docutils literal notranslate"><span class="pre">bytes</span></code> object.
Calling <code class="docutils literal notranslate"><span class="pre">__getitem__</span></code> for the full sequence (as in data[:]) should
either return a <code class="docutils literal notranslate"><span class="pre">bytes</span></code> object with the full sequence, or raise an
<code class="docutils literal notranslate"><span class="pre">UndefinedSequenceError</span></code>.</p>
</li>
</ul>
<p>Subclasses of SequenceDataAbstractBaseClass must call <code class="docutils literal notranslate"><span class="pre">super().__init__()</span></code>
as part of their <code class="docutils literal notranslate"><span class="pre">__init__</span></code> method.</p>
<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.__init__">
<code class="sig-name descname">__init__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.__init__" title="Permalink to this definition">¶</a></dt>
<dd><p>Check if <code class="docutils literal notranslate"><span class="pre">__getitem__</span></code> returns a bytes-like object.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.__len__">
<em class="property">abstract </em><code class="sig-name descname">__len__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.__len__" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.__getitem__">
<em class="property">abstract </em><code class="sig-name descname">__getitem__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">key</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.__getitem__" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.__bytes__">
<code class="sig-name descname">__bytes__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.__bytes__" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.__hash__">
<code class="sig-name descname">__hash__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.__hash__" title="Permalink to this definition">¶</a></dt>
<dd><p>Return hash(self).</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.__eq__">
<code class="sig-name descname">__eq__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">other</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.__eq__" title="Permalink to this definition">¶</a></dt>
<dd><p>Return self==value.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.__lt__">
<code class="sig-name descname">__lt__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">other</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.__lt__" title="Permalink to this definition">¶</a></dt>
<dd><p>Return self&lt;value.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.__le__">
<code class="sig-name descname">__le__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">other</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.__le__" title="Permalink to this definition">¶</a></dt>
<dd><p>Return self&lt;=value.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.__gt__">
<code class="sig-name descname">__gt__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">other</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.__gt__" title="Permalink to this definition">¶</a></dt>
<dd><p>Return self&gt;value.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.__ge__">
<code class="sig-name descname">__ge__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">other</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.__ge__" title="Permalink to this definition">¶</a></dt>
<dd><p>Return self&gt;=value.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.__add__">
<code class="sig-name descname">__add__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">other</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.__add__" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.__radd__">
<code class="sig-name descname">__radd__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">other</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.__radd__" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.__mul__">
<code class="sig-name descname">__mul__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">other</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.__mul__" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.__contains__">
<code class="sig-name descname">__contains__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">item</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.__contains__" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.decode">
<code class="sig-name descname">decode</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">encoding</span><span class="o">=</span><span class="default_value">'utf-8'</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.decode" title="Permalink to this definition">¶</a></dt>
<dd><p>Decode the data as bytes using the codec registered for encoding.</p>
<dl class="simple">
<dt>encoding</dt><dd><p>The encoding with which to decode the bytes.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.count">
<code class="sig-name descname">count</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">sub</span></em>, <em class="sig-param"><span class="n">start</span><span class="o">=</span><span class="default_value">None</span></em>, <em class="sig-param"><span class="n">end</span><span class="o">=</span><span class="default_value">None</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.count" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the number of non-overlapping occurrences of sub in data[start:end].</p>
<p>Optional arguments start and end are interpreted as in slice notation.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.find">
<code class="sig-name descname">find</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">sub</span></em>, <em class="sig-param"><span class="n">start</span><span class="o">=</span><span class="default_value">None</span></em>, <em class="sig-param"><span class="n">end</span><span class="o">=</span><span class="default_value">None</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.find" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the lowest index in data where subsection sub is found.</p>
<p>Return the lowest index in data where subsection sub is found,
such that sub is contained within data[start,end].  Optional
arguments start and end are interpreted as in slice notation.</p>
<p>Return -1 on failure.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.rfind">
<code class="sig-name descname">rfind</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">sub</span></em>, <em class="sig-param"><span class="n">start</span><span class="o">=</span><span class="default_value">None</span></em>, <em class="sig-param"><span class="n">end</span><span class="o">=</span><span class="default_value">None</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.rfind" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the highest index in data where subsection sub is found.</p>
<p>Return the highest index in data where subsection sub is found,
such that sub is contained within data[start,end].  Optional
arguments start and end are interpreted as in slice notation.</p>
<p>Return -1 on failure.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.index">
<code class="sig-name descname">index</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">sub</span></em>, <em class="sig-param"><span class="n">start</span><span class="o">=</span><span class="default_value">None</span></em>, <em class="sig-param"><span class="n">end</span><span class="o">=</span><span class="default_value">None</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.index" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the lowest index in data where subsection sub is found.</p>
<p>Return the lowest index in data where subsection sub is found,
such that sub is contained within data[start,end].  Optional
arguments start and end are interpreted as in slice notation.</p>
<p>Raises ValueError when the subsection is not found.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.rindex">
<code class="sig-name descname">rindex</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">sub</span></em>, <em class="sig-param"><span class="n">start</span><span class="o">=</span><span class="default_value">None</span></em>, <em class="sig-param"><span class="n">end</span><span class="o">=</span><span class="default_value">None</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.rindex" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the highest index in data where subsection sub is found.</p>
<p>Return the highest index in data where subsection sub is found,
such that sub is contained within data[start,end].  Optional
arguments start and end are interpreted as in slice notation.</p>
<p>Raise ValueError when the subsection is not found.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.startswith">
<code class="sig-name descname">startswith</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">prefix</span></em>, <em class="sig-param"><span class="n">start</span><span class="o">=</span><span class="default_value">None</span></em>, <em class="sig-param"><span class="n">end</span><span class="o">=</span><span class="default_value">None</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.startswith" title="Permalink to this definition">¶</a></dt>
<dd><p>Return True if data starts with the specified prefix, False otherwise.</p>
<p>With optional start, test data beginning at that position.
With optional end, stop comparing data at that position.
prefix can also be a tuple of bytes to try.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.endswith">
<code class="sig-name descname">endswith</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">suffix</span></em>, <em class="sig-param"><span class="n">start</span><span class="o">=</span><span class="default_value">None</span></em>, <em class="sig-param"><span class="n">end</span><span class="o">=</span><span class="default_value">None</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.endswith" title="Permalink to this definition">¶</a></dt>
<dd><p>Return True if data ends with the specified suffix, False otherwise.</p>
<p>With optional start, test data beginning at that position.
With optional end, stop comparing data at that position.
suffix can also be a tuple of bytes to try.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.split">
<code class="sig-name descname">split</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">sep</span><span class="o">=</span><span class="default_value">None</span></em>, <em class="sig-param"><span class="n">maxsplit</span><span class="o">=</span><span class="default_value">- 1</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.split" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a list of the sections in the data, using sep as the delimiter.</p>
<dl class="simple">
<dt>sep</dt><dd><p>The delimiter according which to split the data.
None (the default value) means split on ASCII whitespace characters
(space, tab, return, newline, formfeed, vertical tab).</p>
</dd>
<dt>maxsplit</dt><dd><p>Maximum number of splits to do.
-1 (the default value) means no limit.</p>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.rsplit">
<code class="sig-name descname">rsplit</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">sep</span><span class="o">=</span><span class="default_value">None</span></em>, <em class="sig-param"><span class="n">maxsplit</span><span class="o">=</span><span class="default_value">- 1</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.rsplit" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a list of the sections in the data, using sep as the delimiter.</p>
<dl class="simple">
<dt>sep</dt><dd><p>The delimiter according which to split the data.
None (the default value) means split on ASCII whitespace characters
(space, tab, return, newline, formfeed, vertical tab).</p>
</dd>
<dt>maxsplit</dt><dd><p>Maximum number of splits to do.
-1 (the default value) means no limit.</p>
</dd>
</dl>
<p>Splitting is done starting at the end of the data and working to the front.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.strip">
<code class="sig-name descname">strip</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">chars</span><span class="o">=</span><span class="default_value">None</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.strip" title="Permalink to this definition">¶</a></dt>
<dd><p>Strip leading and trailing characters contained in the argument.</p>
<p>If the argument is omitted or None, strip leading and trailing ASCII whitespace.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.lstrip">
<code class="sig-name descname">lstrip</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">chars</span><span class="o">=</span><span class="default_value">None</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.lstrip" title="Permalink to this definition">¶</a></dt>
<dd><p>Strip leading characters contained in the argument.</p>
<p>If the argument is omitted or None, strip leading ASCII whitespace.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.rstrip">
<code class="sig-name descname">rstrip</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">chars</span><span class="o">=</span><span class="default_value">None</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.rstrip" title="Permalink to this definition">¶</a></dt>
<dd><p>Strip trailing characters contained in the argument.</p>
<p>If the argument is omitted or None, strip trailing ASCII whitespace.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.upper">
<code class="sig-name descname">upper</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.upper" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a copy of data with all ASCII characters converted to uppercase.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.lower">
<code class="sig-name descname">lower</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.lower" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a copy of data with all ASCII characters converted to lowercase.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.replace">
<code class="sig-name descname">replace</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">old</span></em>, <em class="sig-param"><span class="n">new</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.replace" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a copy with all occurrences of substring old replaced by new.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.translate">
<code class="sig-name descname">translate</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">table</span></em>, <em class="sig-param"><span class="n">delete</span><span class="o">=</span><span class="default_value">b''</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.translate" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a copy with each character mapped by the given translation table.</p>
<blockquote>
<div><dl class="simple">
<dt>table</dt><dd><p>Translation table, which must be a bytes object of length 256.</p>
</dd>
</dl>
</div></blockquote>
<p>All characters occurring in the optional argument delete are removed.
The remaining characters are mapped through the given translation table.</p>
</dd></dl>

<dl class="py attribute">
<dt id="Bio.Seq.SequenceDataAbstractBaseClass.__abstractmethods__">
<code class="sig-name descname">__abstractmethods__</code><em class="property"> = frozenset({'__getitem__', '__len__'})</em><a class="headerlink" href="#Bio.Seq.SequenceDataAbstractBaseClass.__abstractmethods__" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py class">
<dt id="Bio.Seq.Seq">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Seq.</code><code class="sig-name descname">Seq</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">data</span></em>, <em class="sig-param"><span class="n">length</span><span class="o">=</span><span class="default_value">None</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Seq._SeqAbstractBaseClass</span></code></p>
<p>Read-only sequence object (essentially a string with biological methods).</p>
<p>Like normal python strings, our basic sequence object is immutable.
This prevents you from doing my_seq[5] = “A” for example, but does allow
Seq objects to be used as dictionary keys.</p>
<p>The Seq object provides a number of string like methods (such as count,
find, split and strip).</p>
<p>The Seq object also provides some biological methods, such as complement,
reverse_complement, transcribe, back_transcribe and translate (which are
not applicable to protein sequences).</p>
<dl class="py method">
<dt id="Bio.Seq.Seq.__init__">
<code class="sig-name descname">__init__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">data</span></em>, <em class="sig-param"><span class="n">length</span><span class="o">=</span><span class="default_value">None</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.__init__" title="Permalink to this definition">¶</a></dt>
<dd><p>Create a Seq object.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>data - Sequence, required (string)</p></li>
<li><p>length - Sequence length, used only if data is None (integer)</p></li>
</ul>
</dd>
</dl>
<p>You will typically use Bio.SeqIO to read in sequences from files as
SeqRecord objects, whose sequence will be exposed as a Seq object via
the seq property.</p>
<p>However, you can also create a Seq object directly:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">Seq(&#39;MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_seq</span><span class="p">)</span>
<span class="go">MKQHKAMIVALIVICITAVVAALVTRKDLCEVHIRTGQTEVAVF</span>
</pre></div>
</div>
<p>To create a Seq object with for a sequence of known length but
unknown sequence contents, use None for the data argument and pass
the sequence length for the length argument. Trying to access the
sequence contents of a Seq object created in this way will raise
an UndefinedSequenceError:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_undefined_seq</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="kc">None</span><span class="p">,</span> <span class="mi">20</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_undefined_seq</span>
<span class="go">Seq(None, length=20)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">my_undefined_seq</span><span class="p">)</span>
<span class="go">20</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_undefined_seq</span><span class="p">)</span>
<span class="gt">Traceback (most recent call last):</span>
<span class="c">...</span>
<span class="gr">Bio.Seq.UndefinedSequenceError</span>: <span class="n">Sequence content is undefined</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.Seq.__hash__">
<code class="sig-name descname">__hash__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.__hash__" title="Permalink to this definition">¶</a></dt>
<dd><p>Hash of the sequence as a string for comparison.</p>
<p>See Seq object comparison documentation (method <code class="docutils literal notranslate"><span class="pre">__eq__</span></code> in
particular) as this has changed in Biopython 1.65. Older versions
would hash on object identity.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.Seq.tomutable">
<code class="sig-name descname">tomutable</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.tomutable" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the full sequence as a MutableSeq object.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;MKQHKAMIVALIVICITAVVAAL&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">Seq(&#39;MKQHKAMIVALIVICITAVVAAL&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">tomutable</span><span class="p">()</span>
<span class="go">MutableSeq(&#39;MKQHKAMIVALIVICITAVVAAL&#39;)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.Seq.encode">
<code class="sig-name descname">encode</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">encoding</span><span class="o">=</span><span class="default_value">'utf-8'</span></em>, <em class="sig-param"><span class="n">errors</span><span class="o">=</span><span class="default_value">'strict'</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.encode" title="Permalink to this definition">¶</a></dt>
<dd><p>Return an encoded version of the sequence as a bytes object.</p>
<p>The Seq object aims to match the interface of a Python string.</p>
<p>This is essentially to save you doing str(my_seq).encode() when
you need a bytes string, for example for computing a hash:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;ACGT&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">encode</span><span class="p">(</span><span class="s2">&quot;ascii&quot;</span><span class="p">)</span>
<span class="go">b&#39;ACGT&#39;</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.Seq.complement">
<code class="sig-name descname">complement</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.complement" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the complement sequence by creating a new Seq object.</p>
<p>This method is intended for use with DNA sequences:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;CCCCCGATAG&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span>
<span class="go">Seq(&#39;CCCCCGATAG&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span><span class="o">.</span><span class="n">complement</span><span class="p">()</span>
<span class="go">Seq(&#39;GGGGGCTATC&#39;)</span>
</pre></div>
</div>
<p>You can of course used mixed case sequences,</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;CCCCCgatA-GD&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span>
<span class="go">Seq(&#39;CCCCCgatA-GD&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span><span class="o">.</span><span class="n">complement</span><span class="p">()</span>
<span class="go">Seq(&#39;GGGGGctaT-CH&#39;)</span>
</pre></div>
</div>
<p>Note in the above example, ambiguous character D denotes
G, A or T so its complement is H (for C, T or A).</p>
<p>Note that if the sequence contains neither T nor U, we
assume it is DNA and map any A character to T:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;CGA&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">complement</span><span class="p">()</span>
<span class="go">Seq(&#39;GCT&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;CGAT&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">complement</span><span class="p">()</span>
<span class="go">Seq(&#39;GCTA&#39;)</span>
</pre></div>
</div>
<p>If you actually have RNA, this currently works but we
may deprecate this later. We recommend using the new
complement_rna method instead:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;CGAU&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">complement</span><span class="p">()</span>
<span class="go">Seq(&#39;GCUA&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;CGAU&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">complement_rna</span><span class="p">()</span>
<span class="go">Seq(&#39;GCUA&#39;)</span>
</pre></div>
</div>
<p>If the sequence contains both T and U, an exception is
raised:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;CGAUT&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">complement</span><span class="p">()</span>
<span class="gt">Traceback (most recent call last):</span>
   <span class="c">...</span>
<span class="gr">ValueError</span>: <span class="n">Mixed RNA/DNA found</span>
</pre></div>
</div>
<p>Trying to complement a protein sequence gives a meaningless
sequence:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_protein</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;MAIVMGR&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_protein</span><span class="o">.</span><span class="n">complement</span><span class="p">()</span>
<span class="go">Seq(&#39;KTIBKCY&#39;)</span>
</pre></div>
</div>
<p>Here “M” was interpreted as the IUPAC ambiguity code for
“A” or “C”, with complement “K” for “T” or “G”. Likewise
“A” has complement “T”. The letter “I” has no defined
meaning under the IUPAC convention, and is unchanged.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.Seq.reverse_complement">
<code class="sig-name descname">reverse_complement</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.reverse_complement" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the reverse complement sequence by creating a new Seq object.</p>
<p>This method is intended for use with DNA sequences:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;CCCCCGATAGNR&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span>
<span class="go">Seq(&#39;CCCCCGATAGNR&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span><span class="o">.</span><span class="n">reverse_complement</span><span class="p">()</span>
<span class="go">Seq(&#39;YNCTATCGGGGG&#39;)</span>
</pre></div>
</div>
<p>Note in the above example, since R = G or A, its complement
is Y (which denotes C or T).</p>
<p>You can of course used mixed case sequences,</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;CCCCCgatA-G&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span>
<span class="go">Seq(&#39;CCCCCgatA-G&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span><span class="o">.</span><span class="n">reverse_complement</span><span class="p">()</span>
<span class="go">Seq(&#39;C-TatcGGGGG&#39;)</span>
</pre></div>
</div>
<p>As discussed for the complement method, if the sequence
contains neither T nor U, is is assumed to be DNA and
will map any letter A to T.</p>
<p>If you are dealing with RNA you should use the new
reverse_complement_rna method instead</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;CGA&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">reverse_complement</span><span class="p">()</span>  <span class="c1"># defaults to DNA</span>
<span class="go">Seq(&#39;TCG&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;CGA&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">reverse_complement_rna</span><span class="p">()</span>
<span class="go">Seq(&#39;UCG&#39;)</span>
</pre></div>
</div>
<p>If the sequence contains both T and U, an exception is raised:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;CGAUT&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">reverse_complement</span><span class="p">()</span>
<span class="gt">Traceback (most recent call last):</span>
   <span class="c">...</span>
<span class="gr">ValueError</span>: <span class="n">Mixed RNA/DNA found</span>
</pre></div>
</div>
<p>Trying to reverse complement a protein sequence will give
a meaningless sequence:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_protein</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;MAIVMGR&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_protein</span><span class="o">.</span><span class="n">reverse_complement</span><span class="p">()</span>
<span class="go">Seq(&#39;YCKBITK&#39;)</span>
</pre></div>
</div>
<p>Here “M” was interpretted as the IUPAC ambiguity code for
“A” or “C”, with complement “K” for “T” or “G” - and so on.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.Seq.ungap">
<code class="sig-name descname">ungap</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">gap</span><span class="o">=</span><span class="default_value">'-'</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.ungap" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a copy of the sequence without the gap character(s) (OBSOLETE).</p>
<p>The gap character now defaults to the minus sign, and can only
be specified via the method argument. This is no longer possible
via the sequence’s alphabet (as was possible up to Biopython 1.77):</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;-ATA--TGAAAT-TTGAAAA&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span>
<span class="go">Seq(&#39;-ATA--TGAAAT-TTGAAAA&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span><span class="o">.</span><span class="n">ungap</span><span class="p">(</span><span class="s2">&quot;-&quot;</span><span class="p">)</span>
<span class="go">Seq(&#39;ATATGAAATTTGAAAA&#39;)</span>
</pre></div>
</div>
<p>This method is OBSOLETE; please use my_dna.replace(gap, “”) instead.</p>
</dd></dl>

<dl class="py attribute">
<dt id="Bio.Seq.Seq.__abstractmethods__">
<code class="sig-name descname">__abstractmethods__</code><em class="property"> = frozenset({})</em><a class="headerlink" href="#Bio.Seq.Seq.__abstractmethods__" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py class">
<dt id="Bio.Seq.UnknownSeq">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Seq.</code><code class="sig-name descname">UnknownSeq</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">length</span></em>, <em class="sig-param"><span class="n">alphabet</span><span class="o">=</span><span class="default_value">None</span></em>, <em class="sig-param"><span class="n">character</span><span class="o">=</span><span class="default_value">'?'</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Seq.Seq" title="Bio.Seq.Seq"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Seq.Seq</span></code></a></p>
<p>Read-only sequence object of known length but unknown contents (DEPRECATED).</p>
<p>If you have an unknown sequence, you can represent this with a normal
Seq object, for example:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;N&quot;</span><span class="o">*</span><span class="mi">5</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">Seq(&#39;NNNNN&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">my_seq</span><span class="p">)</span>
<span class="go">5</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_seq</span><span class="p">)</span>
<span class="go">NNNNN</span>
</pre></div>
</div>
<p>However, this is rather wasteful of memory (especially for large
sequences), which is where this class is most useful:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">unk_five</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">5</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">unk_five</span>
<span class="go">UnknownSeq(5, character=&#39;?&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">unk_five</span><span class="p">)</span>
<span class="go">5</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">unk_five</span><span class="p">)</span>
<span class="go">?????</span>
</pre></div>
</div>
<p>You can add unknown sequence together. Provided the characters are the
same, you get another memory saving UnknownSeq:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">unk_four</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">unk_four</span>
<span class="go">UnknownSeq(4, character=&#39;?&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">unk_four</span> <span class="o">+</span> <span class="n">unk_five</span>
<span class="go">UnknownSeq(9, character=&#39;?&#39;)</span>
</pre></div>
</div>
<p>If the characters are different, addition gives an ordinary Seq object:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">unk_nnnn</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">unk_nnnn</span>
<span class="go">UnknownSeq(4, character=&#39;N&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">unk_nnnn</span> <span class="o">+</span> <span class="n">unk_four</span>
<span class="go">Seq(&#39;NNNN????&#39;)</span>
</pre></div>
</div>
<p>Combining with a real Seq gives a new Seq object:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">known_seq</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;ACGT&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">unk_four</span> <span class="o">+</span> <span class="n">known_seq</span>
<span class="go">Seq(&#39;????ACGT&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">known_seq</span> <span class="o">+</span> <span class="n">unk_four</span>
<span class="go">Seq(&#39;ACGT????&#39;)</span>
</pre></div>
</div>
<p>Although originally intended for unknown sequences (thus the class name),
this can be used for homopolymer sequences like AAAAAA, and the biological
methods will respect this:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">homopolymer</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;A&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">homopolymer</span><span class="o">.</span><span class="n">complement</span><span class="p">()</span>
<span class="go">UnknownSeq(6, character=&#39;T&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">homopolymer</span><span class="o">.</span><span class="n">complement_rna</span><span class="p">()</span>
<span class="go">UnknownSeq(6, character=&#39;U&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">homopolymer</span><span class="o">.</span><span class="n">translate</span><span class="p">()</span>
<span class="go">UnknownSeq(2, character=&#39;K&#39;)</span>
</pre></div>
</div>
<dl class="py method">
<dt id="Bio.Seq.UnknownSeq.__init__">
<code class="sig-name descname">__init__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">length</span></em>, <em class="sig-param"><span class="n">alphabet</span><span class="o">=</span><span class="default_value">None</span></em>, <em class="sig-param"><span class="n">character</span><span class="o">=</span><span class="default_value">'?'</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.__init__" title="Permalink to this definition">¶</a></dt>
<dd><p>Create a new UnknownSeq object.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>length - Integer, required.</p></li>
<li><p>alphabet - no longer used, must be None.</p></li>
<li><p>character - single letter string, default “?”. Typically “N”
for nucleotides, “X” for proteins, and “?” otherwise.</p></li>
</ul>
</dd>
</dl>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.UnknownSeq.__len__">
<code class="sig-name descname">__len__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.__len__" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the stated length of the unknown sequence.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.UnknownSeq.__bytes__">
<code class="sig-name descname">__bytes__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.__bytes__" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the unknown sequence as full string of the given length.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.UnknownSeq.__str__">
<code class="sig-name descname">__str__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.__str__" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the unknown sequence as full string of the given length.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.UnknownSeq.__repr__">
<code class="sig-name descname">__repr__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.__repr__" title="Permalink to this definition">¶</a></dt>
<dd><p>Return (truncated) representation of the sequence for debugging.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.UnknownSeq.__add__">
<code class="sig-name descname">__add__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">other</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.__add__" title="Permalink to this definition">¶</a></dt>
<dd><p>Add another sequence or string to this sequence.</p>
<p>Adding two UnknownSeq objects returns another UnknownSeq object
provided the character is the same.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">UnknownSeq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s1">&#39;X&#39;</span><span class="p">)</span> <span class="o">+</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">5</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s1">&#39;X&#39;</span><span class="p">)</span>
<span class="go">UnknownSeq(15, character=&#39;X&#39;)</span>
</pre></div>
</div>
<p>If the characters differ, an UnknownSeq object cannot be used, so a
Seq object is returned:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">UnknownSeq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s1">&#39;X&#39;</span><span class="p">)</span> <span class="o">+</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">5</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;x&quot;</span><span class="p">)</span>
<span class="go">Seq(&#39;XXXXXXXXXXxxxxx&#39;)</span>
</pre></div>
</div>
<p>If adding a string to an UnknownSeq, a new Seq is returned:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">UnknownSeq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">5</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s1">&#39;X&#39;</span><span class="p">)</span> <span class="o">+</span> <span class="s2">&quot;LV&quot;</span>
<span class="go">Seq(&#39;XXXXXLV&#39;)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.UnknownSeq.__radd__">
<code class="sig-name descname">__radd__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">other</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.__radd__" title="Permalink to this definition">¶</a></dt>
<dd><p>Add a sequence on the left.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.UnknownSeq.__mul__">
<code class="sig-name descname">__mul__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">other</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.__mul__" title="Permalink to this definition">¶</a></dt>
<dd><p>Multiply UnknownSeq by integer.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">UnknownSeq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span> <span class="o">*</span> <span class="mi">2</span>
<span class="go">UnknownSeq(6, character=&#39;?&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">3</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span> <span class="o">*</span> <span class="mi">2</span>
<span class="go">UnknownSeq(6, character=&#39;N&#39;)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.UnknownSeq.__rmul__">
<code class="sig-name descname">__rmul__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">other</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.__rmul__" title="Permalink to this definition">¶</a></dt>
<dd><p>Multiply integer by UnknownSeq.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">UnknownSeq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="mi">2</span> <span class="o">*</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span>
<span class="go">UnknownSeq(6, character=&#39;?&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="mi">2</span> <span class="o">*</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">3</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span>
<span class="go">UnknownSeq(6, character=&#39;N&#39;)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.UnknownSeq.__imul__">
<code class="sig-name descname">__imul__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">other</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.__imul__" title="Permalink to this definition">¶</a></dt>
<dd><p>Multiply UnknownSeq in-place.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">UnknownSeq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">seq</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">3</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">seq</span> <span class="o">*=</span> <span class="mi">2</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">seq</span>
<span class="go">UnknownSeq(6, character=&#39;N&#39;)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.UnknownSeq.__getitem__">
<code class="sig-name descname">__getitem__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">index</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.__getitem__" title="Permalink to this definition">¶</a></dt>
<dd><p>Get a subsequence from the UnknownSeq object.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">unk</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">unk</span><span class="p">[:])</span>
<span class="go">NNNNNNNN</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">unk</span><span class="p">[</span><span class="mi">5</span><span class="p">:</span><span class="mi">3</span><span class="p">])</span>

<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">unk</span><span class="p">[</span><span class="mi">1</span><span class="p">:</span><span class="o">-</span><span class="mi">1</span><span class="p">])</span>
<span class="go">NNNNNN</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">unk</span><span class="p">[</span><span class="mi">1</span><span class="p">:</span><span class="o">-</span><span class="mi">1</span><span class="p">:</span><span class="mi">2</span><span class="p">])</span>
<span class="go">NNN</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.UnknownSeq.count">
<code class="sig-name descname">count</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">sub</span></em>, <em class="sig-param"><span class="n">start</span><span class="o">=</span><span class="default_value">None</span></em>, <em class="sig-param"><span class="n">end</span><span class="o">=</span><span class="default_value">None</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.count" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a non-overlapping count, like that of a python string.</p>
<p>This behaves like the python string (and Seq object) method of the
same name, which does a non-overlapping count!</p>
<p>For an overlapping search use the newer count_overlap() method.</p>
<p>Returns an integer, the number of occurrences of substring
argument sub in the (sub)sequence given by [start:end].
Optional arguments start and end are interpreted as in slice
notation.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>sub - a string or another Seq object to look for</p></li>
<li><p>start - optional integer, slice start</p></li>
<li><p>end - optional integer, slice end</p></li>
</ul>
</dd>
</dl>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="s2">&quot;NNNN&quot;</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;N&quot;</span><span class="p">)</span>
<span class="go">4</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;NNNN&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;N&quot;</span><span class="p">)</span>
<span class="go">4</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;N&quot;</span><span class="p">)</span>
<span class="go">4</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;A&quot;</span><span class="p">)</span>
<span class="go">0</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;AA&quot;</span><span class="p">)</span>
<span class="go">0</span>
</pre></div>
</div>
<p>HOWEVER, please note because that python strings and Seq objects (and
MutableSeq objects) do a non-overlapping search, this may not give
the answer you expect:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;NN&quot;</span><span class="p">)</span>
<span class="go">2</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;NNN&quot;</span><span class="p">)</span>
<span class="go">1</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.UnknownSeq.count_overlap">
<code class="sig-name descname">count_overlap</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">sub</span></em>, <em class="sig-param"><span class="n">start</span><span class="o">=</span><span class="default_value">None</span></em>, <em class="sig-param"><span class="n">end</span><span class="o">=</span><span class="default_value">None</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.count_overlap" title="Permalink to this definition">¶</a></dt>
<dd><p>Return an overlapping count.</p>
<p>For a non-overlapping search use the count() method.</p>
<p>Returns an integer, the number of occurrences of substring
argument sub in the (sub)sequence given by [start:end].
Optional arguments start and end are interpreted as in slice
notation.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>sub - a string or another Seq object to look for</p></li>
<li><p>start - optional integer, slice start</p></li>
<li><p>end - optional integer, slice end</p></li>
</ul>
</dd>
</dl>
<p>e.g.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">UnknownSeq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;NN&quot;</span><span class="p">)</span>
<span class="go">3</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;NNN&quot;</span><span class="p">)</span>
<span class="go">2</span>
</pre></div>
</div>
<p>Where substrings do not overlap, should behave the same as
the count() method:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;N&quot;</span><span class="p">)</span>
<span class="go">4</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;N&quot;</span><span class="p">)</span> <span class="o">==</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;N&quot;</span><span class="p">)</span>
<span class="go">True</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;A&quot;</span><span class="p">)</span>
<span class="go">0</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;A&quot;</span><span class="p">)</span> <span class="o">==</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;A&quot;</span><span class="p">)</span>
<span class="go">True</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;AA&quot;</span><span class="p">)</span>
<span class="go">0</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;AA&quot;</span><span class="p">)</span> <span class="o">==</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;AA&quot;</span><span class="p">)</span>
<span class="go">True</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.UnknownSeq.complement">
<code class="sig-name descname">complement</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.complement" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the complement assuming it is DNA.</p>
<p>In typical usage this will return the same unknown sequence:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_nuc</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s1">&#39;N&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_nuc</span>
<span class="go">UnknownSeq(8, character=&#39;N&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_nuc</span><span class="p">)</span>
<span class="go">NNNNNNNN</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_nuc</span><span class="o">.</span><span class="n">complement</span><span class="p">()</span>
<span class="go">UnknownSeq(8, character=&#39;N&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_nuc</span><span class="o">.</span><span class="n">complement</span><span class="p">())</span>
<span class="go">NNNNNNNN</span>
</pre></div>
</div>
<p>If your sequence isn’t actually unknown, and has a nucleotide letter
other than N, the appropriate DNA complement base is used:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;A&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">complement</span><span class="p">()</span>
<span class="go">UnknownSeq(8, character=&#39;T&#39;)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.UnknownSeq.complement_rna">
<code class="sig-name descname">complement_rna</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.complement_rna" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the complement assuming it is RNA.</p>
<p>In typical usage this will return the same unknown sequence. If your
sequence isn’t actually unknown, the appropriate RNA complement base
is used:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;A&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">complement_rna</span><span class="p">()</span>
<span class="go">UnknownSeq(8, character=&#39;U&#39;)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.UnknownSeq.reverse_complement">
<code class="sig-name descname">reverse_complement</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.reverse_complement" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the reverse complement assuming it is DNA.</p>
<p>In typical usage this will return the same unknown sequence:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">UnknownSeq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">example</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">example</span><span class="p">)</span>
<span class="go">NNNNNN</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">example</span><span class="o">.</span><span class="n">reverse_complement</span><span class="p">())</span>
<span class="go">NNNNNN</span>
</pre></div>
</div>
<p>If your sequence isn’t actually unknown, the appropriate DNA
complement base is used:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;A&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">reverse_complement</span><span class="p">()</span>
<span class="go">UnknownSeq(8, character=&#39;T&#39;)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.UnknownSeq.reverse_complement_rna">
<code class="sig-name descname">reverse_complement_rna</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.reverse_complement_rna" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the reverse complement assuming it is RNA.</p>
<p>In typical usage this will return the same unknown sequence. If your
sequence isn’t actually unknown, the appropriate RNA complement base
is used:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;A&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">reverse_complement_rna</span><span class="p">()</span>
<span class="go">UnknownSeq(8, character=&#39;U&#39;)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.UnknownSeq.transcribe">
<code class="sig-name descname">transcribe</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.transcribe" title="Permalink to this definition">¶</a></dt>
<dd><p>Return an unknown RNA sequence from an unknown DNA sequence.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span>
<span class="go">UnknownSeq(10, character=&#39;N&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_dna</span><span class="p">)</span>
<span class="go">NNNNNNNNNN</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span> <span class="o">=</span> <span class="n">my_dna</span><span class="o">.</span><span class="n">transcribe</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span>
<span class="go">UnknownSeq(10, character=&#39;N&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_rna</span><span class="p">)</span>
<span class="go">NNNNNNNNNN</span>
</pre></div>
</div>
<p>In typical usage this will return the same unknown sequence. If your
sequence isn’t actually unknown, but a homopolymer of T, the standard
DNA to RNA transcription is done, replacing T with U:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">9</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;t&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">transcribe</span><span class="p">()</span>
<span class="go">UnknownSeq(9, character=&#39;u&#39;)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.UnknownSeq.back_transcribe">
<code class="sig-name descname">back_transcribe</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.back_transcribe" title="Permalink to this definition">¶</a></dt>
<dd><p>Return an unknown DNA sequence from an unknown RNA sequence.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">20</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span>
<span class="go">UnknownSeq(20, character=&#39;N&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_rna</span><span class="p">)</span>
<span class="go">NNNNNNNNNNNNNNNNNNNN</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span> <span class="o">=</span> <span class="n">my_rna</span><span class="o">.</span><span class="n">back_transcribe</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span>
<span class="go">UnknownSeq(20, character=&#39;N&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_dna</span><span class="p">)</span>
<span class="go">NNNNNNNNNNNNNNNNNNNN</span>
</pre></div>
</div>
<p>In typical usage this will return the same unknown sequence. If your
sequence is actually a U homopolymer, the standard RNA to DNA back
translation applies, replacing U with T:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">9</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;U&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">back_transcribe</span><span class="p">()</span>
<span class="go">UnknownSeq(9, character=&#39;T&#39;)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.UnknownSeq.upper">
<code class="sig-name descname">upper</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.upper" title="Permalink to this definition">¶</a></dt>
<dd><p>Return an upper case copy of the sequence.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">UnknownSeq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">20</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;n&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">UnknownSeq(20, character=&#39;n&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_seq</span><span class="p">)</span>
<span class="go">nnnnnnnnnnnnnnnnnnnn</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">upper</span><span class="p">()</span>
<span class="go">UnknownSeq(20, character=&#39;N&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_seq</span><span class="o">.</span><span class="n">upper</span><span class="p">())</span>
<span class="go">NNNNNNNNNNNNNNNNNNNN</span>
</pre></div>
</div>
<p>See also the lower method.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.UnknownSeq.lower">
<code class="sig-name descname">lower</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.lower" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a lower case copy of the sequence.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">UnknownSeq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">20</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;X&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">UnknownSeq(20, character=&#39;X&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_seq</span><span class="p">)</span>
<span class="go">XXXXXXXXXXXXXXXXXXXX</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">lower</span><span class="p">()</span>
<span class="go">UnknownSeq(20, character=&#39;x&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_seq</span><span class="o">.</span><span class="n">lower</span><span class="p">())</span>
<span class="go">xxxxxxxxxxxxxxxxxxxx</span>
</pre></div>
</div>
<p>See also the upper method.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.UnknownSeq.translate">
<code class="sig-name descname">translate</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">table</span><span class="o">=</span><span class="default_value">'Standard'</span></em>, <em class="sig-param"><span class="n">stop_symbol</span><span class="o">=</span><span class="default_value">'*'</span></em>, <em class="sig-param"><span class="n">to_stop</span><span class="o">=</span><span class="default_value">False</span></em>, <em class="sig-param"><span class="n">cds</span><span class="o">=</span><span class="default_value">False</span></em>, <em class="sig-param"><span class="n">gap</span><span class="o">=</span><span class="default_value">'-'</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.translate" title="Permalink to this definition">¶</a></dt>
<dd><p>Translate an unknown nucleotide sequence into an unknown protein.</p>
<p>If your sequence makes sense as codons (e.g. a poly-A tail AAAAAA),
it will be translated accordingly:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">7</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s1">&#39;A&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">translate</span><span class="p">()</span>
<span class="go">UnknownSeq(2, character=&#39;K&#39;)</span>
</pre></div>
</div>
<p>Otherwise, it will be translated as X for unknown amino acid:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">7</span><span class="p">)</span><span class="o">.</span><span class="n">translate</span><span class="p">()</span>
<span class="go">UnknownSeq(2, character=&#39;X&#39;)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.UnknownSeq.ungap">
<code class="sig-name descname">ungap</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">gap</span><span class="o">=</span><span class="default_value">'-'</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.ungap" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a copy of the sequence without the gap character(s).</p>
<p>The gap character now defaults to the minus sign, and can only
be specified via the method argument. This is no longer possible
via the sequence’s alphabet (as was possible up to Biopython 1.77):</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">UnknownSeq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">20</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s1">&#39;N&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span>
<span class="go">UnknownSeq(20, character=&#39;N&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span><span class="o">.</span><span class="n">ungap</span><span class="p">()</span>  <span class="c1"># using default</span>
<span class="go">UnknownSeq(20, character=&#39;N&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span><span class="o">.</span><span class="n">ungap</span><span class="p">(</span><span class="s2">&quot;-&quot;</span><span class="p">)</span>
<span class="go">UnknownSeq(20, character=&#39;N&#39;)</span>
</pre></div>
</div>
<p>If the UnknownSeq is using the gap character, then an empty Seq is
returned:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_gap</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">20</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;-&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_gap</span>
<span class="go">UnknownSeq(20, character=&#39;-&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_gap</span><span class="o">.</span><span class="n">ungap</span><span class="p">()</span>  <span class="c1"># using default</span>
<span class="go">Seq(&#39;&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_gap</span><span class="o">.</span><span class="n">ungap</span><span class="p">(</span><span class="s2">&quot;-&quot;</span><span class="p">)</span>
<span class="go">Seq(&#39;&#39;)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.UnknownSeq.join">
<code class="sig-name descname">join</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">other</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.join" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a merge of the sequences in other, spaced by the sequence from self.</p>
<p>Accepts either a Seq or string (and iterates over the letters), or an
iterable containing Seq or string objects. These arguments will be
concatenated with the calling sequence as the spacer:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">concatenated</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">5</span><span class="p">)</span><span class="o">.</span><span class="n">join</span><span class="p">([</span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;AAA&quot;</span><span class="p">),</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;TTT&quot;</span><span class="p">),</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;PPP&quot;</span><span class="p">)])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">concatenated</span>
<span class="go">Seq(&#39;AAA?????TTT?????PPP&#39;)</span>
</pre></div>
</div>
<p>If all the inputs are also UnknownSeq using the same character, then it
returns a new UnknownSeq:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">5</span><span class="p">)</span><span class="o">.</span><span class="n">join</span><span class="p">([</span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">3</span><span class="p">),</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">3</span><span class="p">),</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">3</span><span class="p">)])</span>
<span class="go">UnknownSeq(19, character=&#39;?&#39;)</span>
</pre></div>
</div>
<p>Examples taking a single sequence and joining the letters:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="s2">&quot;ACGT&quot;</span><span class="p">)</span>
<span class="go">Seq(&#39;A???C???G???T&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">3</span><span class="p">)</span><span class="o">.</span><span class="n">join</span><span class="p">(</span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">))</span>
<span class="go">UnknownSeq(13, character=&#39;?&#39;)</span>
</pre></div>
</div>
<p>Will only return an UnknownSeq object if all of the objects to be joined are
also UnknownSeqs with the same character as the spacer, similar to how the
addition of an UnknownSeq and another UnknownSeq would work.</p>
</dd></dl>

<dl class="py attribute">
<dt id="Bio.Seq.UnknownSeq.__abstractmethods__">
<code class="sig-name descname">__abstractmethods__</code><em class="property"> = frozenset({})</em><a class="headerlink" href="#Bio.Seq.UnknownSeq.__abstractmethods__" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py class">
<dt id="Bio.Seq.MutableSeq">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Seq.</code><code class="sig-name descname">MutableSeq</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">data</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Seq._SeqAbstractBaseClass</span></code></p>
<p>An editable sequence object.</p>
<p>Unlike normal python strings and our basic sequence object (the Seq class)
which are immutable, the MutableSeq lets you edit the sequence in place.
However, this means you cannot use a MutableSeq object as a dictionary key.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">MutableSeq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">MutableSeq</span><span class="p">(</span><span class="s2">&quot;ACTCGTCGTCG&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;ACTCGTCGTCG&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="p">[</span><span class="mi">5</span><span class="p">]</span>
<span class="go">&#39;T&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="p">[</span><span class="mi">5</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;A&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;ACTCGACGTCG&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="p">[</span><span class="mi">5</span><span class="p">]</span>
<span class="go">&#39;A&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="p">[</span><span class="mi">5</span><span class="p">:</span><span class="mi">8</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;NNN&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;ACTCGNNNTCG&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">my_seq</span><span class="p">)</span>
<span class="go">11</span>
</pre></div>
</div>
<p>Note that the MutableSeq object does not support as many string-like
or biological methods as the Seq object.</p>
<dl class="py method">
<dt id="Bio.Seq.MutableSeq.__init__">
<code class="sig-name descname">__init__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">data</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq.__init__" title="Permalink to this definition">¶</a></dt>
<dd><p>Create a MutableSeq object.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.MutableSeq.data">
<em class="property">property </em><code class="sig-name descname">data</code><a class="headerlink" href="#Bio.Seq.MutableSeq.data" title="Permalink to this definition">¶</a></dt>
<dd><p>Get the data.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.MutableSeq.__setitem__">
<code class="sig-name descname">__setitem__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">index</span></em>, <em class="sig-param"><span class="n">value</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq.__setitem__" title="Permalink to this definition">¶</a></dt>
<dd><p>Set a subsequence of single letter via value parameter.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">MutableSeq</span><span class="p">(</span><span class="s1">&#39;ACTCGACGTCG&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span> <span class="o">=</span> <span class="s1">&#39;T&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;TCTCGACGTCG&#39;)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.MutableSeq.__delitem__">
<code class="sig-name descname">__delitem__</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">index</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq.__delitem__" title="Permalink to this definition">¶</a></dt>
<dd><p>Delete a subsequence of single letter.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">MutableSeq</span><span class="p">(</span><span class="s1">&#39;ACTCGACGTCG&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">del</span> <span class="n">my_seq</span><span class="p">[</span><span class="mi">0</span><span class="p">]</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;CTCGACGTCG&#39;)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.MutableSeq.append">
<code class="sig-name descname">append</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">c</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq.append" title="Permalink to this definition">¶</a></dt>
<dd><p>Add a subsequence to the mutable sequence object.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">MutableSeq</span><span class="p">(</span><span class="s1">&#39;ACTCGACGTCG&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s1">&#39;A&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;ACTCGACGTCGA&#39;)</span>
</pre></div>
</div>
<p>No return value.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.MutableSeq.insert">
<code class="sig-name descname">insert</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">i</span></em>, <em class="sig-param"><span class="n">c</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq.insert" title="Permalink to this definition">¶</a></dt>
<dd><p>Add a subsequence to the mutable sequence object at a given index.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">MutableSeq</span><span class="p">(</span><span class="s1">&#39;ACTCGACGTCG&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">insert</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="s1">&#39;A&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;AACTCGACGTCG&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">insert</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span><span class="s1">&#39;G&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;AACTCGACGGTCG&#39;)</span>
</pre></div>
</div>
<p>No return value.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.MutableSeq.pop">
<code class="sig-name descname">pop</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">i</span><span class="o">=</span><span class="default_value">- 1</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq.pop" title="Permalink to this definition">¶</a></dt>
<dd><p>Remove a subsequence of a single letter at given index.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">MutableSeq</span><span class="p">(</span><span class="s1">&#39;ACTCGACGTCG&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">pop</span><span class="p">()</span>
<span class="go">&#39;G&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;ACTCGACGTC&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">pop</span><span class="p">()</span>
<span class="go">&#39;C&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;ACTCGACGT&#39;)</span>
</pre></div>
</div>
<p>Returns the last character of the sequence.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.MutableSeq.remove">
<code class="sig-name descname">remove</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">item</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq.remove" title="Permalink to this definition">¶</a></dt>
<dd><p>Remove a subsequence of a single letter from mutable sequence.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">MutableSeq</span><span class="p">(</span><span class="s1">&#39;ACTCGACGTCG&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">remove</span><span class="p">(</span><span class="s1">&#39;C&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;ATCGACGTCG&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">remove</span><span class="p">(</span><span class="s1">&#39;A&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;TCGACGTCG&#39;)</span>
</pre></div>
</div>
<p>No return value.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.MutableSeq.reverse">
<code class="sig-name descname">reverse</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq.reverse" title="Permalink to this definition">¶</a></dt>
<dd><p>Modify the mutable sequence to reverse itself.</p>
<p>No return value.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.MutableSeq.complement">
<code class="sig-name descname">complement</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq.complement" title="Permalink to this definition">¶</a></dt>
<dd><p>Modify the mutable sequence to take on its complement.</p>
<p>No return value.</p>
<p>If the sequence contains neither T nor U, DNA is assumed
and any A will be mapped to T.</p>
<p>If the sequence contains both T and U, an exception is raised.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.MutableSeq.reverse_complement">
<code class="sig-name descname">reverse_complement</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq.reverse_complement" title="Permalink to this definition">¶</a></dt>
<dd><p>Modify the mutable sequence to take on its reverse complement.</p>
<p>No return value.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.MutableSeq.extend">
<code class="sig-name descname">extend</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em>, <em class="sig-param"><span class="n">other</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq.extend" title="Permalink to this definition">¶</a></dt>
<dd><p>Add a sequence to the original mutable sequence object.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">MutableSeq</span><span class="p">(</span><span class="s1">&#39;ACTCGACGTCG&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">extend</span><span class="p">(</span><span class="s1">&#39;A&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;ACTCGACGTCGA&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">extend</span><span class="p">(</span><span class="s1">&#39;TTT&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;ACTCGACGTCGATTT&#39;)</span>
</pre></div>
</div>
<p>No return value.</p>
</dd></dl>

<dl class="py method">
<dt id="Bio.Seq.MutableSeq.toseq">
<code class="sig-name descname">toseq</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">self</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq.toseq" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the full sequence as a new immutable Seq object.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="kn">import</span> <span class="n">MutableSeq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_mseq</span> <span class="o">=</span> <span class="n">MutableSeq</span><span class="p">(</span><span class="s2">&quot;MKQHKAMIVALIVICITAVVAAL&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_mseq</span>
<span class="go">MutableSeq(&#39;MKQHKAMIVALIVICITAVVAAL&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_mseq</span><span class="o">.</span><span class="n">toseq</span><span class="p">()</span>
<span class="go">Seq(&#39;MKQHKAMIVALIVICITAVVAAL&#39;)</span>
</pre></div>
</div>
</dd></dl>

<dl class="py attribute">
<dt id="Bio.Seq.MutableSeq.__abstractmethods__">
<code class="sig-name descname">__abstractmethods__</code><em class="property"> = frozenset({})</em><a class="headerlink" href="#Bio.Seq.MutableSeq.__abstractmethods__" title="Permalink to this definition">¶</a></dt>
<dd></dd></dl>

</dd></dl>

<dl class="py exception">
<dt id="Bio.Seq.UndefinedSequenceError">
<em class="property">exception </em><code class="sig-prename descclassname">Bio.Seq.</code><code class="sig-name descname">UndefinedSequenceError</code><a class="headerlink" href="#Bio.Seq.UndefinedSequenceError" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">ValueError</span></code></p>
<p>Sequence contents is undefined.</p>
</dd></dl>

<dl class="py function">
<dt id="Bio.Seq.transcribe">
<code class="sig-prename descclassname">Bio.Seq.</code><code class="sig-name descname">transcribe</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">dna</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.transcribe" title="Permalink to this definition">¶</a></dt>
<dd><p>Transcribe a DNA sequence into RNA.</p>
<p>If given a string, returns a new string object.</p>
<p>Given a Seq or MutableSeq, returns a new Seq object.</p>
<p>e.g.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">transcribe</span><span class="p">(</span><span class="s2">&quot;ACTGN&quot;</span><span class="p">)</span>
<span class="go">&#39;ACUGN&#39;</span>
</pre></div>
</div>
</dd></dl>

<dl class="py function">
<dt id="Bio.Seq.back_transcribe">
<code class="sig-prename descclassname">Bio.Seq.</code><code class="sig-name descname">back_transcribe</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">rna</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.back_transcribe" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the RNA sequence back-transcribed into DNA.</p>
<p>If given a string, returns a new string object.</p>
<p>Given a Seq or MutableSeq, returns a new Seq object.</p>
<p>e.g.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">back_transcribe</span><span class="p">(</span><span class="s2">&quot;ACUGN&quot;</span><span class="p">)</span>
<span class="go">&#39;ACTGN&#39;</span>
</pre></div>
</div>
</dd></dl>

<dl class="py function">
<dt id="Bio.Seq.translate">
<code class="sig-prename descclassname">Bio.Seq.</code><code class="sig-name descname">translate</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">sequence</span></em>, <em class="sig-param"><span class="n">table</span><span class="o">=</span><span class="default_value">'Standard'</span></em>, <em class="sig-param"><span class="n">stop_symbol</span><span class="o">=</span><span class="default_value">'*'</span></em>, <em class="sig-param"><span class="n">to_stop</span><span class="o">=</span><span class="default_value">False</span></em>, <em class="sig-param"><span class="n">cds</span><span class="o">=</span><span class="default_value">False</span></em>, <em class="sig-param"><span class="n">gap</span><span class="o">=</span><span class="default_value">None</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.translate" title="Permalink to this definition">¶</a></dt>
<dd><p>Translate a nucleotide sequence into amino acids.</p>
<p>If given a string, returns a new string object. Given a Seq or
MutableSeq, returns a Seq object.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>table - Which codon table to use?  This can be either a name
(string), an NCBI identifier (integer), or a CodonTable object
(useful for non-standard genetic codes).  Defaults to the “Standard”
table.</p></li>
<li><p>stop_symbol - Single character string, what to use for any
terminators, defaults to the asterisk, “*”.</p></li>
<li><p>to_stop - Boolean, defaults to False meaning do a full
translation continuing on past any stop codons
(translated as the specified stop_symbol).  If
True, translation is terminated at the first in
frame stop codon (and the stop_symbol is not
appended to the returned protein sequence).</p></li>
<li><p>cds - Boolean, indicates this is a complete CDS.  If True, this
checks the sequence starts with a valid alternative start
codon (which will be translated as methionine, M), that the
sequence length is a multiple of three, and that there is a
single in frame stop codon at the end (this will be excluded
from the protein sequence, regardless of the to_stop option).
If these tests fail, an exception is raised.</p></li>
<li><p>gap - Single character string to denote symbol used for gaps.
Defaults to None.</p></li>
</ul>
</dd>
</dl>
<p>A simple string example using the default (standard) genetic code:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">coding_dna</span> <span class="o">=</span> <span class="s2">&quot;GTGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">translate</span><span class="p">(</span><span class="n">coding_dna</span><span class="p">)</span>
<span class="go">&#39;VAIVMGR*KGAR*&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">translate</span><span class="p">(</span><span class="n">coding_dna</span><span class="p">,</span> <span class="n">stop_symbol</span><span class="o">=</span><span class="s2">&quot;@&quot;</span><span class="p">)</span>
<span class="go">&#39;VAIVMGR@KGAR@&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">translate</span><span class="p">(</span><span class="n">coding_dna</span><span class="p">,</span> <span class="n">to_stop</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="go">&#39;VAIVMGR&#39;</span>
</pre></div>
</div>
<p>Now using NCBI table 2, where TGA is not a stop codon:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">translate</span><span class="p">(</span><span class="n">coding_dna</span><span class="p">,</span> <span class="n">table</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>
<span class="go">&#39;VAIVMGRWKGAR*&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">translate</span><span class="p">(</span><span class="n">coding_dna</span><span class="p">,</span> <span class="n">table</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">to_stop</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="go">&#39;VAIVMGRWKGAR&#39;</span>
</pre></div>
</div>
<p>In fact this example uses an alternative start codon valid under NCBI
table 2, GTG, which means this example is a complete valid CDS which
when translated should really start with methionine (not valine):</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">translate</span><span class="p">(</span><span class="n">coding_dna</span><span class="p">,</span> <span class="n">table</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">cds</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="go">&#39;MAIVMGRWKGAR&#39;</span>
</pre></div>
</div>
<p>Note that if the sequence has no in-frame stop codon, then the to_stop
argument has no effect:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">coding_dna2</span> <span class="o">=</span> <span class="s2">&quot;GTGGCCATTGTAATGGGCCGC&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">translate</span><span class="p">(</span><span class="n">coding_dna2</span><span class="p">)</span>
<span class="go">&#39;VAIVMGR&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">translate</span><span class="p">(</span><span class="n">coding_dna2</span><span class="p">,</span> <span class="n">to_stop</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="go">&#39;VAIVMGR&#39;</span>
</pre></div>
</div>
<p>NOTE - Ambiguous codons like “TAN” or “NNN” could be an amino acid
or a stop codon.  These are translated as “X”.  Any invalid codon
(e.g. “TA?” or “T-A”) will throw a TranslationError.</p>
<p>It will however translate either DNA or RNA.</p>
<p>NOTE - Since version 1.71 Biopython contains codon tables with ‘ambiguous
stop codons’. These are stop codons with unambiguous sequence but which
have a context dependent coding as STOP or as amino acid. With these tables
‘to_stop’ must be False (otherwise a ValueError is raised). The dual
coding codons will always be translated as amino acid, except for
‘cds=True’, where the last codon will be translated as STOP.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">coding_dna3</span> <span class="o">=</span> <span class="s2">&quot;ATGGCACGGAAGTGA&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">translate</span><span class="p">(</span><span class="n">coding_dna3</span><span class="p">)</span>
<span class="go">&#39;MARK*&#39;</span>
</pre></div>
</div>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">translate</span><span class="p">(</span><span class="n">coding_dna3</span><span class="p">,</span> <span class="n">table</span><span class="o">=</span><span class="mi">27</span><span class="p">)</span>  <span class="c1"># Table 27: TGA -&gt; STOP or W</span>
<span class="go">&#39;MARKW&#39;</span>
</pre></div>
</div>
<p>It will however raise a BiopythonWarning (not shown).</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">translate</span><span class="p">(</span><span class="n">coding_dna3</span><span class="p">,</span> <span class="n">table</span><span class="o">=</span><span class="mi">27</span><span class="p">,</span> <span class="n">cds</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="go">&#39;MARK&#39;</span>
</pre></div>
</div>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">translate</span><span class="p">(</span><span class="n">coding_dna3</span><span class="p">,</span> <span class="n">table</span><span class="o">=</span><span class="mi">27</span><span class="p">,</span> <span class="n">to_stop</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="gt">Traceback (most recent call last):</span>
   <span class="c">...</span>
<span class="gr">ValueError</span>: <span class="n">You cannot use &#39;to_stop=True&#39; with this table ...</span>
</pre></div>
</div>
</dd></dl>

<dl class="py function">
<dt id="Bio.Seq.reverse_complement">
<code class="sig-prename descclassname">Bio.Seq.</code><code class="sig-name descname">reverse_complement</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">sequence</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.reverse_complement" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the reverse complement sequence of a nucleotide string.</p>
<p>If given a string, returns a new string object.
Given a Seq or a MutableSeq, returns a new Seq object.</p>
<p>Supports unambiguous and ambiguous nucleotide sequences.</p>
<p>e.g.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">reverse_complement</span><span class="p">(</span><span class="s2">&quot;ACTG-NH&quot;</span><span class="p">)</span>
<span class="go">&#39;DN-CAGT&#39;</span>
</pre></div>
</div>
<p>If neither T nor U is present, DNA is assumed and A is mapped to T:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">reverse_complement</span><span class="p">(</span><span class="s2">&quot;A&quot;</span><span class="p">)</span>
<span class="go">&#39;T&#39;</span>
</pre></div>
</div>
</dd></dl>

<dl class="py function">
<dt id="Bio.Seq.complement">
<code class="sig-prename descclassname">Bio.Seq.</code><code class="sig-name descname">complement</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">sequence</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.complement" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the complement sequence of a DNA string.</p>
<p>If given a string, returns a new string object.</p>
<p>Given a Seq or a MutableSeq, returns a new Seq object.</p>
<p>Supports unambiguous and ambiguous nucleotide sequences.</p>
<p>e.g.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">complement</span><span class="p">(</span><span class="s2">&quot;ACTG-NH&quot;</span><span class="p">)</span>
<span class="go">&#39;TGAC-ND&#39;</span>
</pre></div>
</div>
<p>If neither T nor U is present, DNA is assumed and A is mapped to T:</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">complement</span><span class="p">(</span><span class="s2">&quot;A&quot;</span><span class="p">)</span>
<span class="go">&#39;T&#39;</span>
</pre></div>
</div>
<p>However, this may not be supported in future. Please use the
complement_rna function if you have RNA.</p>
</dd></dl>

<dl class="py function">
<dt id="Bio.Seq.complement_rna">
<code class="sig-prename descclassname">Bio.Seq.</code><code class="sig-name descname">complement_rna</code><span class="sig-paren">(</span><em class="sig-param"><span class="n">sequence</span></em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.complement_rna" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the complement sequence of an RNA string.</p>
<div class="doctest highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">complement</span><span class="p">(</span><span class="s2">&quot;ACG&quot;</span><span class="p">)</span>  <span class="c1"># assumed DNA</span>
<span class="go">&#39;TGC&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">complement_rna</span><span class="p">(</span><span class="s2">&quot;ACG&quot;</span><span class="p">)</span>
<span class="go">&#39;UGC&#39;</span>
</pre></div>
</div>
<p>Any T in the sequence is treated as a U.</p>
</dd></dl>

</div>


           </div>
           
          </div>
          <footer>
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
        <a href="Bio.SeqFeature.html" class="btn btn-neutral float-right" title="Bio.SeqFeature module" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right" aria-hidden="true"></span></a>
        <a href="Bio.NaiveBayes.html" class="btn btn-neutral float-left" title="Bio.NaiveBayes module" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left" aria-hidden="true"></span> Previous</a>
    </div>

  <hr/>

  <div role="contentinfo">
    <p>
        &#169; Copyright 1999-2020, The Biopython Contributors.

    </p>
  </div>
    
    
    
    Built with <a href="https://www.sphinx-doc.org/">Sphinx</a> using a
    
    <a href="https://github.com/readthedocs/sphinx_rtd_theme">theme</a>
    
    provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>
        </div>
      </div>

    </section>

  </div>
  


  <div class="rst-versions" data-toggle="rst-versions" role="note" aria-label="versions">
    <span class="rst-current-version" data-toggle="rst-current-version">
      Biopython v: 1.79.dev0
      <span class="fa fa-caret-down"></span>
    </span>
    <div class="rst-other-versions">
      <dl>
        <dt>Versions</dt>
        
          <dd><a href="../../1.77/api/">Previous</a></dd>
        
          <dd><a href="../../latest/api/">Latest</a></dd>
        
          <dd><a href="../../dev/api/">Develop</a></dd>
        
      </dl>
      <dl>
        <dt>Biopython Project</dt>
          <dd>
            <a href="https://biopython.org" class="icon icon-home">  Homepage</a>
          </dd>
          <dd>
            <a href="https://github.com/biopython/biopython" class="fa fa-github">  On GitHub</a>
          </dd>
      </dl>
    </div>
  </div>


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>