

<!DOCTYPE html>
<!--[if IE 8]><html class="no-js lt-ie9" lang="en" > <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js" lang="en" > <!--<![endif]-->
<head>
  <meta charset="utf-8">
  
  <meta name="viewport" content="width=device-width, initial-scale=1.0">
  
  <title>Bio.Seq module &mdash; Biopython 1.76.dev0 documentation</title>
  

  
  
  
  

  
  <script type="text/javascript" src="_static/js/modernizr.min.js"></script>
  
    
      <script type="text/javascript" id="documentation_options" data-url_root="./" src="_static/documentation_options.js"></script>
        <script type="text/javascript" src="_static/jquery.js"></script>
        <script type="text/javascript" src="_static/underscore.js"></script>
        <script type="text/javascript" src="_static/doctools.js"></script>
        <script type="text/javascript" src="_static/language_data.js"></script>
    
    <script type="text/javascript" src="_static/js/theme.js"></script>

    

  
  <link rel="stylesheet" href="_static/css/theme.css" type="text/css" />
  <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="index" title="Index" href="genindex.html" />
    <link rel="search" title="Search" href="search.html" />
    <link rel="next" title="Bio.SeqFeature module" href="Bio.SeqFeature.html" />
    <link rel="prev" title="Bio.NaiveBayes module" href="Bio.NaiveBayes.html" /> 
</head>

<body class="wy-body-for-nav">

   
  <div class="wy-grid-for-nav">
    
    <nav data-toggle="wy-nav-shift" class="wy-nav-side">
      <div class="wy-side-scroll">
        <div class="wy-side-nav-search"  style="background: #10100F" >
          

          
            <a href="index.html">
          

          
            
            <img src="_static/biopython_logo.svg" class="logo" alt="Logo"/>
          
          </a>

          
            
            
              <div class="version">
                1.76.dev0
              </div>
            
          

          
<div role="search">
  <form id="rtd-search-form" class="wy-form" action="search.html" method="get">
    <input type="text" name="q" placeholder="Search docs" />
    <input type="hidden" name="check_keywords" value="yes" />
    <input type="hidden" name="area" value="default" />
  </form>
</div>

          
        </div>

        <div class="wy-menu wy-menu-vertical" data-spy="affix" role="navigation" aria-label="main navigation">
          
            
            
              
            
            
              <p class="caption"><span class="caption-text">API Contents:</span></p>
<ul class="current">
<li class="toctree-l1 current"><a class="reference internal" href="Bio.html">Bio package</a><ul class="current">
<li class="toctree-l2"><a class="reference internal" href="Bio.html#subpackages">Subpackages</a></li>
<li class="toctree-l2 current"><a class="reference internal" href="Bio.html#submodules">Submodules</a><ul class="current">
<li class="toctree-l3"><a class="reference internal" href="Bio.File.html">Bio.File module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.Index.html">Bio.Index module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.LogisticRegression.html">Bio.LogisticRegression module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.MarkovModel.html">Bio.MarkovModel module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.MaxEntropy.html">Bio.MaxEntropy module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.NaiveBayes.html">Bio.NaiveBayes module</a></li>
<li class="toctree-l3 current"><a class="current reference internal" href="#">Bio.Seq module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SeqFeature.html">Bio.SeqFeature module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.SeqRecord.html">Bio.SeqRecord module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.bgzf.html">Bio.bgzf module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.kNN.html">Bio.kNN module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.pairwise2.html">Bio.pairwise2 module</a></li>
<li class="toctree-l3"><a class="reference internal" href="Bio.triefind.html">Bio.triefind module</a></li>
</ul>
</li>
<li class="toctree-l2"><a class="reference internal" href="Bio.html#module-Bio">Module contents</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="BioSQL.html">BioSQL package</a></li>
</ul>

            
          
        </div>
      </div>
    </nav>

    <section data-toggle="wy-nav-shift" class="wy-nav-content-wrap">

      
      <nav class="wy-nav-top" aria-label="top navigation">
        
          <i data-toggle="wy-nav-top" class="fa fa-bars"></i>
          <a href="index.html">Biopython</a>
        
      </nav>


      <div class="wy-nav-content">
        
        <div class="rst-content">
        
          

















<div role="navigation" aria-label="breadcrumbs navigation">

  <ul class="wy-breadcrumbs">
    
      <li><a href="index.html">Docs</a> &raquo;</li>
        
          <li><a href="Bio.html">Bio package</a> &raquo;</li>
        
      <li>Bio.Seq module</li>
    
    
      <li class="wy-breadcrumbs-aside">
        
            
            
              <!-- User defined GitHub URL -->
              <a href="https://github.com/biopython/biopython/blob/master/Bio/Seq.py" class="fa fa-github"> Edit on GitHub</a>
            
          
        
      </li>
    
  </ul>

  
  <div class="rst-breadcrumbs-buttons" role="navigation" aria-label="breadcrumb navigation">
      
        <a href="Bio.SeqFeature.html" class="btn btn-neutral float-right" title="Bio.SeqFeature module" accesskey="n">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="Bio.NaiveBayes.html" class="btn btn-neutral float-left" title="Bio.NaiveBayes module" accesskey="p"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
  </div>
  
  <hr/>
</div>
          <div role="main" class="document" itemscope="itemscope" itemtype="http://schema.org/Article">
           <div itemprop="articleBody">
            
  <div class="section" id="module-Bio.Seq">
<span id="bio-seq-module"></span><h1>Bio.Seq module<a class="headerlink" href="#module-Bio.Seq" title="Permalink to this headline">¶</a></h1>
<p>Provide objects to represent biological sequences with alphabets.</p>
<dl class="simple">
<dt>See also the <a class="reference external" href="http://biopython.org/wiki/Seq">Seq</a> wiki and the chapter in our tutorial:</dt><dd><ul class="simple">
<li><p><a class="reference external" href="http://biopython.org/DIST/docs/tutorial/Tutorial.html">HTML Tutorial</a></p></li>
<li><p><a class="reference external" href="http://biopython.org/DIST/docs/tutorial/Tutorial.pdf">PDF Tutorial</a></p></li>
</ul>
</dd>
</dl>
<dl class="class">
<dt id="Bio.Seq.MutableSeq">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Seq.</code><code class="sig-name descname">MutableSeq</code><span class="sig-paren">(</span><em class="sig-param">data</em>, <em class="sig-param">alphabet=Alphabet()</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>An editable sequence object (with an alphabet).</p>
<p>Unlike normal python strings and our basic sequence object (the Seq class)
which are immutable, the MutableSeq lets you edit the sequence in place.
However, this means you cannot use a MutableSeq object as a dictionary key.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">MutableSeq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="k">import</span> <span class="n">generic_dna</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">MutableSeq</span><span class="p">(</span><span class="s2">&quot;ACTCGTCGTCG&quot;</span><span class="p">,</span> <span class="n">generic_dna</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;ACTCGTCGTCG&#39;, DNAAlphabet())</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="p">[</span><span class="mi">5</span><span class="p">]</span>
<span class="go">&#39;T&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="p">[</span><span class="mi">5</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;A&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;ACTCGACGTCG&#39;, DNAAlphabet())</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="p">[</span><span class="mi">5</span><span class="p">]</span>
<span class="go">&#39;A&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="p">[</span><span class="mi">5</span><span class="p">:</span><span class="mi">8</span><span class="p">]</span> <span class="o">=</span> <span class="s2">&quot;NNN&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;ACTCGNNNTCG&#39;, DNAAlphabet())</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">my_seq</span><span class="p">)</span>
<span class="go">11</span>
</pre></div>
</div>
<p>Note that the MutableSeq object does not support as many string-like
or biological methods as the Seq object.</p>
<dl class="method">
<dt id="Bio.Seq.MutableSeq.append">
<code class="sig-name descname">append</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">c</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq.append" title="Permalink to this definition">¶</a></dt>
<dd><p>Add a subsequence to the mutable sequence object.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">MutableSeq</span><span class="p">(</span><span class="s1">&#39;ACTCGACGTCG&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">append</span><span class="p">(</span><span class="s1">&#39;A&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;ACTCGACGTCGA&#39;)</span>
</pre></div>
</div>
<p>No return value.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.MutableSeq.complement">
<code class="sig-name descname">complement</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq.complement" title="Permalink to this definition">¶</a></dt>
<dd><p>Modify the mutable sequence to take on its complement.</p>
<p>Trying to complement a protein sequence raises an exception.</p>
<p>No return value.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.MutableSeq.count">
<code class="sig-name descname">count</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">sub</em>, <em class="sig-param">start=0</em>, <em class="sig-param">end=9223372036854775807</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq.count" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a non-overlapping count, like that of a python string.</p>
<p>This behaves like the python string method of the same name,
which does a non-overlapping count!</p>
<p>For an overlapping search use the newer count_overlap() method.</p>
<p>Returns an integer, the number of occurrences of substring
argument sub in the (sub)sequence given by [start:end].
Optional arguments start and end are interpreted as in slice
notation.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>sub - a string or another Seq object to look for</p></li>
<li><p>start - optional integer, slice start</p></li>
<li><p>end - optional integer, slice end</p></li>
</ul>
</dd>
</dl>
<p>e.g.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">MutableSeq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_mseq</span> <span class="o">=</span> <span class="n">MutableSeq</span><span class="p">(</span><span class="s2">&quot;AAAATGA&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_mseq</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;A&quot;</span><span class="p">))</span>
<span class="go">5</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_mseq</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;ATG&quot;</span><span class="p">))</span>
<span class="go">1</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_mseq</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;AT&quot;</span><span class="p">)))</span>
<span class="go">1</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_mseq</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;AT&quot;</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="o">-</span><span class="mi">1</span><span class="p">))</span>
<span class="go">1</span>
</pre></div>
</div>
<p>HOWEVER, please note because that python strings, Seq objects and
MutableSeq objects do a non-overlapping search, this may not give
the answer you expect:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="s2">&quot;AAAA&quot;</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;AA&quot;</span><span class="p">)</span>
<span class="go">2</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">MutableSeq</span><span class="p">(</span><span class="s2">&quot;AAAA&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;AA&quot;</span><span class="p">))</span>
<span class="go">2</span>
</pre></div>
</div>
<p>An overlapping search would give the answer as three!</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.MutableSeq.count_overlap">
<code class="sig-name descname">count_overlap</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">sub</em>, <em class="sig-param">start=0</em>, <em class="sig-param">end=9223372036854775807</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq.count_overlap" title="Permalink to this definition">¶</a></dt>
<dd><p>Return an overlapping count.</p>
<p>For a non-overlapping search use the count() method.</p>
<p>Returns an integer, the number of occurrences of substring
argument sub in the (sub)sequence given by [start:end].
Optional arguments start and end are interpreted as in slice
notation.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>sub - a string or another Seq object to look for</p></li>
<li><p>start - optional integer, slice start</p></li>
<li><p>end - optional integer, slice end</p></li>
</ul>
</dd>
</dl>
<p>e.g.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">MutableSeq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">MutableSeq</span><span class="p">(</span><span class="s2">&quot;AAAA&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;AA&quot;</span><span class="p">))</span>
<span class="go">3</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">MutableSeq</span><span class="p">(</span><span class="s2">&quot;ATATATATA&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;ATA&quot;</span><span class="p">))</span>
<span class="go">4</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">MutableSeq</span><span class="p">(</span><span class="s2">&quot;ATATATATA&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;ATA&quot;</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="o">-</span><span class="mi">1</span><span class="p">))</span>
<span class="go">1</span>
</pre></div>
</div>
<p>Where substrings do not overlap, should behave the same as
the count() method:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">MutableSeq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_mseq</span> <span class="o">=</span> <span class="n">MutableSeq</span><span class="p">(</span><span class="s2">&quot;AAAATGA&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_mseq</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;A&quot;</span><span class="p">))</span>
<span class="go">5</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_mseq</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;A&quot;</span><span class="p">)</span> <span class="o">==</span> <span class="n">my_mseq</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;A&quot;</span><span class="p">)</span>
<span class="go">True</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_mseq</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;ATG&quot;</span><span class="p">))</span>
<span class="go">1</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_mseq</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;ATG&quot;</span><span class="p">)</span> <span class="o">==</span> <span class="n">my_mseq</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;ATG&quot;</span><span class="p">)</span>
<span class="go">True</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_mseq</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;AT&quot;</span><span class="p">)))</span>
<span class="go">1</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_mseq</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;AT&quot;</span><span class="p">))</span> <span class="o">==</span> <span class="n">my_mseq</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;AT&quot;</span><span class="p">))</span>
<span class="go">True</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_mseq</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;AT&quot;</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="o">-</span><span class="mi">1</span><span class="p">))</span>
<span class="go">1</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_mseq</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;AT&quot;</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="o">-</span><span class="mi">1</span><span class="p">)</span> <span class="o">==</span> <span class="n">my_mseq</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;AT&quot;</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="o">-</span><span class="mi">1</span><span class="p">)</span>
<span class="go">True</span>
</pre></div>
</div>
<p>HOWEVER, do not use this method for such cases because the
count() method is much for efficient.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.MutableSeq.extend">
<code class="sig-name descname">extend</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">other</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq.extend" title="Permalink to this definition">¶</a></dt>
<dd><p>Add a sequence to the original mutable sequence object.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">MutableSeq</span><span class="p">(</span><span class="s1">&#39;ACTCGACGTCG&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">extend</span><span class="p">(</span><span class="s1">&#39;A&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;ACTCGACGTCGA&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">extend</span><span class="p">(</span><span class="s1">&#39;TTT&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;ACTCGACGTCGATTT&#39;)</span>
</pre></div>
</div>
<p>No return value.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.MutableSeq.index">
<code class="sig-name descname">index</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">item</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq.index" title="Permalink to this definition">¶</a></dt>
<dd><p>Return first occurrence position of a single entry (i.e. letter).</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">MutableSeq</span><span class="p">(</span><span class="s2">&quot;ACTCGACGTCG&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="s2">&quot;A&quot;</span><span class="p">)</span>
<span class="go">0</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="s2">&quot;T&quot;</span><span class="p">)</span>
<span class="go">2</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;T&quot;</span><span class="p">))</span>
<span class="go">2</span>
</pre></div>
</div>
<p>Note unlike a Biopython Seq object, or Python string, multi-letter
subsequences are not supported.  Instead this acts like an array or
a list of the entries. There is therefore no <code class="docutils literal notranslate"><span class="pre">.rindex()</span></code> method.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.MutableSeq.insert">
<code class="sig-name descname">insert</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">i</em>, <em class="sig-param">c</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq.insert" title="Permalink to this definition">¶</a></dt>
<dd><p>Add a subsequence to the mutable sequence object at a given index.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">MutableSeq</span><span class="p">(</span><span class="s1">&#39;ACTCGACGTCG&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">insert</span><span class="p">(</span><span class="mi">0</span><span class="p">,</span><span class="s1">&#39;A&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;AACTCGACGTCG&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">insert</span><span class="p">(</span><span class="mi">8</span><span class="p">,</span><span class="s1">&#39;G&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;AACTCGACGGTCG&#39;)</span>
</pre></div>
</div>
<p>No return value.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.MutableSeq.join">
<code class="sig-name descname">join</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">other</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq.join" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a merge of the sequences in other, spaced by the sequence from self.</p>
<p>Accepts all Seq objects and Strings as objects to be concatenated with the spacer</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">concatenated</span> <span class="o">=</span> <span class="n">MutableSeq</span><span class="p">(</span><span class="s1">&#39;NNNNN&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">join</span><span class="p">([</span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;AAA&quot;</span><span class="p">),</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;TTT&quot;</span><span class="p">),</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;PPP&quot;</span><span class="p">)])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">concatenated</span>
<span class="go">Seq(&#39;AAANNNNNTTTNNNNNPPP&#39;)</span>
</pre></div>
</div>
<p>Throws error if other is not an iterable and if objects inside of the iterable
are not Seq or String objects</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.MutableSeq.pop">
<code class="sig-name descname">pop</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">i=-1</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq.pop" title="Permalink to this definition">¶</a></dt>
<dd><p>Remove a subsequence of a single letter at given index.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">MutableSeq</span><span class="p">(</span><span class="s1">&#39;ACTCGACGTCG&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">pop</span><span class="p">()</span>
<span class="go">&#39;G&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;ACTCGACGTC&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">pop</span><span class="p">()</span>
<span class="go">&#39;C&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;ACTCGACGT&#39;)</span>
</pre></div>
</div>
<p>Returns the last character of the sequence.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.MutableSeq.remove">
<code class="sig-name descname">remove</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">item</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq.remove" title="Permalink to this definition">¶</a></dt>
<dd><p>Remove a subsequence of a single letter from mutable sequence.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">MutableSeq</span><span class="p">(</span><span class="s1">&#39;ACTCGACGTCG&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">remove</span><span class="p">(</span><span class="s1">&#39;C&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;ATCGACGTCG&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">remove</span><span class="p">(</span><span class="s1">&#39;A&#39;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">MutableSeq(&#39;TCGACGTCG&#39;)</span>
</pre></div>
</div>
<p>No return value.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.MutableSeq.reverse">
<code class="sig-name descname">reverse</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq.reverse" title="Permalink to this definition">¶</a></dt>
<dd><p>Modify the mutable sequence to reverse itself.</p>
<p>No return value.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.MutableSeq.reverse_complement">
<code class="sig-name descname">reverse_complement</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq.reverse_complement" title="Permalink to this definition">¶</a></dt>
<dd><p>Modify the mutable sequence to take on its reverse complement.</p>
<p>Trying to reverse complement a protein sequence raises an exception.</p>
<p>No return value.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.MutableSeq.toseq">
<code class="sig-name descname">toseq</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.MutableSeq.toseq" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the full sequence as a new immutable Seq object.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="k">import</span> <span class="n">IUPAC</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_mseq</span> <span class="o">=</span> <span class="n">MutableSeq</span><span class="p">(</span><span class="s2">&quot;MKQHKAMIVALIVICITAVVAAL&quot;</span><span class="p">,</span>
<span class="gp">... </span>                     <span class="n">IUPAC</span><span class="o">.</span><span class="n">protein</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_mseq</span>
<span class="go">MutableSeq(&#39;MKQHKAMIVALIVICITAVVAAL&#39;, IUPACProtein())</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_mseq</span><span class="o">.</span><span class="n">toseq</span><span class="p">()</span>
<span class="go">Seq(&#39;MKQHKAMIVALIVICITAVVAAL&#39;, IUPACProtein())</span>
</pre></div>
</div>
<p>Note that the alphabet is preserved.</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="Bio.Seq.Seq">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Seq.</code><code class="sig-name descname">Seq</code><span class="sig-paren">(</span><em class="sig-param">data</em>, <em class="sig-param">alphabet=Alphabet()</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <code class="xref py py-class docutils literal notranslate"><span class="pre">object</span></code></p>
<p>Read-only sequence object (essentially a string with an alphabet).</p>
<p>Like normal python strings, our basic sequence object is immutable.
This prevents you from doing my_seq[5] = “A” for example, but does allow
Seq objects to be used as dictionary keys.</p>
<p>The Seq object provides a number of string like methods (such as count,
find, split and strip), which are alphabet aware where appropriate.</p>
<p>In addition to the string like sequence, the Seq object has an alphabet
property. This is an instance of an Alphabet class from Bio.Alphabet,
for example generic DNA, or IUPAC DNA. This describes the type of molecule
(e.g. RNA, DNA, protein) and may also indicate the expected symbols
(letters).</p>
<p>The Seq object also provides some biological methods, such as complement,
reverse_complement, transcribe, back_transcribe and translate (which are
not applicable to sequences with a protein alphabet).</p>
<dl class="method">
<dt id="Bio.Seq.Seq.back_transcribe">
<code class="sig-name descname">back_transcribe</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.back_transcribe" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the DNA sequence from an RNA sequence by creating a new Seq object.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="k">import</span> <span class="n">IUPAC</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">messenger_rna</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;AUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAG&quot;</span><span class="p">,</span>
<span class="gp">... </span>                    <span class="n">IUPAC</span><span class="o">.</span><span class="n">unambiguous_rna</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">messenger_rna</span>
<span class="go">Seq(&#39;AUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAG&#39;, IUPACUnambiguousRNA())</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">messenger_rna</span><span class="o">.</span><span class="n">back_transcribe</span><span class="p">()</span>
<span class="go">Seq(&#39;ATGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG&#39;, IUPACUnambiguousDNA())</span>
</pre></div>
</div>
<p>Trying to back-transcribe a protein or DNA sequence raises an
exception:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_protein</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;MAIVMGR&quot;</span><span class="p">,</span> <span class="n">IUPAC</span><span class="o">.</span><span class="n">protein</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_protein</span><span class="o">.</span><span class="n">back_transcribe</span><span class="p">()</span>
<span class="gt">Traceback (most recent call last):</span>
   <span class="c">...</span>
<span class="gr">ValueError</span>: <span class="n">Proteins cannot be back transcribed!</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.Seq.complement">
<code class="sig-name descname">complement</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.complement" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the complement sequence by creating a new Seq object.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="k">import</span> <span class="n">IUPAC</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;CCCCCGATAG&quot;</span><span class="p">,</span> <span class="n">IUPAC</span><span class="o">.</span><span class="n">unambiguous_dna</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span>
<span class="go">Seq(&#39;CCCCCGATAG&#39;, IUPACUnambiguousDNA())</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span><span class="o">.</span><span class="n">complement</span><span class="p">()</span>
<span class="go">Seq(&#39;GGGGGCTATC&#39;, IUPACUnambiguousDNA())</span>
</pre></div>
</div>
<p>You can of course used mixed case sequences,</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="k">import</span> <span class="n">generic_dna</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;CCCCCgatA-GD&quot;</span><span class="p">,</span> <span class="n">generic_dna</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span>
<span class="go">Seq(&#39;CCCCCgatA-GD&#39;, DNAAlphabet())</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span><span class="o">.</span><span class="n">complement</span><span class="p">()</span>
<span class="go">Seq(&#39;GGGGGctaT-CH&#39;, DNAAlphabet())</span>
</pre></div>
</div>
<p>Note in the above example, ambiguous character D denotes
G, A or T so its complement is H (for C, T or A).</p>
<p>Trying to complement a protein sequence raises an exception.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_protein</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;MAIVMGR&quot;</span><span class="p">,</span> <span class="n">IUPAC</span><span class="o">.</span><span class="n">protein</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_protein</span><span class="o">.</span><span class="n">complement</span><span class="p">()</span>
<span class="gt">Traceback (most recent call last):</span>
   <span class="c">...</span>
<span class="gr">ValueError</span>: <span class="n">Proteins do not have complements!</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.Seq.count">
<code class="sig-name descname">count</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">sub</em>, <em class="sig-param">start=0</em>, <em class="sig-param">end=9223372036854775807</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.count" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a non-overlapping count, like that of a python string.</p>
<p>This behaves like the python string method of the same name,
which does a non-overlapping count!</p>
<p>For an overlapping search use the newer count_overlap() method.</p>
<p>Returns an integer, the number of occurrences of substring
argument sub in the (sub)sequence given by [start:end].
Optional arguments start and end are interpreted as in slice
notation.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>sub - a string or another Seq object to look for</p></li>
<li><p>start - optional integer, slice start</p></li>
<li><p>end - optional integer, slice end</p></li>
</ul>
</dd>
</dl>
<p>e.g.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;AAAATGA&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_seq</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;A&quot;</span><span class="p">))</span>
<span class="go">5</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_seq</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;ATG&quot;</span><span class="p">))</span>
<span class="go">1</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_seq</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;AT&quot;</span><span class="p">)))</span>
<span class="go">1</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_seq</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;AT&quot;</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="o">-</span><span class="mi">1</span><span class="p">))</span>
<span class="go">1</span>
</pre></div>
</div>
<p>HOWEVER, please note because python strings and Seq objects (and
MutableSeq objects) do a non-overlapping search, this may not give
the answer you expect:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="s2">&quot;AAAA&quot;</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;AA&quot;</span><span class="p">)</span>
<span class="go">2</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;AAAA&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;AA&quot;</span><span class="p">))</span>
<span class="go">2</span>
</pre></div>
</div>
<p>An overlapping search, as implemented in .count_overlap(),
would give the answer as three!</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.Seq.count_overlap">
<code class="sig-name descname">count_overlap</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">sub</em>, <em class="sig-param">start=0</em>, <em class="sig-param">end=9223372036854775807</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.count_overlap" title="Permalink to this definition">¶</a></dt>
<dd><p>Return an overlapping count.</p>
<p>For a non-overlapping search use the count() method.</p>
<p>Returns an integer, the number of occurrences of substring
argument sub in the (sub)sequence given by [start:end].
Optional arguments start and end are interpreted as in slice
notation.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>sub - a string or another Seq object to look for</p></li>
<li><p>start - optional integer, slice start</p></li>
<li><p>end - optional integer, slice end</p></li>
</ul>
</dd>
</dl>
<p>e.g.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;AAAA&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;AA&quot;</span><span class="p">))</span>
<span class="go">3</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;ATATATATA&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;ATA&quot;</span><span class="p">))</span>
<span class="go">4</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;ATATATATA&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;ATA&quot;</span><span class="p">,</span> <span class="mi">3</span><span class="p">,</span> <span class="o">-</span><span class="mi">1</span><span class="p">))</span>
<span class="go">1</span>
</pre></div>
</div>
<p>Where substrings do not overlap, should behave the same as
the count() method:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;AAAATGA&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_seq</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;A&quot;</span><span class="p">))</span>
<span class="go">5</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;A&quot;</span><span class="p">)</span> <span class="o">==</span> <span class="n">my_seq</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;A&quot;</span><span class="p">)</span>
<span class="go">True</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_seq</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;ATG&quot;</span><span class="p">))</span>
<span class="go">1</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;ATG&quot;</span><span class="p">)</span> <span class="o">==</span> <span class="n">my_seq</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;ATG&quot;</span><span class="p">)</span>
<span class="go">True</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_seq</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;AT&quot;</span><span class="p">)))</span>
<span class="go">1</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;AT&quot;</span><span class="p">))</span> <span class="o">==</span> <span class="n">my_seq</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;AT&quot;</span><span class="p">))</span>
<span class="go">True</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_seq</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;AT&quot;</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="o">-</span><span class="mi">1</span><span class="p">))</span>
<span class="go">1</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;AT&quot;</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="o">-</span><span class="mi">1</span><span class="p">)</span> <span class="o">==</span> <span class="n">my_seq</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;AT&quot;</span><span class="p">,</span> <span class="mi">2</span><span class="p">,</span> <span class="o">-</span><span class="mi">1</span><span class="p">)</span>
<span class="go">True</span>
</pre></div>
</div>
<p>HOWEVER, do not use this method for such cases because the
count() method is much for efficient.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.Seq.endswith">
<code class="sig-name descname">endswith</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">suffix</em>, <em class="sig-param">start=0</em>, <em class="sig-param">end=9223372036854775807</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.endswith" title="Permalink to this definition">¶</a></dt>
<dd><p>Return True if the Seq ends with the given suffix, False otherwise.</p>
<p>This behaves like the python string method of the same name.</p>
<p>Return True if the sequence ends with the specified suffix
(a string or another Seq object), False otherwise.
With optional start, test sequence beginning at that position.
With optional end, stop comparing sequence at that position.
suffix can also be a tuple of strings to try.  e.g.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span><span class="o">.</span><span class="n">endswith</span><span class="p">(</span><span class="s2">&quot;UUG&quot;</span><span class="p">)</span>
<span class="go">True</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span><span class="o">.</span><span class="n">endswith</span><span class="p">(</span><span class="s2">&quot;AUG&quot;</span><span class="p">)</span>
<span class="go">False</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span><span class="o">.</span><span class="n">endswith</span><span class="p">(</span><span class="s2">&quot;AUG&quot;</span><span class="p">,</span> <span class="mi">0</span><span class="p">,</span> <span class="mi">18</span><span class="p">)</span>
<span class="go">True</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span><span class="o">.</span><span class="n">endswith</span><span class="p">((</span><span class="s2">&quot;UCC&quot;</span><span class="p">,</span> <span class="s2">&quot;UCA&quot;</span><span class="p">,</span> <span class="s2">&quot;UUG&quot;</span><span class="p">))</span>
<span class="go">True</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.Seq.find">
<code class="sig-name descname">find</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">sub</em>, <em class="sig-param">start=0</em>, <em class="sig-param">end=9223372036854775807</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.find" title="Permalink to this definition">¶</a></dt>
<dd><p>Find method, like that of a python string.</p>
<p>This behaves like the python string method of the same name.</p>
<p>Returns an integer, the index of the first occurrence of substring
argument sub in the (sub)sequence given by [start:end].</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>sub - a string or another Seq object to look for</p></li>
<li><p>start - optional integer, slice start</p></li>
<li><p>end - optional integer, slice end</p></li>
</ul>
</dd>
</dl>
<p>Returns -1 if the subsequence is NOT found.</p>
<p>e.g. Locating the first typical start codon, AUG, in an RNA sequence:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span><span class="o">.</span><span class="n">find</span><span class="p">(</span><span class="s2">&quot;AUG&quot;</span><span class="p">)</span>
<span class="go">3</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.Seq.index">
<code class="sig-name descname">index</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">sub</em>, <em class="sig-param">start=0</em>, <em class="sig-param">end=9223372036854775807</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.index" title="Permalink to this definition">¶</a></dt>
<dd><p>Like find() but raise ValueError when the substring is not found.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span><span class="o">.</span><span class="n">find</span><span class="p">(</span><span class="s2">&quot;T&quot;</span><span class="p">)</span>
<span class="go">-1</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span><span class="o">.</span><span class="n">index</span><span class="p">(</span><span class="s2">&quot;T&quot;</span><span class="p">)</span>
<span class="gt">Traceback (most recent call last):</span>
           <span class="o">...</span>
<span class="gr">ValueError</span>: <span class="n">substring not found...</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.Seq.join">
<code class="sig-name descname">join</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">other</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.join" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a merge of the sequences in other, spaced by the sequence from self.</p>
<p>Accepts all Seq objects and Strings as objects to be concatenated with the spacer</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">concatenated</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s1">&#39;NNNNN&#39;</span><span class="p">)</span><span class="o">.</span><span class="n">join</span><span class="p">([</span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;AAA&quot;</span><span class="p">),</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;TTT&quot;</span><span class="p">),</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;PPP&quot;</span><span class="p">)])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">concatenated</span>
<span class="go">Seq(&#39;AAANNNNNTTTNNNNNPPP&#39;)</span>
</pre></div>
</div>
<p>Throws error if other is not an iterable and if objects inside of the iterable
are not Seq or String objects</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.Seq.lower">
<code class="sig-name descname">lower</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.lower" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a lower case copy of the sequence.</p>
<p>This will adjust the alphabet if required. Note that the IUPAC
alphabets are upper case only, and thus a generic alphabet must be
substituted.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="k">import</span> <span class="n">Gapped</span><span class="p">,</span> <span class="n">generic_dna</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="k">import</span> <span class="n">IUPAC</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;CGGTACGCTTATGTCACGTAG*AAAAAA&quot;</span><span class="p">,</span>
<span class="gp">... </span>         <span class="n">Gapped</span><span class="p">(</span><span class="n">IUPAC</span><span class="o">.</span><span class="n">unambiguous_dna</span><span class="p">,</span> <span class="s2">&quot;*&quot;</span><span class="p">))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">Seq(&#39;CGGTACGCTTATGTCACGTAG*AAAAAA&#39;, Gapped(IUPACUnambiguousDNA(), &#39;*&#39;))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">lower</span><span class="p">()</span>
<span class="go">Seq(&#39;cggtacgcttatgtcacgtag*aaaaaa&#39;, Gapped(DNAAlphabet(), &#39;*&#39;))</span>
</pre></div>
</div>
<p>See also the upper method.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.Seq.lstrip">
<code class="sig-name descname">lstrip</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">chars=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.lstrip" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a new Seq object with leading (left) end stripped.</p>
<p>This behaves like the python string method of the same name.</p>
<p>Optional argument chars defines which characters to remove.  If
omitted or None (default) then as for the python string method,
this defaults to removing any white space.</p>
<p>e.g. print(my_seq.lstrip(“-“))</p>
<p>See also the strip and rstrip methods.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.Seq.reverse_complement">
<code class="sig-name descname">reverse_complement</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.reverse_complement" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the reverse complement sequence by creating a new Seq object.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="k">import</span> <span class="n">IUPAC</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;CCCCCGATAGNR&quot;</span><span class="p">,</span> <span class="n">IUPAC</span><span class="o">.</span><span class="n">ambiguous_dna</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span>
<span class="go">Seq(&#39;CCCCCGATAGNR&#39;, IUPACAmbiguousDNA())</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span><span class="o">.</span><span class="n">reverse_complement</span><span class="p">()</span>
<span class="go">Seq(&#39;YNCTATCGGGGG&#39;, IUPACAmbiguousDNA())</span>
</pre></div>
</div>
<p>Note in the above example, since R = G or A, its complement
is Y (which denotes C or T).</p>
<p>You can of course used mixed case sequences,</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="k">import</span> <span class="n">generic_dna</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;CCCCCgatA-G&quot;</span><span class="p">,</span> <span class="n">generic_dna</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span>
<span class="go">Seq(&#39;CCCCCgatA-G&#39;, DNAAlphabet())</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span><span class="o">.</span><span class="n">reverse_complement</span><span class="p">()</span>
<span class="go">Seq(&#39;C-TatcGGGGG&#39;, DNAAlphabet())</span>
</pre></div>
</div>
<p>Trying to complement a protein sequence raises an exception:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_protein</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;MAIVMGR&quot;</span><span class="p">,</span> <span class="n">IUPAC</span><span class="o">.</span><span class="n">protein</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_protein</span><span class="o">.</span><span class="n">reverse_complement</span><span class="p">()</span>
<span class="gt">Traceback (most recent call last):</span>
   <span class="c">...</span>
<span class="gr">ValueError</span>: <span class="n">Proteins do not have complements!</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.Seq.rfind">
<code class="sig-name descname">rfind</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">sub</em>, <em class="sig-param">start=0</em>, <em class="sig-param">end=9223372036854775807</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.rfind" title="Permalink to this definition">¶</a></dt>
<dd><p>Find from right method, like that of a python string.</p>
<p>This behaves like the python string method of the same name.</p>
<p>Returns an integer, the index of the last (right most) occurrence of
substring argument sub in the (sub)sequence given by [start:end].</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>sub - a string or another Seq object to look for</p></li>
<li><p>start - optional integer, slice start</p></li>
<li><p>end - optional integer, slice end</p></li>
</ul>
</dd>
</dl>
<p>Returns -1 if the subsequence is NOT found.</p>
<p>e.g. Locating the last typical start codon, AUG, in an RNA sequence:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span><span class="o">.</span><span class="n">rfind</span><span class="p">(</span><span class="s2">&quot;AUG&quot;</span><span class="p">)</span>
<span class="go">15</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.Seq.rindex">
<code class="sig-name descname">rindex</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">sub</em>, <em class="sig-param">start=0</em>, <em class="sig-param">end=9223372036854775807</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.rindex" title="Permalink to this definition">¶</a></dt>
<dd><p>Like rfind() but raise ValueError when the substring is not found.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.Seq.rsplit">
<code class="sig-name descname">rsplit</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">sep=None</em>, <em class="sig-param">maxsplit=-1</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.rsplit" title="Permalink to this definition">¶</a></dt>
<dd><p>Do a right split method, like that of a python string.</p>
<p>This behaves like the python string method of the same name.</p>
<p>Return a list of the ‘words’ in the string (as Seq objects),
using sep as the delimiter string.  If maxsplit is given, at
most maxsplit splits are done COUNTING FROM THE RIGHT.
If maxsplit is omitted, all splits are made.</p>
<p>Following the python string method, sep will by default be any
white space (tabs, spaces, newlines) but this is unlikely to
apply to biological sequences.</p>
<p>e.g. print(my_seq.rsplit(“*”,1))</p>
<p>See also the split method.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.Seq.rstrip">
<code class="sig-name descname">rstrip</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">chars=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.rstrip" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a new Seq object with trailing (right) end stripped.</p>
<p>This behaves like the python string method of the same name.</p>
<p>Optional argument chars defines which characters to remove.  If
omitted or None (default) then as for the python string method,
this defaults to removing any white space.</p>
<p>e.g. Removing a nucleotide sequence’s polyadenylation (poly-A tail):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="k">import</span> <span class="n">IUPAC</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;CGGTACGCTTATGTCACGTAGAAAAAA&quot;</span><span class="p">,</span> <span class="n">IUPAC</span><span class="o">.</span><span class="n">unambiguous_dna</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">Seq(&#39;CGGTACGCTTATGTCACGTAGAAAAAA&#39;, IUPACUnambiguousDNA())</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">rstrip</span><span class="p">(</span><span class="s2">&quot;A&quot;</span><span class="p">)</span>
<span class="go">Seq(&#39;CGGTACGCTTATGTCACGTAG&#39;, IUPACUnambiguousDNA())</span>
</pre></div>
</div>
<p>See also the strip and lstrip methods.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.Seq.split">
<code class="sig-name descname">split</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">sep=None</em>, <em class="sig-param">maxsplit=-1</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.split" title="Permalink to this definition">¶</a></dt>
<dd><p>Split method, like that of a python string.</p>
<p>This behaves like the python string method of the same name.</p>
<p>Return a list of the ‘words’ in the string (as Seq objects),
using sep as the delimiter string.  If maxsplit is given, at
most maxsplit splits are done.  If maxsplit is omitted, all
splits are made.</p>
<p>Following the python string method, sep will by default be any
white space (tabs, spaces, newlines) but this is unlikely to
apply to biological sequences.</p>
<p>e.g.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_aa</span> <span class="o">=</span> <span class="n">my_rna</span><span class="o">.</span><span class="n">translate</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_aa</span>
<span class="go">Seq(&#39;VMAIVMGR*KGAR*L&#39;, HasStopCodon(ExtendedIUPACProtein(), &#39;*&#39;))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">for</span> <span class="n">pep</span> <span class="ow">in</span> <span class="n">my_aa</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;*&quot;</span><span class="p">):</span>
<span class="gp">... </span>    <span class="n">pep</span>
<span class="go">Seq(&#39;VMAIVMGR&#39;, HasStopCodon(ExtendedIUPACProtein(), &#39;*&#39;))</span>
<span class="go">Seq(&#39;KGAR&#39;, HasStopCodon(ExtendedIUPACProtein(), &#39;*&#39;))</span>
<span class="go">Seq(&#39;L&#39;, HasStopCodon(ExtendedIUPACProtein(), &#39;*&#39;))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="k">for</span> <span class="n">pep</span> <span class="ow">in</span> <span class="n">my_aa</span><span class="o">.</span><span class="n">split</span><span class="p">(</span><span class="s2">&quot;*&quot;</span><span class="p">,</span> <span class="mi">1</span><span class="p">):</span>
<span class="gp">... </span>    <span class="n">pep</span>
<span class="go">Seq(&#39;VMAIVMGR&#39;, HasStopCodon(ExtendedIUPACProtein(), &#39;*&#39;))</span>
<span class="go">Seq(&#39;KGAR*L&#39;, HasStopCodon(ExtendedIUPACProtein(), &#39;*&#39;))</span>
</pre></div>
</div>
<p>See also the rsplit method:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="k">for</span> <span class="n">pep</span> <span class="ow">in</span> <span class="n">my_aa</span><span class="o">.</span><span class="n">rsplit</span><span class="p">(</span><span class="s2">&quot;*&quot;</span><span class="p">,</span> <span class="mi">1</span><span class="p">):</span>
<span class="gp">... </span>    <span class="n">pep</span>
<span class="go">Seq(&#39;VMAIVMGR*KGAR&#39;, HasStopCodon(ExtendedIUPACProtein(), &#39;*&#39;))</span>
<span class="go">Seq(&#39;L&#39;, HasStopCodon(ExtendedIUPACProtein(), &#39;*&#39;))</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.Seq.startswith">
<code class="sig-name descname">startswith</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">prefix</em>, <em class="sig-param">start=0</em>, <em class="sig-param">end=9223372036854775807</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.startswith" title="Permalink to this definition">¶</a></dt>
<dd><p>Return True if the Seq starts with the given prefix, False otherwise.</p>
<p>This behaves like the python string method of the same name.</p>
<p>Return True if the sequence starts with the specified prefix
(a string or another Seq object), False otherwise.
With optional start, test sequence beginning at that position.
With optional end, stop comparing sequence at that position.
prefix can also be a tuple of strings to try.  e.g.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;GUCAUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAGUUG&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s2">&quot;GUC&quot;</span><span class="p">)</span>
<span class="go">True</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s2">&quot;AUG&quot;</span><span class="p">)</span>
<span class="go">False</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span><span class="o">.</span><span class="n">startswith</span><span class="p">(</span><span class="s2">&quot;AUG&quot;</span><span class="p">,</span> <span class="mi">3</span><span class="p">)</span>
<span class="go">True</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span><span class="o">.</span><span class="n">startswith</span><span class="p">((</span><span class="s2">&quot;UCC&quot;</span><span class="p">,</span> <span class="s2">&quot;UCA&quot;</span><span class="p">,</span> <span class="s2">&quot;UCG&quot;</span><span class="p">),</span> <span class="mi">1</span><span class="p">)</span>
<span class="go">True</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.Seq.strip">
<code class="sig-name descname">strip</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">chars=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.strip" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a new Seq object with leading and trailing ends stripped.</p>
<p>This behaves like the python string method of the same name.</p>
<p>Optional argument chars defines which characters to remove.  If
omitted or None (default) then as for the python string method,
this defaults to removing any white space.</p>
<p>e.g. print(my_seq.strip(“-“))</p>
<p>See also the lstrip and rstrip methods.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.Seq.tomutable">
<code class="sig-name descname">tomutable</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.tomutable" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the full sequence as a MutableSeq object.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="k">import</span> <span class="n">IUPAC</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;MKQHKAMIVALIVICITAVVAAL&quot;</span><span class="p">,</span>
<span class="gp">... </span>             <span class="n">IUPAC</span><span class="o">.</span><span class="n">protein</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">Seq(&#39;MKQHKAMIVALIVICITAVVAAL&#39;, IUPACProtein())</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">tomutable</span><span class="p">()</span>
<span class="go">MutableSeq(&#39;MKQHKAMIVALIVICITAVVAAL&#39;, IUPACProtein())</span>
</pre></div>
</div>
<p>Note that the alphabet is preserved.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.Seq.transcribe">
<code class="sig-name descname">transcribe</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.transcribe" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the RNA sequence from a DNA sequence by creating a new Seq object.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="k">import</span> <span class="n">IUPAC</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">coding_dna</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;ATGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG&quot;</span><span class="p">,</span>
<span class="gp">... </span>                 <span class="n">IUPAC</span><span class="o">.</span><span class="n">unambiguous_dna</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">coding_dna</span>
<span class="go">Seq(&#39;ATGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG&#39;, IUPACUnambiguousDNA())</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">coding_dna</span><span class="o">.</span><span class="n">transcribe</span><span class="p">()</span>
<span class="go">Seq(&#39;AUGGCCAUUGUAAUGGGCCGCUGAAAGGGUGCCCGAUAG&#39;, IUPACUnambiguousRNA())</span>
</pre></div>
</div>
<p>Trying to transcribe a protein or RNA sequence raises an exception:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_protein</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;MAIVMGR&quot;</span><span class="p">,</span> <span class="n">IUPAC</span><span class="o">.</span><span class="n">protein</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_protein</span><span class="o">.</span><span class="n">transcribe</span><span class="p">()</span>
<span class="gt">Traceback (most recent call last):</span>
   <span class="c">...</span>
<span class="gr">ValueError</span>: <span class="n">Proteins cannot be transcribed!</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.Seq.translate">
<code class="sig-name descname">translate</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">table='Standard'</em>, <em class="sig-param">stop_symbol='*'</em>, <em class="sig-param">to_stop=False</em>, <em class="sig-param">cds=False</em>, <em class="sig-param">gap=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.translate" title="Permalink to this definition">¶</a></dt>
<dd><p>Turn a nucleotide sequence into a protein sequence by creating a new Seq object.</p>
<p>This method will translate DNA or RNA sequences, and those with a
nucleotide or generic alphabet.  Trying to translate a protein
sequence raises an exception.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>table - Which codon table to use?  This can be either a name
(string), an NCBI identifier (integer), or a CodonTable
object (useful for non-standard genetic codes).  This
defaults to the “Standard” table.</p></li>
<li><p>stop_symbol - Single character string, what to use for
terminators.  This defaults to the asterisk, “*”.</p></li>
<li><p>to_stop - Boolean, defaults to False meaning do a full
translation continuing on past any stop codons (translated as the
specified stop_symbol).  If True, translation is terminated at
the first in frame stop codon (and the stop_symbol is not
appended to the returned protein sequence).</p></li>
<li><p>cds - Boolean, indicates this is a complete CDS.  If True,
this checks the sequence starts with a valid alternative start
codon (which will be translated as methionine, M), that the
sequence length is a multiple of three, and that there is a
single in frame stop codon at the end (this will be excluded
from the protein sequence, regardless of the to_stop option).
If these tests fail, an exception is raised.</p></li>
<li><p>gap - Single character string to denote symbol used for gaps.
It will try to guess the gap character from the alphabet.</p></li>
</ul>
</dd>
</dl>
<p>e.g. Using the standard table:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">coding_dna</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;GTGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">coding_dna</span><span class="o">.</span><span class="n">translate</span><span class="p">()</span>
<span class="go">Seq(&#39;VAIVMGR*KGAR*&#39;, HasStopCodon(ExtendedIUPACProtein(), &#39;*&#39;))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">coding_dna</span><span class="o">.</span><span class="n">translate</span><span class="p">(</span><span class="n">stop_symbol</span><span class="o">=</span><span class="s2">&quot;@&quot;</span><span class="p">)</span>
<span class="go">Seq(&#39;VAIVMGR@KGAR@&#39;, HasStopCodon(ExtendedIUPACProtein(), &#39;@&#39;))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">coding_dna</span><span class="o">.</span><span class="n">translate</span><span class="p">(</span><span class="n">to_stop</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="go">Seq(&#39;VAIVMGR&#39;, ExtendedIUPACProtein())</span>
</pre></div>
</div>
<p>Now using NCBI table 2, where TGA is not a stop codon:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">coding_dna</span><span class="o">.</span><span class="n">translate</span><span class="p">(</span><span class="n">table</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>
<span class="go">Seq(&#39;VAIVMGRWKGAR*&#39;, HasStopCodon(ExtendedIUPACProtein(), &#39;*&#39;))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">coding_dna</span><span class="o">.</span><span class="n">translate</span><span class="p">(</span><span class="n">table</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">to_stop</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="go">Seq(&#39;VAIVMGRWKGAR&#39;, ExtendedIUPACProtein())</span>
</pre></div>
</div>
<p>In fact, GTG is an alternative start codon under NCBI table 2, meaning
this sequence could be a complete CDS:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">coding_dna</span><span class="o">.</span><span class="n">translate</span><span class="p">(</span><span class="n">table</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">cds</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="go">Seq(&#39;MAIVMGRWKGAR&#39;, ExtendedIUPACProtein())</span>
</pre></div>
</div>
<p>It isn’t a valid CDS under NCBI table 1, due to both the start codon
and also the in frame stop codons:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">coding_dna</span><span class="o">.</span><span class="n">translate</span><span class="p">(</span><span class="n">table</span><span class="o">=</span><span class="mi">1</span><span class="p">,</span> <span class="n">cds</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="gt">Traceback (most recent call last):</span>
    <span class="o">...</span>
<span class="gr">Bio.Data.CodonTable.TranslationError</span>: <span class="n">First codon &#39;GTG&#39; is not a start codon</span>
</pre></div>
</div>
<p>If the sequence has no in-frame stop codon, then the to_stop argument
has no effect:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">coding_dna2</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;TTGGCCATTGTAATGGGCCGC&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">coding_dna2</span><span class="o">.</span><span class="n">translate</span><span class="p">()</span>
<span class="go">Seq(&#39;LAIVMGR&#39;, ExtendedIUPACProtein())</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">coding_dna2</span><span class="o">.</span><span class="n">translate</span><span class="p">(</span><span class="n">to_stop</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="go">Seq(&#39;LAIVMGR&#39;, ExtendedIUPACProtein())</span>
</pre></div>
</div>
<p>When translating gapped sequences, the gap character is inferred from
the alphabet:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="k">import</span> <span class="n">Gapped</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">coding_dna3</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;GTG---GCCATT&quot;</span><span class="p">,</span> <span class="n">Gapped</span><span class="p">(</span><span class="n">IUPAC</span><span class="o">.</span><span class="n">unambiguous_dna</span><span class="p">))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">coding_dna3</span><span class="o">.</span><span class="n">translate</span><span class="p">()</span>
<span class="go">Seq(&#39;V-AI&#39;, Gapped(ExtendedIUPACProtein(), &#39;-&#39;))</span>
</pre></div>
</div>
<p>It is possible to pass the gap character when the alphabet is missing:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">coding_dna4</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;GTG---GCCATT&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">coding_dna4</span><span class="o">.</span><span class="n">translate</span><span class="p">(</span><span class="n">gap</span><span class="o">=</span><span class="s1">&#39;-&#39;</span><span class="p">)</span>
<span class="go">Seq(&#39;V-AI&#39;, Gapped(ExtendedIUPACProtein(), &#39;-&#39;))</span>
</pre></div>
</div>
<p>NOTE - Ambiguous codons like “TAN” or “NNN” could be an amino acid
or a stop codon.  These are translated as “X”.  Any invalid codon
(e.g. “TA?” or “T-A”) will throw a TranslationError.</p>
<p>NOTE - This does NOT behave like the python string’s translate
method.  For that use str(my_seq).translate(…) instead.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.Seq.ungap">
<code class="sig-name descname">ungap</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">gap=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.ungap" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a copy of the sequence without the gap character(s).</p>
<p>The gap character can be specified in two ways - either as an explicit
argument, or via the sequence’s alphabet. For example:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="k">import</span> <span class="n">generic_dna</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;-ATA--TGAAAT-TTGAAAA&quot;</span><span class="p">,</span> <span class="n">generic_dna</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span>
<span class="go">Seq(&#39;-ATA--TGAAAT-TTGAAAA&#39;, DNAAlphabet())</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span><span class="o">.</span><span class="n">ungap</span><span class="p">(</span><span class="s2">&quot;-&quot;</span><span class="p">)</span>
<span class="go">Seq(&#39;ATATGAAATTTGAAAA&#39;, DNAAlphabet())</span>
</pre></div>
</div>
<p>If the gap character is not given as an argument, it will be taken from
the sequence’s alphabet (if defined). Notice that the returned
sequence’s alphabet is adjusted since it no longer requires a gapped
alphabet:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="k">import</span> <span class="n">IUPAC</span><span class="p">,</span> <span class="n">Gapped</span><span class="p">,</span> <span class="n">HasStopCodon</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_pro</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;MVVLE=AD*&quot;</span><span class="p">,</span> <span class="n">HasStopCodon</span><span class="p">(</span><span class="n">Gapped</span><span class="p">(</span><span class="n">IUPAC</span><span class="o">.</span><span class="n">protein</span><span class="p">,</span> <span class="s2">&quot;=&quot;</span><span class="p">)))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_pro</span>
<span class="go">Seq(&#39;MVVLE=AD*&#39;, HasStopCodon(Gapped(IUPACProtein(), &#39;=&#39;), &#39;*&#39;))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_pro</span><span class="o">.</span><span class="n">ungap</span><span class="p">()</span>
<span class="go">Seq(&#39;MVVLEAD*&#39;, HasStopCodon(IUPACProtein(), &#39;*&#39;))</span>
</pre></div>
</div>
<p>Or, with a simpler gapped DNA example:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="k">import</span> <span class="n">IUPAC</span><span class="p">,</span> <span class="n">Gapped</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;CGGGTAG=AAAAAA&quot;</span><span class="p">,</span> <span class="n">Gapped</span><span class="p">(</span><span class="n">IUPAC</span><span class="o">.</span><span class="n">unambiguous_dna</span><span class="p">,</span> <span class="s2">&quot;=&quot;</span><span class="p">))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">Seq(&#39;CGGGTAG=AAAAAA&#39;, Gapped(IUPACUnambiguousDNA(), &#39;=&#39;))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">ungap</span><span class="p">()</span>
<span class="go">Seq(&#39;CGGGTAGAAAAAA&#39;, IUPACUnambiguousDNA())</span>
</pre></div>
</div>
<p>As long as it is consistent with the alphabet, although it is
redundant, you can still supply the gap character as an argument to
this method:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">Seq(&#39;CGGGTAG=AAAAAA&#39;, Gapped(IUPACUnambiguousDNA(), &#39;=&#39;))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">ungap</span><span class="p">(</span><span class="s2">&quot;=&quot;</span><span class="p">)</span>
<span class="go">Seq(&#39;CGGGTAGAAAAAA&#39;, IUPACUnambiguousDNA())</span>
</pre></div>
</div>
<p>However, if the gap character given as the argument disagrees with that
declared in the alphabet, an exception is raised:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">Seq(&#39;CGGGTAG=AAAAAA&#39;, Gapped(IUPACUnambiguousDNA(), &#39;=&#39;))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">ungap</span><span class="p">(</span><span class="s2">&quot;-&quot;</span><span class="p">)</span>
<span class="gt">Traceback (most recent call last):</span>
   <span class="c">...</span>
<span class="gr">ValueError</span>: <span class="n">Gap &#39;-&#39; does not match &#39;=&#39; from alphabet</span>
</pre></div>
</div>
<p>Finally, if a gap character is not supplied, and the alphabet does not
define one, an exception is raised:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="k">import</span> <span class="n">generic_dna</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;ATA--TGAAAT-TTGAAAA&quot;</span><span class="p">,</span> <span class="n">generic_dna</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span>
<span class="go">Seq(&#39;ATA--TGAAAT-TTGAAAA&#39;, DNAAlphabet())</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span><span class="o">.</span><span class="n">ungap</span><span class="p">()</span>
<span class="gt">Traceback (most recent call last):</span>
   <span class="c">...</span>
<span class="gr">ValueError</span>: <span class="n">Gap character not given and not defined in alphabet</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.Seq.upper">
<code class="sig-name descname">upper</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.Seq.upper" title="Permalink to this definition">¶</a></dt>
<dd><p>Return an upper case copy of the sequence.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="k">import</span> <span class="n">HasStopCodon</span><span class="p">,</span> <span class="n">generic_protein</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">Seq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;VHLTPeeK*&quot;</span><span class="p">,</span> <span class="n">HasStopCodon</span><span class="p">(</span><span class="n">generic_protein</span><span class="p">))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">Seq(&#39;VHLTPeeK*&#39;, HasStopCodon(ProteinAlphabet(), &#39;*&#39;))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">lower</span><span class="p">()</span>
<span class="go">Seq(&#39;vhltpeek*&#39;, HasStopCodon(ProteinAlphabet(), &#39;*&#39;))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">upper</span><span class="p">()</span>
<span class="go">Seq(&#39;VHLTPEEK*&#39;, HasStopCodon(ProteinAlphabet(), &#39;*&#39;))</span>
</pre></div>
</div>
<p>This will adjust the alphabet if required. See also the lower method.</p>
</dd></dl>

</dd></dl>

<dl class="class">
<dt id="Bio.Seq.UnknownSeq">
<em class="property">class </em><code class="sig-prename descclassname">Bio.Seq.</code><code class="sig-name descname">UnknownSeq</code><span class="sig-paren">(</span><em class="sig-param">length</em>, <em class="sig-param">alphabet=Alphabet()</em>, <em class="sig-param">character=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq" title="Permalink to this definition">¶</a></dt>
<dd><p>Bases: <a class="reference internal" href="#Bio.Seq.Seq" title="Bio.Seq.Seq"><code class="xref py py-class docutils literal notranslate"><span class="pre">Bio.Seq.Seq</span></code></a></p>
<p>Read-only sequence object of known length but unknown contents.</p>
<p>If you have an unknown sequence, you can represent this with a normal
Seq object, for example:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;N&quot;</span><span class="o">*</span><span class="mi">5</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">Seq(&#39;NNNNN&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">my_seq</span><span class="p">)</span>
<span class="go">5</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_seq</span><span class="p">)</span>
<span class="go">NNNNN</span>
</pre></div>
</div>
<p>However, this is rather wasteful of memory (especially for large
sequences), which is where this class is most useful:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">unk_five</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">5</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">unk_five</span>
<span class="go">UnknownSeq(5, character=&#39;?&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">len</span><span class="p">(</span><span class="n">unk_five</span><span class="p">)</span>
<span class="go">5</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">unk_five</span><span class="p">)</span>
<span class="go">?????</span>
</pre></div>
</div>
<p>You can add unknown sequence together, provided their alphabets and
characters are compatible, and get another memory saving UnknownSeq:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">unk_four</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">unk_four</span>
<span class="go">UnknownSeq(4, character=&#39;?&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">unk_four</span> <span class="o">+</span> <span class="n">unk_five</span>
<span class="go">UnknownSeq(9, character=&#39;?&#39;)</span>
</pre></div>
</div>
<p>If the alphabet or characters don’t match up, the addition gives an
ordinary Seq object:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">unk_nnnn</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">unk_nnnn</span>
<span class="go">UnknownSeq(4, character=&#39;N&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">unk_nnnn</span> <span class="o">+</span> <span class="n">unk_four</span>
<span class="go">Seq(&#39;NNNN????&#39;)</span>
</pre></div>
</div>
<p>Combining with a real Seq gives a new Seq object:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">known_seq</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;ACGT&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">unk_four</span> <span class="o">+</span> <span class="n">known_seq</span>
<span class="go">Seq(&#39;????ACGT&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">known_seq</span> <span class="o">+</span> <span class="n">unk_four</span>
<span class="go">Seq(&#39;ACGT????&#39;)</span>
</pre></div>
</div>
<dl class="method">
<dt id="Bio.Seq.UnknownSeq.back_transcribe">
<code class="sig-name descname">back_transcribe</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.back_transcribe" title="Permalink to this definition">¶</a></dt>
<dd><p>Return an unknown DNA sequence from an unknown RNA sequence.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">20</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span>
<span class="go">UnknownSeq(20, character=&#39;N&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_rna</span><span class="p">)</span>
<span class="go">NNNNNNNNNNNNNNNNNNNN</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span> <span class="o">=</span> <span class="n">my_rna</span><span class="o">.</span><span class="n">back_transcribe</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span>
<span class="go">UnknownSeq(20, alphabet=DNAAlphabet(), character=&#39;N&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_dna</span><span class="p">)</span>
<span class="go">NNNNNNNNNNNNNNNNNNNN</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.UnknownSeq.complement">
<code class="sig-name descname">complement</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.complement" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the complement of an unknown nucleotide equals itself.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_nuc</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">8</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_nuc</span>
<span class="go">UnknownSeq(8, character=&#39;?&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_nuc</span><span class="p">)</span>
<span class="go">????????</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_nuc</span><span class="o">.</span><span class="n">complement</span><span class="p">()</span>
<span class="go">UnknownSeq(8, character=&#39;?&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_nuc</span><span class="o">.</span><span class="n">complement</span><span class="p">())</span>
<span class="go">????????</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.UnknownSeq.count">
<code class="sig-name descname">count</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">sub</em>, <em class="sig-param">start=0</em>, <em class="sig-param">end=9223372036854775807</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.count" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a non-overlapping count, like that of a python string.</p>
<p>This behaves like the python string (and Seq object) method of the
same name, which does a non-overlapping count!</p>
<p>For an overlapping search use the newer count_overlap() method.</p>
<p>Returns an integer, the number of occurrences of substring
argument sub in the (sub)sequence given by [start:end].
Optional arguments start and end are interpreted as in slice
notation.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>sub - a string or another Seq object to look for</p></li>
<li><p>start - optional integer, slice start</p></li>
<li><p>end - optional integer, slice end</p></li>
</ul>
</dd>
</dl>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="s2">&quot;NNNN&quot;</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;N&quot;</span><span class="p">)</span>
<span class="go">4</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;NNNN&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;N&quot;</span><span class="p">)</span>
<span class="go">4</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;N&quot;</span><span class="p">)</span>
<span class="go">4</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;A&quot;</span><span class="p">)</span>
<span class="go">0</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;AA&quot;</span><span class="p">)</span>
<span class="go">0</span>
</pre></div>
</div>
<p>HOWEVER, please note because that python strings and Seq objects (and
MutableSeq objects) do a non-overlapping search, this may not give
the answer you expect:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;NN&quot;</span><span class="p">)</span>
<span class="go">2</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;NNN&quot;</span><span class="p">)</span>
<span class="go">1</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.UnknownSeq.count_overlap">
<code class="sig-name descname">count_overlap</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">sub</em>, <em class="sig-param">start=0</em>, <em class="sig-param">end=9223372036854775807</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.count_overlap" title="Permalink to this definition">¶</a></dt>
<dd><p>Return an overlapping count.</p>
<p>For a non-overlapping search use the count() method.</p>
<p>Returns an integer, the number of occurrences of substring
argument sub in the (sub)sequence given by [start:end].
Optional arguments start and end are interpreted as in slice
notation.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>sub - a string or another Seq object to look for</p></li>
<li><p>start - optional integer, slice start</p></li>
<li><p>end - optional integer, slice end</p></li>
</ul>
</dd>
</dl>
<p>e.g.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">UnknownSeq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;NN&quot;</span><span class="p">)</span>
<span class="go">3</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;NNN&quot;</span><span class="p">)</span>
<span class="go">2</span>
</pre></div>
</div>
<p>Where substrings do not overlap, should behave the same as
the count() method:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;N&quot;</span><span class="p">)</span>
<span class="go">4</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;N&quot;</span><span class="p">)</span> <span class="o">==</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;N&quot;</span><span class="p">)</span>
<span class="go">True</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;A&quot;</span><span class="p">)</span>
<span class="go">0</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;A&quot;</span><span class="p">)</span> <span class="o">==</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;A&quot;</span><span class="p">)</span>
<span class="go">True</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;AA&quot;</span><span class="p">)</span>
<span class="go">0</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count_overlap</span><span class="p">(</span><span class="s2">&quot;AA&quot;</span><span class="p">)</span> <span class="o">==</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">4</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span><span class="o">.</span><span class="n">count</span><span class="p">(</span><span class="s2">&quot;AA&quot;</span><span class="p">)</span>
<span class="go">True</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.UnknownSeq.join">
<code class="sig-name descname">join</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">other</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.join" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a merge of the sequences in other, spaced by the sequence from self.</p>
<p>Accepts Seq/UnknownSeq objects and Strings as objects to be concatenated with
the spacer</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">concatenated</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">5</span><span class="p">)</span><span class="o">.</span><span class="n">join</span><span class="p">([</span><span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;AAA&quot;</span><span class="p">),</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;TTT&quot;</span><span class="p">),</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;PPP&quot;</span><span class="p">)])</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">concatenated</span>
<span class="go">Seq(&#39;AAA?????TTT?????PPP&#39;)</span>
</pre></div>
</div>
<p>Throws error if other is not an iterable and if objects inside of the iterable
are not Seq/UnknownSeq or String objects.</p>
<p>Will only return an UnknownSeq object of all of the objects to be joined are
also UnknownSeqs with the same character as the spacer, similar to how the
addition of an UnknownSeq and another UnknownSeq would work.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.UnknownSeq.lower">
<code class="sig-name descname">lower</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.lower" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a lower case copy of the sequence.</p>
<p>This will adjust the alphabet if required:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="k">import</span> <span class="n">IUPAC</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">UnknownSeq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">20</span><span class="p">,</span> <span class="n">IUPAC</span><span class="o">.</span><span class="n">extended_protein</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">UnknownSeq(20, alphabet=ExtendedIUPACProtein(), character=&#39;X&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_seq</span><span class="p">)</span>
<span class="go">XXXXXXXXXXXXXXXXXXXX</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">lower</span><span class="p">()</span>
<span class="go">UnknownSeq(20, alphabet=ProteinAlphabet(), character=&#39;x&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_seq</span><span class="o">.</span><span class="n">lower</span><span class="p">())</span>
<span class="go">xxxxxxxxxxxxxxxxxxxx</span>
</pre></div>
</div>
<p>See also the upper method.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.UnknownSeq.reverse_complement">
<code class="sig-name descname">reverse_complement</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.reverse_complement" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the reverse complement of an unknown sequence.</p>
<p>The reverse complement of an unknown nucleotide equals itself:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">UnknownSeq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="k">import</span> <span class="n">generic_dna</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">example</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">6</span><span class="p">,</span> <span class="n">generic_dna</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">example</span><span class="p">)</span>
<span class="go">NNNNNN</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">example</span><span class="o">.</span><span class="n">reverse_complement</span><span class="p">())</span>
<span class="go">NNNNNN</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.UnknownSeq.transcribe">
<code class="sig-name descname">transcribe</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.transcribe" title="Permalink to this definition">¶</a></dt>
<dd><p>Return an unknown RNA sequence from an unknown DNA sequence.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">10</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span>
<span class="go">UnknownSeq(10, character=&#39;N&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_dna</span><span class="p">)</span>
<span class="go">NNNNNNNNNN</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span> <span class="o">=</span> <span class="n">my_dna</span><span class="o">.</span><span class="n">transcribe</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_rna</span>
<span class="go">UnknownSeq(10, alphabet=RNAAlphabet(), character=&#39;N&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_rna</span><span class="p">)</span>
<span class="go">NNNNNNNNNN</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.UnknownSeq.translate">
<code class="sig-name descname">translate</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">**kwargs</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.translate" title="Permalink to this definition">¶</a></dt>
<dd><p>Translate an unknown nucleotide sequence into an unknown protein.</p>
<p>e.g.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">9</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;N&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_seq</span><span class="p">)</span>
<span class="go">NNNNNNNNN</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_protein</span> <span class="o">=</span> <span class="n">my_seq</span><span class="o">.</span><span class="n">translate</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_protein</span>
<span class="go">UnknownSeq(3, alphabet=ProteinAlphabet(), character=&#39;X&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_protein</span><span class="p">)</span>
<span class="go">XXX</span>
</pre></div>
</div>
<p>In comparison, using a normal Seq object:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">Seq</span><span class="p">(</span><span class="s2">&quot;NNNNNNNNN&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_seq</span><span class="p">)</span>
<span class="go">NNNNNNNNN</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_protein</span> <span class="o">=</span> <span class="n">my_seq</span><span class="o">.</span><span class="n">translate</span><span class="p">()</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_protein</span>
<span class="go">Seq(&#39;XXX&#39;, ExtendedIUPACProtein())</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_protein</span><span class="p">)</span>
<span class="go">XXX</span>
</pre></div>
</div>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.UnknownSeq.ungap">
<code class="sig-name descname">ungap</code><span class="sig-paren">(</span><em class="sig-param">self</em>, <em class="sig-param">gap=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.ungap" title="Permalink to this definition">¶</a></dt>
<dd><p>Return a copy of the sequence without the gap character(s).</p>
<p>The gap character can be specified in two ways - either as an explicit
argument, or via the sequence’s alphabet. For example:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">UnknownSeq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="k">import</span> <span class="n">Gapped</span><span class="p">,</span> <span class="n">generic_dna</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">20</span><span class="p">,</span> <span class="n">Gapped</span><span class="p">(</span><span class="n">generic_dna</span><span class="p">,</span> <span class="s2">&quot;-&quot;</span><span class="p">))</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span>
<span class="go">UnknownSeq(20, alphabet=Gapped(DNAAlphabet(), &#39;-&#39;), character=&#39;N&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span><span class="o">.</span><span class="n">ungap</span><span class="p">()</span>
<span class="go">UnknownSeq(20, alphabet=DNAAlphabet(), character=&#39;N&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_dna</span><span class="o">.</span><span class="n">ungap</span><span class="p">(</span><span class="s2">&quot;-&quot;</span><span class="p">)</span>
<span class="go">UnknownSeq(20, alphabet=DNAAlphabet(), character=&#39;N&#39;)</span>
</pre></div>
</div>
<p>If the UnknownSeq is using the gap character, then an empty Seq is
returned:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">my_gap</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">20</span><span class="p">,</span> <span class="n">Gapped</span><span class="p">(</span><span class="n">generic_dna</span><span class="p">,</span> <span class="s2">&quot;-&quot;</span><span class="p">),</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;-&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_gap</span>
<span class="go">UnknownSeq(20, alphabet=Gapped(DNAAlphabet(), &#39;-&#39;), character=&#39;-&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_gap</span><span class="o">.</span><span class="n">ungap</span><span class="p">()</span>
<span class="go">Seq(&#39;&#39;, DNAAlphabet())</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_gap</span><span class="o">.</span><span class="n">ungap</span><span class="p">(</span><span class="s2">&quot;-&quot;</span><span class="p">)</span>
<span class="go">Seq(&#39;&#39;, DNAAlphabet())</span>
</pre></div>
</div>
<p>Notice that the returned sequence’s alphabet is adjusted to remove any
explicit gap character declaration.</p>
</dd></dl>

<dl class="method">
<dt id="Bio.Seq.UnknownSeq.upper">
<code class="sig-name descname">upper</code><span class="sig-paren">(</span><em class="sig-param">self</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.UnknownSeq.upper" title="Permalink to this definition">¶</a></dt>
<dd><p>Return an upper case copy of the sequence.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Alphabet</span> <span class="k">import</span> <span class="n">generic_dna</span>
<span class="gp">&gt;&gt;&gt; </span><span class="kn">from</span> <span class="nn">Bio.Seq</span> <span class="k">import</span> <span class="n">UnknownSeq</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span> <span class="o">=</span> <span class="n">UnknownSeq</span><span class="p">(</span><span class="mi">20</span><span class="p">,</span> <span class="n">generic_dna</span><span class="p">,</span> <span class="n">character</span><span class="o">=</span><span class="s2">&quot;n&quot;</span><span class="p">)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span>
<span class="go">UnknownSeq(20, alphabet=DNAAlphabet(), character=&#39;n&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_seq</span><span class="p">)</span>
<span class="go">nnnnnnnnnnnnnnnnnnnn</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">my_seq</span><span class="o">.</span><span class="n">upper</span><span class="p">()</span>
<span class="go">UnknownSeq(20, alphabet=DNAAlphabet(), character=&#39;N&#39;)</span>
<span class="gp">&gt;&gt;&gt; </span><span class="nb">print</span><span class="p">(</span><span class="n">my_seq</span><span class="o">.</span><span class="n">upper</span><span class="p">())</span>
<span class="go">NNNNNNNNNNNNNNNNNNNN</span>
</pre></div>
</div>
<p>This will adjust the alphabet if required. See also the lower method.</p>
</dd></dl>

</dd></dl>

<dl class="function">
<dt id="Bio.Seq.back_transcribe">
<code class="sig-prename descclassname">Bio.Seq.</code><code class="sig-name descname">back_transcribe</code><span class="sig-paren">(</span><em class="sig-param">rna</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.back_transcribe" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the RNA sequence back-transcribed into DNA.</p>
<p>If given a string, returns a new string object.</p>
<p>Given a Seq or MutableSeq, returns a new Seq object with an RNA alphabet.</p>
<p>Trying to transcribe a protein or DNA sequence raises an exception.</p>
<p>e.g.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">back_transcribe</span><span class="p">(</span><span class="s2">&quot;ACUGN&quot;</span><span class="p">)</span>
<span class="go">&#39;ACTGN&#39;</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Bio.Seq.complement">
<code class="sig-prename descclassname">Bio.Seq.</code><code class="sig-name descname">complement</code><span class="sig-paren">(</span><em class="sig-param">sequence</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.complement" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the complement sequence of a nucleotide string.</p>
<p>If given a string, returns a new string object.
Given a Seq or a MutableSeq, returns a new Seq object with the same
alphabet.</p>
<p>Supports unambiguous and ambiguous nucleotide sequences.</p>
<p>e.g.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">complement</span><span class="p">(</span><span class="s2">&quot;ACTG-NH&quot;</span><span class="p">)</span>
<span class="go">&#39;TGAC-ND&#39;</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Bio.Seq.reverse_complement">
<code class="sig-prename descclassname">Bio.Seq.</code><code class="sig-name descname">reverse_complement</code><span class="sig-paren">(</span><em class="sig-param">sequence</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.reverse_complement" title="Permalink to this definition">¶</a></dt>
<dd><p>Return the reverse complement sequence of a nucleotide string.</p>
<p>If given a string, returns a new string object.
Given a Seq or a MutableSeq, returns a new Seq object with the same
alphabet.</p>
<p>Supports unambiguous and ambiguous nucleotide sequences.</p>
<p>e.g.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">reverse_complement</span><span class="p">(</span><span class="s2">&quot;ACTG-NH&quot;</span><span class="p">)</span>
<span class="go">&#39;DN-CAGT&#39;</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Bio.Seq.transcribe">
<code class="sig-prename descclassname">Bio.Seq.</code><code class="sig-name descname">transcribe</code><span class="sig-paren">(</span><em class="sig-param">dna</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.transcribe" title="Permalink to this definition">¶</a></dt>
<dd><p>Transcribe a DNA sequence into RNA.</p>
<p>If given a string, returns a new string object.</p>
<p>Given a Seq or MutableSeq, returns a new Seq object with an RNA alphabet.</p>
<p>Trying to transcribe a protein or RNA sequence raises an exception.</p>
<p>e.g.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">transcribe</span><span class="p">(</span><span class="s2">&quot;ACTGN&quot;</span><span class="p">)</span>
<span class="go">&#39;ACUGN&#39;</span>
</pre></div>
</div>
</dd></dl>

<dl class="function">
<dt id="Bio.Seq.translate">
<code class="sig-prename descclassname">Bio.Seq.</code><code class="sig-name descname">translate</code><span class="sig-paren">(</span><em class="sig-param">sequence</em>, <em class="sig-param">table='Standard'</em>, <em class="sig-param">stop_symbol='*'</em>, <em class="sig-param">to_stop=False</em>, <em class="sig-param">cds=False</em>, <em class="sig-param">gap=None</em><span class="sig-paren">)</span><a class="headerlink" href="#Bio.Seq.translate" title="Permalink to this definition">¶</a></dt>
<dd><p>Translate a nucleotide sequence into amino acids.</p>
<p>If given a string, returns a new string object. Given a Seq or
MutableSeq, returns a Seq object with a protein alphabet.</p>
<dl class="simple">
<dt>Arguments:</dt><dd><ul class="simple">
<li><p>table - Which codon table to use?  This can be either a name
(string), an NCBI identifier (integer), or a CodonTable object
(useful for non-standard genetic codes).  Defaults to the “Standard”
table.</p></li>
<li><p>stop_symbol - Single character string, what to use for any
terminators, defaults to the asterisk, “*”.</p></li>
<li><p>to_stop - Boolean, defaults to False meaning do a full
translation continuing on past any stop codons
(translated as the specified stop_symbol).  If
True, translation is terminated at the first in
frame stop codon (and the stop_symbol is not
appended to the returned protein sequence).</p></li>
<li><p>cds - Boolean, indicates this is a complete CDS.  If True, this
checks the sequence starts with a valid alternative start
codon (which will be translated as methionine, M), that the
sequence length is a multiple of three, and that there is a
single in frame stop codon at the end (this will be excluded
from the protein sequence, regardless of the to_stop option).
If these tests fail, an exception is raised.</p></li>
<li><p>gap - Single character string to denote symbol used for gaps.
Defaults to None.</p></li>
</ul>
</dd>
</dl>
<p>A simple string example using the default (standard) genetic code:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">coding_dna</span> <span class="o">=</span> <span class="s2">&quot;GTGGCCATTGTAATGGGCCGCTGAAAGGGTGCCCGATAG&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">translate</span><span class="p">(</span><span class="n">coding_dna</span><span class="p">)</span>
<span class="go">&#39;VAIVMGR*KGAR*&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">translate</span><span class="p">(</span><span class="n">coding_dna</span><span class="p">,</span> <span class="n">stop_symbol</span><span class="o">=</span><span class="s2">&quot;@&quot;</span><span class="p">)</span>
<span class="go">&#39;VAIVMGR@KGAR@&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">translate</span><span class="p">(</span><span class="n">coding_dna</span><span class="p">,</span> <span class="n">to_stop</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="go">&#39;VAIVMGR&#39;</span>
</pre></div>
</div>
<p>Now using NCBI table 2, where TGA is not a stop codon:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">translate</span><span class="p">(</span><span class="n">coding_dna</span><span class="p">,</span> <span class="n">table</span><span class="o">=</span><span class="mi">2</span><span class="p">)</span>
<span class="go">&#39;VAIVMGRWKGAR*&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">translate</span><span class="p">(</span><span class="n">coding_dna</span><span class="p">,</span> <span class="n">table</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">to_stop</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="go">&#39;VAIVMGRWKGAR&#39;</span>
</pre></div>
</div>
<p>In fact this example uses an alternative start codon valid under NCBI
table 2, GTG, which means this example is a complete valid CDS which
when translated should really start with methionine (not valine):</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">translate</span><span class="p">(</span><span class="n">coding_dna</span><span class="p">,</span> <span class="n">table</span><span class="o">=</span><span class="mi">2</span><span class="p">,</span> <span class="n">cds</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="go">&#39;MAIVMGRWKGAR&#39;</span>
</pre></div>
</div>
<p>Note that if the sequence has no in-frame stop codon, then the to_stop
argument has no effect:</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">coding_dna2</span> <span class="o">=</span> <span class="s2">&quot;GTGGCCATTGTAATGGGCCGC&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">translate</span><span class="p">(</span><span class="n">coding_dna2</span><span class="p">)</span>
<span class="go">&#39;VAIVMGR&#39;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">translate</span><span class="p">(</span><span class="n">coding_dna2</span><span class="p">,</span> <span class="n">to_stop</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="go">&#39;VAIVMGR&#39;</span>
</pre></div>
</div>
<p>NOTE - Ambiguous codons like “TAN” or “NNN” could be an amino acid
or a stop codon.  These are translated as “X”.  Any invalid codon
(e.g. “TA?” or “T-A”) will throw a TranslationError.</p>
<p>It will however translate either DNA or RNA.</p>
<p>NOTE - Since version 1.71 Biopython contains codon tables with ‘ambiguous
stop codons’. These are stop codons with unambiguous sequence but which
have a context dependent coding as STOP or as amino acid. With these tables
‘to_stop’ must be False (otherwise a ValueError is raised). The dual
coding codons will always be translated as amino acid, except for
‘cds=True’, where the last codon will be translated as STOP.</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">coding_dna3</span> <span class="o">=</span> <span class="s2">&quot;ATGGCACGGAAGTGA&quot;</span>
<span class="gp">&gt;&gt;&gt; </span><span class="n">translate</span><span class="p">(</span><span class="n">coding_dna3</span><span class="p">)</span>
<span class="go">&#39;MARK*&#39;</span>
</pre></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">translate</span><span class="p">(</span><span class="n">coding_dna3</span><span class="p">,</span> <span class="n">table</span><span class="o">=</span><span class="mi">27</span><span class="p">)</span>  <span class="c1"># Table 27: TGA -&gt; STOP or W</span>
<span class="go">&#39;MARKW&#39;</span>
</pre></div>
</div>
<p>It will however raise a BiopythonWarning (not shown).</p>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">translate</span><span class="p">(</span><span class="n">coding_dna3</span><span class="p">,</span> <span class="n">table</span><span class="o">=</span><span class="mi">27</span><span class="p">,</span> <span class="n">cds</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="go">&#39;MARK&#39;</span>
</pre></div>
</div>
<div class="highlight-default notranslate"><div class="highlight"><pre><span></span><span class="gp">&gt;&gt;&gt; </span><span class="n">translate</span><span class="p">(</span><span class="n">coding_dna3</span><span class="p">,</span> <span class="n">table</span><span class="o">=</span><span class="mi">27</span><span class="p">,</span> <span class="n">to_stop</span><span class="o">=</span><span class="kc">True</span><span class="p">)</span>
<span class="gt">Traceback (most recent call last):</span>
   <span class="c">...</span>
<span class="gr">ValueError</span>: <span class="n">You cannot use &#39;to_stop=True&#39; with this table ...</span>
</pre></div>
</div>
</dd></dl>

</div>


           </div>
           
          </div>
          <footer>
  
    <div class="rst-footer-buttons" role="navigation" aria-label="footer navigation">
      
        <a href="Bio.SeqFeature.html" class="btn btn-neutral float-right" title="Bio.SeqFeature module" accesskey="n" rel="next">Next <span class="fa fa-arrow-circle-right"></span></a>
      
      
        <a href="Bio.NaiveBayes.html" class="btn btn-neutral float-left" title="Bio.NaiveBayes module" accesskey="p" rel="prev"><span class="fa fa-arrow-circle-left"></span> Previous</a>
      
    </div>
  

  <hr/>

  <div role="contentinfo">
    <p>
        &copy; Copyright 1999-2017, The Biopython Contributors

    </p>
  </div>
  Built with <a href="http://sphinx-doc.org/">Sphinx</a> using a <a href="https://github.com/rtfd/sphinx_rtd_theme">theme</a> provided by <a href="https://readthedocs.org">Read the Docs</a>. 

</footer>

        </div>
      </div>

    </section>

  </div>
  


  <script type="text/javascript">
      jQuery(function () {
          SphinxRtdTheme.Navigation.enable(true);
      });
  </script>

  
  
    
   

</body>
</html>